---
title: "FG1 - Learning Objectives"
author: "Written by: Klaas, Simon, Siebren, Christa ... "
output: 
  html_document:
    css: custom.css
    toc: true
    toc_float: true
    collapsed: false
    number_sections: false
    toc_depth: 3
editor_options: 
  chunk_output_type: inline
---
<style type="text/css">
.title {
  display: none;
}

#getting-started img {
  margin-right: 10px;
}

</style>

<div class="row" style="padding-top:20px;">
<br>

```{r setup, include=FALSE}
knitr::opts_chunk$set(message=FALSE,warning=FALSE, cache=TRUE)
```

> ### Goal:
In this practical course we will eventually examine epigenetic marks around differentially expressed genes and how these marks are different.  


<br>
**In <span style="color: red;">red</span> are things that we could also remove, they might for example be disturbing from the bigger picture or it is a list that we might expand or shorten**  
<br>

> ### Global Learning Objectives  
1. Understand how data are generated from a ChIP-sequencing experiment.   
2. Know how to display and browse data in the UCSC Genome Browser and formulate some basic hypothesis based on observations using this tool.  
3. Familiarize yourself with several basic Bioconductor packages.   
4. Quantify and visualize signal distribution and signal strength of histone ChIP-seq in R.   
5. Summarize genome-wide data ChIP-data into clustered heatmap.   
6. Relate marker genes from cell clusters detected in single-cell RNA-seq experiment to differences between cells at the epigenetic level. 

> ### Week 1 Learning Objectives: seq2science, UCSC Genome browser  
1. Know how output from the sequencer looks like  and what processing steps should be taken to obtain trustworthy data that we visualize and that we can perform calculations (and statistics) on.  
2. Understand how to move around ("*browsing*") in the UCSC genome browser and know how to go to regions and genes of interest.  
3. Recognize the following elements in the UCSC genome browser: *chromosome, centromere, telomere, gene, exon, intron, <span style="color: red;">utr</span>, promoter, enhancer/repressor*.   
4. Define the following terms as they relate to the UCSC genome browser: *browsing, track, coverage, peaks, <span style="color: red;">summit</span>*.    
5. Recognize and differentiate the profile of <span style="color: red;">*ATAC-seq, RNA-seq and histone ChIP-seq* datasets in the genome browser based on their coverage profile and their location with respect to genes (H4K4me3, H3K27ac, H3K36me3, H3K9me3 histone ChIP-seq</span>).   

Q1: We want to know what kind of histone mark can be found in the promoter of ... gene   
Q2: Find histone mark covering the enhancer located 20kb upstream of ....  

> ### Week 2 Learning Objectives: Bioconductor and comparison among marks  
1. Move around in GRanges container  
2. Use GenomicFeatures to obtain 
3. Find and count the overlap among histone ChIP-seq peaks and promoters or genic regions   
- `countOverlaps(query, subject)`, `findOverlaps(query, subject)`  
4. Plotting summarized coverage of histone marks around TSS and over genic regions <span style="color: red;">--> or this one to week 3</span>    

In week 1 we have estimated which mark is enriched in promoters, gene bodies or other intergenic regions, this week we will add a number to that estimate.  
Q1: We want quantify in what region the H3Kme3 signal is enriched.  
Q2: We want to do the same for H3K27ac and for H3K36me3.  
Q3: What is the average coverage profile of different marks around landmark elements like TAG, TSS and TSS

> ### Week 3 Learning Objectives: Comparison among cells   
1. visualize histone marks of one cell around genomic landmarks in heatmap  
2. be able to cluster heatmap data  
3. instead of different marks, use same mark from different cell types  

Q1: Do histone marks show regional overlap? --> plot different marks in a heatmap and cluster the data  

Q2: Want to identify differentially marked regions and common marked regions among different cell types.  

> ### Week 4 Learning Objectives: relate gene expression differences to epigenetic changes  
1. go from list with DE genes per cell type to TSSs  
2. Plot histone markings around these genes in different cell types  

Q1: What histone marks can be found around DE genes of different cell types    


Q2: 
