---
title: "1. UCSC Genome Browser"
output:
  html_document: 
    collapsed: no
    css: custom.css
    highlight: pygments
    theme: cerulean
    toc: yes
    toc_float: yes
editor_options:
  chunk_output_type: inline
---

<style type="text/css">
.title {
  display: none;
}

</style>


```{r setup, include=FALSE}
#library(learnr)
knitr::opts_chunk$set(message=FALSE,warning=FALSE, cache=TRUE)
```

## 1.1 Learning Objectives  
<br>
**This week we will look at how sequencing data is processed and familiarize ourselves with the UCSC Genome browser to visually inspect (epi-)genomics data.**   
<br>

> 1. Know how raw reads (from eg ChIP-seq) are processed to obtain trustworthy and interpretable data.  
2. Understand how to move ("*browse*") through the UCSC genome browser and know how to go to regions and genes of interest.  
3. Recognize the following elements/regions in the UCSC genome browser: chromosome, <span style="color: red;">centromere, telomere</span>, gene, exon, intron, utr, promoter, enhancer/repressor.   
4. Tell apart UCSC genome browser tracks of H3K27ac, H3K27me3, H3K36me3, H3K4me1, H3K4me3 and H3K9me3 histone ChIP-seq and ATAC-seq signal (and ChIP-seq input) based on their coverage profile and their location with respect to genes and other annotated genomic/chromosomal elements. 
5. Can interpret - and thus understand - ChIP- and ATAC-seq signal in the UCSC genome browser.  
- These refer to Global learning objectives #1-3.   

<br>  

***  
<br> 

## 1.2 Genomic methods recap  

- What is ChIP-seq? What is ATAC-seq? Presentation with ChIP-seq steps? Refer to book?  
- What is a library and what do we mean with 'library preparation'? What's the use of adapters?    
- Sequencing samples. What happens in sequencer? What is the output?  
- Brief genomics recap/pre-quiz?  

<br>  

***  
<br>  

## 1.3 From raw data to science   
<center>  
![](images/week1/2_seq2science/whatbioinformaticiancsdo.png){width=60%}  
</center>  

<br><br>  

### 1.3.1 Sequencer output     
The sequencer determines the sequence of the DNA molecules in your sample and produces massive amounts of data. The read sequences are written to `.fastq` files that show 4 lines per DNA sequence. This is an example with 2 DNA sequences (8 lines):  
<br>
</center>  
![](images/week1/2_seq2science/fastq_example3.png)   
</center>  
<br>

The lines in the fastq file tell us:   
 
- line 1: start with '@'-sign and contains sequence ID given by the sequencer including sequencer id, flowcell id and sample barcode   
- line 2: the read DNA sequence in ACGTN-alphabet  
- line 3: '+' (originally reported the sequence ID again)  
- line 4: quality score for each nucleotide   

Fastq files have the extension .fq or .fastq, or are compressed as .fq.gz or .fastq.gz  

<br>  

***  
<br> 

### 1.3.2 Post-sequencing processing steps   
The chemical reaction and quality of the optical measurement that occur in the sequencer are not perfect and depend among others on the nucleotide composition of all samples, the length of the DNA molecules and the concentration of the samples. The Phred quality score  (line 3) represents the nucleotide identification accuracy or base call accuray. This encoded score can be used downstream to remove nucleotides or complete reads with low quality.  
  
In adddition, you want to know where the DNA molecule came from in your reference genome and, depending on your experimental set-up, likely want to identify regions with more reads than expected if reads were randomly distributed over the genome. E.g. identify peaks of TF binding, peaks of histone variants, expressed genes, or interacting chromosomal regions.   

It is the bio-informaticians (or your) task to:  

1. **Remove adapters and poor-quality nucleotides**: The adapters should be removed, as they were not part of the original sample and low-quality nt's are more likely to be wrong and we therefore remove those as well. Can result in that reads are removed completely because of their overall low quality or short length.   
<br>
2. **Align reads to the genome**: Identify the likely location in the reference genome where the original DNA molecule came from. Since the first publication of the human genome, successive versions (also called *assemblies* or *builds*) have been released with improvements in quality thanks to technological advancements as well as improvements in the representativeness of the genome with respect to the genetic diversity found in the human population, including historically underrepresentated populations. We will work with the (one of the) latests relasese referred to as GRCh38 (Genome Reference Consortium Human Build 38). <span style="color: red;">A large number of alignment softwares are available... </span>. The alignment procedure often adds flags for reads aligned to a unique region or to multiple regions which you can use to filter.  
<br>
3. **Filter artefacts and reads aligning to multiple regions**: Remove duplicated regions, remove PCR artefacts and optionally remove reads aligning non-uniquely.   
<br>
4. **Visualize the data**: .Bigwig files are created which can be uploaded to the UCSC Genome Browser which we will look at in more detail today as well.  
<br>
5. **Call peaks**: Computationally identify regions with significantly higher number of reads than expected. There are different tools out there to perform this task MACS2 is one of them and was used to call peaks in the datasets that we will be using. Peak calling is a mathemetical/statistical procedure that depends on set thresholds (p-value, false-discovery rate), the control sample used. It is always wise to visually inspect the resulting peaks in the UCSC genome browser.  
<br>
6. **Read counting**: Count reads that fall within genes, other genomic elements or peak regions. These countings and the actual peak locations are used for eg. differential gene expression in differentiated vs undifferentiated. Or control versus treatment. A well-known, widely used tool for this purpose is DESeq2.  
<br>
7. Optionally **DNaseI footprinting**, **variant calling**, **novel gene identification**: Depending on your experimental set up ... For DNaseI-seq, we can identify small depleted regions within peaks that correspond to the nucleotides that interact with the TF. This interaction protects these nucleotides from DNaseI attacks. These so-called footprints can be used to identify TF motifs and predict the likely TF bound. Although this sounds simple, and for a stable binding factor like CTCF it may be the case, it is not always that simple in reality.    
<br>

<span style="color: red;">Visual depiction of the different steps -- or make assignment to put them in the right order with quality checks in between?</span> E.g.:  
![](images/week1/2_seq2science/seq2science_siebrenspptx.png){width=70%}\    
[Sec2science / adapted from Siebrens presentation / shows RNA-seq pipeline ](https://vanheeringen-lab.github.io/seq2science/content/workflows.html)      
<br>  

![](images/week1/2_seq2science/encode_chipseq_readmapping.png){width=50%}\   
[The ENCODE Project, ChIP-seq read mapping ](https://www.encodeproject.org/pipelines/ENCPL220NBH/)   
<br>  

![](images/week1/2_seq2science/Furay_2012_naturereviews_fig2_chipseqanalysis.png){width=70%}\   
[Furay, 2012, Nat Rev](https://pubmed.ncbi.nlm.nih.gov/23090257/)  

<br>  

***  
<br> 

### 1.3.3 Quality evaluation    
It is important to check the quality of your data along the way:   

- FASTQC (Qualty Check) software visaully summarizes the attached Phred quality scores.  
- Evaluate the data yourself in the genome browser and compare replicate samples, samples of different histone marks, compare to a reference dataset or compare to the input material.   
- Clustering of samples; can reveal label swaps, failed experiments or contaminations.   

<br>  

***  
<br>  

### <span style="color: red;">1.3.4 Output files: bigwigs, peaks & counts</span>    
<span style="color: red;">Showing examples of how the output may look like or do this along the way in the numbered list above?  

<br>  

***  
<br> 

### <span style="color: red;">1.3.5 Alternatively: use FAIR data   
<span style="color: red;">Several reserach consortia provide easy access and clear description of their data through data portals. These files can be downloaded or immediately uploaded to the UCSC Genome browser using the URL.   
</span> 

<br>  

***  
<br> 

### 1.3.6 Exercises   
Ideas:  

- Put steps in right order with options for quality checks.  
- Recognize a type of file?  
- Evaluate good/bad fastqc output?  

<br>  

***  
<br>  

## 1.4 Data visualization: UCSC Genome Browser     

> UCSC GB sessions for making these exercises:  
>  
- Default should look like (if it doesn't, I need to adjust the text: http://genome-euro.ucsc.edu/s/ctoenhake/hg38_default    
- Genes and ruler only: http://genome-euro.ucsc.edu/s/ctoenhake/hg38_genesonly   
- Monocyte data (blueprint in colors, encode in black): http://genome-euro.ucsc.edu/s/ctoenhake/hg38_monos_blueprint_and_encode    

#### 1.4.1 UCSC Genome browser intro   
Genome browsers are great tools to display genomics data. You can walk, or *browse*, through all chromosomes, look at the location of genes, extract DNA sequences, examine pre-loaded data or upload your own data.  
<br>

> If you didn't do so already, watch the following youtube video's to get familiar with the UCSC Genome Browser:  
>  
1. [UCSC Genome Browser Basics](https://www.youtube.com/watch?v=DNXI-M9oQl8)  
2. [UCSC Genome Browser Tracks](https://www.youtube.com/watch?v=1Kpy5G_MSG8)  
>
> **Optionally**, finding the coordinates of a sequence using blat, handy for eg. primers, and using the Table Browser to, among others, obtain DNA, RNA or protein sequences:  
>  
3. [UCSC Genome Browser Blat](https://www.youtube.com/watch?v=E6DDNGDXH8A)   
4. [UCSC Genome Browser Table Browser](https://www.youtube.com/watch?v=Ueqt5Wgre9c&ab)  

<br>

### 1.4.2 Go to the UCSC Genome browser  
**First, we will familiarize ourselves with browsing through the UCSC genome browser and the recognition of genes, their coding elements and their orientation.**  

<span style="color: royalblue; font-family: Arial; font-size:18px;">ACTION: Go to the UCSC Genome Browser, [genome.ucsc.edu](genome.ucsc.edu).</span>   
<br> 
The starting page has been updated since the video was made but still holds the same blue bar at the top with **Genomes**, **Tools**, etc:  

![](images/week1/3b_ucsc_gb_looking_at_genes/ucsc_genome_browser_fig1_startpage_topmenubar.png){width=100%}\    
<br>

> *Tip*: In case you have been here before, it is wise to reset user settings such that you don't continue with settings from a previous project. For this, hover over "Genome Browser" and select "Reser All User Settings".  

<span style="color: royalblue; font-family: Arial; font-size:18px;">ACTION: Go to the "Genomes" tab.</span>     
<br>

> *Tip*: A pop-up may appear now or one stap later, redirected you to the official European mirror site if those servers are geographically closer, that's ok.  
>  
- European mirror site, servers are located at the Universität Bielefeld Center for Biotechnology in Bielefeld, Germany: [http://genome-euro.ucsc.edu](http://genome-euro.ucsc.edu)  
- Asian mirror site, located at the RIKEN Yokohama Campus, Japan: [http://genome-asia.ucsc.edu](http://genome-asia.ucsc.edu)  

<br>
On the left you can browse and select the genome you are interested in (not a drop-down menu anymore as in the video). By default, the human genome is selected.   
<br>
<span style="color: royalblue; font-family: Arial; font-size:18px;">ACTION: Make sure that "Dec. 2013 (GRCh38/hg38)" is selected under "Human Assembly".</span>     
<br>
We can go to a position of our choice by entering the gene or chromosomal coordinates under **Position/Search Term** or immediatly go to the 'Current position' by clicking on **GO** on the right-hand side or **Genome Browser** in the menu bar on top.  
<br>
<span style="color: royalblue; font-family: Arial; font-size:18px;">ACTION: Go to *ACTB* gene (gene coding for non-muscular actin).</span>     

> *Hint* your browser should be showing chr7:5,527,148-5,530,601.     

As discussed in the video, in the top of the window we have navigation and zoom bottons. Right below we can read the chromosomal coordinates of and length of the gene and below that we see a miniature chromosome with the characteristic gray/black pattern from karyotyping. A red |-sign gives us an idea of our current positioning in the chromosome.   

![](images/week1/3b_ucsc_gb_looking_at_genes/ucsc_gb_actb_locationinchromosome.png){width=100%}\     
<br>
```{r Q1_actin_location, eval = F, echo=T, class.source = "exercise"}
     question("Question 1: What chromosome is ACTB located on? And is it located on the short or long arm of this chromosome?",
              answer("chr7, short arm", correct = TRUE),
              answer("chr7, long arm", message = "Where do you see the red |-sign in the miniature chromosome on top?"),
              answer("chr17, short arm", message = "Review the UCSC Genome Browser Basics video, or find *chr#* in the browser window"),
              answer("chr17, long arm", message = "Review the UCSC Genome Browser Basics video, or find *chr#* in the browser window"),
              allow_retry = TRUE
              )
```
<br>
The data window displays various *tracks* or *track sets*. Tracks runs the complete length of the chromosome. Data includes most often:   

- **positional information**, of a gene, SNP or cir-regulatory element for example, often displayed as horizontal bars the thickness and color of which can hold additional information depending on the type of track.  
- **signal information**, of gene expression, detection of histone marks or degree of conservation. Often displayed as histograms but can also be summarized in vertical bars with different levels of color intensity encoding the signal intensity.   

<br>
<span style="color: red;"> Question asign what the name of the track with gene locations is or poicutre with c/b/c/d/e/f etc. instead of track name and asking which one diplays the chromosomal coordinates and which one the gene locations?</span> 

<br>
The method of display and whether individual tracks are displayed can be changed by:  

1. Right-mouse click on the grey, vertical bar at the left-most area of the Genome Browser shows you options `hide`, `dense`, `squish`, `pack` and `full`. Additional adjustments to the track can be made by selecting `Configure [track name]` in this menu.  
2. Scroll to the area below the data window, here you see a possibly overwhelming list of potential tracks, most are `hidden`. Using the pull down menu you can change display settings of multiple tracks at once. Always make sure to hit the `refresh` button below the browser window to ensure all changes are performed.  

<br>
<span style="color: royalblue; font-family: Arial; font-size:18px;">ACTION: (1) hide all tracks and tracksets except for "ruler", "GENCODE v32". And (2) for "GENCODE v32" via `Configure GENCODE v32`, make sure the option `splice variants` is unchecked.</span>     
<br>
Your GB session should look like this: ![](images/week1/3b_ucsc_gb_looking_at_genes/ucsc_gb_actb_genetracksonly.png){width=100%}\      
<br>
We see the complete *ACTB* gene with exons as horizontally stretched rectangles and introns as horintal lines. The rectangles with largest height within exons are the *coding* exons, the other regions are the 5- and 3UTR. The arrowheads inside the introns indicate the coding diretion.   
<br>
More information about *ACTB* can be obtained by clicking on the gene in the browser:  
![](images/week1/3b_ucsc_gb_looking_at_genes/ucsc_gb_actb_geneinformation.png){width=100%}\    
<br>
```{r ACTB_exons, eval = F, echo=T, class.source = "exercise"}
question("Question 2: How many exons does the *ACTB* gene have?",
          answer("5", message = "exons are displayed as thicker lines in the gene, introns as lines with arrows in them."),
          answer("6", correct = TRUE),
          answer("7", message = "one exon can contain coding and non-coding (UTR) elements" ),
          answer("8", message = "one exon can contain coding and non-coding (UTR) elements"),
           allow_retry = TRUE)
```
<br>  
![ACT locus for question 3](images/week1/3b_ucsc_gb_looking_at_genes/ucsc_gb_actb_question_TSSquestion.png){width=100%}\        
<br>     
```{r ACTB_start, eval = F, echo=T, class.source = "exercise"}
question("Question 3: Which letter points to the start codon?",
              answer("A", message = "Look at the directionality of the gene, this is more likely the end of the gene."),
              answer("B", message = "Look at the directionality of the gene, this is more likely the end of the coding exons."),
              answer("C", correct = TRUE),
              answer("D", message = "Be aware of the small 5UTR in this exon!"),
              answer("E", message = "This is more likely the TSS."))
```

```{r ACTB_statements, eval = F, echo=T, class.source = "exercise"}
     question("Question 4: What statement about the *ACTB* gene is correct? (Select ALL that apply.) ",
              answer("ACTB gene is encoded on the minus-strand", correct = TRUE),
              answer("The TTS of ACTB gene has a higher coordinate than its TSS", message = "Look at the direction of arrows in the introns."),
              answer("The coding size of this gene is appr 2kb", correct = TRUE),
              answer("If we want to look at the promoter of this gene, we have to move slightly to the left", correct = TRUE),
              answer("This gene is not transcribed", message = "you don't have data to say this is correct")
)

```
<br>

<span style="color: royalblue; font-family: Arial; font-size:18px;">ACTION: Go to the gene *SPI1*, this gene encodes the myeloid and B-cell–specific transcription factor PU.1 [Pham 2012 Blood](https://doi.org/10.1182/blood-2012-01-402453).</span>      

* Copy this sentence and fill in the blanks with numbers in the code:   

SPI1 gene is located on chr [...]. It spans [...] bp and is encoded on the [plus/minus] strand. It has [...] exons and [...] introns. It is an [zinc finger domain TF/ETS-domain transcription factor]. The first downstream protein-gene is [MYBPC3/SLC39A13]

```{r spti, exercise =T, eval = F, exercise.eval=FALSE}
# Fill in the blanks with numbers in the code:  
SPI1 gene is located on chr ...  . It spans ... bp and is encoded on the [plus/minus] strand. It has ... coding exons and ... introns. It is an [zinc finger domain TF/ETS-domain transcription factor]. The first downstream protein-gene is [MYBPC3/SLC39A13].  
```


**Wonderfull, you are now able to recognize genes, their location within the chromosome, and thei coding elements as well as their orientation. These are important reference points used when we evaluate the quality of epigenomics data and to formulate hypothesis that we can test later on.**  

***  
<br>  

### 1.4.3 Viewing ENCODE/BLUEPRINT data   

**Yes! Finally time for some real data! You will inspect the characteristic histone PTM and DNaseI-seq combinations, what their location in relation to genes and - ultimately - what this may say about the gene's transcription state.**   
<br>
To upload the data we have already put them together in a so-called track hub, hosted by our own server. We can add this to our browser session by going to the tab "My Data" in the top menu and select "Track Hubs". In the "My hubs" tab you can paste a URL that directs to the data you want to upload.  
<br>
![Go to Track Hubs](images/week1/3c_ucsc_gb_looking_at_data/ucsc_gb_encode_fig1_gototrackhub.png){width=100%}\    
![My Hubs window](images/week1/3c_ucsc_gb_looking_at_data/ucsc_gb_encode_fig2_trackhub.png){width=100%}\     
<br>
<span style="color: royalblue; font-family: Arial; font-size:18px;">ACTION: Use the description above and the trackhub URL [.....] to upload the required data.</span>     
<br>  
Note that, after adding the trackhub, an additional section is added below the browser, specfic to the tracks in this trackhub.  
<br>
This trackhub contains histone PTM ChIP-seq and ATAC-seq data of monocytes, to be precise of **CD14+, CD16- classical monocytes**. 
<br>

> *Background*: CD14 and CD16 are cell surface markers of the **CD (Cluster of Differentiation) gene family** which we will come across more often during this practical. These cell surface molecules and their differential expression are used for immunophenotyping of cells. CD molecules can function as receptors, ligands and, for example, activate signalling cascades, mediate cell adhesion, cell migration, regulate gene expression etc. [HUGO, genegoups, CD](https://www.genenames.org/data/genegroup/#!/group/471)  

<br> 
We have the following datasets:  

- ChIP-seq data histone PTMs:  

    * H3K9me3 (gray)   
    * H3K27me3 (gray)  
    * H3K4me3(red)   
    * H3K4me1(yellow)   
    * H3K27me1 (peach)  
    * H3K36me3 (green)   
    
- DNaseI-seq data in the blue track.   
- polyA-selected, stranded RNA-seq data, again in green.  
- Input ChIP-seq in black.  

> *Tip*: What we haven't discussed so far is the y-axis at the left side of the track with *signal* information such as the ones just uploaded. Using this y-axis we can evaluate signal intensities. The height is currently set to 30, 50 or 100 <span style="color: red;">Reads/million?</span>. You can change it by right-mouse click on the trakname, select the `Configure [trackname]` option and change the `min` and `max` of the **Vertical viewing range**. Hit `APPLY` afterwards.  

<br>
<span style="color: royalblue; font-family: Arial; font-size:18px;">ACTION: Go back to the ACTB locus and zoom out 3x to see the regions surrounding the *ACTB* locus.</span>     

> *Hint*: your browser should be showing chr7:5,523,694-5,534,055 approximately, a few bases more or less is not a problem.   

Use the RNA-seq data...   
```{r ACTB_expression, eval = F, echo=T, class = "bg-warning"}
     question("Question 5: would you conclude that the ACTB gene is active?",
              answer("Yes", correct = TRUE),
              answer("No", message = "Look at the signal in RNA-seq data")
     )
```
<br>
<span style="color: red;">Explain peaks?.... again?</span>  
<br>
```{r ACTB_ATACseq, eval = F, echo=T, class = "bg-warning"}
question("Question 6: Which of the following regions points to a DNaseI-seq peak?)",
  answer("chr7:5530101-5530501", message = "Nice try but this region has a H3K27ac ChIP-seq peak."),
  answer("chr7:5530501-5530800", correct = TRUE, message = "Yes. Note that this DNaseI-seq peak is located at the *ACTB* TSS/promoter"),
  answer("chr7:5532001-5532500", message = "Try again. If anything, you may be looking at the H3K4me1 signal.", ),
  answer("None of the these", message = "Try again. Please look again"),
  allow_retry = TRUE
     )
```

```{r ACTB_histone_marks_one, eval = F, echo=T, class.source = "exercise"}
question("Question 7: Which histone marks, are found at the ACTB promoter? (Select ALL that apply)",
  answer("H3K9me3", message = "wrong"),
  answer("H3K27me3", message = "wrong"),
  answer("H3K4me3", correct = TRUE, message = "correct"),
  answer("H3K4me1", message = "wrong"),
  answer("H3K27ac", correct = TRUE, message = "correct"),
  answer("H3K36me3", message = "wrong"),
  answer("none", message = "wrong"))
```

```{r ACTB_histone_marks_two, eval = F, echo=T, class.source = "exercise"}
question("Question 8: The DNaseI-seq peak located *upstream* of the *ACTB* could be an enhancer. What observation gives additional supports for this statement?",
  answer("Absence of H3K27me3", message = "Absence of H3K27me3 does not necessarily mean the region is an enhancer."),
  answer("Absence of H3K4me1 signal in this region.", message = "wrong"),
  answer("Presence of H3K4me1 signal in this region.", correct = TRUE ),
  answer("The location of this DNaseI-seq peak suggests it is as it lies outside a gene", message = "wrong"))
```
<br>

<span style="color: royalblue; font-family: Arial; font-size:18px;">ACTION:Zoom out 3x and then another 10x.</span>     

> *Hint*: your browser should be showing chr7:5,373,445-5,684,304 approximately, a few bases more or less is not a problem.   

A prominant H3K27me3 peaks comes in the viewing window now. It covers  the promoter of *FSCN1* gene.  

```{r FSCN1_bivalent_promoter, eval = F, echo=T}
question("Question 9: What additional histone marks, mark this region? (Select ALL that apply.",
         answer("H3K9me3", message = "wrong"),
         answer("H3K27me3", message = "wrong"),
         answer("H3K4me3", correct = T, message = "correct"),
         answer("H3K4me1", correct = T, message = "correct"),
         answer("H3K27ac", message = "wrong"),
         answer("H3K36me3", message = "wrong"),
         answer("none", message = "wrong")
         )
```
<br>
This is an exmaple of a bivalent promoter.  
<br>
```{r FSCN1_expression, eval = F, echo=T}
question("Question 10: Based on the histone marks, would you predict that *FSCN1* is transcribed?",
         answer("Yes", message = "wrong"),
         answer("No", correct = T)
         )
```
<br>
So far we have seen active promoters and bivalent promoters. Let's look at a silenced gene.

<span style="color: royalblue; font-family: Arial; font-size:18px;">ACTION: Go to the ACTA1 gene, [encoding muscular actin](https://www.genecards.org/cgi-bin/carddisp.pl?gene=ACTA1&keywords=acta1). Zoom out 10x.</span>     

> *Hint* you should be at ±  chr1:229,418,420-229,446,919.  

<br>
```{r ACTA1_marking_one, eval = F, echo=T}
question("Question 11: Which histone mark mediates repression of *ACTA1* in these cells?",
         answer("H3K9me3", message = "wrong"),
         answer("H3K27me3", correct = T),
         answer("H3K4me3",  message = "correct"),
         answer("H3K4me1", message = "correct"),
         answer("H3K27ac", message = "wrong"),
         answer("H3K36me3", message = "wrong"),
         answer("none", message = "wrong"),
         allow_retry = TRUE
         )
```

```{r ACTA1_marking_two, eval = F, echo=T}
question("Question 12: Which other observations support that this gene is not active? (Select ALL that apply)",
         answer("Absence of H3K36me3 marking covering the gene body.", message = "H3K36me3 is indeed lacking but other observations are correct as well."),
         answer("Absence of signal in RNA-seq signal", message = "RNA-seq is indeed lacking but other observations are correct as well."),
         answer("Absence of H3K4me3 signal at the *ACTA1* promoter",  message = "correct"),
         answer("All the above", correct = TRUE, message = "Indeed, the absence of signals in RNA-seq, H3K36me3 *and* H3K4me3 all support that this gene is not expressed. Observe that the H3K27me3 signal covers a relatively broad domain." ),
         allow_retry = TRUE
         )
```
<br>

<span style="color: royalblue; font-family: Arial; font-size:18px;">ACTION: Go to the MYOD1 gene, [encoding TF for muscle cell differentiation](https://www.genecards.org/cgi-bin/carddisp.pl?gene=MYOD1&keywords=myod1). Zoom out 3x.</span>     

> *Hint* you should be at ±  chr11:17,717,005-17,724,702.    

<br>
```{r MYOD1_marking, eval = F, echo=T}
question("Question 12: Which mark is likely mediates the silencing of *MYOD1* in these cells?",
         answer("H3K9me3", message = "wrong"),
         answer("H3K27me3", correct = T),
         answer("H3K4me3",  message = "correct"),
         answer("H3K4me1", message = "correct"),
         answer("H3K27ac", message = "wrong"),
         answer("H3K36me3", message = "wrong"),
         answer("none", message = "wrong"),
         allow_retry = TRUE
         )
```
<br>

Other H3K37me3 marked promoters:  

- SLCA4A6 gene, encoding a serotonin (neurotransmitter) ... bivalent? Has bit of H3K4me1.  
- OCE2 (eye color) - also bivalent?  
- NEUROG2 (tf for noerogenesis) - also bivalent?   
- OLIG2 (tf neuro/oligodendrite genesis) - also bivalent?  

Also look at a broad H3K9me3 domain? I see some but those have H3K4me3 and H3K36me3 marking as well.... 

- Background:  

    * H3K9me3 spans braod domains of 10ks of bp's. Constitutive hetrochromatin and less often facultative heterochromatin.  
    * H3K9me3 and H3K27me3 are mechanistically differen.  
    * Not part of bivalent promoters?  
    * reviews: [Becker 2017 Trens in Genetics](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4698194/); [Ninova 2019 Genomics](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6803365/)   

- Example locations:  

    * chr19 broad domains with KRAB-ZNF encoding genes. Spanning 0.4-1Mb chr19:28,115,617-58,617,616 [Vogel 2005 Genome Res](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1665633/)  
    * 
H3K9me3 is associated with non-coding regions in the genome as well as repressing lineage-inappropriate genes [Becker 2015 Trends in Genetics](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4698194/)  

### 1.4.4 Recognizing blinded genomics data  
So far you have browsed through ChIP-, DNaseI- and RNA-seq data. These tracks had the correct track names attached to them.   
<br>
<span style="color: royalblue; font-family: Arial; font-size:18px;">ACTION: upload the following track hub [URL].</span>   
<br>
Our collegue accidently let the tubes fall before the tubes were properly labelled (always label your tubes first before adding your sample!). We know ChIPs were performed for H3K9me3, H3K27me3, H3K4me3, H3K36me3, H3K4me1 and H3K27ac. Which track (labelled a-f) is which?  


## Old questions, ignore please!!  
Let's go to the **CD14** gene.  
![](images/week1/ucsc_genome_browser_fig2_genomespage_cd14.png){width=80%}\    
<br>  

<br>

```{r Q1_cd14_location, eval = F, echo=T, class.source = "exercise"}
     question("What chromosome is CD14 located on? And is it located on the short or long arm of this chromosome?",
              answer("chr5, short arm", message = "Where do you see the red |-sign in the miniature chromosome on top?"),
              answer("chr5, long arm", correct =TRUE),
              answer("chr14, short arm", message = "Review the UCSC Genome Browser Basics video"),
              answer("chr14, long arm", message = "Review the UCSC Genome Browser Basics video, or find *chr#* in the browser"),
              allow_retry = TRUE
              )
```
<br>

Focus on the **GENCODE v32 gene annotation. This image is a screenshot of the CD14 gene:  
![](images/week1/ucsc_genome_browser_fig3_q2_V3.png)\    
<br>  

Focus on the top gene, <span style="color: red;"> Or remove isoforms?</span>  

```{r Q2_gene_elements, eval = F, echo=T, class.source = "exercise"}
quiz(caption = "Quiz 2", 
     question("What element(s) become, once this gene is transcribed, part of the mature CD14 mRNA?",
              answer("only D", message = "utrs are also part of the mRNA"),
              answer("D, H and I", message = "most mRNAs have an UTR on both the 3' and 5' side"),
              answer("C, D and G", message = "almost correct, but focus on the top gene" ),
              answer("C, D, H and I", correct = TRUE),
              allow_retry = TRUE),
     
     question("Which arrows point to the start- and stopcodon?",
              answer("A points to the startcodon, J to the stopcodon", message = "the startcodon is most often not the same as the TSS"),
              answer("B points to the startcodon, F to the stopcodon", message = "look at the directionality of the gene"),
              answer("F points to the startcodon, B to the stopcodon", correct = TRUE),
              answer("J points to the startcodon, A to the stopcodon", message = "the startcodon is most often not the same as the TSS"),
              allow_retry = TRUE),
     
     question("What statement about the configuration of CD14 gene is correct? (Select ALL that apply) ",
              answer("CD14 gene is encoded on the minus-strand", correct = TRUE),
              answer("The TTS of CD14 gene has a higher coordinate than its TSS", message = "look at the direction of arrows in the introns"),
              answer("Transcription of this gene can result in at least two different mRNA isoforms that differ in their 3UTR length", message = "look at the directionality of the gene"),
              answer("If we want to look at the promoter of this gene, we have to move slightly to the left", correct = TRUE),
              answer("This gene is not transcribed", message = "you don't have data to say this is correct")
              )
)

```




