---
title: "Prepare files for week 2"
author: "Christa Toenhake"
date: "13/11/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r libraries, message=FALSE, warning=FALSE}
library(tidyverse)
library(GenomicRanges)
library(rtracklayer)
library(org.Hs.eg.db)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene 
```

```{r prepare monocyt data}
# this is a preperatory chunk 
# add  exercise.setup = "mono-data" to chunks that require this data to be evailable
# H3k4me1 
mono_h3k4me1 <- rtracklayer::import("data/blueprint/bed/C000S5H2.ERX547981.H3K4me1.bwa.GRCh38.broad.20150527.chr19.bed", format = "broadPeak")
# H3K4me3  
mono_h3k4me3 <- rtracklayer::import("data/blueprint/bed/C000S5H2.ERX547984.H3K4me3.bwa.GRCh38.20150527.chr19.bed", format = "narrowPeak")
# H3K9me3 
mono_h3k9me3 <- rtracklayer::import("data/blueprint/bed/C000S5H2.ERX547982.H3K9me3.bwa.GRCh38.broad.20150527.chr19.bed", format = "broadPeak")
# H3K27ac 
mono_h3k27ac <- rtracklayer::import("data/blueprint/bed/C000S5H2.ERX547980.H3K27ac.bwa.GRCh38.20150527.chr19.bed", format = "narrowPeak")
# H3K27me3   
mono_h3k27me3 <- rtracklayer::import("data/blueprint/bed/C000S5H2.ERX547983.H3K27me3.bwa.GRCh38.broad.20150527.chr19.bed", format = "broadPeak")
# H3K36me3
mono_h3k36me3 <- rtracklayer::import("data/blueprint/bed/C000S5H2.ERX547979.H3K36me3.bwa.GRCh38.broad.20150527.chr19.bed", format = "broadPeak")
# create list
mono_list  <- GRangesList(mono_h3k4me1, mono_h3k4me3, mono_h3k9me3, mono_h3k27ac, mono_h3k27me3, mono_h3k36me3)
# add names to each element in the list
names(mono_list) <- c("h3k4me1", "h3k4me3", "h3k9me3", "h3k27ac", "h3k27me3", "h3k36me3")
save(mono_h3k4me1, mono_h3k4me3, mono_h3k9me3, mono_h3k27ac, mono_h3k27me3, mono_h3k36me3, mono_list, file =  "data/prepared_rds/blueprint_monocyte_chr19_granges.RData")
```

```{r prepare-anno, include = F}
seqlevels(txdb) <- "chr19" 

# generate list of genomic annotations
proximal.promoter.cutoff=1000L # Specify the cutoff in bases to classify proximal promoter or enhencer. Peaks that reside within proximal.promoter.cutoff upstream from or overlap with transcription start site are classified as proximal promoters. The default is 1000 bases.
immediate.downstream.cutoff=1000L # Specify the cutoff in bases to classify immediate downstream region or enhancer region. Peaks that reside within immediate.downstream.cutoff downstream of gene end but not overlap 3 prime UTR are classified as immediate downstream. 
#' The default is 1000 bases.

transcripts_chr19 <-  unique(transcripts(txdb, columns=NULL))
exons_chr19 <- exons(txdb, columns=NULL)
introns_chr19 <- unique(unlist(intronsByTranscript(txdb)))
fiveUTRs_chr19 <- unique(unlist(fiveUTRsByTranscript(txdb)))
threeUTRs_chr19 <- unique(unlist(threeUTRsByTranscript(txdb)))
promoters_chr19 <- unique(promoters(txdb, upstream=proximal.promoter.cutoff, downstream=0))
immediateDownstream_chr19 <- unique(flank(transcripts_chr19, width=immediate.downstream.cutoff, start=FALSE, use.names=FALSE))

# make list of annotations
annotation_chr19 <- list(exons_chr19, introns_chr19, fiveUTRs_chr19, threeUTRs_chr19, promoters_chr19, immediateDownstream_chr19)

# remove all metadata information
annotation_chr19 <- lapply(annotation_chr19, function(.anno){mcols(.anno)<-NULL; .anno})

# add names
names(annotation_chr19)[1:6] <- c("Exons", "Introns", "fiveUTRs", "threeUTRs", "Promoters", "immediateDownstream")

# convert to GRangeslist
annotation_chr19_gr <- GRangesList(annotation_chr19)

# to make 'intergenicRegions', add up all above annotated regions and identify the remaining gaps
# we ignore strand of the annotations because peaks are unstranded
newAnno <- c(unlist(annotation_chr19_gr))
newAnno.rd <- newAnno
strand(newAnno.rd) <- "*"
newAnno.rd <- reduce(trim(newAnno.rd))
intergenicRegions_chr19 <- gaps(newAnno.rd, end=seqlengths(txdb))
intergenicRegions_chr19 <- intergenicRegions_chr19[strand(intergenicRegions_chr19)=="*"]
names(intergenicRegions_chr19) <- NULL

# add intergenicRegions to annotation list
annotation_chr19_gr$intergenicRegions <- intergenicRegions_chr19

rm(newAnno.rd, newAnno, annotation_chr19, transcripts_chr19)

# save the final GRangesList object
saveRDS(annotation_chr19_gr, file = "data/prepared_rds/txdb_annotation_chr19_grangeslist.rds")
```

```{r prepare txdb, echo =T}
genes <- unique(genes(txdb, filter = list(tx_chrom = "chr19")))
transcripts<-  unique(transcripts(txdb, filter = list(tx_chrom = "chr19")))
promoters_def <- unique(promoters(txdb, filter = list(tx_chrom = "chr19"))) # upstream=2000, downstream=200
promoters_3k <- unique(promoters(txdb, filter = list(tx_chrom = "chr19"), upstream = 3000, downstream = 3000)) 

# save above GRanges objects in RData file
save(genes, transcripts, promoters_def, promoters_3k, file =  "data/prepared_rds/txdb_annotation_chr19_genes_tx_pr_tss_granges.RData")

# map ENTREZ gene id to gene SYMBOL
entrezid_symbols <- AnnotationDbi::select(org.Hs.eg.db, keys = genes$gene_id, columns = c("ENTREZID", "SYMBOL"), keytype = "ENTREZID")
# save the final GRangesList object
saveRDS(entrezid_symbols, file = "data/prepared_rds/gene_entrezids_genesymbols_df.rds")

```

```{r encode cres}
library(plyr)
# Supplementary Table 10 from 'The ENCODE consortium, 2020, nature'
# https://www.nature.com/articles/s41586-020-2493-4#MOESM12
# table has same format as .bed file so probably 0-based
# r works 1-based, therefore have to set 'start.in.df.are.0based = T' 
encode <- makeGRangesFromDataFrame(read.delim("data/encode/encode_nature2020_supldata10_chr19.txt", stringsAsFactors = F, header = T), keep.extra.columns = T, starts.in.df.are.0based=T)
names(encode) <- encode$cCRE_accession

encode_ucsc <- rtracklayer::import("data/encode/encode_cCREs_ucscgenomebrowser_chr19.bed")
names(encode_ucsc) <- encode_ucsc$name

# add itemRBG from ucsc file to encode publication file
encode$ucsc_rgb <- encode_ucsc[match(encode$cCRE_accession, encode_ucsc$name),]$itemRgb

# check that each name as the right itemRGB
encode_labels <- arrange(as.data.frame(mcols(encode[, c("cCRE_accession", "ucsc_rgb")])), desc(cCRE_accession))
encodeucsc_labels <- arrange(as.data.frame(mcols(encode_ucsc[, c("name", "itemRgb")])), desc(name))
all(encode_labels$cCRE_accession == encodeucsc_labels$name)
all(encode_labels$ucsc_rgb == encodeucsc_labels$itemRgb)
head(encodeucsc_labels)
head(encode_labels)
all_labels <- merge(encode_labels, encodeucsc_labels, by.x = "cCRE_accession", by.y = "name", all.x = T, all.y = T)
identical(all_labels$ucsc_rgb,all_labels$itemRgb)
rm(all_labels, encode_labels, encodeucsc_labels)

# translate itemRGB into a categorie 
# ref: http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg38&g=encodeCcreCombined
itemRGBs <- data.frame(names(table(encode_ucsc$itemRgb)))
colnames(itemRGBs) <- "hex"
itemRGBs$hex <- as.character(itemRGBs$hex)
itemRGBs$color <- c("blue", "red", "orange", "pink", "yellow")
itemRGBs$ucsc_labels <- c("CTCF", "prom", "enhP", "K4m3", "enhD")
#itemRGBs

# add categorie from ucsc website to encode publication file
# based on reported itemRGB
encode$ucsc_label <- as.factor(encode$ucsc_rgb)
encode$ucsc_label <- as.character(mapvalues(encode$ucsc_label, from = c("#00B0F0", "#FF0000", "#FFA700", "#FFAAAA", "#FFCD00"), to = c("CTCF", "prom", "enhP", "K4m3", "enhD")))
encode
table(encode$group, encode$ucsc_label)

# make simplfied object for tutorial
encode_chr19 <- encode[, c("cCRE_accession", "ucsc_label")]
show(encode_chr19)
# save this object
saveRDS(encode_chr19, file = "data/prepared_rds/encode_ccres_chr19.rds")
```

