<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Script</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Functional Genomics 2020</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="DetailedWorkflow.html">Script</a>
</li>
<li>
  <a href="Week1.html">Week 1 - UCSC Genome Browser</a>
</li>
<li>
  <a href="Week2.html">Week 2 - Bioconductor</a>
</li>
<li>
  <a href="intro_bioconductor.html">Datacamp Bioconductor</a>
</li>
<li>
  <a href="Links.html">Links</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Script</h1>

</div>


<style type="text/css">
.title {
  display: none;
}

#getting-started img {
  margin-right: 10px;
}

</style>
<div class="row" style="padding-top: 30px;">



<div id="script" class="section level1">
<h1>Script</h1>
<div id="week-1" class="section level2">
<h2>Week 1</h2>
<div id="chip-seq-recapdiscussion" class="section level3">
<h3>1.1 ChIP-seq recap/discussion</h3>
<p><strong>Discuss</strong>:</p>
<ul>
<li>What is ChIP-seq? Presentation with ChIP-seq steps?<br />
</li>
<li>Sequencing samples. What happens in sequencer? What is the output?<br />
<br></li>
</ul>
</div>
<div id="from-data-outputted-by-the-sequencer-to-science" class="section level3">
<h3>1.2 From data outputted by the sequencer to science</h3>
<p><a href="https://vanheeringen-lab.github.io/seq2science/index.html">Seq2science</a><br />
<em>fastq file, bam processing, bigwig, peak calling, count table</em></p>
<p><strong>Discuss</strong>:</p>
<ul>
<li>Modify Siebrens pptx.<br />
</li>
<li>Fastq files, what does the data look like?<br />
</li>
<li>4 lines of code with:
<ol style="list-style-type: decimal">
<li>unique_read_name (starts with @-sign)<br />
</li>
<li>nucleotide sequence<br />
</li>
<li><code>+</code> (previously, this line showed the same as line 1)<br />
</li>
<li>nucleotide quality score (Phred encoded)</li>
</ol></li>
<li>The neccessary processing steps:
<ol style="list-style-type: lower-alpha">
<li>Read quality and adapter trimming<br />
</li>
<li>Aligning fastq reads to reference genome<br />
</li>
<li>BAM sieving.<br />
</li>
</ol>
<ul>
<li>What considerations are important? PCR duplicates, multi mapping reaads<br />
</li>
<li>Discuss use of input sample<br />
</li>
</ul>
<ol start="4" style="list-style-type: lower-alpha">
<li>Peak calling. Use of control samples. Narrow and Broad peaks?<br />
</li>
<li>Count table: What does it look?<br />
<br></li>
</ol></li>
</ul>
</div>
<div id="look-at-the-read-density-yourself-in-a-genome-browser" class="section level3">
<h3>1.3 Look at the read density yourself in a genome browser</h3>
<p><em>Functional elements, ChIP signals, peaks, hypotheses about enrichment.</em></p>
<p><strong>Discuss</strong>:</p>
<ul>
<li>UCSC Genome Browser
<ul>
<li>how to switch off tracks. Hide all tracks except for <strong>GENCODE v32</strong> and <strong>Base Position</strong>.<br />
</li>
<li>how to browse trough the data<br />
</li>
<li>how to upload custom data. Upload the <span style="color: red;">processed chip-seq data of histone marks in <strong>heatlhy cells</strong>, input and a <strong>failed track</strong> as .bigwig file. Or use trackhub?</span>
<ul>
<li>bigwig is a special data format to display dense, continuous data as graphs. The main advantage of this format is that only the portions of the files needed to dispaly a particular regoin are transferred making it considerably faster in a genome browsing tool <a href="https://genome.ucsc.edu/goldenPath/help/bigWig.html">ref</a>.<br />
</li>
</ul></li>
<li>As functional genomic elements: <span style="color: red;">upload custom track or use <strong>ENCODE cCREs</strong> track?</span></li>
</ul></li>
</ul>
<p>Now browse through the data.</p>
<p>Peaks are …</p>
<p><strong>Assignments/Questions</strong>:</p>
<ol style="list-style-type: decimal">
<li>One of the tracks represents the input, which one and why?<br />
</li>
<li>What kind of data is visualized in track “A”?<br />
</li>
<li>Browse to these four regions, which one of them would you consider to display a peak (tip, make screenshot and paste in a temporary pptx. to compare the images)<br />
</li>
<li>Of the marks visualized which ones are associated with gene expression and which ones with repression (from lecture)?</li>
</ol>
<p><strong>Discuss</strong>:<br />
Peak calling software is used to identify peaks within a certain range of confidence. <span style="color: red;">and using replictes?</span></p>
<p><strong>Assignments/Questions</strong>:</p>
<ol start="5" style="list-style-type: decimal">
<li>Which of the following influences the number of peaks identified:<br />
</li>
</ol>
<ul>
<li>P-value threshold, the lower the fewer peaks<br />
</li>
<li>read depth<br />
</li>
<li>Q-value/false-discovery rate, the<br />
</li>
<li>the type of control sample …<br />
</li>
<li></li>
</ul>
<p><strong>Discuss</strong>:</p>
<p>Upload peaks called by the software.</p>
<p><strong>Assignments/Questions</strong>:</p>
<ol start="6" style="list-style-type: decimal">
<li>What are different functional genomic elements you have heard of during lectures?</li>
</ol>
<p>The custom track contains five tracks, numbered 1-5, representing different genomic elements.</p>
<ol start="7" style="list-style-type: decimal">
<li>Which # represent what kind of element?<br />
</li>
<li>Browse through the browser, estimate for each histone mark whether it is enriched in one or several of the genomic elements (1-5)</li>
</ol>
</div>
</div>
<div id="week-2" class="section level2">
<h2>Week 2</h2>
<div id="introduction-to-bioconductor" class="section level3">
<h3>2.1 Introduction to Bioconductor</h3>
<p><strong>Bioconductor Packages:</strong></p>
<ul>
<li><a href="https://www.bioconductor.org/">Bioconductor</a> provides tools for analysis of high throughput genomic data in R</li>
<li>Has its own repository, own way to install packages, each release designed with specific version of R<br />
</li>
<li>Install using BiocManager, the special installer for Bioconductor packages. The advantage over the ‘standard’ <code>install.pakages</code> command is that Bioconductor has different release cycles than regular R and BiocManager therefore recognizes the newest version better. Using <code>install.packages</code> you risk installing older versions (with more unfixed bugs and fewer features). In addition, BiocManager allows users to install packages from the ‘devel’ branch (under development), thus cutting-edge while <code>install.packages</code> points to the ‘release’ branch. Lastly, Bioconducter packages are heavily dependend on each other. It is therefor very important that they are from the same relase. Biocmanager checks that packages are from the same release.</li>
</ul>
<p><strong>Some important Bioconductor packages</strong></p>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Package</th>
<th>Purpose</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="http://bioconductor.org/packages/release/bioc/html/BSgenome.html">BSgenome</a></td>
<td>infrastructure and functions for related genome packages with full genome sequences for many species</td>
</tr>
<tr class="even">
<td><a href="https://www.bioconductor.org/packages/release/bioc/html/Biostrings.html">Biostrings</a></td>
<td>provides containters and functions for fast manipulation of large biological sequences (aka genome data)</td>
</tr>
<tr class="odd">
<td><a href="https://bioconductor.org/packages/release/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html">GenomicRanges</a></td>
<td>handles genomic intervals</td>
</tr>
<tr class="even">
<td><a href="https://www.bioconductor.org/packages/release/bioc/html/GenomicFeatures.html">GenomicFeatures</a></td>
<td>functions to retrieve and manage genomic features from public databases</td>
</tr>
<tr class="odd">
<td><a href="https://bioconductor.org/packages/release/bioc/html/GenomicAlignments.html">GenomicAlignments</a></td>
<td>containers to store short sequence alignments for read counting, computing the coverage, junction detection, and working with the nucleotide content of the alignments</td>
</tr>
</tbody>
</table>
<pre class="r"><code>library(tidyverse) # compilation of packages for usein data science. These packages are developed such that they nicely work together for data import, tidying, manipulation, visualization and programming  
library(GenomicRanges) # functions and data structures (containters) to handle genomic intervals which are datasets with notation chr-start-end. 
library(GenomicFeatures) # for functions to retrieve and manage genomic features (eg genes, exons, transcripts, promoters) from public databases  
library(GenomicAlignments) # containers to store short sequence alignments  </code></pre>
</div>
<div id="genomicranges" class="section level3">
<h3>2.2 GenomicRanges</h3>
<p><strong>Discuss</strong>:</p>
<ul>
<li>how to import them and how to make them from scratch<br />
</li>
<li>convert dataframe to GRanges object with <code>as()</code>
<ul>
<li>dataframe should have <code>start</code> <code>end</code> and <code>chr</code><br />
</li>
<li><code>gr &lt;- as(dataframe, &quot;GRanges&quot;)</code><br />
</li>
</ul></li>
<li>minimally requires chr, start and end column from dataframe in input<br />
</li>
<li>also has metadata, <code>seqnames</code> and <code>seqinfo</code><br />
</li>
<li>see ist of available accessors use: <code>methods(class = &quot;GRanges&quot;)</code>
<ul>
<li><code>seqnames(gr)</code> - chromosome names<br />
</li>
<li><code>ranges(gr)</code> - returns IRanges object<br />
</li>
<li><code>mcols(gr)</code> - display metadata per range<br />
</li>
<li><code>seqinfo(gr)</code> - to store sequence information<br />
</li>
<li><code>genome(gr)</code>- genome name<br />
</li>
<li>both are <strong>setter</strong> and <strong>getter</strong> functions<br />
</li>
<li>can be inherited by other S4 objects</li>
</ul></li>
</ul>
</div>
<div id="grangeslist" class="section level3">
<h3>2.3 GRangesList</h3>
<ul>
<li>container of collection of GRanges<br />
</li>
<li>construct with:
<ul>
<li><code>as(mylist, &quot;GRangesList&quot;)</code><br />
</li>
<li><code>GRangesList(myGranges1, myGranges2, ...)</code></li>
</ul></li>
<li>convert back with <code>unlist()</code><br />
</li>
<li>accessers found by <code>methods(class = &quot;GRangesList&quot;)</code><br />
</li>
<li>handy for eg. transcripts bij gene, exons by genes, read alignments, sliding windows
<ul>
<li><code>slidingWindows()</code> - makes sliding windows<br />
</li>
</ul></li>
<li><code>exonsBy()</code> function returns a GRangesList <br> Overlaps<br />
</li>
<li><code>countOverlaps(query, subject)</code><br />
</li>
<li><code>findOverlaps(query, subject)</code><br />
</li>
<li><code>subsetByOverlap(query, subject)</code><br />
</li>
<li><code>overlapsAny()</code><br />
</li>
<li>query and subject can be GRanges and GRangesList</li>
</ul>
</div>
<div id="genomicfeatures" class="section level3">
<h3>2.4 GenomicFeatures</h3>
<ul>
<li>manipulate TxDb objects from UCSC or BioMart<br />
</li>
<li>Transcript Database Objects</li>
<li>Obtain TxDb object by:
<ul>
<li>use <code>loadDB</code> as in <code>loadDB(system.file(&quot;extdata&quot;, &quot;hg19_knownGene_sample.sqlite&quot;, package=&quot;GenomicFeatures&quot;))</code><br />
</li>
<li>load TxDB annotation package like: <code>library(TxDb.Hsapiens.UCSC.hg19.knownGene)</code><br />
<br> Functions to manipulate TxDb object:<br />
</li>
</ul></li>
<li><code>genes()</code> function extracts genomic features from TxDb-like object.
<ul>
<li>returns <code>GRanges</code> object<br />
</li>
</ul></li>
<li>other functions to extract features are <code>transcripts()</code>, <code>cds()</code>, <code>exons()</code>, and <code>promoters()</code>
<ul>
<li>input is TxDb object</li>
<li>define <code>filter =</code> to filter a condition on a column as list<br />
</li>
<li>define <code>columns =</code> to select column names as list</li>
</ul></li>
</ul>
</div>
<div id="custom-data" class="section level3">
<h3>2.5 Custom data</h3>
<ul>
<li>reading BAM files<br />
</li>
<li>reading peak files and regions .BED files</li>
</ul>
</div>
<div id="section" class="section level3">
<h3></h3>
</div>
</div>
<div id="week-3" class="section level2">
<h2>Week 3</h2>
<div id="introduction" class="section level3">
<h3>Introduction</h3>
<p>We will be looking at data from BLUEPRINT, a consortium of European universities and research institutes aimed to systematically study/uncover the epigenomic landscape of cells found in the blood.</p>
<p>‘Normal’ epigenome of a given cell. Producding reference epigenomes.</p>
<p>Physiology and pathology (health and disease, development, lineage specification, onset and progression of disease, ) are both epigenetically regulated.</p>
<p>We will look at the chromatin-based regulation of gene expression, thus the relationship between histone signatures (combination of histone marks) and gene expression</p>
<p>Epigenomic information, such as covalent histone modifications, DNA accessibility and DNA methylation</p>
<p><strong>BLUEPRINT</strong><br />
- reference genemes of cells found in the blood<br />
- healthy individuals and with blood-based diseases including autoimmune diseases, leukemias.</p>
<div id="use-of-thses-insights" class="section level4">
<h4>Use of thses insights:</h4>
<ul>
<li>epi-drugs<br />
</li>
<li>case stratification, improved diagnosis and progrnosis</li>
</ul>
</div>
</div>
<div id="dataset" class="section level3">
<h3>Dataset</h3>
<ol style="list-style-type: decimal">
<li>lymphocytic leukemia?</li>
</ol>
<ul>
<li>On the one hand, pathological tissues and cells exhibit epigenetic imprints of the developmental or differentiation stages from which they originate and, on the other hand, they acquire disease-specific epigenetic alterations</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>innate immune cells/innate immune memory?</li>
</ol>
<ul>
<li>comparing healthy with cancerous cells<br />
</li>
<li>environmental stimuli can determine a cell’s future response/behavior by altering its epigenetic state, and thus alter susceptibility to disease</li>
</ul>
</div>
<div id="work" class="section level3">
<h3>Work</h3>
<ul>
<li>looking at peaks<br />
</li>
<li>assessing enrichment<br />
</li>
<li>computing coverage</li>
</ul>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
