<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Week 2: Exploratory &amp; Overlap analysis</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


<link rel="stylesheet" href="custom.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">FG 2020</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="fg1.html">fg1</a>
</li>
<li>
  <a href="fg2.html">fg2</a>
</li>
<li>
  <a href="prepdata.html">Appendix</a>
</li>
<li>
  <a href="example1_sequence_short_fastqc.html">FASTQC_exmpl1</a>
</li>
<li>
  <a href="example2_sequence_short_fastqc.html">FASTQC_exmpl2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Week 2: Exploratory &amp; Overlap analysis</h1>

</div>


<div id="introduction" class="section level2">
<h2>2.1 Introduction</h2>
<p><br> In week 1 we discussed the major steps involved in obtaining genomics data, from experiment to raw data to normalized signal and peaks. You examined histone PTM ChIP-, DNaseI-, and RNA-seq data of monocytes in the UCSC genome browser and searched for regions with increased signal, <em>peaks</em>. You looked at the position of these peaks in the chromosome and with respect to genes as well as the co-occurrence of different marks.<br />
<br> These observations showed examples of marks associated with a particular functional elements (e.g. active enhancer, active promoter, silenced domains). Manual annotation of each peak region would be an overkill and dangerously subjective. <br> <strong>This week and next week you will perform computational analyses to annotate peaks and answer common questions including:</strong> “<em>How many peaks do I have?</em>”, “<em>Is this mark statistically enriched at promoters?</em>”, “<em>Which and how often do marks co-occur?</em>”,“<em>The expression of which gene could be affected by this mark?</em>”, and “<em>What is the signal of the mark around a particular element of interest like the TSS?</em>”<br />
<br></p>
<div id="learning-objectives" class="section level3">
<h3>2.1.1 Learning Objectives</h3>
<blockquote>
<p>At the end of week 2 you are able to:</p>
<ol style="list-style-type: decimal">
<li>Import ChIP-seq peaks into a GRanges object in r.<br />
</li>
<li>Perform exploratory data analysis on GRanges objects with ChIP-seq peaks using plyranges and ggplot.</li>
<li>Detect and count overlap between two GRanges objects.<br />
</li>
<li>Plot this overlap in a venndiagram.<br />
</li>
<li>Statistically test for enrichment of histone marks in a particular genomic region.<br />
</li>
<li>Visually compare gene expression of genes with different histone marks at their promoters.<br />
</li>
<li>Summarize the genome-wide overlap among multiple marks in a upset plot.<br />
</li>
</ol>
<ul>
<li>These refer to global learning objectives #4-#7.</li>
</ul>
</blockquote>
</div>
<div id="exercises" class="section level3">
<h3>Exercises</h3>
<p>In this tutorial you will have multiple-choice and checkbox questions like in <strong>fg1</strong> but also complete-the-code exercises. The latter can be evaluated on the computed outcome (e.g a computed p-value), in other cases, the code itself is checked. <br> For short lines of code, you have to decide which function or object to use and how to write this in code. For some longer lines of codes, we left blanks as "___" that indicate where in the code you should add a function or object name.<br />
<br></p>
<ul>
<li>Hit <code>Start Over</code> to remove any adjustments you made and start again.<br />
</li>
<li>Hit <code>Hint</code> (when provided) for clues.<br />
</li>
<li>Hit <code>Run Code</code> to preview your code and the output.<br />
</li>
<li>Hit <code>Submit Answer</code> (if present) to submit the code. As ‘submitting’ not always shows the code output, it is advised to use <code>Run Code</code> first.</li>
</ul>
<p>Although we ask for specific answers or completion of specific code, you are encouraged to test your own code and use <code>Run Code</code> and <code>Start Over</code> extensively. I.e., remove the pre-coded code and write your own code for a different graph or different summary of the data. Use <code>Run Code</code> to preview the results. Unfortunately these self-thought of codes can not be evaluated. For the evaluation you need to use our pre-coded code. Hit <code>Start Over</code> to get this pre-coded code.</p>
<p><br></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quiz</span>(<span class="at">caption =</span> <span class="st">&quot;Example questions:&quot;</span>,</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>     <span class="fu">question</span>(<span class="st">&quot;What number is the letter A in the English alphabet?&quot;</span>,</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>        <span class="fu">answer</span>(<span class="st">&quot;8&quot;</span>),</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>        <span class="fu">answer</span>(<span class="st">&quot;14&quot;</span>),</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>        <span class="fu">answer</span>(<span class="st">&quot;1&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>        <span class="fu">answer</span>(<span class="st">&quot;23&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>     ),</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>     <span class="fu">question</span>(<span class="st">&quot;Where are you right now? (select ALL that apply)&quot;</span>,</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>          <span class="fu">answer</span>(<span class="st">&quot;Planet Earth&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>          <span class="fu">answer</span>(<span class="st">&quot;Pluto&quot;</span>),</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>          <span class="fu">answer</span>(<span class="st">&quot;At a computing device&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>          <span class="fu">answer</span>(<span class="st">&quot;In the Milky Way&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>          <span class="at">incorrect =</span> <span class="st">&quot;Incorrect. You&#39;re on Earth, in the Milky Way, at a computer.&quot;</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><strong>Example coding exercise:</strong></p>
<blockquote>
<p><strong>Example exercise</strong>: Print the head of object <code>monocytes_h3k4me3</code>.</p>
</blockquote>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># print the head of `monocytes_h3k4me3`</span></span></code></pre></div>
<div id="examplecode-hint">
<p><strong>Hint:</strong> You may want to use the <code>head(...)</code> function.</p>
</div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(monocytes_h3k4me3)</span></code></pre></div>
<pre><code>## GRanges object with 6 ranges and 6 metadata columns:
##       seqnames        ranges strand |                            name     score
##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; |                     &lt;character&gt; &lt;numeric&gt;
##   [1]    chr19 245884-246269      * | 3337.macs2_peak_call_peak_22416       317
##   [2]    chr19 266835-267653      * | 3337.macs2_peak_call_peak_22417      1805
##   [3]    chr19 291204-291424      * | 3337.macs2_peak_call_peak_22418        55
##   [4]    chr19 295007-295206      * | 3337.macs2_peak_call_peak_22419        55
##   [5]    chr19 341785-345382      * | 3337.macs2_peak_call_peak_22420      2297
##   [6]    chr19 408554-409469      * | 3337.macs2_peak_call_peak_22421       105
##       signalValue    pValue    qValue      peak
##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;
##   [1]    14.05594    33.902  31.78678       190
##   [2]    50.02259  183.1329 180.54498       544
##   [3]     4.96092   7.42337   5.53701       137
##   [4]     4.96092   7.42337   5.53701       126
##   [5]    60.35784 232.49678 229.72572      2533
##   [6]     7.02797  12.54858  10.58345       762
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>
</div>
<div id="chip-seq-peaks-in-r" class="section level2">
<h2>2.2 ChIP-seq peaks in R</h2>
<div id="data-formats" class="section level3">
<h3>2.2.1 Data formats</h3>
<p>In genomics you often work with <strong>interval</strong> data. Think of peaks, genes, exons, … any genomic region reported with the <strong>genomic coordinates: chr, start, and end</strong>. As we discussed in week 1, these often come in a tabular format with the basic information about the location and some other information.<br />
<br> Common file formats include BED format for peak files and GFF for gene annotations. In practice, peak files come in two flavors, both cohering to the BED format but with small differences in columns 6-7. These are called <strong>narrowPeak</strong> and <strong>broadPeak</strong> files. This is because different data types have different peak shapes and peak-calling takes that into account. If you have narrowPeak files, this means that during peak calling, settings were used that fit ChIP-seq datasets with sharp, narrow enrichment signals. For broadPeak files, peak calling settings were used to detect broad domains of (overall lower) enrichment. We will come back to this difference shortly.</p>
<blockquote>
<h4 id="background-info-file-formats"><strong>Background info</strong>: File formats</h4>
<details>
<summary>
What do BED and GFF files hold and look like?
</summary>
<p><strong>Browser Extensible Data - BED format</strong>:</p>
<ul>
<li>Used for peaks, motif locations or other custom intervals.<br />
</li>
<li>Has 3 required columns: chromosome, start and end position.<br />
</li>
<li>6 or 7 optional columns, in case of peaks:<br />
4th: peak name<br />
5th: peak score (-10log(q-value) * 10, rounded down to integer value))<br />
6th: strand to denote orientation (if applicable, otherwise “*” or “.” if unstranded)<br />
7th: signalValue<br />
8th: p-value to denote statistical significance, given as -log10(p-value)<br />
9th: q-value statistical significance using false discoveray rate, given as -log10(q-value)<br />
10th (only for sharp peaks not broad domains): location of peak summit relative to the “start” coordinate.</li>
</ul>
<p>Example:<br />
<img src="images/wk1_3_file_format_bed.png" style="width:80.0%" /><br />
Further reading on the file formats on the UCSC Genome Browser FAQ page: <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED</a>, and the <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format12">narrowPeak BED</a> and <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format13">broadPeak BED</a>.<br />
<br> <strong>General Feature Format - GFF format</strong>:</p>
<ul>
<li>Common file format for storing gene annotations not only including genes but also transcripts/splice variants, cDNA sequences, exons, rRNA, ncRNA, etc.<br />
</li>
<li>Begins with meta-data in headerlines, starting with #<br />
</li>
<li>Records reported in 9 fixed columns<br />
</li>
<li>Column 9 can contains various attirbutes (eg gene symbol, the transcript to which the exon belongs).<br />
</li>
<li>Downloaded for example from <a href="https://www.ensembl.org/Homo_sapiens/Info/Index">Ensembl</a>.</li>
</ul>
<p>Example:<br />
<img src="images/wk1_3_file_format_gff.png" style="width:80.0%" /><br />
Further reading on the gff file formats: <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format3">UCSC Genome Browser</a>, or <a href="https://m.ensembl.org/info/website/upload/gff3.html">ENSEMBL</a>.<br />
<br></p>
</blockquote>
</div>
<div id="peaks-as-data.frame" class="section level3">
<h3>2.2.2 Peaks as data.frame</h3>
<p><br> Peak data can be imported into a data.frame with:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>object_name <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;location/of/peak_file.bed&quot;</span>)</span></code></pre></div>
<p><br> The data.frame data structure is, however, not the most efficient way to work with interval data. E.g. a simple manipulation such as shifting all reported intervals 2 bp to the right, requires you to manipulate the “start” and “end” columns at the same time.<br />
<br> Interval data such as genomic peaks can be more efficiently handled with the IRanges package which works with a data structure especially developed for <strong>ranges of integers</strong>: IRanges objects.</p>
</div>
<div id="introducing-iranges-and-granges" class="section level3">
<h3>2.2.3 Introducing IRanges and GRanges</h3>
<p>To construct an IRanges object you need to define at least two of the following three values:</p>
<ol style="list-style-type: decimal">
<li>a starting coordinate<br />
</li>
<li>a finishing coordinate<br />
</li>
<li>the width of the interval.</li>
</ol>
<p><br> The GRanges objects of the GenomicRanges package are very similar but require a additional <strong>sequence name</strong> (in other words a chromosome) for every interval and an optional <strong>strand</strong> column.<br />
<br> Here we create GRanges object called <code>gr</code> with 3 interals on chr1:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create `gr`</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>gr_chr1 <span class="ot">&lt;-</span>  <span class="fu">GRanges</span>(<span class="at">seqnames =</span> <span class="fu">c</span>(<span class="st">&quot;chr1&quot;</span>), <span class="co"># argument is recycled  </span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">ranges =</span> <span class="fu">IRanges</span>(<span class="at">start=</span><span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">10</span>,<span class="dv">25</span>), <span class="at">width =</span> <span class="fu">c</span>(<span class="dv">13</span>,<span class="dv">17</span>,<span class="dv">6</span>)),</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">strand =</span> <span class="fu">c</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;+&quot;</span>, <span class="st">&quot;+&quot;</span>),</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">color =</span> <span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>))</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co"># add names to `gr_chr1`</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gr_chr1) <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;interval&quot;</span>, LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co"># print the object:</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>gr_chr1</span></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 1 metadata column:
##              seqnames    ranges strand |       color
##                 &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
##   interval_A     chr1      5-17      - |       black
##   interval_B     chr1     10-26      + |         red
##   interval_C     chr1     25-30      + |       green
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<p><br> When we visualize these intervals we see three ‘blocks’ along the horizontal axis: <img src="fg2_files/figure-html/plot%20gr-1.png" width="672" /></p>
</div>
<div id="handy-functions-for-granges" class="section level3">
<h3>2.2.4 Handy functions for GRanges</h3>
<p>There are several functions to retrieve and set the values in GRanges objects. Here is an image of a GRanges object called <code>gr</code> with the respective functions. You will use these functions soon.</p>
<p><img src="images/wk2_2_GRanges_fields_my_gr.png" style="width:100.0%" /><br />
<br> A list of the above functions (and more) for GRanges can also be called with <code>methods(class = "...")</code>. This also command works for other objects as well. You just need to figure out the class of the object in question using <code>class([object_name])</code>.<br />
<br> You can print a summary of the GRanges object with <code>show(object_name)</code> and <code>print(object_name)</code> (this works for many other R objects as well).<br />
<br></p>
</div>
<div id="slots-in-granges-objects" class="section level3">
<h3>2.2.5 Slots in GRanges objects</h3>
<p>Notice that <code>gr</code> holds 3 slots:</p>
<ol style="list-style-type: decimal">
<li>the genomic coordinates in a <strong>GRanges</strong> slot<br />
</li>
<li>2 columns as <strong>metadata</strong> (metadata is data about the data)<br />
</li>
<li>a genome in the <strong>seqinfo</strong> slot</li>
</ol>
<p>The main differences between the metadata and the genomic coordinates are:</p>
<table>
<colgroup>
<col width="42%" />
<col width="57%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Genomic coordinates</th>
<th align="left">Metadata columns</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Printed on the left-hand side of the |-sign</td>
<td align="left">Printed on the right-hand side of the |-sign</td>
</tr>
<tr class="even">
<td align="left">Extract using <code>granges(object_name)</code></td>
<td align="left">Extract as DataFrame with <code>mcols(object_name)</code> or <code>object_name$column_name</code> for a specific column</td>
</tr>
<tr class="odd">
<td align="left">Restricted to variables <code>seqnames</code>, <code>ranges</code> and <code>strand</code></td>
<td align="left">Almost anything can be stored in the metadata</td>
</tr>
</tbody>
</table>
<p><br> Information about the genome is stored in the <strong>seqinfo</strong> part of the object. In the example you see that the peaks are located in human genome 38 which holds 455 sequences in total of which 1 circular (mitochondrial genome). (Sequences that can not be confidently placed on a specific chromosome is stored in chrUn, chr<em>N</em>_random and chr<em>N</em>_region <a href="http://genome.ucsc.edu/FAQ/FAQdownloads#download10">explanation on UCSC GB website</a>.)</p>
</div>
<div id="importing-peak-files-as-granges" class="section level3">
<h3>2.2.6 Importing peak files as GRanges</h3>
<p>To parse peak files directly into GRanges you use the <code>import()</code> function from the rtracklayer package, for example:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load the required library, is usually done once at the start of your analysis.  </span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rtracklayer)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># import monocyte H3K4me3 peak locations</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>monocytes_h3k4me3 <span class="ot">&lt;-</span> <span class="fu">import</span>(<span class="st">&quot;C000S5H2.ERX547984.H3K4me3.bwa.GRCh38.20150527.chr19.bed&quot;</span>, <span class="at">format =</span> <span class="st">&quot;narrowPeak&quot;</span>)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co"># (you could also use rtracklayer::import(&quot;C000S5H2.ERX547984.H3K4me3.bwa.GRCh38.20150527.chr19.bed&quot;, format = &quot;narrowPeak&quot;) and skip loading the library)</span></span></code></pre></div>
<p><br> As mentioned above (2.2.1), different data types have different peak shapes resulting in “narrowPeak” and “broadPeak” files. H3K4me3 ChIP-seq generally shows sharp(er) enrichment that can reach high enrichment values. For <code>import()</code> to function properly, we defined this format in the <code>format =</code> parameter.<br />
<br> H3K4me1 ChIP-seq cover broad(er) domains and peak calling took that into account, resulting in a “broadPeak” peak file.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Which function imports the H3K4me1 dataset into a GRanges object?&quot;</span>,</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;import(</span><span class="sc">\&quot;</span><span class="st">C000S5H2.ERX547981.H3K4me1.bwa.GRCh38.broad.20150527.chr19.bed</span><span class="sc">\&quot;</span><span class="st">, format = </span><span class="sc">\&quot;</span><span class="st">narrowPeak</span><span class="sc">\&quot;</span><span class="st">)&quot;</span>, <span class="at">message =</span> <span class="st">&quot;check the file format.&quot;</span>),</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;import(</span><span class="sc">\&quot;</span><span class="st">C000S5H2.ERX547981.H3K4me1.bwa.GRCh38.broad.20150527.chr19.bed</span><span class="sc">\&quot;</span><span class="st">, format = </span><span class="sc">\&quot;</span><span class="st">BED</span><span class="sc">\&quot;</span><span class="st">)&quot;</span>, <span class="at">message =</span> <span class="st">&quot;BED is not the right format.&quot;</span>),</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;import(</span><span class="sc">\&quot;</span><span class="st">C000S5H2.ERX547981.H3K4me1.bwa.GRCh38.broad.20150527.chr19.bed</span><span class="sc">\&quot;</span><span class="st">, format = </span><span class="sc">\&quot;</span><span class="st">broadPeak</span><span class="sc">\&quot;</span><span class="st">)&quot;</span>, <span class="at">correct =</span>T),</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;read.table(</span><span class="sc">\&quot;</span><span class="st">C000S5H2.ERX547981.H3K4me1.bwa.GRCh38.broad.20150527.chr19.bed</span><span class="sc">\&quot;</span><span class="st">, format = </span><span class="sc">\&quot;</span><span class="st">broadPeak</span><span class="sc">\&quot;</span><span class="st">)&quot;</span>, <span class="at">message =</span> <span class="st">&quot;read.table parses the data into a data.frame while we asked for a GRanges object.&quot;</span>),</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">allow_retry =</span> <span class="cn">TRUE</span>,</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">random_answer_order =</span> <span class="cn">TRUE</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>         )</span></code></pre></div>
<p><br> Let’s take a quick look at <code>monocytes_h3k4me3</code>.</p>
</div>
<div id="exercise-1" class="section level3">
<h3>Exercise 1</h3>
<blockquote>
<p><strong>Exercise 1a</strong>: Print the head of <code>monocytes_h3k4me3</code>.</p>
</blockquote>
<div id="q1_head_h3k4me3-hint">
<p><strong>Hint:</strong> Use the <code>head(...)</code> function.</p>
</div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(monocytes_h3k4me3)</span></code></pre></div>
<pre><code>## GRanges object with 6 ranges and 6 metadata columns:
##       seqnames        ranges strand |                            name     score
##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; |                     &lt;character&gt; &lt;numeric&gt;
##   [1]    chr19 245884-246269      * | 3337.macs2_peak_call_peak_22416       317
##   [2]    chr19 266835-267653      * | 3337.macs2_peak_call_peak_22417      1805
##   [3]    chr19 291204-291424      * | 3337.macs2_peak_call_peak_22418        55
##   [4]    chr19 295007-295206      * | 3337.macs2_peak_call_peak_22419        55
##   [5]    chr19 341785-345382      * | 3337.macs2_peak_call_peak_22420      2297
##   [6]    chr19 408554-409469      * | 3337.macs2_peak_call_peak_22421       105
##       signalValue    pValue    qValue      peak
##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;
##   [1]    14.05594    33.902  31.78678       190
##   [2]    50.02259  183.1329 180.54498       544
##   [3]     4.96092   7.42337   5.53701       137
##   [4]     4.96092   7.42337   5.53701       126
##   [5]    60.35784 232.49678 229.72572      2533
##   [6]     7.02797  12.54858  10.58345       762
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_result</span>(</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a> <span class="fu">pass_if</span>(<span class="sc">~</span><span class="fu">identical</span>(.result, <span class="fu">head</span>(monocytes_h3k4me3)))</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a> )</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quiz</span>(<span class="at">caption =</span> <span class="st">&quot;&quot;</span>, </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">question</span>(<span class="st">&quot;How many metadata columns does the &#39;monocytes_h3k4me3&#39; object have?&quot;</span>,</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;2&quot;</span>),</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;3&quot;</span>),</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;6&quot;</span>, <span class="at">correct =</span> T, <span class="at">message =</span> <span class="st">&quot;The original .BED files lacked column headers. The `rtracklayer::import()` function appendended variable names and converted the data to the data type (eg numeric, integer, character etc.) according to the file format we defined.&quot;</span>),   </span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;9&quot;</span>), </span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">allow_retry =</span> <span class="cn">TRUE</span>),</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">question</span>(<span class="st">&quot;How many different chromosomes are reported in this file? *Hint*: look in seqinfo part of the object.&quot;</span>,</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;1&quot;</span>, <span class="at">correct =</span> T, <span class="at">message =</span> <span class="st">&quot;We have restricted all peak files to chromosome 19 to limit their file size. That is why &#39;seqinfo&#39; holds only one sequence&quot;</span>),</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;24&quot;</span>),</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;455&quot;</span>),</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">allow_retry =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="briefly-summarizing" class="section level3">
<h3>Briefly summarizing</h3>
<p>You have the following information in your object <code>monocytes_h3k4me3</code>:</p>
<table>
<colgroup>
<col width="14%" />
<col width="21%" />
<col width="21%" />
<col width="42%" />
</colgroup>
<thead>
<tr class="header">
<th>Column</th>
<th>Section in GRanges object</th>
<th>Variable</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1.</td>
<td>GRanges</td>
<td>seqnames</td>
<td>chromosome name of peak location.</td>
</tr>
<tr class="even">
<td>2.</td>
<td>GRanges</td>
<td>ranges</td>
<td>start and enc position of peak.</td>
</tr>
<tr class="odd">
<td>3.</td>
<td>GRanges</td>
<td>strand</td>
<td>if applicable, orientation of the peak, otherwise *.</td>
</tr>
<tr class="even">
<td>4.</td>
<td>Metadata</td>
<td>name</td>
<td>peak name, given by the peak caller.</td>
</tr>
<tr class="odd">
<td>5.</td>
<td>Metadata</td>
<td>score</td>
<td>-10log(qvalue) * 10, rounded down to integer value.</td>
</tr>
<tr class="even">
<td>6.</td>
<td>Metadata</td>
<td>signalValue</td>
<td>enrichment signal at the peak summit.</td>
</tr>
<tr class="odd">
<td>7.</td>
<td>Metadata</td>
<td>pValue</td>
<td>-log10(p-value), (<em>e.g.if p-value = 1e-10, this value is 10</em>), roughly put, the significance of the enrichment.</td>
</tr>
<tr class="even">
<td>8.</td>
<td>Metadata</td>
<td>qValue</td>
<td>-log10(qvalue), the false discovery rate determined by swapping test and control.</td>
</tr>
<tr class="odd">
<td>9.</td>
<td>Metadata</td>
<td>peak</td>
<td>location of peak summit relative to the peak start.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="exploratory-data-analysis" class="section level2">
<h2>2.3 Exploratory data analysis</h2>
<div id="peak-counts" class="section level3">
<h3>2.3.1 Peak counts</h3>
<p>A common first step in genomics data analysis is to make look at some basic characteristics of our data by making exploratory summaries and plots.</p>
</div>
<div id="exercise-2a" class="section level3">
<h3>Exercise 2a</h3>
<blockquote>
<p><strong>Exercise 2a</strong>: Determine the number of H3K4me3 and H3K4me1 peaks in monocytes.</p>
<ul>
<li>The peak data are stored in the R objects <code>monocytes_h3k4me3</code> and <code>monocytes_h3k4me1</code>.</li>
</ul>
</blockquote>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># How many peaks are stored in monocytes_h3k4me3?</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>monocytes_h3k4me3</span></code></pre></div>
<div id="q2a_exploreh3k4me3-hint">
<p><strong>Hint:</strong> You may want to use the <code>length()</code> function.</p>
</div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(monocytes_h3k4me3)</span></code></pre></div>
<pre><code>## [1] 2838</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># And how many H3K4me1 peaks do you have?</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(monocytes_h3k4me1)</span></code></pre></div>
<pre><code>## [1] 3282</code></pre>
</div>
<div id="exercise-2b" class="section level3">
<h3>Exercise 2b</h3>
<blockquote>
<p><strong>Exercise 2b</strong>: And what is the distribution of peak sizes for these marks?</p>
</blockquote>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># What is the distribution of H3K4me3 peak sizes?   </span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">___</span>(monocytes_h3k4me3))</span></code></pre></div>
<div id="q2b_widthh3k4me3-hint">
<p><strong>Hint:</strong> You can retrieve peak widths with the <code>width()</code> function.</p>
</div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">width</span>(monocytes_h3k4me3))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   200.0   389.2   879.5  1198.6  1661.2 12705.0</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># And what is the distribution of H3K4me1 peak sizes?   </span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>monocytes_h3k4me1</span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">width</span>(monocytes_h3k4me1))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##     190     775    1662    2642    3046   32063</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quiz</span>(<span class="at">caption =</span> <span class="st">&quot;&quot;</span>, </span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">question</span>(<span class="st">&quot;What are the median peak sizes for H3K4me3 and H3K4me1 ChIP-seq peaks in the current dataset? (in monocytes, restricted to chromosome19)&quot;</span>,</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;median H3K4me3 peak = 879.5 bp&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;median H3K4me3 peak = 1662 bp&quot;</span>),</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;median H3K4me3 peak = 1198.6 bp&quot;</span>),   </span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;median H3K4me3 peak = 2642 bp&quot;</span>),</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;median H3K4me1 peak = 879.5 bp&quot;</span>),</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;median H3K4me1 peak = 1662 bp&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;median H3K4me1 peak = 1198.6 bp&quot;</span>),   </span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;median H3K4me1 peak = 2642 bp&quot;</span>), </span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">allow_retry =</span> <span class="cn">TRUE</span>),</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">question</span>(<span class="st">&quot;And what are the widths of the *largest* peaks in the current H3K4me3 and H3K27me1 ChIP-seq datasets of monocytes?&quot;</span>,</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;largest H3K4me3 peak = 8237 bp&quot;</span>, <span class="at">message =</span> <span class="st">&quot;8237 bp is the width of the largest H3K27ac peak on chr19&quot;</span> ),</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;largest H3K4me3 peak = 12705 bp&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;largest H3K4me3 peak = 73112 bp&quot;</span>, <span class="at">message =</span> <span class="st">&quot;73112 bp is the width of the largest H3K36me3 peak on chr19&quot;</span> ),   </span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;largest H3K4me1 peak = 32063 bp&quot;</span>, <span class="at">correct =</span> T ),</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;largest H3K4me1 peak = 86201 bp&quot;</span>, <span class="at">message =</span> <span class="st">&quot;86201 bp is the width of the largest H3K27me3 peak on chr19&quot;</span>),</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;largest H3K4me1 peak = 136766 bp&quot;</span>, <span class="at">message =</span> <span class="st">&quot;136766 bp is the width of the largest H3K9me3 peak on chr19&quot;</span>),</span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>             <span class="at">allow_retry =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
<p><br> Notice the larger median and maximum peak size for H3K4me1 peaks compared to H3K4me3 peaks. This is related to these datasets having relative “broad domain” and “narrow peaks” of enrichment, respectively.</p>
</div>
<div id="plotting-summaries-with-ggplot2" class="section level3">
<h3>2.3.2 Plotting summaries with ggplot2</h3>
<p>You will use the ggplot2 package to plot the number of peaks for each dataset, to compare the average peak sizes and to plot the total number or fraction of bp’s covered by these peaks on chromosome 19.<br />
<br></p>
<blockquote>
<h4 id="a-very-brief-refresher-on-ggplot2-ting">A very brief refresher on ggplot2-ting:</h4>
<p>With <strong>ggplot2</strong> you generally begin with the <code>ggplot(data = ..)</code> function and you add so-called layers with <code>geom..()</code> functions. Example geom functions include <code>geom_point()</code> for a scatter plot, <code>geom_histogram()</code> for a histogram, <code>geom_density()</code> for a smoothened version of the histogram, and <code>geom_bar()</code> or <code>geo_col()</code> for a barplot.<br />
<br> The <strong>geom functions</strong> take a <code>mapping = aes(x = ..., y = ..., ...)</code> argument that defines how variables in the dataset are mapped to visual properties (aes for “aesthetic mapping”). Depending on the geom function, you can map variables to the x- and y-axis, to the shape of symbols, the line type of line graphs, to the fill color of symbols and more.<br />
<br> <strong>Scales</strong> can be customized with <code>scale_[]_[]</code> functions. These can be used to for example, change a color palette, set limits to the values included in the mapping, or log transform an axis.<br />
<br> Modifications of the plot appearance (including background colors, thick labels, thick lines, axis labels, titles) can be achieved by combinations of several functions.<br />
<br> A template code for a ggplot2 plot:<br />
<code>ggplot(data = &lt;DATA&gt;)+</code><br />
<code>geom_&lt;FUNCTION&gt;(mapping = aes(&lt;MAPPING&gt;))+</code><br />
<code>&lt;SCALE_FUNCTION&gt;+</code> <code>&lt;THEME_FUNCTION&gt;+</code> <code>&lt;GGTITLE&gt;</code><br />
<br> <strong>References</strong>:<br />
<em><a href="https://r4ds.had.co.nz/data-visualisation.html">R for Datascience, chapter 3 Data visualization</a></em><br />
<em><a href="https://github.com/rstudio/cheatsheets/raw/master/data-visualization-2.1.pdf">ggplot2 cheatsheet</a></em></p>
</blockquote>
</div>
<div id="exercise-3" class="section level3">
<h3>Exercise 3</h3>
<blockquote>
<p>Make a barplot showing the number of peaks per ChIP experiment.</p>
<ul>
<li>All chip-seq peaks are concatenated in the GRanges object <code>monocytes_all</code>.</li>
</ul>
</blockquote>
<blockquote>
<p><strong>Exercise 3a</strong>: Print a summary of <code>monocytes_all</code> using <code>show()</code> and identify the variable (column in metadata) that you can use for counting the number of peaks per chip (this will be the statistic used in the barplot).</p>
</blockquote>
<div id="monocytesall-hint">
<p><strong>Hint:</strong> Use the <code>show()</code> function</p>
</div>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(monocytes_all)</span></code></pre></div>
<pre><code>## GRanges object with 16949 ranges and 7 metadata columns:
##            seqnames            ranges strand |                            name
##               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |                     &lt;character&gt;
##    h3k4me1    chr19     266829-267802      * | 3334.macs2_peak_call_peak_41377
##    h3k4me1    chr19     290879-291479      * | 3334.macs2_peak_call_peak_41378
##    h3k4me1    chr19     294458-297235      * | 3334.macs2_peak_call_peak_41379
##    h3k4me1    chr19     337956-345843      * | 3334.macs2_peak_call_peak_41380
##    h3k4me1    chr19     392533-392941      * | 3334.macs2_peak_call_peak_41381
##        ...      ...               ...    ... .                             ...
##   h3k36me3    chr19 58566381-58567633      * | 3332.macs2_peak_call_peak_25740
##   h3k36me3    chr19 58568654-58571389      * | 3332.macs2_peak_call_peak_25741
##   h3k36me3    chr19 58578016-58579087      * | 3332.macs2_peak_call_peak_25742
##   h3k36me3    chr19 58581528-58584711      * | 3332.macs2_peak_call_peak_25743
##   h3k36me3    chr19 58585677-58586540      * | 3332.macs2_peak_call_peak_25744
##                score signalValue    pValue    qValue      peak        chip
##            &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt; &lt;character&gt;
##    h3k4me1        75     5.21238   9.15226   7.58659      &lt;NA&gt;     h3k4me1
##    h3k4me1        26     3.31169   3.98867    2.6193      &lt;NA&gt;     h3k4me1
##    h3k4me1        60     4.59427   7.50804    6.0026      &lt;NA&gt;     h3k4me1
##    h3k4me1       147     6.93584  16.61078  14.79942      &lt;NA&gt;     h3k4me1
##    h3k4me1        24     3.20502    3.7854   2.42508      &lt;NA&gt;     h3k4me1
##        ...       ...         ...       ...       ...       ...         ...
##   h3k36me3        22     2.91169   3.55852   2.28818      &lt;NA&gt;    h3k36me3
##   h3k36me3        31      3.3764   4.52293   3.18835      &lt;NA&gt;    h3k36me3
##   h3k36me3        25     3.12015    3.8119    2.5179      &lt;NA&gt;    h3k36me3
##   h3k36me3        60     4.43644   7.51895   6.01861      &lt;NA&gt;    h3k36me3
##   h3k36me3        24     3.07961    3.7475    2.4578      &lt;NA&gt;    h3k36me3
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Which variable will you use for counting the number of peaks per ChIP-seq dataset?&quot;</span>,</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;score&quot;</span>),</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;seqnames&quot;</span>),</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;peak&quot;</span>), </span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;chip&quot;</span>, <span class="at">correct =</span>T),</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">allow_retry =</span> T)</span></code></pre></div>
<p>As GRanges objects are not compatible with ggplot2, you have to extract the metadata and write into a a data.frame</p>
</div>
<div id="exercise-3b" class="section level3">
<h3>Exercise 3b</h3>
<blockquote>
<p><strong>Exercise 3b</strong>: Complete the code below to extract the metadata from the object <code>monocytes_all</code>.</p>
<ul>
<li>The function <code>as.data.frame()</code> is used to convert the resulting ‘DFrame’ object into a ‘data.frame’.</li>
</ul>
</blockquote>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># extract metadata </span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>monocytes_metadata <span class="ot">&lt;-</span> <span class="fu">__</span>(__)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="co"># convert to a data.frame</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>monocytes_metadata <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(monocytes_metadata)</span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="co"># extract metadata </span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>monocytes_metadata <span class="ot">&lt;-</span> <span class="fu">mcols</span>(monocytes_all)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="co"># convert to a data.frame</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>monocytes_metadata <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(monocytes_metadata)</span></code></pre></div>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_code</span>()</span></code></pre></div>
</div>
<div id="exercise-3c" class="section level3">
<h3>Exercise 3c</h3>
<blockquote>
<p><strong>Exercise 3c</strong>: Complete the code below to plot the number of peaks per ChIP in a bar chart using ggplot2 plotting.</p>
</blockquote>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the number of peaks per chip</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">__</span>(<span class="at">data =</span> __)<span class="sc">+</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">__</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">y=</span>__) )<span class="sc">+</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_colorblind</span>()<span class="sc">+</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_calc</span>()<span class="sc">+</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)<span class="sc">+</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Number of peaks per ChIP, monocytes, chr19&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the number of peaks per chip</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> monocytes_metadata)<span class="sc">+</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">mapping=</span><span class="fu">aes</span>(<span class="at">y=</span>chip))<span class="sc">+</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)<span class="sc">+</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Number of peaks per ChIP, monocytes, chr19&quot;</span>)</span></code></pre></div>
<p><img src="fg2_files/figure-html/barplotpeakcount-solution-1.png" width="672" /></p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_code</span>()</span></code></pre></div>
<p><em>For which marks do you observe the most peaks? And for which the fewest number of peaks? Did they, in the genome browser, show a sharp peaks or broad domains of enrichment?</em><br />
<br> The range of peak counts can tell you whether your experiment and peak calling performed as expected. If they show extremely high or low values, it is likely something is off (something you should have already noted in the Genome Browser). It can also function as an extra control step to check that all peaks are indeed imported in R (and no strange things happened to your files along the way).<br />
<br> In this case, the number of peaks is correct and you can continue with the analysis. <br></p>
</div>
<div id="exercise-4" class="section level3">
<h3>Exercise 4</h3>
<blockquote>
<p>Visualize the differences in peak size between the different marks in a boxplot.</p>
<p><strong>Exercise 4a</strong>: Finish the code below to add the ‘peak_size’ variable to <code>monocytes_metadata</code>. This new variable holds the peak widths.</p>
</blockquote>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># add variable &#39;peak_size&#39; to monocytes_metadata with the width of the peaks from monocytes_all </span></span></code></pre></div>
<div id="addpeaksize-hint">
<p><strong>Hint:</strong> Use <code>monocytes_metadata$peak_size &lt;- width(monocytes_all)</code></p>
</div>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>monocytes_metadata<span class="sc">$</span>peak_size <span class="ot">&lt;-</span> <span class="fu">width</span>(monocytes_all)</span></code></pre></div>
<blockquote>
<p><strong>Exercise 4b</strong>: Now plot the peak size distribution as a boxplot. Make sure to hit “Run Code” to show the plot in the console.</p>
</blockquote>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the peak_size per chip as boxplot</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fu">__</span>(<span class="at">data =</span> __)<span class="sc">+</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">__</span>(<span class="at">mapping =</span> <span class="fu">__</span>(<span class="at">x =</span> __, <span class="at">y =</span> __), <span class="at">fill =</span> <span class="st">&quot;gray&quot;</span>, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">bins =</span> <span class="dv">100</span>)<span class="sc">+</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span></code></pre></div>
<div id="plotpeaksize-hint">
<p><strong>Hint:</strong> Use <code>ggplot(data = monocytes_metadata)+</code></p>
</div>
<div id="plotpeaksize-hint">
<p><strong>Hint:</strong> and <code>geom_boxplot(mapping = aes(x = chip, y = ...))</code></p>
</div>
<div id="plotpeaksize-hint">
<p><strong>Hint:</strong> and y as <code>geom_boxplot(mapping = aes(x = chip, y = peak_size))</code></p>
</div>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> monocytes_metadata)<span class="sc">+</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> chip, <span class="at">y =</span> peak_size), <span class="at">fill =</span> <span class="st">&quot;gray&quot;</span>, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>)<span class="sc">+</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span></code></pre></div>
<p><img src="fg2_files/figure-html/plotpeaksize-solution-1.png" width="672" /></p>
<blockquote>
<p><strong>Exercise 4c</strong>: This doesn’t look very informative, let’s log10-transform the y-axis (again, hit “Run Code” to view the plot):</p>
</blockquote>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the peak_size per chip as boxplot, log10-transform the y-axis</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">__</span>(<span class="at">data =</span> __)<span class="sc">+</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">__</span>(<span class="at">mapping =</span> <span class="fu">__</span>(<span class="at">x =</span> __, <span class="at">y =</span> __), <span class="at">fill =</span> <span class="st">&quot;gray&quot;</span>, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>)<span class="sc">+</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">&quot;__&quot;</span>)</span></code></pre></div>
<div id="plotpeaksizelog-hint">
<p><strong>Hint:</strong> Specify “log10” for the <code>trans =</code> parameter</p>
</div>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> monocytes_metadata)<span class="sc">+</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> chip, <span class="at">y =</span> peak_size), <span class="at">fill =</span> <span class="st">&quot;gray&quot;</span>, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">&quot;log10&quot;</span>)</span></code></pre></div>
<p><img src="fg2_files/figure-html/plotpeaksizelog-solution-1.png" width="672" /></p>
<p>As mentioned in section 2.2.1 and 2.2.6, different data types have different peak shapes. H3K4me3 and H3K4me1 ChIP-seq peaks are resp. <strong>narrow</strong>, and the <strong>broad</strong> peaks.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quiz</span>(<span class="at">caption =</span> <span class="st">&quot;Use the boxplot above.&quot;</span>, </span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>     <span class="fu">question</span>(<span class="st">&quot;Which ChIP-seq dataset has/have broad peak shapes?&quot;</span>,</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>),</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;3K36me3&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">allow_retry =</span> <span class="cn">TRUE</span>),</span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>     <span class="fu">question</span>(<span class="st">&quot;which have narrow peak shapes?&quot;</span>,</span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;3K36me3&quot;</span>),</span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>),</span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>),</span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">allow_retry =</span> <span class="cn">TRUE</span>,</span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">message =</span> <span class="st">&quot;You could have gotten this information from the [BLUEPRINT ChIP-Seq Analysis Pipeline description](http://dcc.blueprint-epigenome.eu/#/md/chip_seq_grch38) (remember: FAIR principles!). Under the section **Peak Calling** you can read that peak calling was performed with MACS2 and that the -broad flag was added depending on the mark in question.&quot;</span> )</span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><br></p>
<blockquote>
<details>
<summary>
In summary, marks with narrow and with broad peaks:
</summary>
<table>
<thead>
<tr class="header">
<th align="left">narrowPeak file</th>
<th align="left">broadPeak file</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">H3K27ac</td>
<td align="left">H3K27me3</td>
</tr>
<tr class="even">
<td align="left">H3K4me3</td>
<td align="left">H3K36me3</td>
</tr>
<tr class="odd">
<td align="left">H3K9/14ac</td>
<td align="left">H3K9me3</td>
</tr>
<tr class="even">
<td align="left">H2A.Zac</td>
<td align="left">H3K4me1</td>
</tr>
</tbody>
</table>
</details>
</blockquote>
</div>
<div id="plotting-summarized-values-using-dplyr-ggplot" class="section level3">
<h3>2.3.3 Plotting summarized values using dplyr &amp; ggplot</h3>
<blockquote>
<p>In this section you will visualize the fraction that each ChIP-seq peak set covers in total on chromosome 19.</p>
</blockquote>
<p>To generate this plot, you have to calculate the total number of bps covered by each peak dataset. This can be achieved by summing all the width values.<br />
<br> GRanges objects follow the <strong>tidy data principle</strong>: each row of a Ranges object corresponds to an interval, and each column will represent a variable about that interval, and generally each object will represent a single unit of an observation (like gene annotations). You can use <strong>dplyr</strong>-like functions from the <code>plyranges</code> package to manipulate these objects and use the <strong>pipe</strong> operator <code>%&gt;%</code> to combine functions in a workflow.</p>
<blockquote>
<h4 id="refresher-dplyr"><strong>Refresher</strong>: dplyr</h4>
<details>
<summary>
Common dplyr-functions and examples of their plyranges-relatives on GRanges objects
</summary>
<table>
<colgroup>
<col width="23%" />
<col width="38%" />
<col width="38%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Purpose</th>
<th align="left">Example on GRanges</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>select()</code></td>
<td align="left">subset variables (=columns)</td>
<td align="left"><code>select(my_gr, GC)</code></td>
</tr>
<tr class="even">
<td align="left"><code>group_by()</code></td>
<td align="left">group data into rows with the same value for the specified variable.</td>
<td align="left"><code>my_gr %&gt;% group_by(strand)</code>.</td>
</tr>
<tr class="odd">
<td align="left"><code>filter()</code></td>
<td align="left">subset observations (= rows)</td>
<td align="left"><code>filter(my_gr, GC &lt; 0.4)</code> or <code>my_gr %&gt;% group_by(strand) %&gt;%  filter(GC == max(GC))</code>.</td>
</tr>
<tr class="even">
<td align="left"><code>summarize()</code></td>
<td align="left">Summarise variables, often per group</td>
<td align="left"><code>group_by(my_gr, strand) %&gt;% summarize(n = n(), gc = max(GC))</code></td>
</tr>
<tr class="odd">
<td align="left"><code>mutate()</code></td>
<td align="left">Add a new varialbe</td>
<td align="left"></td>
</tr>
</tbody>
</table>
See also the <a href="https://rstudio.com/wp-content/uploads/2015/02/data-wrangling-cheatsheet.pdf">dplyr cheatsheet</a>
</details>
</blockquote>
</div>
<div id="exercise-5a" class="section level3">
<h3>Exercise 5a</h3>
<blockquote>
<p>To plot the fraction of bps covered by each chip-seq peak dataset first calculate this fraction.</p>
</blockquote>
<blockquote>
<p><strong>Exercise 5a</strong>: Finish the code below using <code>group_by()</code> and <code>summarize()</code> to calculate</p>
<ol style="list-style-type: lower-alpha">
<li>the total number of bps covered by each peak dataset and<br />
</li>
<li>as a fraction of total bps in chromosome 19 (=58617616 bp)</li>
</ol>
</blockquote>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the total number of bps covered by each chipseq dataset</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="co"># and as a fraction of total bps in chromosome 19</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>monocytes_metadata <span class="sc">%&gt;%</span> </span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  ___ <span class="sc">%&gt;%</span></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">total_bps =</span> ___ ,</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">fraction_chr19 =</span> ___ <span class="sc">/</span><span class="dv">58617616</span>)</span></code></pre></div>
<div id="plotpeaksize-hint">
<p><strong>Hint:</strong> Use <code>monocytes_metadata %&gt;% group_by(chip)</code></p>
</div>
<div id="plotpeaksize-hint">
<p><strong>Hint:</strong> piped as <code>monocytes_metadata %&gt;% group_by(chip) %&gt;% summarize(total_bps = sum(peak_size))</code></p>
</div>
<div id="plotpeaksize-hint">
<p><strong>Hint:</strong> and and the second summary variable defined as <code>monocytes_metadata %&gt;% group_by(chip) %&gt;% summarize(total_bps = sum(peak_size), fraction_chr19 = total_bps /58617616)</code></p>
</div>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>monocytes_metadata <span class="sc">%&gt;%</span> </span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(chip) <span class="sc">%&gt;%</span> </span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">total_bps =</span> <span class="fu">sum</span>(peak_size),</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">fraction_chr19 =</span> total_bps <span class="sc">/</span><span class="dv">58617616</span>)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 3
##   chip     total_bps fraction_chr19
##   &lt;chr&gt;        &lt;int&gt;          &lt;dbl&gt;
## 1 h3k27ac    2229872         0.0380
## 2 h3k27me3   5946197         0.101 
## 3 h3k36me3  11225842         0.192 
## 4 h3k4me1    8671082         0.148 
## 5 h3k4me3    3401660         0.0580
## 6 h3k9me3   11944274         0.204</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_code</span>()</span></code></pre></div>
</div>
<div id="exercise-5b" class="section level3">
<h3>Exercise 5b</h3>
<blockquote>
<p><strong>Exercise 5b</strong>: To plot the <code>fraction_chr19</code> as bar chart, pipe the output of the code below into ggplot.<br />
We instruct the bar function to plot the actual values with: <code>stat = "identity"</code>.<br />
Hit “Run Code” to plot the plot in the console.</p>
</blockquote>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the fraction covered in a bar chart</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>monocytes_metadata <span class="sc">%&gt;%</span>  <span class="fu">group_by</span>(chip) <span class="sc">%&gt;%</span> </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">total_bps =</span> <span class="fu">sum</span>(peak_size),  </span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">fraction_chr19 =</span> <span class="fu">sum</span>(peak_size) <span class="sc">/</span><span class="dv">58617616</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(.)<span class="sc">+</span> </span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">__</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> __, <span class="at">y =</span> __, <span class="at">fill =</span> chip), <span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>monocytes_metadata <span class="sc">%&gt;%</span>  <span class="fu">group_by</span>(chip) <span class="sc">%&gt;%</span> </span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">total_bps =</span> <span class="fu">sum</span>(peak_size),  </span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">fraction_chr19 =</span> <span class="fu">sum</span>(peak_size) <span class="sc">/</span><span class="dv">58617616</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(.)<span class="sc">+</span> </span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> chip, <span class="at">y =</span> fraction_chr19, <span class="at">fill =</span> chip), <span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="fg2_files/figure-html/plotsummarizewidth-solution-1.png" width="672" /></p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_code</span>()</span></code></pre></div>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Which type of peaks covers relativley more bp in chromosomes 9?&quot;</span>,</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;narrow peak ChIP-seq peak datasets&quot;</span>),</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;broad domain ChIP-seq peak datasets&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">allow_retry =</span> T)</span></code></pre></div>
<p><strong>These summary characteristics may suggest that H3K4me3 is relatively sparse (relative compared to the other marks) but that doesn’t mean it isn’t functional. Several analyses can inform us about the functionality of ChIP-seq marks. Let’s start with their enrichment in promoters.</strong>. <br></p>
<hr />
<p><br></p>
</div>
</div>
<div id="overlap-analysis" class="section level2">
<h2>2.4 Overlap analysis</h2>
<div id="detect-overlap-with-genomicranges" class="section level3">
<h3>2.4.1 Detect overlap with GenomicRanges</h3>
<p>To understand the function of a ChIPped histone mark or factor often want to know whether it is enriched in a particular genomic element. To find out, you need to calculate a quantitative summary of its genome-wide distribution across the different elements. By comparing this distribution to the genome-wide coverage of these different elements (null hypothesis), we can test for enrichment of the mark or factor.<br />
<br> The genomic elements are defined in a reference file. You can use various reference files for this exercise: from transcript-oriented objects, to functional elements from a database like ENCODE, to a custom one.<br />
<br> In the coming exercise, you will first quantify the overlap between H3K4me3 peaks and promoters and, subsequently, between H3K4me3 peaks and several different genomic elements simultaneously.<br />
<br> To identify and count overlap between two sets of GRanges objects (e.g. H3K4me3 peaks and promoters) you can use function from the GenomicRanges package:<br />
<br> <strong>countOverlaps</strong>: <code>countOverlaps(query, subject)</code> returns a integer vector with <em>the number of overlaps</em> for each element in the <code>query</code>.<br />
<br> <strong>subsetByOverlaps</strong>: <code>subsetByOverlaps(query, subject)</code> extracts the elements in the <code>query</code> that overlap with at least one element in the subject.<br />
<br> <strong>findOverlaps</strong>: <code>findOverlaps(query, subject)</code> returns a <code>Hits</code> object containing the index pairings for the overlapping elements.</p>
<ul>
<li>The columns with indexes can be accessed through <code>queryHits(overlap_object)</code> and <code>subjectHits(overlap_object)</code>. These can subsequently be used to extract the corresponding peaks or regions from the objects that were used as inputs.<br />
</li>
<li>If a peak in one of the inputs overlaps with multiple peaks in the other, its index will appear multiple times in the output.</li>
</ul>
<p><br></p>
</div>
<div id="overlap-promoters-vs-h3k4me3-peaks" class="section level3">
<h3>2.4.2 Overlap promoters vs h3k4me3 peaks</h3>
<p>To determine the overlap among H3K4me3 peaks and promoters, you first need to obtain promoter interval regions as GRanges object. We have done that for you with the following code that:</p>
<ol style="list-style-type: decimal">
<li>Loads the <code>TxDb.Hsapiens.UCSC.hg38.knownGene</code> database package.<br />
</li>
<li>Extract promoter regions with the <code>promoters()</code> function. This function outputs a GRanges object with intervals around the TSS.<br />
</li>
<li>By default, <code>promoters()</code> works on transcripts but we are often more interested in promoters of genes therefore we add the <code>genes()</code> function.</li>
</ol>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load the txdb package which holds transcript-based gene models of hg38 genome  </span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TxDb.Hsapiens.UCSC.hg38.knownGene)</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">&lt;-</span> TxDb.Hsapiens.UCSC.hg38.knownGene </span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevels</span>(txdb) <span class="ot">&lt;-</span> <span class="st">&quot;chr19&quot;</span> <span class="co"># limits the database to chromosome 19</span></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a><span class="co"># extract promoter coordinates at the genes level</span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>promoters_chr19 <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">promoters</span>(<span class="fu">genes</span>(txdb), <span class="at">upstream=</span><span class="dv">2000</span>, <span class="at">downstream=</span><span class="dv">200</span>)) </span></code></pre></div>
</div>
<div id="exercise-6a" class="section level3">
<h3>Exercise 6a</h3>
<blockquote>
<p><strong>Exercise 6a:</strong> How many promoter are stored in <code>promoters_chr19</code>?</p>
</blockquote>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co">#  How many regions do you have are stored in `promoters_chr19`? </span></span></code></pre></div>
<div id="q6a_proms-hint">
<p><strong>Hint:</strong> You may want to use the <code>length()</code> function.</p>
</div>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(promoters_chr19)</span></code></pre></div>
<pre><code>## [1] 1723</code></pre>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_result</span>(</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a> <span class="fu">pass_if</span>(<span class="sc">~</span><span class="fu">identical</span>(.result, <span class="fu">length</span>(promoters_chr19)))</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a> )</span></code></pre></div>
</div>
<div id="exercise-6b" class="section level3">
<h3>Exercise 6b</h3>
<blockquote>
<p><strong>Exercise 6b:</strong> Use code to identify the width of the promoter windows. The output of this code should be one integer value.</p>
</blockquote>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># determine the promoter width</span></span></code></pre></div>
<div id="q6b_proms-hint">
<p><strong>Hint:</strong> first run a code to show a summary of the widths <code>summary(width(object))</code></p>
</div>
<div id="q6b_proms-hint">
<p><strong>Hint:</strong> obtain the correct output format with <code>unique()</code></p>
</div>
<div id="q6b_proms-hint">
<p><strong>Hint:</strong> <code>unique(width(promoters_chr19))</code></p>
</div>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_result</span>(</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a> <span class="fu">pass_if</span>(<span class="sc">~</span><span class="fu">identical</span>(.result, <span class="dv">2200</span>))</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a> )</span></code></pre></div>
</div>
<div id="exercise-6c" class="section level3">
<h3>Exercise 6c</h3>
<blockquote>
<p>We want to know how many of the promoters overlap with a H3K4me3 ChIP-seq peak and vice versa.</p>
</blockquote>
<blockquote>
<p><strong>Exercise 6c:</strong> Use <code>findOverlaps()</code> to determine the overlap between <code>monocytes_h3k4me3</code> (query) and <code>promoters_chr19</code> (subject).</p>
<ul>
<li>You need to hit “Run Code” to be able to see the output of <code>show()</code></li>
</ul>
</blockquote>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co">#exercise.setup = &quot;prepare_exercise6&quot;}</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Find overlap between monocytes_h3k4me3 peaks and promoters</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> <span class="fu">___</span>(<span class="at">query =</span> ___, <span class="at">subject =</span> ___)</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a><span class="co"># print the overlap output</span></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(___)</span></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Find overlap between monocytes_h3k4me3 peaks and promoters</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(<span class="at">query =</span> monocytes_h3k4me3, <span class="at">subject =</span> promoters_chr19)</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="co"># print the overlap output</span></span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(overlap)</span></code></pre></div>
<pre><code>## Hits object with 1366 hits and 0 metadata columns:
##          queryHits subjectHits
##          &lt;integer&gt;   &lt;integer&gt;
##      [1]         3        1603
##      [2]         5        1056
##      [3]         6         406
##      [4]         7         629
##      [5]         8         629
##      ...       ...         ...
##   [1362]      2836        1339
##   [1363]      2836        1413
##   [1364]      2837        1339
##   [1365]      2837        1413
##   [1366]      2838        1413
##   -------
##   queryLength: 2838 / subjectLength: 1723</code></pre>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_code</span>()</span></code></pre></div>
<p><br> The Hits object “overlap” reports the indexes of <code>monocytes_h3k4me3</code> and <code>promoters_chr19</code> that overlap. If a peak or a promoter overlaps several times, each overlap will be reported in a new row.<br />
<br></p>
</div>
<div id="exercise-6d" class="section level3">
<h3>Exercise 6d</h3>
<blockquote>
<p><strong>Exercise 6d:</strong> What fraction of promoters overlaps with H3K4me3 peaks?</p>
</blockquote>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the unique promoter indexes</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>unique_promoters <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">___</span>(overlap))</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="co"># How many of these unique promoters do you have?</span></span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>unique_promoter_count <span class="ot">&lt;-</span> <span class="fu">___</span>(unique_promoters)</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the resulting number</span></span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>unique_promoter_count</span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a><span class="co"># What fraction of promoters is part of the overlap?  </span></span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a>unique_promoter_count<span class="sc">/</span><span class="fu">__</span>(promoters)</span></code></pre></div>
<div id="q6d_overlap-hint">
<p>Use <code>queryHits()</code> and <code>subjectHits()</code> to extract the indexes of overlapping peaks and promoters from <code>overlap</code>, respectively.</p>
</div>
<div id="q6d_overlap-hint">
<p>Use <code>unique()</code> to minimize this output to unique peaks.</p>
</div>
<div id="q6d_overlap-hint">
<p>Use <code>length()</code> to count the number of unique indexes of the query and the subject.</p>
</div>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the unique promoter indexes</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>unique_promoters <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">subjectHits</span>(overlap))</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a><span class="co"># How many of these unique promoters do you have?</span></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>unique_promoters_count <span class="ot">&lt;-</span> <span class="fu">length</span>(unique_promoters)</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the resulting number</span></span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>unique_promoters_count</span></code></pre></div>
<pre><code>## [1] 1150</code></pre>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co"># What fraction of promoters is part of the overlap?  </span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>unique_promoters_count<span class="sc">/</span><span class="fu">length</span>(promoters_chr19)</span></code></pre></div>
<pre><code>## [1] 0.6674405</code></pre>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question_text</span>(</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Enter the number of promoters that are part of the overlap:&quot;</span>,</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;1723&quot;</span>, <span class="at">message =</span> <span class="st">&quot;This is the total number of promoters. Find the ones part of the overlap with subjectHits() unique() and length()&quot;</span>),</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;1150&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;1366&quot;</span>, <span class="at">message =</span> <span class="st">&quot;This is the length of the overlap object, select the subject and reduce with unique.&quot;</span>),</span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;1087&quot;</span>, <span class="at">message =</span> <span class="st">&quot;This is number of unique H3K4me3 peaks that are part of the overlap. Make sure you use subjectHits() and not queryHits() function.&quot;</span>),</span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co"># enter the fraction of promoters that are part of the overlap, round to 3 decimal places:  </span></span></code></pre></div>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(unique_promoters_count<span class="sc">/</span><span class="fu">length</span>(promoters_chr19),<span class="dv">3</span>)</span></code></pre></div>
<pre><code>## [1] 4.092</code></pre>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_result</span>(</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pass_if</span>(<span class="sc">~</span><span class="fu">identical</span>(.result,<span class="fu">round</span>(unique_promoters_count<span class="sc">/</span><span class="fu">length</span>(promoters_chr19),<span class="dv">3</span>))))</span></code></pre></div>
</div>
<div id="exercise-6e" class="section level3">
<h3>Exercise 6e</h3>
<blockquote>
<p><strong>Exercise 6e:</strong> And what fraction of the h3k4me3 peaks are part of the overlap?</p>
</blockquote>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the unique peaks</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>unique_peaks <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">___</span>(overlap))</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a><span class="co"># How many of these unique peaks do you have?</span></span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>unique_peaks_count <span class="ot">&lt;-</span> <span class="fu">___</span>(unique_peaks)</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the resulting number</span></span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>unique_peaks_count</span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a><span class="co"># What fraction of peaks is part of the overlap?  </span></span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a>unique_peaks_count<span class="sc">/</span><span class="fu">__</span>(__)</span></code></pre></div>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the unique peaks</span></span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>unique_peaks <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">queryHits</span>(overlap))</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a><span class="co"># How many of these unique peaks do you have?</span></span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>unique_peaks_count <span class="ot">&lt;-</span> <span class="fu">length</span>(unique_peaks)</span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the resulting number</span></span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>unique_peaks_count</span></code></pre></div>
<pre><code>## [1] 1692</code></pre>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="co"># What fraction of peaks is part of the overlap?  </span></span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>unique_peaks_count<span class="sc">/</span><span class="fu">length</span>(monocytes_h3k4me3)</span></code></pre></div>
<pre><code>## [1] 0.5961945</code></pre>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question_text</span>(</span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Enter the number of H3K4me3 peaks that are part of the overlap:&quot;</span>,</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;2838&quot;</span>, <span class="at">message =</span> <span class="st">&quot;This is the total number of H3K4me3 peaks. Find the ones part of the overlap with subjectHits() unique() and length()&quot;</span>),</span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;1087&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;1366&quot;</span>, <span class="at">message =</span> <span class="st">&quot;This is the length of the overlap object, select the subject and reduce the output with unique().&quot;</span>),</span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">answer</span>(<span class="st">&quot;1150&quot;</span>, <span class="at">message =</span> <span class="st">&quot;This is number of unique promoter regions that are part of the overlap. Make sure you use queryHits() and not the subjectHits() function.&quot;</span>),</span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># enter the fraction of promoters that are part of the overlap, round to 3 decimal places:  </span></span></code></pre></div>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(unique_peaks_count<span class="sc">/</span><span class="fu">length</span>(monocytes_h3k4me3),<span class="dv">3</span>)</span></code></pre></div>
<pre><code>## [1] 0.383</code></pre>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_result</span>(</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pass_if</span>(<span class="sc">~</span><span class="fu">identical</span>(.result,<span class="fu">round</span>(unique_peaks_count<span class="sc">/</span><span class="fu">length</span>(monocytes_h3k4me3),<span class="dv">3</span>))))</span></code></pre></div>
</div>
<div id="exercise-6f" class="section level3">
<h3>Exercise 6f</h3>
<p>Plot the overlap among promoters and H3K4me3 peaks with the <code>plot.pairwise.venn()</code> function of the <code>VennDiagram</code> package.</p>
<p>We Use the minimum of the ‘unique_peaks_count’ and ‘unique_promoters_count’ as the number of ‘common peaks’ for our venn diagram.</p>
<blockquote>
<p><strong>Exercise 6f:</strong><br />
Complte the the <code>plot.pairwise.venn()</code> code below to visualize these counts in a venn diagram. The <code>plot.pairsiwe.venn()</code> function needs two “areas” that represent the total counts (overlapping and not-overlapping) of each element, and a count for the overlap.</p>
<ul>
<li>Hit “Run Code” to ensure that the venn diagram is plotted.</li>
</ul>
</blockquote>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># determine the number of common counts</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>common_counts <span class="ot">&lt;-</span> <span class="fu">min</span>(unique_promoters_count, unique_peaks_count)</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a><span class="co"># call a new plotting area</span></span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.newpage</span>()</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the overlap in a venn diagram</span></span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a><span class="fu">draw.pairwise.venn</span>( </span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">area1=</span>___, <span class="co"># total count for area 1</span></span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a>           <span class="at">area2=</span>___, <span class="co"># total count for area 2</span></span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a>           <span class="at">cross.area=</span>common_counts, <span class="co"># count for the overlap.</span></span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a>           <span class="at">category=</span><span class="fu">c</span>(<span class="st">&quot;H3K4me3&quot;</span>, <span class="st">&quot;Promoters&quot;</span>), </span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a>           <span class="at">fill=</span><span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;gray&quot;</span>), </span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>           <span class="at">cat.cex=</span><span class="fl">1.2</span>)</span></code></pre></div>
<div id="q6f_plotvenn-hint">
<p>Use <code>length()</code> for area1 and area2</p>
</div>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a><span class="co"># determine the number of common counts</span></span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>common_counts <span class="ot">&lt;-</span> <span class="fu">min</span>(unique_promoters_count, unique_peaks_count)</span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a><span class="co"># call a new plotting area</span></span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.newpage</span>()</span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the overlap in a venn diagram</span></span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a><span class="fu">draw.pairwise.venn</span>( </span>
<span id="cb83-10"><a href="#cb83-10" aria-hidden="true" tabindex="-1"></a>   <span class="at">area1=</span><span class="fu">length</span>(monocytes_h3k4me3),</span>
<span id="cb83-11"><a href="#cb83-11" aria-hidden="true" tabindex="-1"></a>   <span class="at">area2=</span><span class="fu">length</span>(promoters_chr19), </span>
<span id="cb83-12"><a href="#cb83-12" aria-hidden="true" tabindex="-1"></a>   <span class="at">cross.area=</span>common_counts, </span>
<span id="cb83-13"><a href="#cb83-13" aria-hidden="true" tabindex="-1"></a>   <span class="at">category=</span><span class="fu">c</span>(<span class="st">&quot;H3K4me3&quot;</span>, <span class="st">&quot;Promoters&quot;</span>), </span>
<span id="cb83-14"><a href="#cb83-14" aria-hidden="true" tabindex="-1"></a>   <span class="at">fill=</span><span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;gray&quot;</span>), </span>
<span id="cb83-15"><a href="#cb83-15" aria-hidden="true" tabindex="-1"></a>   <span class="at">cat.cex=</span><span class="fl">1.2</span>)</span></code></pre></div>
<p><img src="fg2_files/figure-html/q6f_plotvenn-solution-1.png" width="672" /></p>
<pre><code>## (polygon[GRID.polygon.276], polygon[GRID.polygon.277], polygon[GRID.polygon.278], polygon[GRID.polygon.279], text[GRID.text.280], text[GRID.text.281], text[GRID.text.282], text[GRID.text.283], text[GRID.text.284])</code></pre>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_code</span>()</span></code></pre></div>
<p><strong>This looks like a considerable enrichment of H3K4me3 in promoter regions. Let’s test this.</strong></p>
</div>
<div id="testing-for-enrichment-by-overlap" class="section level3">
<h3>2.4.3 Testing for enrichment by overlap</h3>
</div>
<div id="exercise-7-is-there-a-significant-enrichment-of-h3k4me3-in-promoters" class="section level3">
<h3>Exercise 7: Is there a significant enrichment of h3k4me3 in promoters?</h3>
<p><br> To answer this question, we compare the fraction of promoters with a H3K4me3 peak with the chromosome 19-wide fraction of promoters. <br> If H3K4me3 is not enriched at promoters, we would expect that the fraction of promoters with a H3K4me3 peak is in the same range as the fraction of promoters on chromosome 19. (This is our <strong>null hypothesis of no enrichment</strong>). <br> As some promoters may overlap we <em>reduced</em> <code>promoters</code> to non-overlapping intervals to calculate the chromosome 19-wide fraction of promoters. We do that by running:</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="co"># obtain reduced promoter regions</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>promoters_chr19_reduced <span class="ot">&lt;-</span> <span class="fu">reduce</span>(promoters_chr19)</span></code></pre></div>
<p>Similarly to exercise 3 we then extract the metadata from this object and write it into a data.frame:</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co"># retrieve metadata and convert to dataframe</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>promoters_metadata <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">mcols</span>(promoters_chr19_reduced))</span></code></pre></div>
</div>
<div id="exercise-7a" class="section level3">
<h3>Exercise 7a</h3>
<blockquote>
<p><strong>Exercise 7a</strong>: Finish the code below to define the “promoter_size” variable and calculate the fraction of bps covered by <code>promoters_chr19_reduced</code> in chromosome 19 (chr19 = 58617616 bp).</p>
</blockquote>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># define the variable &#39;promoter_size&#39;</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>promoters_metadata<span class="sc">$</span>promoter_size <span class="ot">&lt;-</span> <span class="fu">___</span>(promoters_chr19_reduced)</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the  fraction of bps covered &#39;promoters_chr19_reduced`</span></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>promoters_metadata <span class="sc">%&gt;%</span></span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">total_bps =</span> ___ ,</span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">fraction_chr19 =</span> ___ <span class="sc">/</span><span class="dv">58617616</span>)</span></code></pre></div>
<div id="q7a_promfraction-hint">
<p><strong>Hint:</strong> Look back at exercise 5a.</p>
</div>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="co"># define the variable &#39;promoter_size&#39;</span></span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>promoters_metadata<span class="sc">$</span>promoter_size <span class="ot">&lt;-</span> <span class="fu">width</span>(promoters_chr19_reduced)</span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the  fraction of bps covered &#39;promoters_chr19_reduced`</span></span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a>promoters_metadata <span class="sc">%&gt;%</span></span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">total_bps =</span> <span class="fu">sum</span>(promoter_size),</span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">fraction_chr19 =</span> total_bps <span class="sc">/</span><span class="dv">58617616</span>)</span></code></pre></div>
<pre><code>##   total_bps fraction_chr19
## 1   3622268     0.06179487</code></pre>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="co"># enter the fraction of promoters in chromosome 19, round of to 3 decimals</span></span></code></pre></div>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fl">0.06179487</span>    ,<span class="dv">3</span>)</span></code></pre></div>
<pre><code>## [1] 0.062</code></pre>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_result</span>(</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pass_if</span>(<span class="sc">~</span><span class="fu">identical</span>(.result,<span class="fu">round</span>(<span class="fl">0.06179487</span>,<span class="dv">3</span>))))</span></code></pre></div>
<p><br> Promoters make up 6% of this chromosome but appr. 66% of all promoters overlap with h3k4me3 peaks. This looks like an enrichment.</p>
</div>
<div id="exercise-7b" class="section level3">
<h3>Exercise 7b</h3>
<p>You can test whether the observed fraction is indeed larger than expected with a binomial test, in r we can use the function <code>binom.test(x, p)</code> for this:</p>
<ul>
<li><code>x</code> = vector with number of successes (= number of promoters with H3K4me3 peak) and number of failures (= number of promoters without H3K4me3 peak)</li>
<li><code>p</code> = expected probability of success, in this case the fraction of promoters in chromosome 19</li>
</ul>
<p><br></p>
<blockquote>
<p><em>Background</em>: The binomial test is run when an experiment has two possible outcomes (i.e. success/failure) and you have an idea about what the probability of success is. Success in this case is overlap and our expectation is that 20% of the cases show overlap. The test calculates the probability of getting a desired outcome with a specific sample size.</p>
</blockquote>
<blockquote>
<p><strong>Exercise 7b</strong>: Use a binomial test to test for enrichment of H3K4me3 in promoters.</p>
<ul>
<li>Use the function <code>binom.test(x = c(successes, failures), p, alternative = "greater")</code> to call the test.</li>
<li>Use the information from exercise 6a and 6d to calculate the number of successes and failures.</li>
<li>Set <code>alternative = "greater"</code> because we test for <em>enrichment</em> and our alternative hypothesis is that the true probability is <em>larger</em> than the expected probability.</li>
</ul>
</blockquote>
<p><br></p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the indexes from the overlap object to identify promoters with H3K4me3 peak</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>promoters_with_h3k4me3 <span class="ot">&lt;-</span> promoters_chr19[ <span class="fu">unique</span>(<span class="fu">___</span>(overlap)) ]</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform negative selection using the same indexes to find promoters without H3K4me3 peak</span></span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a><span class="co"># the - sign removes the lines with the respective indexes from the original object</span></span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a>promoters_without_h3k4me3 <span class="ot">&lt;-</span> promoters_chr19[ <span class="sc">-</span><span class="fu">unique</span>(<span class="fu">___</span>(overlap)) ]</span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the x for your test</span></span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a><span class="co"># hint: use the function length()</span></span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a>x_test <span class="ot">&lt;-</span> <span class="fu">c</span>(___, ___)</span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the p for your test</span></span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a><span class="co"># hint: exercise 7a</span></span>
<span id="cb95-14"><a href="#cb95-14" aria-hidden="true" tabindex="-1"></a>p_test <span class="ot">&lt;-</span> __</span>
<span id="cb95-15"><a href="#cb95-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-16"><a href="#cb95-16" aria-hidden="true" tabindex="-1"></a><span class="co"># binomial test for enrichment of h3k4me3 peaks in promoters:</span></span>
<span id="cb95-17"><a href="#cb95-17" aria-hidden="true" tabindex="-1"></a>binomtest_result <span class="ot">&lt;-</span> <span class="fu">binom.test</span>(<span class="at">x =</span> x_test_prom_with_h3k4me3, <span class="at">p =</span> p_test_prom_with_h3k4me3, <span class="at">alternative =</span> <span class="st">&quot;greater&quot;</span> )</span>
<span id="cb95-18"><a href="#cb95-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-19"><a href="#cb95-19" aria-hidden="true" tabindex="-1"></a><span class="co"># report the test output</span></span>
<span id="cb95-20"><a href="#cb95-20" aria-hidden="true" tabindex="-1"></a>binomtest_result</span></code></pre></div>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the indexes from the overlap object to identify promoters with H3K4me3 peak</span></span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>promoters_with_h3k4me3 <span class="ot">&lt;-</span> promoters_chr19[ <span class="fu">unique</span>(<span class="fu">subjectHits</span>(overlap)) ]</span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform negative selection using the same indexes to find promoters without H3K4me3 peak</span></span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a><span class="co"># the - sign removes the lines with the respective indexes from the original object</span></span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a>promoters_without_h3k4me3 <span class="ot">&lt;-</span> promoters_chr19[ <span class="sc">-</span><span class="fu">unique</span>(<span class="fu">subjectHits</span>(overlap)) ]</span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the x for your test</span></span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a><span class="co"># hint: use the function length()</span></span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a>x_test_prom_with_h3k4me3 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">length</span>(promoters_with_h3k4me3), <span class="fu">length</span>(promoters_without_h3k4me3))</span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the p for your test</span></span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a><span class="co"># hint: exercise 7a</span></span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a>p_test_prom_with_h3k4me3 <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fl">0.06179487</span>,<span class="dv">3</span>)</span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a><span class="co"># binomial test for enrichment of h3k4me3 peaks in promoters:</span></span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a>binomtest_result <span class="ot">&lt;-</span> <span class="fu">binom.test</span>(<span class="at">x =</span> x_test_prom_with_h3k4me3, <span class="at">p =</span> p_test_prom_with_h3k4me3, <span class="at">alternative =</span> <span class="st">&quot;greater&quot;</span>)</span>
<span id="cb96-19"><a href="#cb96-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-20"><a href="#cb96-20" aria-hidden="true" tabindex="-1"></a><span class="co"># report the test output</span></span>
<span id="cb96-21"><a href="#cb96-21" aria-hidden="true" tabindex="-1"></a>binomtest_result</span></code></pre></div>
<pre><code>## 
##  Exact binomial test
## 
## data:  x_test_prom_with_h3k4me3
## number of successes = 1150, number of trials = 1723, p-value &lt; 2.2e-16
## alternative hypothesis: true probability of success is greater than 0.062
## 95 percent confidence interval:
##  0.6482792 1.0000000
## sample estimates:
## probability of success 
##              0.6674405</code></pre>
</div>
<div id="exercise-7c" class="section level3">
<h3>Exercise 7c</h3>
<blockquote>
<p><strong>Exercise 7c</strong> Based on this test, do you conclude that H3K4me3 is enriched in promoter regions?</p>
</blockquote>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Is H3K4me3 enriched promoters?&quot;</span>,</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;*Yes*&quot;</span>, <span class="at">correct =</span> T, <span class="at">message =</span> <span class="st">&quot;The binomial test shows a p-value &lt; 2.2e-16 and a confidence interval that excludes the expected 0.062 We therefore reject the H0 of no enrichment.&quot;</span>),</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;*No*&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Incorrect. Look at the p-value of the previous test.&quot;</span>)</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="overlap-promoters-exons-introns-and-intergenic-regions-vs-h3k4me3-peaks" class="section level3">
<h3>2.4.4 Overlap promoters, exons, introns and intergenic regions vs h3k4me3 peaks</h3>
<p>Instead of looking only at the overlap with promoters, we can also calculate the distribution of H3K4me3 peaks over various genomic features. To achieve this with functions like <code>findOverlaps</code> would require several, successive overlap analyses that would clutter up the code. Luckily, special packages have been developed for ChIP-seq analyses that performe this task. One of these packages is <code>genomation</code>. We have installed and loaded that package by:</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;genomation&quot;</span>)</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genomation)</span></code></pre></div>
<p><br> You can use the following function to calculate the peak distribution of over exons, introns, promoter, intergenic regions:<br />
<code>annotateWithGeneParts(target = [peaks_as_GRanges], feature = [features_as_GRangesList])</code><br />
<br> An important difference with the analysis above is that <code>annotateWithGeneParts()</code> uses transcript-level features. In exercise 6 and 7 we used gene-level features, ie. one promoter per gene. <code>annotateWithGeneParts()</code> defines one promoter per transcript.<br />
<br> The features have been loaded from gencode (downloaded from the UCSC genome browser) and read into a GRangesList object with the genomation-function <code>readTranscriptFeatures()</code>. With <code>up.flank</code> and <code>down.flank</code> we defined the up- and downstream boundaries of promoters around the TSSs.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>gencode_chr19 <span class="ot">&lt;-</span> <span class="fu">readTranscriptFeatures</span>(<span class="st">&quot;data/encode/gencodev32.chr19.bed&quot;</span>, <span class="at">unique.prom =</span> <span class="cn">TRUE</span>, <span class="at">up.flank =</span> <span class="dv">2000</span>, <span class="at">down.flank =</span> <span class="dv">200</span>) </span></code></pre></div>
</div>
<div id="exercise-8a" class="section level3">
<h3>Exercise 8a</h3>
<blockquote>
<p><strong>Exercise 8a</strong>: Use annotateWithGeneParts to determine the overlap between H3K4me3 peaks and promoters, exons, introns and intergenic regions.</p>
</blockquote>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="co"># overlap of h3k4me3 with promoters, exons, introns and intergenic regions</span></span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>annotation_h3k4me3 <span class="ot">&lt;-</span> <span class="fu">___</span>(<span class="at">target =</span> __, <span class="at">feature =</span> __)</span></code></pre></div>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>annotation_h3k4me3 <span class="ot">&lt;-</span> <span class="fu">annotateWithGeneParts</span>(<span class="at">target =</span> monocytes_h3k4me3, <span class="at">feature =</span> gencode_chr19)</span></code></pre></div>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_code</span>()</span></code></pre></div>
</div>
<div id="exercise-8b" class="section level3">
<h3>Exercise 8b</h3>
<p>You can visualize the result with:<br />
<code>plotTargetAnnotation(x = [annotateWithGeneParts output], main = "....")</code></p>
<blockquote>
<p><strong>Exercise 8b</strong>: Plot the output of exercise 8a</p>
</blockquote>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># visualize the output of exercise 8a</span></span></code></pre></div>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotTargetAnnotation</span>(<span class="at">x =</span> annotation_h3k4me3, <span class="at">main =</span> <span class="st">&quot;H3K4me3 over gene parts&quot;</span>)</span></code></pre></div>
<p><img src="fg2_files/figure-html/q8b_genomationoverlap-solution-1.png" width="672" /></p>
</div>
<div id="exercise-8c" class="section level3">
<h3>Exercise 8c</h3>
<p>Rerun the analysis of exercise 8a and 8b for the other marks (H3K4me1, H3K36me3, H3K27me3, H3K27ac, H3K9me3). You can use the following code section run these plots.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quiz</span>(<span class="at">caption =</span><span class="st">&quot;&quot;</span>, </span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>     <span class="fu">question</span>(<span class="st">&quot;Which mark is harldy found in intergenic regions?&quot;</span>,</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>),</span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>),</span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>),</span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">allow_retry =</span> T),</span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a>     <span class="fu">question</span>(<span class="st">&quot;For which three marks 50% or more of the peaks are overlapping with promoters, according to these analysis?&quot;</span>,</span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K4me1&quot;</span>, <span class="at">correct =</span> T)</span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>),</span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>),</span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>),</span>
<span id="cb106-14"><a href="#cb106-14" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>, <span class="at">correct =</span> T)</span>
<span id="cb106-15"><a href="#cb106-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">allow_retry =</span> T,</span>
<span id="cb106-16"><a href="#cb106-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">random_answer_order =</span> T),</span>
<span id="cb106-17"><a href="#cb106-17" aria-hidden="true" tabindex="-1"></a>     <span class="fu">question</span>(<span class="st">&quot;Which 2 marks have the 25% or more of peaks overlapping with intergenic regions?&quot;</span>,</span>
<span id="cb106-18"><a href="#cb106-18" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>),</span>
<span id="cb106-19"><a href="#cb106-19" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K4me1&quot;</span>)</span>
<span id="cb106-20"><a href="#cb106-20" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>),</span>
<span id="cb106-21"><a href="#cb106-21" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb106-22"><a href="#cb106-22" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb106-23"><a href="#cb106-23" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>)</span>
<span id="cb106-24"><a href="#cb106-24" aria-hidden="true" tabindex="-1"></a>              <span class="at">allow_retry =</span> T,</span>
<span id="cb106-25"><a href="#cb106-25" aria-hidden="true" tabindex="-1"></a>              <span class="at">random_answer_order =</span> T)</span></code></pre></div>
<p>Realize that this transcript-oriented analyses gives you slightly different results compared to the gene-centered one in exercise 8a and 8b. For now, we will stick to the gene-centered values and look at how the presence of histone marks at the TSS is related to the expression of the downstream gene.</p>
</div>
<div id="what-are-the-genes-with-a-h3k4me3-peak-in-their-promoter" class="section level3">
<h3>2.4.5 What are the genes with a H3K4me3 peak in their promoter?</h3>
<p>To answer this question, we first need to obtain the genes associated with the promoters that have a H3K4me3 peak. To do so, we can use the indexes from the overlap object (exercise 6) to subset the original <code>promoters_chr19</code> object and extract the corresponding geneIDs.</p>
</div>
<div id="exercise-9" class="section level3">
<h3>Exercise 9</h3>
<blockquote>
<p><strong>Exercise 9</strong>: How many different genes are associated withthe H3K4me3-overlapping promoters?</p>
</blockquote>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="co"># the objects `overlap` and `promoters_chr19` are loaded for you </span></span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a><span class="co"># subset for promoters that are part of the overlap with H3K4me3 in monocytes </span></span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>promoters_with_h3k4me3  <span class="ot">&lt;-</span> </span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a><span class="co"># convert the metadata of this object into a dataframe</span></span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a><span class="co"># count the length of unique geneIDs using summarize() and ndistinct()</span></span>
<span id="cb107-7"><a href="#cb107-7" aria-hidden="true" tabindex="-1"></a><span class="fu">as.data.frame</span>(<span class="fu">mcols</span>(promoters_with_h3k4me3)) <span class="sc">%&gt;%</span> </span>
<span id="cb107-8"><a href="#cb107-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">___</span>(<span class="at">unique_geneids =</span> ___ )  </span></code></pre></div>
<div id="q9a_genesids_overlap-hint">
<p>Use the ‘gene_id’ column in the object <code>promoters_chr19</code></p>
</div>
<div id="q9a_genesids_overlap-hint">
<p>Use <code>subjectHits(overlap)</code> to subset <code>promoters_chr19</code></p>
</div>
<div id="q9a_genesids_overlap-hint">
<p><code>promoters_chr19[subjectHits(overlap)]</code></p>
</div>
<div id="q9a_genesids_overlap-hint">
<p>To get a vector of geneIDs, you need to</p>
</div>
<div id="q9a_genesids_overlap-hint">
<p><code>length(unique(promoters_with_h3k4me3[,1]))</code></p>
</div>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="co"># the objects `overlap` and `promoters_chr19` are loaded for you </span></span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a><span class="co"># subset for promoters that are part of the overlap with H3K4me3 in monocytes </span></span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a>promoters_with_h3k4me3 <span class="ot">&lt;-</span> promoters_chr19[<span class="fu">subjectHits</span>(overlap)]</span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a><span class="co"># count the length of unique geneIDs in this object using summarize() and ndistinct()</span></span>
<span id="cb108-7"><a href="#cb108-7" aria-hidden="true" tabindex="-1"></a><span class="fu">as.data.frame</span>(<span class="fu">mcols</span>(promoters_with_h3k4me3)) <span class="sc">%&gt;%</span> </span>
<span id="cb108-8"><a href="#cb108-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">unique_geneids =</span> <span class="fu">n_distinct</span>(gene_id)) </span></code></pre></div>
<pre><code>##   unique_geneids
## 1           1150</code></pre>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_result</span>(</span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pass_if</span>(<span class="sc">~</span><span class="fu">identical</span>(.result,<span class="dv">1150</span>)))</span></code></pre></div>
</div>
</div>
<div id="comparing-gene-expression" class="section level2">
<h2>2.5 Comparing gene expression</h2>
<div id="genes-with-h3k4me3-at-their-promoter" class="section level3">
<h3>2.5.1 Genes with H3K4me3 at their promoter</h3>
<p>Can we observe a difference in gene expression between genes with and without H3K4me3 in their promoters?</p>
</div>
<div id="exericse-10-plot-the-expression-of-genes-with-and-without-h3k4me3-in-their-promoters" class="section level3">
<h3>Exericse 10: Plot the expression of genes with and without H3K4me3 in their promoters</h3>
<p>We obtained the gene expression quantification from the BLUEPRINT data portal <a href="http://dcc.blueprint-epigenome.eu/#/experiments/ERX157053">ref</a> and filtered it for genes present on chromosome 19. These data are loaded in object <code>quantification_chr19</code> for you.<br />
<br></p>
<blockquote>
<p><strong>Exercise 10a</strong>: first use dplyr::mutate(NEW_VARIABLE = ….) to define a new column <code>h3k4me3_promoter</code> which groups genes based on their overlap (TRUE/FALSE) with H3K4me3 peaks.</p>
</blockquote>
<p><br></p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="co"># The object quantification_chr19 holds the RNA-seq quantification for genes on chromosome 19. </span></span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a><span class="co"># View the structure to identify the variable you will use in mutate. </span></span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(quantification_chr19)</span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a new variable that identifies if a gene is present in the overlap or not.  </span></span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a>quantification_chr19_new <span class="ot">&lt;-</span> quantification_chr19 <span class="sc">%&gt;%</span> </span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">___</span>(<span class="at">h3k4me3_promoter =</span> ___ <span class="sc">%in%</span> <span class="fu">mcols</span>(promoters_with_h3k4me3)[,<span class="dv">1</span>] )</span></code></pre></div>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a><span class="co"># The object quantification_chr19 holds the RNA-seq quantification for genes on chromosome 19. </span></span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a><span class="co"># View the structure to identify the variable you will use in mutate. </span></span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(quantification_chr19)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    1635 obs. of  3 variables:
##  $ entrezgene_id: chr  &quot;126393&quot; &quot;115703&quot; &quot;5976&quot; &quot;56928&quot; ...
##  $ TPM          : num  0 0.16 4.54 3.35 7.19 0.03 0.01 0 2.86 1.33 ...
##  $ FPKM         : num  0 0.73 20.39 15.06 32.35 ...</code></pre>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a new variable that identifies if a gene is present in the overlap or not.  </span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>quantification_chr19_new <span class="ot">&lt;-</span> quantification_chr19 <span class="sc">%&gt;%</span> </span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">h3k4me3_promoter =</span> entrezgene_id <span class="sc">%in%</span> promoters_with_h3k4me3<span class="sc">$</span>gene_id )</span></code></pre></div>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_code</span>()</span></code></pre></div>
<blockquote>
<p><strong>Exercise 9b</strong>: Use ggplot() and geom_boxplot() to visualize the FPKM (y-axis) per gene group.</p>
</blockquote>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use ggplot() and geom_boxplot() to visualize the FPKM per gene group in the dataset `quantification_chr19_new`   </span></span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(___)<span class="sc">+</span></span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> __, <span class="at">y =</span> __))<span class="sc">+</span></span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_calc</span>()</span></code></pre></div>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(quantification_chr19_new)<span class="sc">+</span></span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> h3k4me3_promoter, <span class="at">y =</span> FPKM))<span class="sc">+</span></span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_calc</span>()</span></code></pre></div>
<p><img src="fg2_files/figure-html/q10b_plot_genequant-solution-1.png" width="672" /></p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_code</span>()</span></code></pre></div>
<p>This doesn’t look very informative. RNA-seq quantification has a high range. We must therefore log-transoform the y-axis.</p>
<blockquote>
<p><strong>Exercise 10c</strong>: Repeat the plot of 9b but lgo10-transform the y-axis.</p>
<ul>
<li>add 1 to all FPKM values to prevent log10(0).<br />
</li>
<li>also add another ‘layer’, visualizing the same data but as jitter (points).</li>
</ul>
</blockquote>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Repeat the plot of 9b but lgo10-transform the y-axis    </span></span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(quantification_chr19_new)<span class="sc">+</span></span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> h3k4me3_promoter, <span class="at">y =</span> (FPKM<span class="sc">+</span><span class="dv">1</span>)))<span class="sc">+</span></span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> h3k4me3_promoter, <span class="at">y =</span> (FPKM<span class="sc">+</span><span class="dv">1</span>)), <span class="at">point=</span> <span class="fl">0.5</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>)<span class="sc">+</span> <span class="co"># alpha makes the points transparant, ensuring that you still see the violin shape.  </span></span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_calc</span>()<span class="sc">+</span></span>
<span id="cb119-6"><a href="#cb119-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">&quot;___&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Repeat the plot of 9b but lgo10-transform the y-axis    </span></span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(quantification_chr19_new)<span class="sc">+</span></span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> h3k4me3_promoter, <span class="at">y =</span> (FPKM<span class="sc">+</span><span class="dv">1</span>)))<span class="sc">+</span></span>
<span id="cb120-5"><a href="#cb120-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> h3k4me3_promoter, <span class="at">y =</span> (FPKM<span class="sc">+</span><span class="dv">1</span>)), <span class="at">point=</span> <span class="fl">0.5</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>)<span class="sc">+</span> </span>
<span id="cb120-6"><a href="#cb120-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_calc</span>()<span class="sc">+</span></span>
<span id="cb120-7"><a href="#cb120-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">&quot;log10&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: Ignoring unknown parameters: point</code></pre>
<p><img src="fg2_files/figure-html/q10c_plot_genequant-solution-1.png" width="672" /></p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_code</span>()</span></code></pre></div>
<p>There is a clear difference in gene expression between the two gene sets with presence of H3K4me3 being associated with higher expression. However, H3K4me3 peak presence clearly does not explain everything.<br />
<br><br />
What if we also take H3K27me3 into account. What trend do we observe then?</p>
</div>
<div id="genes-with-h3k27me3-promoter" class="section level3">
<h3>2.5.2 Genes with H3K27me3 @ promoter</h3>
</div>
<div id="exercise-11-plot-the-expression-of-genes-with-and-without-h3k27me3-at-their-promoters" class="section level3">
<h3>Exercise 11: Plot the expression of genes with and without H3K27me3 at their promoters</h3>
<blockquote>
<p><strong>Exercise 11</strong>: To do this, you must</p>
<ol style="list-style-type: decimal">
<li>identify overlap between H3K27me3 peaks and promoters<br />
</li>
<li>obtain promoters that are reported in the overlap<br />
</li>
<li>define a new variable called “h3k27me3_promoter” in <code>quantification_chr19</code> that groups genes based on the overlap of their promoter with H3K27me3 peak. Instead of writing this into a new object, you can pipe the output data.frame immediately into step 4.<br />
</li>
<li>plot the FPKM per gene set.</li>
</ol>
<ul>
<li>Use the previous exercises if you need hints to code these steps.</li>
</ul>
</blockquote>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. identify the overlap between monocytes_h3k27me3 and promoters_chr19</span></span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>overlap_h3k27me3_promoters <span class="ot">&lt;-</span> <span class="fu">___</span>(<span class="at">query =</span> ___, <span class="at">subject =</span> ___)</span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-4"><a href="#cb123-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. obtain promoters that are reported in the overlap  </span></span>
<span id="cb123-5"><a href="#cb123-5" aria-hidden="true" tabindex="-1"></a>promoters_with_h3k27me3 <span class="ot">&lt;-</span> promoters_chr19[<span class="fu">___</span>(overlap_h3k27me3_promoters)]</span>
<span id="cb123-6"><a href="#cb123-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb123-7"><a href="#cb123-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. define a &quot;h3k27me3_promoter&quot; in `quantification_chr19`, pipe the output in your ggplot command  </span></span>
<span id="cb123-8"><a href="#cb123-8" aria-hidden="true" tabindex="-1"></a>quantification_chr19 <span class="sc">%&gt;%</span>  </span>
<span id="cb123-9"><a href="#cb123-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">___</span>(<span class="at">h3k27me3_promoter =</span> ____) <span class="sc">%&gt;%</span> </span>
<span id="cb123-10"><a href="#cb123-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(.) <span class="sc">+</span></span>
<span id="cb123-11"><a href="#cb123-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> ___, <span class="at">y =</span> (FPKM<span class="sc">+</span><span class="dv">1</span>)))<span class="sc">+</span></span>
<span id="cb123-12"><a href="#cb123-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> ___, <span class="at">y =</span> (FPKM<span class="sc">+</span><span class="dv">1</span>)), <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">size =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb123-13"><a href="#cb123-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_calc</span>()<span class="sc">+</span></span>
<span id="cb123-14"><a href="#cb123-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">&quot;log10&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. identify the overlap between monocytes_h3k27me3 and promoters_chr19</span></span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a>overlap_h3k27me3_promoters <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(<span class="at">query =</span> monocytes_h3k27me3, <span class="at">subject =</span> promoters_chr19)</span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. obtain promoters that are reported in the overlap  </span></span>
<span id="cb124-6"><a href="#cb124-6" aria-hidden="true" tabindex="-1"></a>promoters_with_h3k27me3 <span class="ot">&lt;-</span> promoters_chr19[<span class="fu">subjectHits</span>(overlap_h3k27me3_promoters)]</span>
<span id="cb124-7"><a href="#cb124-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-8"><a href="#cb124-8" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. define a &quot;h3k27me3_promoter&quot; in `quantification_chr19`, pipe the output in your ggplot command  </span></span>
<span id="cb124-9"><a href="#cb124-9" aria-hidden="true" tabindex="-1"></a>quantification_chr19 <span class="sc">%&gt;%</span>  </span>
<span id="cb124-10"><a href="#cb124-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">h3k27me3_promoter =</span> entrezgene_id <span class="sc">%in%</span> promoters_with_h3k27me3<span class="sc">$</span>gene_id) <span class="sc">%&gt;%</span> </span>
<span id="cb124-11"><a href="#cb124-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(.) <span class="sc">+</span></span>
<span id="cb124-12"><a href="#cb124-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> h3k27me3_promoter, <span class="at">y =</span> (FPKM<span class="sc">+</span><span class="dv">1</span>)))<span class="sc">+</span></span>
<span id="cb124-13"><a href="#cb124-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> h3k27me3_promoter, <span class="at">y =</span> (FPKM<span class="sc">+</span><span class="dv">1</span>)), <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">size =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb124-14"><a href="#cb124-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_calc</span>()<span class="sc">+</span></span>
<span id="cb124-15"><a href="#cb124-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">&quot;log10&quot;</span>)</span></code></pre></div>
<p><img src="fg2_files/figure-html/q11_H3K27me3_prom-solution-1.png" width="672" /></p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_code</span>()</span></code></pre></div>
<p>This time the trend is the other way around; presence of H3K27me3 is associated with lower gene expression. H3K4me3 and H3K27me3 can also co-occur. What gene expression level is associate with promoters that have both marks?</p>
</div>
<div id="bivalent-promoters" class="section level3">
<h3>2.5.3 Bivalent promoters</h3>
<p>This time you will make 4 gene sets, depending on the overlap of promoters with H3K4me3 and/or H3K27me3 peaks.</p>
</div>
<div id="exercise-12" class="section level3">
<h3>Exercise 12</h3>
<blockquote>
<p><strong>Exercise 12a</strong>: To do this, you must define a new “promoter_mark” in <code>quantification_chr19</code> that groups genes based in 4 groups of promoters based on them being or not being reported in ‘promoters_with_only_h3k4me3’ or ‘promoters_with_only_h3k27me3’. For this, we use <code>case_when()</code> within <code>mutate()</code> and define new category names based on the conditions as shown below.</p>
</blockquote>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. define the variable &quot;promoter_mark&quot; in `quantification_chr19`</span></span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a><span class="co"># define the level name between the &quot;&quot;-signs.  </span></span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a>quantification_chr19_new <span class="ot">&lt;-</span> quantification_chr19 <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(</span>
<span id="cb126-4"><a href="#cb126-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">promoter_mark =</span> <span class="fu">case_when</span>(</span>
<span id="cb126-5"><a href="#cb126-5" aria-hidden="true" tabindex="-1"></a>    entrezgene_id <span class="sc">%in%</span> promoters_with_h3k4me3<span class="sc">$</span>gene_id <span class="sc">&amp;</span> entrezgene_id <span class="sc">%in%</span> promoters_with_h3k27me3<span class="sc">$</span>gene_id <span class="sc">~</span> <span class="st">&quot;____&quot;</span>,  <span class="co"># promoters with both marks</span></span>
<span id="cb126-6"><a href="#cb126-6" aria-hidden="true" tabindex="-1"></a>    entrezgene_id <span class="sc">%in%</span> promoters_with_h3k4me3<span class="sc">$</span>gene_id <span class="sc">&amp;</span> <span class="sc">!</span>(entrezgene_id <span class="sc">%in%</span> promoters_with_h3k27me3<span class="sc">$</span>gene_id) <span class="sc">~</span> <span class="st">&quot;____&quot;</span>, <span class="co"># promoters with only h3k4me3</span></span>
<span id="cb126-7"><a href="#cb126-7" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span>(entrezgene_id <span class="sc">%in%</span> promoters_with_h3k4me3<span class="sc">$</span>gene_id) <span class="sc">&amp;</span> entrezgene_id <span class="sc">%in%</span> promoters_with_h3k27me3<span class="sc">$</span>gene_id <span class="sc">~</span> <span class="st">&quot;____&quot;</span>, <span class="co"># and so forth... </span></span>
<span id="cb126-8"><a href="#cb126-8" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span>(entrezgene_id <span class="sc">%in%</span> promoters_with_h3k4me3<span class="sc">$</span>gene_id <span class="sc">|</span> entrezgene_id <span class="sc">%in%</span> promoters_with_h3k27me3<span class="sc">$</span>gene_id) <span class="sc">~</span> <span class="st">&quot;____&quot;</span>) ) </span>
<span id="cb126-9"><a href="#cb126-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-10"><a href="#cb126-10" aria-hidden="true" tabindex="-1"></a><span class="co"># print a table with the number of promoters per promoter mark</span></span></code></pre></div>
<div id="q12_bivalent_prom-hint">
<p>An example of level names would be:<br />
<code>quantification_chr19 %&gt;%  mutate(promoter_mark = case_when(</code> <code>entrezgene_id %in% promoters_with_h3k4me3$gene_id &amp; entrezgene_id %in% promoters_with_h3k27me3$gene_id ~ "bivalent",</code> <code>entrezgene_id %in% promoters_with_h3k4me3$gene_id &amp; !(entrezgene_id %in% promoters_with_h3k27me3$gene_id) ~ "H3K4me3_only",</code> <code>!(entrezgene_id %in% promoters_with_h3k4me3$gene_id) &amp; entrezgene_id %in% promoters_with_h3k27me3$gene_id ~ "H3K27me3_only",</code> <code>!(entrezgene_id %in% promoters_with_h3k4me3$gene_id | entrezgene_id %in% promoters_with_h3k27me3$gene_id) ~ "no_mark") )</code></p>
</div>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a>quantification_chr19_new <span class="ot">&lt;-</span> quantification_chr19 <span class="sc">%&gt;%</span>  <span class="fu">mutate</span>(<span class="at">promoter_mark =</span> <span class="fu">case_when</span>(</span>
<span id="cb127-3"><a href="#cb127-3" aria-hidden="true" tabindex="-1"></a>    entrezgene_id <span class="sc">%in%</span> promoters_with_h3k4me3<span class="sc">$</span>gene_id <span class="sc">&amp;</span> entrezgene_id <span class="sc">%in%</span> promoters_with_h3k27me3<span class="sc">$</span>gene_id <span class="sc">~</span> <span class="st">&quot;bivalent&quot;</span>, </span>
<span id="cb127-4"><a href="#cb127-4" aria-hidden="true" tabindex="-1"></a>    entrezgene_id <span class="sc">%in%</span> promoters_with_h3k4me3<span class="sc">$</span>gene_id <span class="sc">&amp;</span> <span class="sc">!</span>(entrezgene_id <span class="sc">%in%</span> promoters_with_h3k27me3<span class="sc">$</span>gene_id) <span class="sc">~</span> <span class="st">&quot;H3K4me3_only&quot;</span>, </span>
<span id="cb127-5"><a href="#cb127-5" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span>(entrezgene_id <span class="sc">%in%</span> promoters_with_h3k4me3<span class="sc">$</span>gene_id) <span class="sc">&amp;</span> entrezgene_id <span class="sc">%in%</span> promoters_with_h3k27me3<span class="sc">$</span>gene_id <span class="sc">~</span> <span class="st">&quot;H3K27me3_only&quot;</span>, </span>
<span id="cb127-6"><a href="#cb127-6" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span>(entrezgene_id <span class="sc">%in%</span> promoters_with_h3k4me3<span class="sc">$</span>gene_id <span class="sc">|</span> entrezgene_id <span class="sc">%in%</span> promoters_with_h3k27me3<span class="sc">$</span>gene_id) <span class="sc">~</span> <span class="st">&quot;no_mark&quot;</span>) ) </span>
<span id="cb127-7"><a href="#cb127-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-8"><a href="#cb127-8" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(quantification_chr19_new<span class="sc">$</span>promoter_mark)</span></code></pre></div>
<pre><code>## 
## H3K27me3_only  H3K4me3_only      bivalent       no_mark 
##           203           791           324           317</code></pre>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_result</span>(</span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pass_if</span>(<span class="sc">~</span><span class="fu">identical</span>(.result,<span class="fu">table</span>(quantification_chr19_new<span class="sc">$</span>promoter_mark)))</span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<blockquote>
<p><strong>Exercise 12b</strong>: plot the gene expression as log10(FPKM+1) in a boxplot with extra geom_jittter, per group of genes.</p>
</blockquote>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pipe the output into your ggplot</span></span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(___) <span class="sc">+</span></span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> ___, <span class="at">y =</span> (FKPM<span class="sc">+</span><span class="dv">1</span>)))<span class="sc">+</span></span>
<span id="cb130-4"><a href="#cb130-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> ___, <span class="at">y =</span> (FKPM<span class="sc">+</span><span class="dv">1</span>)), <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">size =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb130-5"><a href="#cb130-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_calc</span>()<span class="sc">+</span></span>
<span id="cb130-6"><a href="#cb130-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">&quot;log10&quot;</span>)</span></code></pre></div>
<div id="q12b_bivalent_prom-hint">
<p><code>ggplot(quantification_chr19_new)+ geom_boxplot(mapping = aes(x = promoter_mark, y = (FPKM+1)))+</code> <code>geom_jitter(mapping = aes(x = promoter_mark, y = (FPKM+1)), alpha = 0.8, size = 0.5)+</code> <code>theme_calc()+</code> `scale_y_continuous(trans = “log10”)``</p>
</div>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(quantification_chr19_new) <span class="sc">+</span></span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> promoter_mark, <span class="at">y =</span> (FPKM<span class="sc">+</span><span class="dv">1</span>)))<span class="sc">+</span></span>
<span id="cb131-4"><a href="#cb131-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> promoter_mark, <span class="at">y =</span> (FPKM<span class="sc">+</span><span class="dv">1</span>)), <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">size =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb131-5"><a href="#cb131-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_calc</span>()<span class="sc">+</span></span>
<span id="cb131-6"><a href="#cb131-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">&quot;log10&quot;</span>)</span></code></pre></div>
<p><img src="fg2_files/figure-html/q12b_bivalent_prom-solution-1.png" width="672" /></p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Which of the following statments are correct?&quot;</span>,</span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;There are more promoters with only H3K4me3 than with H3K4me3 combined with another mark.&quot;</span>, <span class="at">message =</span> <span class="st">&quot;We have to evaluate more marks to determine whether H3K4me3 is indeed found more often by itself at a promoter than combined with another mark.&quot;</span>),</span>
<span id="cb132-3"><a href="#cb132-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K27me3 is not doing anything as the associated genes have the same expression level as genes without marks.&quot;</span>),</span>
<span id="cb132-4"><a href="#cb132-4" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;Gene expression is associated with differnt histone mark (combinations).&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb132-5"><a href="#cb132-5" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K4me3 defines promoters.&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K4me3 does not define promoters. Promoters are defined as the regions near the TSS.&quot;</span>),</span>
<span id="cb132-6"><a href="#cb132-6" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;There are more promoters with chromatin markings than without, in chromosome 19 of monocytes.&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb132-7"><a href="#cb132-7" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K4me3 is more often found at promoters than H3K27me3, in chromosome 19 of monocytes.&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb132-8"><a href="#cb132-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">allow_retry =</span> T,</span>
<span id="cb132-9"><a href="#cb132-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">random_answer_order =</span> T</span>
<span id="cb132-10"><a href="#cb132-10" aria-hidden="true" tabindex="-1"></a>         )</span></code></pre></div>
<p><strong>So far we have only looked at individual overlaps but chromatin states are actually defined by the combinatorial occupancy of different marks and factors. Let’s analyse the chromosome19-wide overlap of the marks in our dataset and identify the occuring combinations.</strong></p>
</div>
</div>
<div id="summarizing-overlap-among-marks" class="section level2">
<h2>2.6 Summarizing overlap among marks</h2>
<p>You will make an UpSet plot, which is an elegant alternative to the VennDiagram, that visualizes the size of each detected combination of histone marks. Combination in our case refers to overlap. This <strong>size</strong> of the combined set can be defined in various ways, depending on how you compare your regions. In the <code>ComplexHeatmap</code> package they define three ‘modes’ for it: distinct, intersect and union. You will work with the results from the setting ‘distinct’. Below is a visual summary of these modes.</p>
<p><img src="images/wk2_4_complexheatmap_modes.png" />/<br />
<a href="https://jokergoo.github.io/ComplexHeatmap-reference/book/upset-plot.html#input-data">Ref</a>: <em>Zuguang Gu, ComplexHeatmap Complete Reference, last revised on 2020-10-27</em><br />
<br> The <code>make_comb_mat([list_of_grangesobjects])</code> is used to detect overlapping marks:</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>monocytes_combinationmatrix <span class="ot">&lt;-</span> <span class="fu">make_comb_mat</span>(monocytes_list) <span class="co"># takes a while! </span></span></code></pre></div>
<p><code>make_comb_mat()</code> returns a “combination matrix” object. Special functions can be used to view its information and filter the reported combinations:</p>
<ul>
<li>set_name() prints the names of the objects from the original list<br />
</li>
<li>set_size() prints the total number of bps covered by the intervals in each object<br />
</li>
<li>comb_size() reports the total number of bps covered by the respective combination of intervals.
<ul>
<li>The combinations are encoded in a binary fashion of 0s and 1s.<br />
</li>
<li>A 1 means the mark in question is present in the combination.<br />
</li>
<li>These 0s and 1s follow the order given by set_name().</li>
</ul></li>
</ul>
<div id="exericse-13a" class="section level3">
<h3>Exericse 13a</h3>
<blockquote>
<p><strong>Exercise 13a</strong>: Examine the <code>monocytes_combinationmatrix</code> object with the obove mentioned functions.</p>
</blockquote>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="co"># What are the names of the sets?</span></span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a><span class="fu">__</span>(monocytes_combinationmatrix)</span>
<span id="cb134-3"><a href="#cb134-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-4"><a href="#cb134-4" aria-hidden="true" tabindex="-1"></a><span class="co"># What are the sizes of the sets? (in million bps)</span></span>
<span id="cb134-5"><a href="#cb134-5" aria-hidden="true" tabindex="-1"></a><span class="fu">___</span>(monocytes_combinationmatrix)<span class="sc">/</span><span class="fl">1e-6</span></span>
<span id="cb134-6"><a href="#cb134-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-7"><a href="#cb134-7" aria-hidden="true" tabindex="-1"></a><span class="co"># What are the sizes of the detected set combinations? (in million bps)</span></span>
<span id="cb134-8"><a href="#cb134-8" aria-hidden="true" tabindex="-1"></a><span class="fu">___</span>(monocytes_combinationmatrix)<span class="sc">/</span><span class="fl">1e-6</span></span></code></pre></div>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a><span class="co"># What are the names of the sets?</span></span>
<span id="cb135-3"><a href="#cb135-3" aria-hidden="true" tabindex="-1"></a><span class="fu">set_name</span>(monocytes_combinationmatrix)</span></code></pre></div>
<pre><code>## [1] &quot;h3k4me1&quot;  &quot;h3k4me3&quot;  &quot;h3k9me3&quot;  &quot;h3k27ac&quot;  &quot;h3k27me3&quot; &quot;h3k36me3&quot;</code></pre>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="co"># What are the sizes of the sets? (in million bps)</span></span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set_size</span>(monocytes_combinationmatrix)<span class="sc">/</span><span class="fl">1e-6</span></span></code></pre></div>
<pre><code>##      h3k4me1      h3k4me3      h3k9me3      h3k27ac     h3k27me3     h3k36me3 
## 8.671082e+12 3.401660e+12 1.194427e+13 2.229872e+12 5.946197e+12 1.122584e+13</code></pre>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="co"># What are the sizes of the detected set combinations?  </span></span>
<span id="cb139-2"><a href="#cb139-2" aria-hidden="true" tabindex="-1"></a><span class="fu">comb_size</span>(monocytes_combinationmatrix)<span class="sc">/</span><span class="fl">1e-6</span></span></code></pre></div>
<pre><code>##       100000       010000       001000       000100       000010       000001 
## 3.275890e+12 4.856900e+10 9.053924e+12 9.591000e+09 3.933451e+12 7.657355e+12 
##       110000       101000       100100       100010       100001       011000 
## 1.110241e+12 8.048600e+10 3.517410e+11 5.158870e+11 1.065497e+12 2.177000e+09 
##       010100       010010       010001       001100       001010       001001 
## 1.315490e+11 8.376000e+09 2.013000e+09 4.940000e+08 7.743140e+11 1.733578e+12 
##       000101       000011       111000       110100       110010       110001 
## 1.204810e+11 1.749000e+10 3.577100e+10 1.221020e+12 4.414880e+11 1.206580e+11 
##       101100       101010       101001       100110       100101       100011 
## 9.082000e+09 3.764100e+10 7.475000e+09 1.211000e+09 1.647000e+11 7.222000e+09 
##       011100       011010       010101       001101       001011       111100 
## 1.266000e+09 3.391000e+09 5.178800e+10 7.910000e+08 1.326770e+11 1.015500e+10 
##       111010       111001       110110       110101       110011       101110 
## 5.677000e+10 5.210000e+08 1.186500e+10 1.407130e+11 7.340000e+08 2.240000e+08 
##       101101       101011       111110       111101       110111       101111 
## 5.330000e+08 8.890000e+08 1.941000e+09 1.010000e+08 5.530000e+08 7.300000e+07</code></pre>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quiz</span>(<span class="at">caption =</span><span class="st">&quot;&quot;</span>, </span>
<span id="cb141-2"><a href="#cb141-2" aria-hidden="true" tabindex="-1"></a>     <span class="fu">question</span>(<span class="st">&quot;Which mark covers most bps?&quot;</span>,</span>
<span id="cb141-3"><a href="#cb141-3" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>),</span>
<span id="cb141-4"><a href="#cb141-4" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>),</span>
<span id="cb141-5"><a href="#cb141-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>),</span>
<span id="cb141-6"><a href="#cb141-6" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>, <span class="at">correct =</span> T, <span class="at">message =</span> <span class="st">&quot;other marks have more peaks, but this marks spans broad domains.&quot;</span>),</span>
<span id="cb141-7"><a href="#cb141-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">allow_retry =</span> T),</span>
<span id="cb141-8"><a href="#cb141-8" aria-hidden="true" tabindex="-1"></a>     <span class="fu">question</span>(<span class="st">&quot;And which combination of marks apparantly covers most bps?&quot;</span>,</span>
<span id="cb141-9"><a href="#cb141-9" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K4me3 and H3K27me3&quot;</span>),</span>
<span id="cb141-10"><a href="#cb141-10" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K9me3 and H3K27me3&quot;</span>),</span>
<span id="cb141-11"><a href="#cb141-11" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K9me3 and H3K4me3&quot;</span>),</span>
<span id="cb141-12"><a href="#cb141-12" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;H3K9me3 and H3K36me3&quot;</span>, <span class="at">correct =</span> T, <span class="at">message =</span> <span class="st">&quot;These two marks generally cover large windows in the genome. Their co-occupancy has been associated weakly transcribed genes. A large domain of H3K36me3 H3K9me3 bivalent chromatin is found at the cluster of zinc finger genes on chromosome 19 we saw in week1. (ref: </span></span>
<span id="cb141-13"><a href="#cb141-13" aria-hidden="true" tabindex="-1"></a><span class="st">                     Mauser et al. 2017 Epigentics and Chromatin)&quot;</span>),</span>
<span id="cb141-14"><a href="#cb141-14" aria-hidden="true" tabindex="-1"></a>              <span class="at">allow_retry =</span> T,</span>
<span id="cb141-15"><a href="#cb141-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">random_answer_order =</span> T)</span>
<span id="cb141-16"><a href="#cb141-16" aria-hidden="true" tabindex="-1"></a>     )</span></code></pre></div>
<p>The combination H3K4me1 and H3K4me3 is the 3rd most widespread combination. This combination might well represent signal due to some cross-reactivity of the antibody (-ies) or co-chipping of these regions because of enhancer-promoter pairing. It would help if we could also look at the signal besides its presence/absence. Will do so in week3.</p>
</div>
<div id="exericse-13b" class="section level3">
<h3>Exericse 13b</h3>
<blockquote>
<p><strong>Exercise 13b</strong>: Filter <code>monocytes_combinationmatrix</code> for combinations of 100kb and more. And plot the filtered combination matrix with <code>UpSet()</code>.</p>
</blockquote>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="co"># filter monocytes_combinationmatrix for combinations covering at least 100kb</span></span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a>monocytes_combmatrix_filt <span class="ot">&lt;-</span> monocytes_combinationmatrix[<span class="fu">comb_size</span>(monocytes_combinationmatrix) <span class="sc">&gt;</span> ___ ]</span>
<span id="cb142-3"><a href="#cb142-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-4"><a href="#cb142-4" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the result in an upset plot:  </span></span>
<span id="cb142-5"><a href="#cb142-5" aria-hidden="true" tabindex="-1"></a><span class="fu">___</span>(monocytes_combmatrix_filt)</span></code></pre></div>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="co"># solution</span></span>
<span id="cb143-2"><a href="#cb143-2" aria-hidden="true" tabindex="-1"></a><span class="co"># filter monocytes_combinationmatrix for combinations covering at least 200kb</span></span>
<span id="cb143-3"><a href="#cb143-3" aria-hidden="true" tabindex="-1"></a>monocytes_combmatrix_filt <span class="ot">&lt;-</span> monocytes_combinationmatrix[<span class="fu">comb_size</span>(monocytes_combinationmatrix) <span class="sc">&gt;</span> <span class="dv">100000</span>]</span>
<span id="cb143-4"><a href="#cb143-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-5"><a href="#cb143-5" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the result in an upset plot:  </span></span>
<span id="cb143-6"><a href="#cb143-6" aria-hidden="true" tabindex="-1"></a><span class="fu">UpSet</span>(monocytes_combmatrix_filt)</span></code></pre></div>
<p><img src="fg2_files/figure-html/q12b_upsetrplot-solution-1.png" width="672" /></p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grade_code</span>()</span></code></pre></div>
</div>
<div id="exericse-13c" class="section level3">
<h3>Exericse 13c</h3>
<blockquote>
<p><strong>Exercise 13c</strong>: Run the code below, which changes the labels at the top and rightside.</p>
</blockquote>
<p><img src="fg2_files/figure-html/q13c_niceupset-1.png" width="672" /></p>
<p>This makes the plot a bit better readable. You can also add additional plots to the top and bottom of this graph.</p>
</div>
<div id="exericse-13d" class="section level3">
<h3>Exericse 13d</h3>
<blockquote>
<p><strong>Exercise 13d</strong>: Run the code below, calculating the distance to the nearest TSS and plotting those values in the</p>
</blockquote>
<p>First we filter the original combination matrix and then extract all the matrices into a GRanges object. The <code>lapply()</code> command returns a list of these GRanges objects.</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="co"># filter monocytes_combinationmatrix for combinations covering at least 200kb</span></span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a>monocytes_combmatrix_filt <span class="ot">&lt;-</span> monocytes_combinationmatrix[<span class="fu">comb_size</span>(monocytes_combinationmatrix) <span class="sc">&gt;</span> <span class="dv">100000</span>]</span>
<span id="cb145-3"><a href="#cb145-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-4"><a href="#cb145-4" aria-hidden="true" tabindex="-1"></a><span class="co"># extract different combinations they are turned into a list of GRanges objects  </span></span>
<span id="cb145-5"><a href="#cb145-5" aria-hidden="true" tabindex="-1"></a>monocytes_combmatrix_sets <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">comb_name</span>(monocytes_combmatrix_filt), <span class="cf">function</span>(nm) <span class="fu">extract_comb</span>(monocytes_combmatrix_filt, nm))</span></code></pre></div>
<p><img src="fg2_files/figure-html/q13d_add_distance_to_tss-1.png" width="672" /></p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a> <span class="fu">question</span>(<span class="st">&quot;Use the plot above, which statements do you agree with?&quot;</span>,</span>
<span id="cb146-2"><a href="#cb146-2" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;There are more promoters with only H3K4me3 than with H3K4me3 combined with another mark.&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Based on these marks, we would say no. Compare with the output from question 11. Taking more marks along let us see that H3K4me3 only is sparse.&quot;</span>),</span>
<span id="cb146-3"><a href="#cb146-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;Column I (letters at the bottom), marks promoters&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb146-4"><a href="#cb146-4" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K4me3 defines promoters.&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K4me3 does not define promoters. Promoters are defined as the regions near the TSS.&quot;</span>),</span>
<span id="cb146-5"><a href="#cb146-5" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;Column M marks the second-most widespread combination and could be promoters or enhancers.&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb146-6"><a href="#cb146-6" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;Column D, with H3K36me3 only marking, likely represents silences genes&quot;</span>),</span>
<span id="cb146-7"><a href="#cb146-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">allow_retry =</span> T,</span>
<span id="cb146-8"><a href="#cb146-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">random_answer_order =</span> T</span>
<span id="cb146-9"><a href="#cb146-9" aria-hidden="true" tabindex="-1"></a>         )</span></code></pre></div>
</div>
</div>
<div id="so-far" class="section level2">
<h2>So far,…</h2>
<p>So far we have examined peak locations, overlap with features and the association between the absence or presence of a mark and gene expression. This binary approach is a very simplistic model and we have been ignoring the signal strength altogether. As in, <em>Are high and lowly expressed genes associated with equal levels of H3K4me3 at their promoter or not? And what about H3K36me3 across the gene body?</em>. The measure of strength is also taken along in the definition of different chromatin states, see for example this paper on the <a href="https://doi.org/10.1038/nmeth.1906">chromHMM tool</a>. Positional and signal information are combined to learn these states. Next week you will examine the relationship between signal strength and gene expression.</p>
<p>But what also should be said, you have come very far already. Very well done, keep up the good work!!</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
