<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Written by: Klaas, Simon, Siebren, Christa …" />


<title>Functional Genomics 1 practical 2020</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="custom.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Functional Genomics 2020</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="p2_learningObjectives.html">Learning Objectives</a>
</li>
<li>
  <a href="p4_week1.html">UCSC Genome Browser</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Functional Genomics 1 practical 2020</h1>
<h4 class="author">Written by: Klaas, Simon, Siebren, Christa …</h4>

</div>


<div class="row" style="padding-top: 20px;">



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This is the computer practical for the <a href="https://www.ru.nl/courseguides/science/vm/osirislinks/bb/nwi-bb064b/">Functional Genomics 1</a> course. It builds upon R programming skills taught in the first year during <a href="https://www.ru.nl/courseguides/science/vm/osirislinks/bp/nwi-bp031/">Genomics and Big Data</a> and a basic understanding of the R programming language is required. We will also use statistics from the course <a href="https://www.ru.nl/courseguides/science/vm/osirislinks/bp/nwi-bp012c/">Statistics 1</a>.</p>
<p>The <strong>purpose</strong> of this practical is to teach you common bio-informatics analyses performed to analyse epigenomics data. You may already be familiar with the output of these analyses from scientific publications or lectures. We will focus on <strong>how to detect, quantify, annotate and visualise epigenetic marks</strong>. At the end of our analyses, we will also relate epigenetic markings to gene activity as well as candidate TFs that might establish them.</p>
<p>For this practical, we will use <strong>data for white blood cells</strong> from the <a href="http://dcc.blueprint-epigenome.eu/#/home">the BLUEPRINT project</a>, namely: monocytes, T cells, B cells and natural killer cells. BLUEPRINT is a <em>FP7 project aiming to produce a blueprint of haemopoetic epigenomes.</em> We have chosen these datasets because they are well-documenten and easy to find through the BLUEPRINT data portal, and not because we want to teach you the hematopoietic system.<br />
<br></p>
<blockquote>
<h3 id="research-goals">Research goals</h3>
<ol style="list-style-type: decimal">
<li>Use combinations of histone PTM data and chromatin accessibility to identify functional genomic elements (eg promoter, enhancers etc) and predict gene activity (active or repressed).<br />
</li>
<li>Check obove mentioned predictions on gene activity in single-cell RNA-seq data.<br />
</li>
<li>Vice versa, use the epigenetic data to explain gene expression differences detected in a single-cell RNA-seq experiment.<br />
</li>
<li>Predict TFs that could mediate cell-specific gene expression.</li>
</ol>
</blockquote>
<p><br></p>
<blockquote>
<h3 id="global-learning-objectives">Global learning objectives</h3>
<ol style="list-style-type: decimal">
<li>Understand how useful data is obtained from a ChIP-sequencing experiment.<br />
</li>
<li>Know how to display and browse through genomics data in the UCSC Genome Browser and formulate some basic hypotheses using this tool.<br />
</li>
<li>Tell apart different genomic datasets based on their pattern int he UCSC Genome Browser.<br />
</li>
<li>Quantify and visually summarize the genomic distribution of histone ChIP-seq peaks and signal strength in r.<br />
</li>
<li>Statistically test for enrichment of histone marks in functional genomic elements.<br />
</li>
<li>Summarize genome-wide ChIP-seq signal in a heatmap and use this heatmap to identify identify functional genomic elements and predict gene activity.<br />
</li>
<li>Check hypotheses about the epigenetic landscape based on single-cell RNA-seq data with actual epigenomics data, and vice versa (check hypotheses on gene activity in single-cel RNA-seq data).<br />
</li>
<li>Formulate hypotheses about (cell-specific) TF function using histone ChIP-seq and ATAC-seq data.</li>
</ol>
</blockquote>
<p><a href = "https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001046"> <img src="images/plosbiol_2011_ENCODE_userguid_fig1a.png" style="width:80.0%" /> </a><br />
<a href="https://doi.org/10.1371/journal.pbio.1001046">The ENCODE Project Consortium, 2011, PLoS Biology</a><br />
<br></p>
</div>
<div id="structure-of-this-practical" class="section level2">
<h2>Structure of this practical</h2>
<p><strong>Week 1</strong>: Raw ChIP-seq data processing and visualize epigenomics data in the UCSC Genome browser.<br />
Global learning objectives #1-3.</p>
<p><strong>Week 2</strong>: Quantify the (co-)occurrences of epigenetic marks and use this to annotate genomic regions.<br />
Global learning objectives #4-5.</p>
<p><strong>Week 3</strong>: Compare epigenetic profiles among different cell types.<br />
Global learning objectives #5-6.</p>
<p><strong>Week 4</strong>: Synthesis: epigenomics, scRNA-seq and motif finding.<br />
Global learning objectives #7-8.<br />
<br></p>
</div>
<div id="need-help" class="section level2">
<h2>Need help?</h2>
<p>Use the virtual classroom space in Brightspace/Zoom meeting when you need advice. A lecturer will be present there during practical hours.</p>
<p>Additional links to resoures can be found under <a href="#Resources">links</a> <br></p>
</div>
<div id="computational-environment" class="section level2">
<h2>Computational environment</h2>
<p>For this practical it is not required to install R and R studio. Instead, we provide you with the correct environment and required packages. In case you want to repeat the practical at home/on your own computer you should make sure you install the following packages:</p>
<ul>
<li>tidyverse ..</li>
<li>blabla <br></li>
</ul>
</div>
<div id="background-info" class="section level2">
<h2>Background info</h2>
<div id="using-r" class="section level3">
<h3>Using R</h3>
<p>We will code in R, which you have already familiarized yourself with during the Genomics &amp; Big Data course.</p>
<div id="what-does-an-r-package-or-function-do" class="section level5">
<h5>What does an R package or function do?</h5>
<ul>
<li>Use the built-in RStudio help interface to find a particular function<br />
</li>
<li>Googling <em>[function or package name] ‘r’</em><br />
</li>
<li>Go to the ‘Packages’ tab to find user guides and other documentation on the package<br />
</li>
<li>Look into the ‘Help’ tab of the function</li>
</ul>
</div>
<div id="i-get-an-error" class="section level5">
<h5>I get an error?</h5>
<ul>
<li>Read the error, might give you hints</li>
<li>Google the error + r<br />
</li>
<li><a href="https://stackoverflow.com">Stackoverflow</a> is great, Google will often take you there because someone has already asked your question, and someone else has answered, usually many people have answered your question many ways.<br />
<br></li>
</ul>
</div>
</div>
<div id="table-with-cell-markers-and-tfs" class="section level3">
<h3>Table with cell markers and TFs</h3>
<p>Here’s a table with cell surface proteins, cytokines and TFs that are specific to certain cell types.</p>
<p>Some of the cell surface markers were used to isolate these cells from the bulk using (MACS and) FACS.</p>
<table style="width:100%;">
<colgroup>
<col width="16%" />
<col width="16%" />
<col width="16%" />
<col width="16%" />
<col width="16%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th>Cell</th>
<th>Marker</th>
<th>Localization and/or function</th>
<th>Gene(s)</th>
<th>Also found on</th>
<th>Ref</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Monocytes</strong></td>
<td>CD14</td>
<td>plasma membrane</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=CD14">CD14</a></td>
<td>macrophages</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/24671955/">Schidl 2014 Blood</a></td>
</tr>
<tr class="even">
<td>,,</td>
<td>CCR2</td>
<td>plasma membrane</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=CCR2">CCR2</a></td>
<td>??</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/24671955/">Schidl 2014 Blood</a></td>
</tr>
<tr class="odd">
<td>,,</td>
<td>CD163</td>
<td>plasma membrane</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=CD163">CD163</a></td>
<td>??</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/24671955/">Schidl 2014 Blood</a></td>
</tr>
<tr class="even">
<td>,,</td>
<td>VCAN</td>
<td>extracellular space</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=VCAN">VCAN</a></td>
<td>??</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/24671955/">Schidl 2014 Blood</a></td>
</tr>
<tr class="odd">
<td>,,</td>
<td>S100A8</td>
<td>??</td>
<td>??</td>
<td>??</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/24671955/">Schidl 2014 Blood</a></td>
</tr>
<tr class="even">
<td>,,</td>
<td>S100A9</td>
<td>??</td>
<td>??</td>
<td>??</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/24671955/">Schidl 2014 Blood</a></td>
</tr>
<tr class="odd">
<td>,,</td>
<td>S100A12</td>
<td>??</td>
<td>??</td>
<td>??</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/24671955/">Schidl 2014 Blood</a></td>
</tr>
<tr class="even">
<td>,,</td>
<td>CEBPb</td>
<td>??</td>
<td>??</td>
<td>??</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/24671955/">Schidl 2014 Blood</a></td>
</tr>
<tr class="odd">
<td>,,</td>
<td>PU.1</td>
<td>nucleus, transcription factor</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=SPI1&amp;keywords=SPI1">SPI1</a></td>
<td>myeloid cells and B cells</td>
<td></td>
</tr>
<tr class="even">
<td><strong>NK cells</strong></td>
<td>CD16</td>
<td>plasma membrane</td>
<td>??</td>
<td>macrophages, neutrophils</td>
<td></td>
</tr>
<tr class="odd">
<td><strong>T cells</strong> (CD4+ and CD8+)</td>
<td>CD3</td>
<td>plasma membrane</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=CD3D">CD3-delta</a>, <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=CD3G">-gamma</a>, <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=CD3E">-epsilon</a></td>
<td>??</td>
<td></td>
</tr>
<tr class="even">
<td>,,</td>
<td>CD45RA</td>
<td>plasma membrane</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=PTPRC">PTPRC</a></td>
<td>B cells, NK cells, neutrophils, monocytes, …</td>
<td></td>
</tr>
<tr class="odd">
<td><strong>CD4+ T cells</strong></td>
<td>CD4</td>
<td>plasma membrane</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=CD4">CD4 gene</a></td>
<td>monocytes, macrophages, dendritic cells</td>
<td></td>
</tr>
<tr class="even">
<td>,,</td>
<td>ThPOK</td>
<td>nucleus, transcription factor</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=ZBTB7B">ZBTB7B</a></td>
<td>CD4+ T cells and B cells</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/29677476/">Taniuchi 2018 Ann Rev Immunology</a></td>
</tr>
<tr class="odd">
<td>,,</td>
<td>T-bet</td>
<td>nucleus, transcription factor</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=TBX21">TBX21</a></td>
<td>..</td>
<td>..</td>
</tr>
<tr class="even">
<td>,,</td>
<td>GATA-3</td>
<td>nucleus, transcription factor</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=GATA3">GATA3</a></td>
<td><strong>CD8 T cells??</strong></td>
<td>..</td>
</tr>
<tr class="odd">
<td>,,</td>
<td>RunX3</td>
<td>nucleus, transcription factor</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>,,</td>
<td>INF-gamma</td>
<td>cytokine, secreted</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=IFNG">INFG</a></td>
<td>..</td>
<td>..</td>
</tr>
<tr class="odd">
<td>,,</td>
<td>IL-4</td>
<td>cytokine, secreted</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=IL4">IL4</a></td>
<td>..</td>
<td>..</td>
</tr>
<tr class="even">
<td><strong>CD8+ T cells</strong></td>
<td>CD8A</td>
<td>plasma membrane</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=CD8A">CD8A</a></td>
<td>CD8+ T cells, B cells, NK cells</td>
<td>..scRNA-seq marker</td>
</tr>
<tr class="odd">
<td><strong>B cells</strong></td>
<td>PU.1</td>
<td>nucleus, transcription factor</td>
<td><a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=SPI1&amp;keywords=SPI1">SPI1</a></td>
<td>myeloid cells and B cells</td>
<td></td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
<div id="hematopoiesis-scheme" class="section level3">
<h3>Hematopoiesis scheme</h3>
<p><a href = "https://www.blueprint-epigenome.eu/UserFiles/File/cell%20types%20images%20etc/Haematopoiesis_original.pdf"> <img src="images/BP_Haematopoiesis_original.png" /> </a> <a href="https://www.blueprint-epigenome.eu/UserFiles/File/cell%20types%20images%20etc/Haematopoiesis_original.pdf">BLUEPRINT, hematopoiesis</a><br />
<br></p>
</div>
<div id="white-blood-cells-in-current-dataset" class="section level3">
<h3>White blood cells in current dataset</h3>
<p>Blood cells originate from common, multipotent hemopoietic stem cells in the bone marrow which divide to keep their population stocked or differentiate into more specific progenitors of the myeloid or lymphoid lineages. These progenitors undergo a limited number of divisions and differentiate along the way into mature blood cells: white blood cells, red blood cells and platelets.</p>
<p>White blood cells encompass a varied group of cells. Each type has its own developmental process controlled by cytokines, cytokine receptors, transcription factors and chromatin remodellers and with distinct morphologies and functions at maturity. Of all the different white blood cells, we will focus on monocytes, NK cells, T cells and B cells in our analyses.</p>
<p><strong>Monocytes</strong> make up 2-8% of the white blood cells, derive from the myeloid lineage and are part of the innate immune system (first line of defense, limited/general specificity to particular antigens and can’t multiply furhter). They can enter peripheral tissue and differentiate into macrophages which are good at engulfing larger particles but also in dendritic cells which are highly adapted to present foreign antigens to cells of the adaptive immune system.</p>
<p><strong>Natural killer cells</strong> or <strong>NK lymphocytes</strong> come from the lymphoid lineage. They are part of the innate immune system with high cytotoxic capabilities and an important first line of defense against virus infections and the formation of abnormal tissues such as cancers.</p>
<p><strong>T cells</strong> or <strong>T lymphocytes</strong> come from the lymphoid lineage as well and can recognize antigens presented by infected cells or special-antigen presenting cells using their T cell receptor. They can either kill the infected cell immediately - cytotoxic activity - or recruit and/or activate other immune cells including B cells - helper activity. These cytotoxic T lymphocytes are recognized by CD8 molecule on their membrane and helper T lymphocytes by CD4.</p>
<p><strong>B cells</strong> or <strong>B lymphocytes</strong> also come from the lymphoid lineage and are part of the adaptive (second line, cells differentiated to target specific antigens). Upon activation they differentiate into an antibody-producing plasma cell or a memory B cell.</p>
<p>Besides white blood cells we also find <strong>red blood cells</strong> and <strong>platelets</strong> in the blood. Both originate from the myeloid lineage and have lost their nucleus during development but contain RNA including mRNA. Mature red blood cells mediate transport of O2 and CO2 in the blood. Platelets originate from megakaryocytes in the bone marrow and release coagulation factors in response to a bleeding or injury of the blood vessel wall. <br></p>
</div>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<div id="resources-used-to-compile-this-practicaltutorial" class="section level3">
<h3>Resources used to compile this practical/tutorial</h3>
<ul>
<li><a href="https://vanheeringen-lab.github.io/seq2science/index.html">seq2science</a></li>
<li>BLUEPRINT project<br />
</li>
<li>…<br />
<br></li>
</ul>
</div>
</div>
<div id="old-text" class="section level2">
<h2>Old text</h2>
<ol style="list-style-type: decimal">
<li><p>We will use the <strong>UCSC Genome Browser</strong> to visualize the data, look at the distribution of the signal, formulate hypothesis/-es and - roughly - evaluate the quality of the experiment (of note: obvously, this is not the only quality check).</p></li>
<li>Next, we will transform our observations into numbers and thus <strong>quantify</strong> the signal intensity and distribution of the epigenetic marks we are looking at.
<ul>
<li>For this purpose we will use packages from the <a href="http://bioconductor.org/">Bioconductor</a> project, a extensive and cutting edge repository of R packages used for the analysis of high througput sequencing data.<br />
</li>
<li>We’ve limited these analyses to <span style="color: red;">chr21/protein coding genes/used random 5M reads instead of the original dataset (called <em>downsampling</em>)</span> to save running time on the server. More reads increases the power of the analyses and confidence of tests <span style="color: red;">but blabla</span>.</li>
</ul></li>
<li><p>Combine data from different marks and cells and link differences and profiles to functional consequences with respect to gene expression.</p></li>
</ol>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
