<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Written by: Klaas, Simon, Siebren, Christa …" />


<title>FG1 - Learning Objectives</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="custom.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Functional Genomics 2020</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="p2_learningObjectives.html">Learning Objectives</a>
</li>
<li>
  <a href="p3_script.html">Script</a>
</li>
<li>
  <a href="p4_week1.html">W1</a>
</li>
<li>
  <a href="p5_week2.html">W2</a>
</li>
<li>
  <a href="p6_intro_bioconductor.html">DC Bioconductor</a>
</li>
<li>
  <a href="p7_links.html">Links</a>
</li>
<li>
  <a href="p8_seurat_pbmc_tutorial.html">PBMC3k</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">FG1 - Learning Objectives</h1>
<h4 class="author">Written by: Klaas, Simon, Siebren, Christa …</h4>

</div>


<style type="text/css">
.title {
  display: none;
}
.author {
  display: none;
}

#getting-started img {
  margin-right: 10px;
}

</style>
<div class="row" style="padding-top:10px;">
<p><br></p>
<ul>
<li>Question-/Problem-driven learning –&gt; formulate candidate questions per week.<br />
</li>
<li>Example dataset: monocytes, T cells and B cells (and NK cells? and platelets?).
<ul>
<li>NK cells can become a question for the heatmap, what they most look like? Or is this going too much in biology already?<br />
</li>
</ul></li>
<li>Start in UCSC genome browser with one cell type (eg monocytes). Introduce same data of other cell types in week 2.<br />
<br></li>
</ul>
<div id="overall" class="section level2">
<h2>Overall</h2>
<blockquote>
<h4 id="goal-of-this-practical">Goal of this practical</h4>
<ul>
<li>Our aim is to identify clusters of differentially marked genes among various blood cells. <em>Importantly</em>, we want to know in what extend these clusters overlap (and could explain) differential gene expression detected in a related single-cell RNA-seq dataset.<br />
<br></li>
<li>We want to identify the major/significant epigenetic and transcriptional differences among varioud blood cells. How can we discriminate these cells at the epigenetically? Which TFs drive the expression of important marker genes of a given blood cell?</li>
</ul>
</blockquote>
<blockquote>
<h4 id="global-learning-objectives">Global learning objectives</h4>
<ol style="list-style-type: decimal">
<li>Understand how data are generated from a ChIP-sequencing experiment.<br />
</li>
<li>Know how to display and browse through genomics data in the UCSC Genome Browser and formulate some basic hypotheses based on observations using this tool.<br />
</li>
<li>Tell apart UCSC genome browser tracks of H3K27ac, H3K27me3, H3K36me3, H3K4me1 and H3K4me3 histone ChIP-seq and ATAC-seq signal. <span style="color: red;">and BS-seq, and input?</span><br />
</li>
<li>Understand and be able to navigate within the containters provided by <span style="color: red;">several…GenomicRanges, GenomicFeatures</span> Bioconductor packages.<br />
</li>
<li>Count, summarize/quantify and visualize histone ChIP-seq signal distribution and signal strength in R.<br />
</li>
<li>Summarize genome-wide ChIP-seq signal in heatmap and cluster this heatmap.<br />
</li>
<li>Assign regulatory elements to a gene (and be able to reflect on pros and cons of this procedure).<br />
</li>
<li>Summarize the overlap between marker genes from cell clusters detected in single-cell RNA-seq experiment to differences between cells at the epigenetic level.<br />
</li>
<li>Or, Identify DNA motifs and TFs that are active in the differentially marked regions.</li>
</ol>
</blockquote>
<hr />
</div>
<div id="week-1" class="section level2">
<h2>Week 1</h2>
<ul>
<li><em>seq2science, UCSC Genome browser</em><br />
</li>
<li>Global objectives #1-3.<br />
</li>
<li><p>Perform in H5P?<br />
<br></p></li>
<li>Discuss ChIP-seq in presentation<br />
</li>
<li>Follow seq2science workflow<br />
</li>
<li><p>Peakcalling: note that it is not black/white, not absolute, thresholds are set. NOT into technicalities and mathematics of read piling/peak models, narrow-wide peaks<br />
<br></p></li>
</ul>
<blockquote>
<h4 id="learning-objectives-after-this-tutorial-the-student-should-understandbe-able-to">Learning Objectives (<em>after this tutorial the student should understand/be able to…</em>)</h4>
<ol style="list-style-type: decimal">
<li>Know how ChIP-seq is performed experimentally (incl why &amp; what controls are needed).<br />
</li>
<li>Know how raw ChIP-seq read output is processed to obtain trustworthy data.<br />
</li>
<li>Understand how to mov (“<em>browsing</em>”) around in the UCSC genome browser and know how to go to regions and genes of interest.<br />
</li>
<li>Recognize the following elements/regions in the UCSC genome browser: <em>chromosome, centromere, telomere, gene, exon, intron, <span style="color: red;">utr</span>, promoter, enhancer/repressor</em>.<br />
</li>
<li>Can interpret - and thus understand - ChIP- and ATAC-seq signal in the UCSC genome browser.<br />
</li>
<li>Recognize and differentiate the profile of <em>ATAC-seq, RNA-seq and histone ChIP-seq</em> datasets in the genome browser based on their coverage profile and their location with respect to genes (H3K27ac, H3K27me3, H3K36me3, H3K4me1 and H3K4me3).</li>
</ol>
</blockquote>
<blockquote>
<h4 id="working-questions">Working questions</h4>
<ol style="list-style-type: decimal">
<li>We want to know what the peak (what histone mark) in the promoter of … [gene] …. is.<br />
</li>
<li>Find histone mark covering the enhancer located 20kb upstream of …. [gene] ….<br />
</li>
<li><em>Or</em> Now find the nearest upstream H3K27ac mark. Where is it located?<br />
</li>
<li><em>Summarizing</em> What marks are generally found at promoters, at gene bodies, active enhancers and at repressed domains?</li>
</ol>
</blockquote>
<hr />
</div>
<div id="week-2" class="section level2">
<h2>Week 2</h2>
<ul>
<li><em>Bioconductor, count peak distribution and comparison among marks</em><br />
</li>
<li>Global objectives #4-5.</li>
</ul>
<p>In week 1 we have estimated which mark is enriched in promoters, gene bodies or other intergenic regions. This week we will put a number to these estimates. More specifically,..<br />
<br></p>
<blockquote>
<h4 id="learning-objectives">Learning Objectives</h4>
<ol style="list-style-type: decimal">
<li>Move around in GRanges container.<br />
</li>
<li>Obtain annotation of your reference genome of interest and know how to extract the features important to you.<br />
</li>
<li>Find and count the overlap among histone ChIP-seq peaks and genomic elements or another peak set. (<code>countOverlaps(query, subject)</code>, <code>findOverlaps(query, subject)</code>)<br />
</li>
<li>Plot the quantification (ggplot + geom_bar).<br />
</li>
<li>Determine whether or not enrichment is significant (test?).</li>
</ol>
</blockquote>
<blockquote>
<h4 id="working-questions-1">Working questions</h4>
<ol style="list-style-type: decimal">
<li>In what genomic region do we find most of the H3K4me3 peaks? And is this a significant enrichment over the background (genome)?<br />
</li>
<li>And what if we look at H3K27ac or H3K36me3?<br />
</li>
<li>Do these numbers differ (significantly?) when we look at T cells, B cells or NK cells or memory T cells or macrophages?<br />
</li>
<li>What marks overlap and what marks hardly overlap? <span style="color: red;"> too vague, need idea of output</span></li>
</ol>
</blockquote>
<p><span style="color: red;">Would be nice to add a summarizing question that evokes thoughts about signal intensity (differences) as bridge for week 3. Like question 4 but better or improvement thereof.</span></p>
<hr />
</div>
<div id="week-3" class="section level2">
<h2>Week 3</h2>
<ul>
<li><em>Comparing coverage/signal intensity in CoveragePlot and heatmap among regions and cells</em><br />
</li>
<li>Global objectives #5-6</li>
<li>Heatmap using <span style="color: red;">package ngs.plot, EnrichedHeatmap, deepTools, ???</span><br />
</li>
<li>Use heatmap to compare marks or cells? Or both? How many columns?</li>
</ul>
<p>In week 1 we looked at the localization of histone marks (peaks) and we quantified this in week 2 of peaks. You might have seen that some histone marks show partial overlap with other marks. In addition, up till now we have worked with the presence or absence of <em>peaks</em>. This binary measure is not sufficient to summarize the diversity in <em>signal intensities</em> we observed before in the UCSC genome browser.</p>
<p><span style="color: red;">Go into singal intensity differences here and/or into different markings among cells? or both?</span><br />
<br></p>
<blockquote>
<h4 id="learning-objectives-1">Learning Objectives</h4>
<ol style="list-style-type: decimal">
<li>Be able to visually summarize read coverage (<em>signal</em>) of different histone marks around genomic landmarks (TSS or over gene body)<br />
</li>
<li>Understand and interpret clustered heatmap of histone marks and or cells. <span style="color: red;">Improve! How should heatmap look like?</span><br />
</li>
<li>How to assign histone ChIP-seq peaks to genes (and understand the assumptions/decisions you make when you do this).</li>
</ol>
</blockquote>
<blockquote>
<h4 id="working-questions-2">Working questions</h4>
<ol style="list-style-type: decimal">
<li>Does H3K36me3 signal start at the TSS or at the ATG/start codon?<br />
</li>
<li><em>Or</em> Does H3K36me3 signal increase or decrease downstream from ATG/start (vs upstream)?<br />
</li>
<li>Are H3K4me3 and H3K27ac signal higher or weaker directly upstream of the TSS compared to directly downstream of the TSS? Do these marks behave the same?<br />
</li>
<li>Do NK cells look more like monocytes or like T cells? Based on histone markings we have here.<br />
</li>
<li>We have this cluster of H3K4me3 + H3K27me3 marked promoters, what genes belong to these promoters?<br />
</li>
<li>The [TF] is essential for T cell phenotype. What cluster do you find this gene’s promotor?<br />
</li>
<li>Same for enhancers?</li>
</ol>
</blockquote>
<hr />
</div>
<div id="week-4" class="section level2">
<h2>Week 4</h2>
<ul>
<li><em>overlap marker genes scRNA-seq to detected epigenetic differences</em> or <em>find or count motif occurrences using pattern matching from GRanges</em><br />
</li>
<li>Use PBMC3k dataset<br />
</li>
<li><p>Get UMAP from Satija lab.<br />
<br></p></li>
<li>We continue with clusters that differentially mark T cells from monoyctes [..] and assign them to closest gene.<br />
</li>
<li>Open/active regions only?<br />
</li>
<li><p>Compare these to + and - marker genes from scRNA-seq</p></li>
</ul>
<blockquote>
<h4 id="learning-objectives-2">Learning Objectives</h4>
<ol style="list-style-type: decimal">
<li>Can interpret UMAP from scRNA-seq.lk<br />
</li>
<li>Understand the added value of scRNA-seq over bulk rna-seq (lecture not this workshop)<br />
</li>
<li>Compare gene lists obtained from the two techniques</li>
</ol>
</blockquote>
<blockquote>
<h4 id="working-questions-3">Working Questions</h4>
<ol style="list-style-type: decimal">
<li>We have this cluster of actively marked regions in monocytes. Are these genes indeed expressed?<br />
</li>
<li>Where in the heatmap do we find the marker genes back that were detected in the sc-RNA-seq experiment?<br />
</li>
<li>These TFs were found markers for resp. monocytes, T-cell, B-cells in the scRNA-seq experiment. Can these findings be supported by our ChIP-seq data?</li>
</ol>
</blockquote>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
