<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Written by: Klaas, Simon, Siebren, Christa …" />


<title>FG1 - Learning Objectives</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="custom.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Functional Genomics 2020</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="p2_learningObjectives.html">Learning Objectives</a>
</li>
<li>
  <a href="p4_week1.html">UCSC Genome Browser</a>
</li>
<li>
  <a href="Glossary.html">Glossary</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">FG1 - Learning Objectives</h1>
<h4 class="author">Written by: Klaas, Simon, Siebren, Christa …</h4>

</div>


<style type="text/css">
.title {
  display: none;
}
.author {
  display: none;
}

#getting-started img {
  margin-right: 10px;
}

</style>
<div class="row" style="padding-top:10px;">
<p><br></p>
<div id="some-general-notes" class="section level4">
<h4>Some general notes</h4>
<ul>
<li>Question-/Problem-driven learning –&gt; formulate candidate questions per week.<br />
</li>
<li>Example dataset: monocytes, CD4 and CD8 T cells, B cells and NK cells.<br />
</li>
<li>Start in UCSC genome browser with one cell type (eg monocytes). Introduce same data of other cell types in week 2.<br />
<br></li>
</ul>
</div>
<div id="overall" class="section level2">
<h2>Overall</h2>
<div id="focusparadigmshypotheses-from-lectures" class="section level4">
<h4>Focus/paradigms/hypotheses from lectures</h4>
<ul>
<li>Specific functional genomic regions (eg promoter, enhancers etc) are associated with distinct combinations of histone modifications. Also, chromatin and heterochromatin are also characterized by distinct histone modifications.<br />
</li>
<li>Active and repressed genes are associated with distinct sets of histone modifications. Recognition of a histone modification may lead to regulation of gene transcription. Eg The basal transcription factor TFIID interacts H3K4me3 via its PHD finger domain.<br />
</li>
<li>Vice versa, using epigenetic data (eg presence/absence of histone marks and chromatin accessibility) we can predict the transcriptional state of a gene or functionality of a region.<br />
</li>
<li>Heatmaps based on genomic coordinates - in which the intensity of the color indicates the strength of the signal - allow for a very compact and powerful visualisation and therefore interpretation of epigenomics data.<br />
</li>
<li>Presence of bivalent chromatin domains marks develomentally regulated genes, thought to mark genes taht will need to be switche don in the near future and are ‘poised’ for transcription.<br />
<br></li>
</ul>
<blockquote>
<h4 id="research-goals-of-this-practical">‘Research goals’ of this practical</h4>
<ul>
<li>Use combinations of histone PTM data and chromatin accessibility to identify functional genomic elements (eg promoter, enhancers etc) and predict gene activity (active or repressed) using heatmaps.<br />
</li>
<li>Check obove mentioned predictions on gene activity in single-cell RNA-seq data.<br />
</li>
<li>Vice versa, use the epigenetic data to explain gene expression differences detected in a single-cell RNA-seq experiment.<br />
</li>
<li>Predict TFs that could mediate cell-specific gene expression.</li>
</ul>
</blockquote>
<blockquote>
<h4 id="global-learning-objectives">Global learning objectives</h4>
<ol style="list-style-type: decimal">
<li>Understand how we come from a ChIP-sequencing experiment to useful data.</li>
<li>Know how to display and browse through genomics data in the UCSC Genome Browser and formulate some basic hypotheses using this tool.<br />
</li>
<li>Tell apart different genomics datasets based on their pattern int he UCSC Genome Browser.<br />
</li>
<li>Quantify and visually summarize the genomic distribution of histone ChIP-seq peaks and signal strength in r.<br />
</li>
<li>Statistically test for enrichment of histone marks in functional genomic elements.<br />
</li>
<li>Summarize genome-wide ChIP-seq signal in a heatmap and use this heatmap to identify identify functional genomic elements and predict gene activity.<br />
</li>
<li>Check hypotheses about the epigenetic landscape based on single-cell RNA-seq data with actual epigenomics data, and vice versa (check hypotheses on gene activity in single-cel RNA-seq data).<br />
</li>
<li>Formulate hypotheses about (cell-specific) TF function using histone ChIP-seq and ATAC-seq data.</li>
</ol>
</blockquote>
</div>
<div id="additional-goals-may-inclule" class="section level4">
<h4>Additional goals may inclule</h4>
<ul>
<li>Use combinations of histone PTMs to identify heterochromatin and euchromatin regions in these cells.<br />
</li>
<li>Find/Hypothesize/suggest TFs that could drive the expression of important marker genes of a given blood cell? (by looking at motif enriched in promoters and enhancers and TF gene expression)<br />
</li>
<li>Presence of bivalent chromatin domains marks at genes that are suggested to be ‘poised’ for transcription.<br />
<br></li>
</ul>
<hr />
</div>
</div>
<div id="week-1" class="section level2">
<h2>Week 1</h2>
<ul>
<li>Focus: how we make data out of a ChIP-seq experiment, how to use the UCSC Genome Browser, and familiarize ourselves with the UCSC Genome Browser visualization.<br />
</li>
<li><em>seq2science, UCSC Genome browser</em><br />
</li>
<li>Global objectives #1-3.<br />
</li>
<li><p>Perform in H5P?<br />
<br></p></li>
<li>Discuss ChIP-seq in presentation<br />
</li>
<li>Follow seq2science workflow<br />
</li>
<li><p>Peakcalling: note that it is not black/white, not absolute, thresholds are set. NOT into technicalities and mathematics of read piling/peak models, narrow-wide peaks<br />
<br></p></li>
</ul>
<blockquote>
<h4 id="learning-objectives-after-this-tutorial-the-student-should-understandbe-able-to">Learning Objectives (<em>after this tutorial the student should understand/be able to…</em>)</h4>
<ol style="list-style-type: decimal">
<li>Know how raw read output (from eg ChIP-esq) is processed to obtain trustworthy data.<br />
</li>
<li>Understand how to mov (“<em>browsing</em>”) around in the UCSC genome browser and know how to go to regions and genes of interest.<br />
</li>
<li>Recognize the following elements/regions in the UCSC genome browser: <em>chromosome, centromere, telomere, gene, exon, intron, <span style="color: red;">utr</span>, promoter, enhancer/repressor</em>.<br />
</li>
<li>Tell apart UCSC genome browser tracks of H3K27ac, H3K27me3, H3K36me3, H3K4me1 and H3K4me3 (and H3K9me3) histone ChIP-seq and ATAC-seq signal (and ChIP-seq input) based on their coverage profile and their location with respect to genes and other annotated genomic/chromosomal elements.</li>
<li>Can interpret - and thus understand - ChIP- and ATAC-seq signal in the UCSC genome browser.</li>
</ol>
</blockquote>
<blockquote>
<h4 id="working-questions">Working questions</h4>
<ol style="list-style-type: decimal">
<li>What histone mark marks the promoter of TNF gene? Let’s go and look for it ;).<br />
</li>
<li>Now your turn, what about CD3 is this gene active in monocytes? (no it is not, also logical, it is part of the T cel receptor present on the cellsurface of T cells)<br />
</li>
<li>Look at the enhancer located 20kb upstream of …. [gene] …. Based on the histone and chromatin accessibility data would you say this enhancer is active or inactive? What histone marks do you find around it?<br />
</li>
<li><em>Summarizing</em> What marks are generally found at promoters, at gene bodies, active enhancers and at repressed domains?</li>
</ol>
</blockquote>
<hr />
</div>
<div id="week-2" class="section level2">
<h2>Week 2</h2>
<ul>
<li>Focus: <em>Specific functional genomic regions (eg promoter, enhancers etc, (and their activity ?)) are associated with distinct combinations of histone PTMs.</em><br />
</li>
<li><em>Bioconductor, count peak distribution and compare marks among regions</em><br />
</li>
<li>Global objectives #4-5.</li>
</ul>
<p>In week 1 we have estimated which mark is enriched in promoters, gene bodies or enhancers. This week we will put numbers to these estimates. More specifically,..<br />
<br></p>
<p>Question from my side, what would be a good first step to look at for students?:<br />
- Count the occurrences of a histone PTM per genomic element and then state something about the enrichment of that mark in a particular genomic element?<br />
- Or, count the overlap of histone PTMs with each other and with ATAC-seq signals and assign a function to these combinations (eg active enhancer, repressed enhancer, active promoter, gene, bivalent promoter)<br />
- introduce heatmap in this week (question 8) or do it next week?</p>
<blockquote>
<h4 id="learning-objectives">Learning Objectives</h4>
<ol style="list-style-type: decimal">
<li>Move around in GRanges object which holds and can handle tables with genomic coordinates.<br />
</li>
<li>Obtain annotation of your reference genome of interest (in TxDb object) and know how to extract the features important to you.<br />
</li>
<li>Perform intersections with the r objects from (1) and (2).<br />
</li>
<li>Find and count the overlap (using intersection) among histone ChIP-seq peaks and genomic elements (<code>countOverlaps(query, subject)</code>, <code>findOverlaps(query, subject)</code>).<br />
</li>
<li>Plot this quantification (ggplot + geom_bar).</li>
<li>Statistically test for enrichment of histone marks in (a) particular genomic region(s).<br />
</li>
<li>Optionally, quantify and plot the overlap among histone marks (a tiny bit like chromHMM but much less fancy, ggplot + geom_bar with index at the bottom for the different combinations or plot the tabular output as heatmap with nubmer display).</li>
</ol>
</blockquote>
<blockquote>
<h4 id="working-questions-1">Working questions</h4>
<ol style="list-style-type: decimal">
<li>Look in the benome browser for the H3K4me3 in monocytes or look back at your answer of question 5 in week 1. This time we will count all the overlap between H3K4me3 peaks and different genomic elements to find out in which region most of the H3K4me3 peaks occur?</li>
<li>Is this enrichment in promoters more than you would expect by chance?<br />
</li>
<li>And what if we look at H3K27ac or H3K36me3?<br />
</li>
<li>Do these numbers differ (significantly?) when we look at T cells instead of monocytes?<br />
</li>
<li>Now we will count the overlap among histone marks in monocytes. What genomic elements are associated with each combination? (one question but will take some time for students)</li>
</ol>
</blockquote>
<p>The bar graph or heatmap we made for question 4 does not take along peak height (signal strength) which could hold additional information does not allow a locus-specific comparison among different cell types. For that we would need to, somehow, include the genomic location int he plot.</p>
<p>Luckily, we have another option to plot the occurrences and co-occurrences of histone marks around a chosen genomic landmark (eg TSS) that <em>does</em> take along the genomic locus and signal strength: a heatmap. We will do so next week.</p>
<p><br> #### Alternative workflow<br />
Based on</p>
<ul>
<li>Computational Genomics with R Book from Altuna Akalin,creative commons license, chapters 6 and 9 <a href="http://compgenomr.github.io/book/">booklink</a> -<br />
</li>
<li>Bioconductor 2018 conference, workshop GenomicRanges <a href="https://bioconductor.github.io/BiocWorkshops/solving-common-bioinformatic-challenges-using-genomicranges.html">workshop link</a><br />
</li>
<li>Bioconductor 2020 conference, workshop Integrated ChIP-seq Data Analysis <a href="http://integratedchipseqanalysis-workshop-kqumrdos.bioc.cancerdatasci.org/files/vignettes/IntegratedChIPseqAnalysisWorkshop.html#Load_required_libraries">workshop link</a><br />
</li>
<li>ChIPpeakAnno tutorial/vignette <a href="https://bioconductor.org/packages/release/bioc/vignettes/ChIPpeakAnno/inst/doc/ChIPpeakAnno.html">vignette link</a></li>
</ul>
<p><strong>Required packages</strong>:</p>
<ul>
<li>ChIPpeakAnno<br />
</li>
<li>GenomicRanges<br />
</li>
<li>rtracklayer<br />
</li>
<li>UpSetR</li>
</ul>
<ol style="list-style-type: decimal">
<li><p>Familiarize with *Ranges structure and language and GRanges and GRangesList objects<br />
Functions:</p></li>
<li><p>Import MACS2 peaks into R<br />
Functions: <code>toGRanges()</code></p></li>
<li>Optionally: find overlap between replicates<br />
</li>
<li>Count and visualize distribution of peaks over genomic different marks in exon, intron, enhancer, proximal promoter, 5’ UTR and 3’ UTR</li>
<li>Annotate peaks to genes<br />
</li>
<li><p>Determine and visualize the distance of a peak to a genomic feature</p></li>
<li>How many promoters are active in monocytes? And what about CD4 T cells<br />
</li>
</ol>
<ol style="list-style-type: lower-alpha">
<li>Familiarize with GRanges object<br />
</li>
<li>Import peaks data into R<br />
</li>
<li>Identify the condfident peaks among two replicates, plot in venn diagram, continue with these for the analysis</li>
</ol>
<hr />
</div>
<div id="week-3" class="section level2">
<h2>Week 3</h2>
<ul>
<li>Focus: <em>Histone PTMs positioning is cell-specific</em> or <em>Different cell types show distinct, cell-specific histone enrichment patterns (associated with cell-specific gene expression and the cell-specific activity of TFs (if we want to go into motifs))</em><br />
</li>
<li>Global objectives #4-6</li>
<li>Heatmap using <span style="color: red;">package ngs.plot, EnrichedHeatmap, deepTools, ???</span><br />
</li>
<li>Use heatmap to compare marks or cells? Or both? Do first all marks in one cell</li>
</ul>
<p>In week 1 we looked at the localization of histone marks (peaks) and we quantified this in week 2 of peaks. You might have seen that some histone marks show partial overlap with other marks. This week we will summarize and, importantly, compare histone markings genome-wide <em>among different cell types</em><br />
<br></p>
<blockquote>
<h4 id="learning-objectives-1">Learning Objectives</h4>
<ol style="list-style-type: decimal">
<li>Be able to visually summarize read coverage (<em>signal</em>) of different histone marks around genomic landmarks (coverage plot around TSS or over gene body).<br />
</li>
<li>Summarize genome-wide ChIP-seq signal <em>of different histone marks</em> in a heatmap and cluster this heatmap (for 1 or 2 cell types only, not more. Centered on identified ATAC-summits or any peak midpoint?) and use this heatmap to identify/annotate functional genomic elements.<br />
</li>
<li>In a related way, use a heatmap to identify differentially marked promoters among different cell types (centered at TSSs, plot H3K4me3 signal).<br />
</li>
<li>And identify differently marked enhancers among different cell types (centered on ATAC peaks not in promoters(?), using H3K4me1 and/or H3K27ac).<br />
</li>
<li>Assign enhancers to genes (and understand the assumptions/decisions you make when you do this).</li>
</ol>
</blockquote>
<blockquote>
<h4 id="working-questions-2">Working questions</h4>
<ol style="list-style-type: decimal">
<li>Does H3K36me3 signal start at the TSS or at the ATG/start codon and does its signal increase or decrease downstream from ATG/start (vs upstream)? First look in the GB to formualte a hypothesis. Then plot.<br />
</li>
<li>Are H3K4me3 and H3K27ac signal higher or weaker directly upstream of the TSS compared to directly downstream of the TSS? Do these marks behave the same?<br />
</li>
<li>What histone marks are found around ATAC peaks and do these reflect different functional genomic elements?<br />
</li>
<li>Do NK cells look more like monocytes or like T cells? Based on H3K4me3 signal in promoters.<br />
</li>
<li>Find the top 3/5/10 differently marked promoters in each cell type?<br />
</li>
<li>The [TF] is essential for T cell phenotype. Based on histone markings at the promoter, would you argue that it is exclusively expressed in T cells or not (only compare with the cells we have data for here).</li>
</ol>
</blockquote>
<hr />
</div>
<div id="week-4" class="section level2">
<h2>Week 4</h2>
<ul>
<li>Focus: <em>Interpret single-cell RNA-seq data, formulate hypotheses about epigenome and test hypothesis by plotting the data for these genes</em><br />
</li>
<li>or <em>find or count motif occurrences using pattern matching from GRanges</em><br />
</li>
<li><p>Use PBMC3k dataset, get UMAP and top marker genes per cluster/cell type from Satija lab.<br />
<br></p></li>
<li>We continue with clusters that differentially mark T cells from monoyctes [..] and assign them to closest gene.<br />
</li>
<li>Open/active regions only?<br />
</li>
<li><p>Compare these to + and - marker genes from scRNA-seq<br />
<br></p></li>
</ul>
<p><strong>nb</strong>:<br />
- Platelets are anuclear but rich in RNAs including mRNAs (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4110657/">Schubert et al. 2014 Blood</a>)</p>
<blockquote>
<h4 id="learning-objectives-2">Learning Objectives</h4>
<ol style="list-style-type: decimal">
<li>Can interpret UMAP and gene epression plots (FeaturePlots) from scRNA-seq.<br />
</li>
<li>Understand the added value of scRNA-seq over bulk rna-seq (lecture not this workshop).<br />
</li>
<li>Use the methods and plotting strategies from the first three weeks to test</li>
</ol>
</blockquote>
<blockquote>
<h4 id="working-questions-3">Working Questions</h4>
<ol style="list-style-type: decimal">
<li>As we discussed before, CD3 is part of the T cell receptor and should be expressed by all T cells. Can you confirm this with the scRNA-seq data?<br />
</li>
<li>What is the most likely explanation for the lack of CD3 epxression in some dots (a dot is a cell) that are clustered as T cell?<br />
</li>
<li>From week 3 we had a list of top differently marked promoters per cell type. For which are you able to confirm cell specific gene expression?<br />
</li>
<li>These [file] are the top marker genes from scRNA-seq experiment. Which of the following hypotheses about the epigenetic markings of these genes, do you agree with?<br />
</li>
<li>Let’s look at ChIP-seq signals in around their TSSs using a heatmap (hopefully there are some bivalent promoters). What is the most likely explanation for detecting repressive marks?</li>
</ol>
</blockquote>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
