<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>1. UCSC Genome Browser</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="custom.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">FG 2020</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="p3_week1.html">Week 1</a>
</li>
<li>
  <a href="p4_week2_v4.html">Week 2</a>
</li>
<li>
  <a href="prepdata.html">Appendix</a>
</li>
<li>
  <a href="example1_sequence_short_fastqc.html">FASTQC_exmpl1</a>
</li>
<li>
  <a href="example2_sequence_short_fastqc.html">FASTQC_exmpl2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">1. UCSC Genome Browser</h1>

</div>


<style type="text/css">
.title {
  display: none;
}

</style>
<div id="learning-objectives" class="section level2">
<h2>1.1 Learning Objectives</h2>
<p><br> <strong>This week’s focus will be on how sequencing data is processed and acquint ourselves with the UCSC Genome browser as a tool to visually inspect (epi-)genomics data.</strong><br />
<br></p>
<blockquote>
<p>At the end of week 1 you are able to</p>
<ol style="list-style-type: decimal">
<li>Explain the steps taken to go from a ChIP-seq experiment, to raw data, and finally to data usable for analysis.<br />
</li>
<li>Recognize the following elements in the UCSC genome browser: chromosome, centromere, telomere, gene, exon, intron, utr, promoter, and (candidate) enhancer/repressors.<br />
</li>
<li>Go to regions of interest in the genome browser.<br />
</li>
<li>Explain the difference between ‘uptream’ and ‘downstream’.<br />
</li>
<li>Identify peaks in the coverage tracks of a histone ChIP-seq experiment in the genome browser.<br />
</li>
<li>Identify tracks of H3K27ac, H3K27me3, H3K36me3, H3K4me1, H3K4me3 and H3K9me3 histone ChIP-seq and DNaseI-seq signal (and ChIP-seq input) based on their coverage profile and their location with respect to genes and other annotated genomic/chromosomal elements.<br />
</li>
</ol>
<ul>
<li>These refer to Global learning objectives #1-3.</li>
</ul>
</blockquote>
</div>
<div id="sample----raw-data" class="section level2" data-progressive="TRUE">
<h2 data-progressive="TRUE">1.2 Sample --&gt; raw data</h2>
<p>Briefly review ChIP-seq, library preperation for next-generation sequencing and analysis of the resulting data with the following video:<br />
<br> <strong>ChIP-seq by StatQuest</strong>, duration of 8:30:<br />
<a href="https://www.youtube.com/watch?v=nkWGmaYRues&amp;ab_channel=StatQuestwithJoshStarmer">ChIP-seq by SatQuest</a><br />
<br></p>
<details>
<summary>
Click here for additional information about Illumina sequencing.
</summary>
<blockquote>
<p><em>Background</em>: There are several manufacturers of sequencing machines and reagents. <strong>Illumina</strong> is a common one. The Illumina technology is summarized in the following videos (optional):</p>
<ul>
<li><a href="https://www.youtube.com/watch?v=mLZqS0RYqCk&amp;ab_channel=RobEdwards">Introduction to Illumina Sequencing, Rob Edwards, duration 2:45</a><br />
</li>
<li><a href="https://www.youtube.com/watch?reload=9&amp;v=womKfikWlxM&amp;ab_channel=Illumina">Illumina Sequencing Technology, Illumina, duration 5:03</a></li>
</ul>
<p>In summary, Illumina sequencing involves:</p>
<ol style="list-style-type: decimal">
<li>Library preperation (also called sample preperation):
<ol style="list-style-type: lower-roman">
<li>Extract DNA from your sample.<br />
</li>
<li>Fragment this DNA.<br />
</li>
<li>Ligate <strong>adapters</strong> to the fragment these contain a sample specific sequence (= <strong>barcod</strong> or <strong>index</strong>), regions complementary to oligo’s that are found on the flow cell (these oligo’s are called P5 and P7) and primer bindings sites for sequencing (Rd1 SP and Rd2 SP):<br />
<img src="images/wk1_1_illuminalibrary.png" style="width:80.0%" /><br />
</li>
</ol></li>
<li>Load library on the <strong>flow cell</strong>:
<ol style="list-style-type: lower-roman">
<li>The flow cell is a glass slide with P5 and P7 oligo’s attached to the surface.<br />
</li>
<li>The DNA fragments are denatures and single stranded fragments anneal to P5 and P7 oligo’s present on the flow cell.<br />
</li>
</ol></li>
<li><strong>Cluster</strong> growth:
<ol style="list-style-type: lower-roman">
<li>Fragments are amplified, the P5 and P7 oligo’s are used as the primer for this process ensuring that the final molecule is attached to the flow cell.<br />
</li>
<li>Multiple amplification cycles result in a cluster of fragments.<br />
</li>
<li>This procedure occurs for millions of fragments resulting in millions of clusters.<br />
</li>
<li>The reverse strands are cleaved, ensuring that the same strand is used for sequencing within each cluster.<br />
</li>
</ol></li>
<li>Sequencing by synthesis:
<ol style="list-style-type: lower-roman">
<li>Sequencing primer (Rd1 SP) is added and 4 fluorescently labelled nucleotides compete for incorporation in the growing strand.<br />
</li>
<li>Image acquisition: A laser excites the clusters and the signal from the incorporated nt is measured.<br />
</li>
<li>The emmission wave length and signal intensity determine the <strong>base call</strong> and the <strong>quality</strong> of the reading.<br />
</li>
<li>A special index read primer is used to sequence the barcode.<br />
</li>
<li>For paired-end sequencing, the above procedure is repeated with the reverse strand.<br />
<a href = "https://hbctraining.github.io/Intro-to-ChIPseq/lessons/02_QC_FASTQC.html"> <img src="images/wk1_1_illuminasequencing.png" style="width:80.0%" alt="Intro-to-ChIPseq, hbctraining" /><br />
</a></li>
</ol></li>
</ol>
</blockquote>
</details>
<p><br> <!--Currently Illumina also has the nextSeq 500 series (besides MiSeq and HiSeq mentioned in the video). One of the main difference being that the MiSeq and HiSeq use four fluorescent colors (G = blue ,C = green, A = yellow and T = red) while the nextSeq series use 2-color chemistry (green and red). The effect being that with each cycle for the NextSeq, only two images need to be taken, As ar read as clusters appearing in both images and Gs are read as clusters that are not present in either of the two.--></p>
<p><br> The <strong>sequencer</strong> analyzes a set number of bases, eg 42, 50 or 100 bp, called a <strong>read</strong>. The resulting <strong>base calls</strong> and their <strong>quality scores</strong> are stored in <strong>fastq files</strong> which have the .fq or .fastq extension (or compressed as .fq.gz or .fastq.gz). These are always organized in the same manner with 4 lines per read. This is an example with 2 DNA sequences (8 lines):<br />
<br> <img src="images/wk1_2_fastq_layout.png" style="width:100.0%" /><br />
<br></p>
<ol style="list-style-type: decimal">
<li>Header starts with ‘@’-sign and contains a unique sequence ID (the element up to the fist space). This header often follows a set substructure. Element after the space often includes the sample barcode.<br />
</li>
<li>Called bases in ACGTN-alphabet<br />
</li>
<li>Midline, can be empty or hold ‘+’<br />
</li>
<li>Quality score for each nucleotide, also called <strong>Phred score</strong>.</li>
</ol>
<p><br> The Phred scores represents the probability (p) that the reported call is incorrect. You do not need to learn this encoding by heart but they will come back in the Quality Control step of our analysis, therefore we discuss briefly how they are encoded:</p>
<table>
<thead>
<tr class="header">
<th align="center">Phred quality score</th>
<th align="center">Probability of incorrect base call</th>
<th align="center">Accuracy</th>
<th align="center">Characterer</th>
<th align="left"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">0</td>
<td align="center">1 in 1</td>
<td align="center">0%</td>
<td align="center">!</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="center">10</td>
<td align="center">1 in 10</td>
<td align="center">90%</td>
<td align="center">+</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="center">20</td>
<td align="center">1 in 100</td>
<td align="center">99%</td>
<td align="center">5</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="center">30</td>
<td align="center">1 in 1000</td>
<td align="center">99.9%</td>
<td align="center">?</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="center">40</td>
<td align="center">1 in 10000</td>
<td align="center">99.99%</td>
<td align="center">J</td>
<td align="left"></td>
</tr>
</tbody>
</table>
<p>Here’s another visualization of how Phred scores are encoded :<br />
<img src="images/wk1_2_phred_quality_scores_TorstenSeemannpresentation.png" style="width:80.0%" /></p>
<details>
<summary>
Here’s the complete encoding of Phred scrores 0-40.
</summary>
<blockquote>
<p><em>Background</em>:<br />
<img src="images/wk1_1_phred_ascii.png" style="width:80.0%" /><br />
<!--ref: Amixon TRAINING, Bioinformatics for beginners - File formats Part 2. - Short reads, written on 01/07/2013, visited on 29/11/2020, through: https://www.omixon.com/bioinformatics-for-beginners-file-formats-part-2-short-reads/--></p>
</blockquote>
</details>
<p><br> Before we can start our data analysis and identify for example regions marked by activating and repressing histone marks, we need to process the raw data. This includes cleaning up, mapping, visualizing the sequenced reads and identifying genomic regions where reads tend to pile up (in other words “peaks”). Which is what we will discuss next.<br />
<br></p>
</div>
<div id="raw-data----science" class="section level2" data-progressive="TRUE">
<h2 data-progressive="TRUE">1.3 Raw data --&gt; science</h2>
<div id="post-sequencer-processing" class="section level3">
<h3>1.3.1 Post-sequencer processing</h3>
<p>The sequencer determines the sequence of the DNA molecules in your sample and produces massive amounts of data. In an ideal world, these data would 100% reflect reality. But….. It doesn’t and it is often your or a bio-informaticians task to discriminate valuable signal from artefacts and noise, the latter being accumulations of reads in regions that don’t actually contain binding sites of the protein or mofidified histone of interest.<br />
<br> <img src="images/wk1_3_intro_whatbioinformaticiancsdo.png" style="width:60.0%" /> <br />
<br> A general pipeline for ChIP sequencing data involves several following steps. In theory, all of them can be performed in R but the first steps are commonly performed by tools outside R (often on Linux server). The resulting peakfiles, files with read depth and/or read counts are then imported in R.</p>
<p><br> Here we will discuss the steps that are necessary to turn the raw before we can start our analyses in R:<br />
<br></p>
<div id="run-a-quality-check-on-the-raw-data" class="section level4">
<h4>1. Run a quality check on the raw data</h4>
<p>We run the raw fastq files through <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FASTQC</a> software to get a quick impression of whether our data has any technical problems of which we should be aware before doing any further analysis. This tools outputs a HTML report with graphs and tables summarizing, among others the base call accuracy, library base composition, duplication levels and the presence of overrepresented sequences.</p>
<p>The <strong>Per base sequence quality</strong> plot is one of the most important plots generated. It summarizes the base call accuracy (Y-axis) for each cycle (X-axis) across the library. See this example:<br />
<br> <img src="images/wk1_3_fastqc_per_base_quality_vb1.png" style="width:80.0%" /></p>
<p>This exmaple shows the average Phred quality for a library sequenced to 72 bp. The majority of bases have a good quality, the quality drops for the last two bases and we should (in this case) remove these from the end of the read.</p>
<p>For most runs, quality should be good for most reads through the whole run. If quality is low, we can use the other plots and tables to try to understand why this is.</p>
<ul>
<li>Good (Illumina) quality is generally Phred &gt; 28 - 30<br />
</li>
<li>Concerning (Illumina) quality is Phred &lt; 20</li>
</ul>
<p>If the QC does not support major technical problems like a systematic base bias that occured during sequencing or contamination during sample prep with foreign DNA, we can continue.</p>
<blockquote>
<p><strong>Exercise 1</strong><br />
Use the following two FASTQC reports to answer the following questions.</p>
<ul>
<li><a href="https://ctoen.github.io/FunctGen/example1_sequence_short_fastqc.html">FASTQC report 1</a><br />
</li>
<li><a href="https://ctoen.github.io/FunctGen/example2_sequence_short_fastqc.html">FASTQC report 2</a></li>
</ul>
</blockquote>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quiz</span>(</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">question</span>(<span class="st">&quot;What is the sequence length?&quot;</span>,</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;45&quot;</span>),</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;40&quot;</span>, <span class="at">correct =</span> T),   </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;Can&#39;t be answered using these reports&quot;</span>), </span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">allow_retry =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">random_answer_order =</span> <span class="cn">TRUE</span>),</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">question</span>(<span class="st">&quot;Which shows the **best per base sequence quality**?&quot;</span>,</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;report 1&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;report 2&quot;</span>),</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">allow_retry =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
<p><br></p>
<blockquote>
<p><em>Background</em>: For detailed discussion on FASTQC reports see the FASTQC website of the Babraham institute, and their tutorial on Quality Control in Sequencing Experiments <a href="https://www.bioinformatics.babraham.ac.uk/training.html#sequenceqc">ref</a>.</p>
</blockquote>
<p><br></p>
</div>
<div id="trim-and-quality-filter-reads" class="section level4">
<h4>2. Trim and quality-filter reads</h4>
<p>In this cleaning step we predominantly:</p>
<ul>
<li>Remove reads composed solely of adapters without insert.<br />
</li>
<li>Remove reads with an overall low-quality.<br />
</li>
<li>Remove or mask low-quality bases within predominantly high quality reads (if these occur, they are often the last one or two bases for Illumina sequencing). This can result in the complete removal of some reads because of their overall low quality or short length.</li>
</ul>
<p><br></p>
</div>
<div id="align-reads-to-the-genome" class="section level4">
<h4>3. Align reads to the genome</h4>
<p>Next Generation Sequence (NGS) alignment means that we identify the likely location in the reference genome where the original DNA molecule came from. This process is called <strong>aligning</strong> and is performed by <strong>alignment tools</strong>, also called <strong>read mappers</strong>. Well known examples include STAR, BWA and bowtie2. In essence, these tools perform the same tasks but with different algorithms/strategies.<br />
<br> Since the publication of the human genome, successive versions have been released (also called <strong>assemblies</strong> or <strong>builds</strong>). The improvements in quality are thanks to technological advancements as well as improvements in the representativeness of the genome with respect to the genetic diversity found in the human population. We will work with (one of) the latests relaese referred to as <strong>GRCh38</strong> or <strong>hg38</strong> (Genome Reference Consortium Human Build 38).<br />
<br> The aligner thus requires as input the quality-filtered and trimmed fastq files together with the reference genome.<br />
<br> The outputted alignment is stored in (<em>huge!</em> ) TAB-delimited text Sequence Alignment Map files, <strong>SAM files</strong>. The compressed version of SAM files are called <strong>BAM files</strong> (Binary Alginment Map).<br />
<br></p>
<details>
<summary>
Click here for additional information about SAM/BAM file format.
</summary>
<blockquote>
<p><em>Background</em>: The SAM/BAM files contains a header with information about the mapping procedure. This is followed by the read alignments with 1 line per read. The latter contains information about the read sequence, how much of it was mapped to the reference genome (eg. the whole read or only part of it), how well it alinged, and lots of other information, dependent on the aligner.<br />
<br> To give you a feel for how SAM files looks like, here’s an example of the header and one record:<br />
<img src="images/wk1_3_bam_header_example_sandrews_tutorial201911.png" style="width:100.0%" /><br />
<!-- ref: Simon Andrews, Babraham Bioinformatics, Quality control for Sequencing Experiments pdf, v2019-11 --> <img src="images/wk1_3_bam_record_example_sandrews_tutorial201911.png" style="width:100.0%" /><br />
<!-- ref: Simon Andrews, Babraham Bioinformatics, Quality control for Sequencing Experiments pdf, v2019-11 --> This example tells you among others that the library was mapped with <a href="http://daehwankimlab.github.io/hisat2/manual/">hisat2</a> to the mouse reference genome GRCm38 with certain additional settings. The single displayed read is mapping to chr18, starting at position 57944851.<br />
<br> In summary:<br />
<img src="images/wk1_3_BAM_fields_broadinstitute.png" style="width:100.0%" /><br />
Each record holds at least 11 columns (exact number depends on the aligner and post-alignment analytics performed on the file):</p>
<table>
<colgroup>
<col width="33%" />
<col width="33%" />
<col width="33%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">ColumnNR</th>
<th align="left">Description</th>
<th align="left">Example</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1.</td>
<td align="left">Read name</td>
<td align="left">HWI-D00436:394:CBGLBANXX:1:1101:1222:1861</td>
</tr>
<tr class="even">
<td align="left">2.</td>
<td align="left">12 “bits” flag summarizing of the alignment. Webtools to explain the meaning of each flag: <a href="https://www.samformat.info/sam-format-flag">samformat</a>, or <a href="https://broadinstitute.github.io/picard/explain-flags.html">Broad Institute</a></td>
<td align="left">16</td>
</tr>
<tr class="odd">
<td align="left">3.</td>
<td align="left">Chromosome in the reference genome the read aligns</td>
<td align="left">chr18</td>
</tr>
<tr class="even">
<td align="left">4.</td>
<td align="left">Position within the chromosome the read aligns</td>
<td align="left">5794485</td>
</tr>
<tr class="odd">
<td align="left">5.</td>
<td align="left">Mapping quality score, different programs use different scores</td>
<td align="left">60</td>
</tr>
<tr class="even">
<td align="left">6.</td>
<td align="left">CIGAR string, is a summary of the alignment per nt</td>
<td align="left">50M</td>
</tr>
<tr class="odd">
<td align="left">7.</td>
<td align="left">Chromosome where the paired read aligns</td>
<td align="left">chr18 (if there is no paired read 0 or *)</td>
</tr>
<tr class="even">
<td align="left">8.</td>
<td align="left">Position within where the paried read aligns</td>
<td align="left">5794685 (or 0 if there is no paired read)</td>
</tr>
<tr class="odd">
<td align="left">9.</td>
<td align="left">Length of the complete fragment, from the first mapped base of read 1 to the last mapped base of read 2.</td>
<td align="left">200 (0 if there is no pair)</td>
</tr>
<tr class="even">
<td align="left">10.</td>
<td align="left">Called bases</td>
<td align="left">AAAAGATCTCTTGAT…</td>
</tr>
<tr class="odd">
<td align="left">11.</td>
<td align="left">Phred quality score of the read</td>
<td align="left">GGGGGGGGGGGGGGG…</td>
</tr>
</tbody>
</table>
</blockquote>
</details>
<p><br></p>
</div>
<div id="filter-artefacts-and-reads-aligning-to-multiple-regions" class="section level4">
<h4>4. Filter artefacts and reads aligning to multiple regions</h4>
<p>From the initial alignment we often identify and remove PCR artefacts, reads aligning with a low mapping quality (column 5) and optionally remove reads aligning non-uniquely. These actions can be performed with various tools. Commonly used ones are <a href="http://samtools.sourceforge.net/">SAMtools</a> and <a href="https://broadinstitute.github.io/picard/">Picard tools</a>. The input and output of these steps are both BAM files.<br />
<br></p>
</div>
<div id="visualize-read-depth" class="section level4">
<h4>5. Visualize read depth</h4>
Files are created that count how many reads align to each nucleotide in the reference genome. Nucleotides with a relative high score pinpoint the likely locations of the ChIPped factor or histone PTM; to a region of accessible chromatin in ATAC-seq; or to the region that was likely transcribed and detected in RNA-seq.<br />
<br> These files can be uploaded to the UCSC Genome Browser which we will look at in more detail today as well. Various file types can store this kind of data. We wil use bigwig files (.bw file extension), these are fast and relatively small but we can’t view them.<br />
<br>
<details>
<summary>
A fictional example of alignment and a read depth graph
</summary>
<img src="images/wk1_3_read_depth_Pavlopoulos2018.png" style="width:80.0%" /><br />
<!-- ref: Pavlopoulos et al (2013) BioData Mining 6, Article number: 13 (2013), figure 6 -->
</details>
<p><br></p>
</div>
<div id="call-peaks" class="section level4">
<h4>6. Call peaks</h4>
<p>Reads are scattered through the genome but locations where the used antibody recognizes its target (eg. DNA-binding TF, histone, histone variant or histone PTM) will collect more reads. <strong>Peak calling</strong> is the computational process to identify these regions, often with respect to a control dataset where no antibody or no specific antibody was used, resp. <strong>input control</strong> and <strong>IgG control</strong> . A commonly used tool for this step is <a href="10.1186/gb-2008-9-9-r137">MACS2</a>. MACS2 was also used to call peaks in the datasets that we will be using.<br />
<br> Peak calling is a statistical procedure and never black and white. It depends among others on thresholds used (p-value, false-discovery rate) and the control sample. It is always wise to visually inspect the resulting peaks in the UCSC genome browser. If the called peaks do not correspond to the sites with high read coverage, you may need to adjust the settings for peak calling.<br />
<br> Peaks are repored as regions, or <strong>genomic intervals</strong> with their genomic coordinates: <strong>chr start end</strong>. As well as a <strong>score</strong> that indicates the strength of the signal observed for this peak. We will look at these files more closesly in week 2.<br />
<br></p>
</div>
<div id="read-counting" class="section level4">
<h4>7. Read counting</h4>
<p>Count reads that fall within genes, other genomic elements or peak regions. These counts and the actual peak locations are used for eg. differential gene expression analyses in RNA-seq or differential TF-binding in ChIP-seq (comparing differentiated vs undifferentiated or control vs treatment). A well-known, widely used tool for this purpose is DESeq2.<br />
<br> Sample clustering can be included here as additional QC. It should identify which samples are closely related and if they don’t, this might point to a label swap or more severe issue like not-working stimulus in one of the replicates.<br />
<br> Here’s an example of such a clustering, showing how different each sample is. Samples 15754-56 should be moslty similar and different from 15757-59, and vice versa:<br />
<span style="color: red;">Siebren: what’s the color legend representing?</span><br />
<img src="images/wk1_3_sample_clustering_siebrenspptx.svg" style="width:60.0%" /></p>
<blockquote>
<h4 id="optionally-dnasei-footprinting-variant-calling-novel-gene-identification-and-more">8. Optionally: DNaseI footprinting, variant calling, novel gene identification and more</h4>
<p>Depending on your experimental set up additional steps are performed outside R. For DNaseI-seq, we can identify small depleted regions within peaks that correspond to the nucleotides that interact with the TF. This interaction protects these nucleotides from DNaseI attacks. These so-called footprints can be used to identify TF motifs and predict the likely TF bound. Although this sounds simple, and for a stable binding factor like CTCF it may be the case, it is not always that simple in reality.</p>
</blockquote>
<p><br></p>
</div>
</div>
<div id="common-causes-of-noise-and-artefacts" class="section level3">
<h3>1.3.2 Common causes of noise and artefacts</h3>
<p>Artefact and noise in your data can arise from various causes. Here’s a list of common <strong>causes</strong>:</p>
<ol style="list-style-type: decimal">
<li>Synthetic DNA sequences<br />
</li>
<li>Poor base call accuracy<br />
</li>
<li>Amplification bias during library preparation<br />
</li>
<li>Genomic repeats and Low complexity regions</li>
</ol>
<blockquote>
<p><strong>Exercise 2</strong>: Below is a list of descriptions and methods to handle these arefacts. Read these carefully and combine what you think is the most likely description and method to handle the above mentioned artefacts.</p>
</blockquote>
<p><strong>Descriptions</strong>:</p>
<ol style="list-style-type: upper-alpha">
<li>The amount of nucleic acids you generally obtain from your cells is often too low to be directly analyzed and one or more steps of PCR are needed (unless you use special, amplification-free protocols and sequencing methods). Not all DNA or RNA molecules behave the same in this reaction. Some are amplified more than others.<br />
</li>
<li>These include adapters ligated to the DNA molecules to allow their amplification with uniform primers during library prep and in the sequencer. These adapter sequences come with the kit or are designed by the researcher self.<br />
</li>
<li>The chemical reaction and quality of the optical measurement that occur in the sequencer are not perfect and depend among others on the nucleotide composition of all samples, the length of the DNA molecules and the concentration of the samples. Each nt is therefore given a quality score that reflects the accuracy of the measurement.</li>
</ol>
<p><strong>Methods</strong>:</p>
<ol style="list-style-type: lower-alpha">
<li>We can control for this by using the input sample as the control during peak calling, paired-end sequencing and remove all fragments with exactly the same coordinates for read 1 and read 2, and/or the use of a unique molecular identifier (UMI) appended to our adapters that labels each unique DNA fragment.<br />
</li>
<li>Remove low-quality nt’s from reads or remove reads altoghether if all their nt’s are of low quality.<br />
</li>
<li>Filter these out of our dataset right after we get our data from the sequencer.</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">quiz</span>(<span class="at">caption =</span> <span class="st">&quot;Which answers summarize the appropriate combinations of artefact - description - method?&quot;</span>,</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">question</span>(<span class="st">&quot;For (1) *Synthetic DNA sequences*&quot;</span>,</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;Description B; Method c&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;Description C; Method b&quot;</span>), </span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;Description A; Method c&quot;</span>),</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;Description C; Method a&quot;</span>), </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">allow_retry =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">random_answer_order =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">question</span>(<span class="st">&quot;For (2) *Base call accuracy*&quot;</span>,</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;Description B; Method b&quot;</span>), </span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;Description B; Method c&quot;</span>),</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;Description C; Method b&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;Description A; Method b&quot;</span>), </span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>             <span class="at">allow_retry =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>             <span class="at">random_answer_order =</span> <span class="cn">TRUE</span>),</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">question</span>(<span class="st">&quot;For (3) *Amplification bias during library preparation*&quot;</span>,</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;Description B; Method b&quot;</span>),</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;Description C; Method b&quot;</span>),</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;Description B; Method c&quot;</span>),</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>             <span class="fu">answer</span>(<span class="st">&quot;Description A; Method a&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>             <span class="at">allow_retry =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>             <span class="at">random_answer_order =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><br></p>
<p>Lastly, when trying to pinpoint the location in the reference genome where the DNA molecule most likely was derived off, we are confronted with <strong>Genomic repeats and Low complexity regions</strong>. Repetitive sequences such as short tandem repeats and interspersed repeats make up roughly 50% of the human genome. Reads produced by sequencers are generally shorter than the repeats, thus mapping them becomes ambiguous. When dealing with a read that maps to multiple places, there are usually 3 choices: (1) Discard them; (2) Take the best alignment, if there are multiple best hits take one randomly; (3) Report all alignments or report up to a certain number. The strategy you choose depends on the sample, the technology, your research question and the reference species you work with.</p>
<p><span style="color: red;">Exercise 2: Choose the right order of steps that is used for ChIP-seq data analysis from the following list of options</span></p>
<p><img src="images/wk1_3_Furay_2012_naturereviews_fig2_chipseqanalysis.png" style="width:80.0%" /> <br />
<a href="https://pubmed.ncbi.nlm.nih.gov/23090257/">Furay, 2012, Nat Rev</a></p>
<!--
- **Amplification bias during the library preparation:** The amount of nucleic acids you generally obtain from your cells is often too low to be directly analyzed and one or more steps of PCR are needed (unless you use special, amplification-free protocols and sequencing methods). Not all DNA or RNA molecules behave the same in this reaction. Some are amplified more than others and this can generate a bias. We can control for amplification bias with:  
  
  + Use input sample, for example as control in peak calling  
  + Paired-end sequencing 
  + Add a Unique Molecular Identifier, UMI, to the barcodes, to 

  
- **Synthetic DNA sequence:** These include adapters ligated to the DNA molecules to allow their amplification with uniform primers during library prep and in the sequencer. These adapter sequences come with the kit or are designed by the researcher self. Whether they are custom or commercial, they should be removed because they were not part of the original DNA or RNA material.    
- **Base call accuracy:** The chemical reaction and quality of the optical measurement that occur in the sequencer are not perfect and depend among others on the nucleotide composition of all samples, the length of the DNA molecules and the concentration of the samples. Each nt is therefore given a quality score that reflects the accuracy of the measurement or base call accuracyas it is also called.  
- Lastly, when trying to pinpoint the location in the reference genome where the DNA mo lececule most likely was derived off, we are confronted with:  
  
  * **Genomic repeats.** The origin of reads coming from these reigons is difficult to pinpoint. But also, the reference genome might contain more or less number of repeats than the original sample.  
  * **Low complexity regions**. These are regions that contain a lot of sequence similarity, eg telomeres. Because of this the quality of the reference genome is generally low in these regions. -->
<p><br></p>
</div>
<div id="alternatively-use-fair-data" class="section level3">
<h3>1.3.3 Alternatively: use FAIR data</h3>
<p>In 2014 a core set of principles were drafted in order to optimize the reusability of research data, the resulting <strong>FAIR principles</strong> were <a href="https://www.nature.com/articles/sdata201618">published</a> in 2016. They represent a set of guidelines and best practices to ensure that data or any digital object are <strong>Findable</strong>, <strong>Accessible</strong>, <strong>Interoperable</strong> and <strong>Re-usable</strong> with special emphasis on enhancing the ability of machines to automatically find and use data.</p>
<p><img src="images/wk1_3_fairdata_fosteropenscience.png" style="width:80.0%" /><br />
<!--The Open Science Training Handbook, via https://book.fosteropenscience.eu/en/,--></p>
<ul>
<li><p><strong>Findable</strong>: The first thing to be in place to make data reusable is the possibility to find them. It should be easy to find the data and the metadata for both humans and computers. Automatic and reliable discovery of datasets and services depends on machine-readable persistent identifiers and metadata.</p></li>
<li><p><strong>Accessible</strong>: The (meta)data should be retrievable by their identifier using a standardized and open communications protocol, possibly including authentication and authorisation. Also, metadata should be available even when the data are no longer available.</p></li>
<li><p><strong>Interoperable</strong>: The data should be able to be combined with and used with other data or tools. The format of the data should therefore be open and interpretable for various tools, including other data records. The concept of interoperability applies both at the data and metadata level. For instance, the (meta)data should use vocabularies that follow FAIR principles.</p></li>
<li><p><strong>Re-usable</strong>: Ultimately, FAIR aims at optimizing the reuse of data. To achieve this, metadata and data should be well-described so that they can be replicated and/or combined in different settings. Also, the reuse of the (meta)data should be stated with (a) clear and accessible license(s).</p></li>
</ul>
<p>An extensive list of databases for genomics research can be found <a href="https://sites.google.com/site/anshulkundaje/idataset">here</a>.</p>
<p><br></p>
</div>
</div>
<div id="data-visualization-ucsc-genome-browser" class="section level2">
<h2>1.4 Data visualization: UCSC Genome Browser</h2>
<blockquote>
<p>UCSC GB sessions for making these exercises (tip: use right-mouse click to not loose this tutorial page):</p>
<ul>
<li>Default should look like (if it doesn’t, I need to adjust the text): <a href="http://genome-euro.ucsc.edu/s/ctoenhake/hg38_default" class="uri">http://genome-euro.ucsc.edu/s/ctoenhake/hg38_default</a><br />
</li>
<li>Genes and ruler only: <a href="http://genome-euro.ucsc.edu/s/ctoenhake/hg38_genesonly" class="uri">http://genome-euro.ucsc.edu/s/ctoenhake/hg38_genesonly</a><br />
</li>
<li>Monocyte wiggle data (blueprint in colors, encode in black): <a href="http://genome-euro.ucsc.edu/s/ctoenhake/hg38_monos_blueprint_and_encode" class="uri">http://genome-euro.ucsc.edu/s/ctoenhake/hg38_monos_blueprint_and_encode</a><br />
</li>
<li>Monocyte wiggle + peak data, blueprint only: <a href="http://genome-euro.ucsc.edu/s/ctoenhake/hg38_monos_blueprint_wiglle_and_peaks" class="uri">http://genome-euro.ucsc.edu/s/ctoenhake/hg38_monos_blueprint_wiglle_and_peaks</a></li>
</ul>
</blockquote>
</div>
<div id="ucsc-genome-browser" class="section level2">
<h2>1.4 UCSC Genome Browser</h2>
<div id="ucsc-genome-browser-intro" class="section level3">
<h3>1.4.1 UCSC Genome browser intro</h3>
<p>Genome browsers are great tools to display genomics data. You can walk, or <strong>browse</strong>, through all chromosomes, look at the location of genes, extract DNA sequences, examine pre-loaded data or upload your own data. The latter comes in quite handy when you want to, for example, have a visual (personal) confirmation of data quality, evaluate the effect of different settings for processing the raw reads, and/or confirm as computationally detected associations.<br />
<br></p>
<p>Watch the following youtube video’s to get familiar with the UCSC Genome Browser:</p>
<ol style="list-style-type: decimal">
<li><a href="https://www.youtube.com/watch?v=DNXI-M9oQl8">UCSC Genome Browser Basics (duration: 08.40)</a><br />
</li>
<li><a href="https://www.youtube.com/watch?v=1Kpy5G_MSG8">UCSC Genome Browser Tracks (duration: 09.14)</a></li>
</ol>
<p><br></p>
<p><strong>Checklist</strong>:</p>
<ul>
<li>Watching video 1 you should be able to:
<ul>
<li>Access your genome &amp; assembly of interest in the UCSC Genome Browser.<br />
</li>
<li>Find the chromosomal coordinates of the region currently in view or of a region you just selected (with click+drag).<br />
</li>
<li>Go to a gene or region of interest.<br />
</li>
<li>Zoom in and out.<br />
</li>
<li>Recognize UTRs, exons, introns, and the direction of transcription.<br />
</li>
<li>Find the length of a region or gene.<br />
</li>
</ul></li>
<li>After watching video 2 you should be able to:
<ul>
<li>Define what we mean with a track.<br />
</li>
<li>Adjust the display of tracks: eg. full, pack, hide etc.<br />
</li>
<li>Display tracks from the list of preloaded tracks below the browser (important: hit <code>refresh</code>).</li>
</ul></li>
</ul>
<p><br> Additional important actions and terms that were not explicitly discussed in the videos:</p>
<ul>
<li><p><strong>Selecting a region in the genome browser</strong>: hold shift-key and click/drag with your left mouse key over the region. A pop-up appears to ‘Zoom in’, ‘Add highlight’ or ‘Cancel’ (cancel = do nothing).<br />
<img src="images/wk1_4_gb_drag_and_select.png" style="width:60.0%" /></p></li>
<li><p><strong>Upstream</strong> and <strong>downstream</strong> refer to a position relative to the coding strand of a gene. Upstream means to the 5’ end and beyond. Downstream means to the 3’ end and beyond. Thus, depending on the orientation of the gene, upstream can mean that you move in the direction of the start of the chromosome or to the end of the chromosome.<br />
<img src="images/wk1_4_upstream_downstream_munca.jpg" style="width:80.0%" /><br />
<!--https://www.mun.ca/biology/scarr/iGen3_05-03.html--></p></li>
</ul>
<!-- *Optional videos*: finding the coordinates of a sequence using blat, handy for eg. primers, and using the Table Browser to, among others, obtain DNA, RNA or protein sequences:  
1. [UCSC Genome Browser Blat, 9m39s](https://www.youtube.com/watch?v=E6DDNGDXH8A)   
2. [UCSC Genome Browser Table Browser, 7m19s](https://www.youtube.com/watch?v=Ueqt5Wgre9c&ab) -->
<p><br></p>
</div>
<div id="go-to-the-ucsc-genome-browser" class="section level3">
<h3>1.4.2 Go to the UCSC Genome browser</h3>
<p><strong>Let’s go to the UCSC genome browser….</strong></p>
<p><span style="color: goldenrod; font-family: Arial; font-size:16px;"><strong>Action</strong>: Go to the <a href="https://genome.ucsc.edu/index.html">UCSC Genome Browser</a> (blue text is a link, use right-mouse click to open in new window).</span><br />
<br> The starting page has been updated since the video was made but still holds the same blue bar at the top with <strong>Genomes</strong>, <strong>Tools</strong>, etc.:<br />
<img src="images/wk1_4b_ucsc_genome_browser_topmenubar.png" style="width:80.0%" /> </p>
<p><br></p>
<blockquote>
<p><em>Tip</em>: In case you have been here before, it is wise to reset user settings such that you don’t continue with settings from a previous project. For this, hover over “Genome Browser” and select “Reser All User Settings”.</p>
</blockquote>
<blockquote>
<p><em>Tip</em>: A pop-up may appear now or one stap later, redirected you to the official European mirror site if those servers are geographically closer, that’s ok.</p>
</blockquote>
<p><br> <span style="color: goldenrod; font-family: Arial; font-size:16px;"><strong>Action</strong>: To access the human genome, select the “Genomes” tab.</span><br />
<br> On the left you can browse and select the genome you are interested in (not a drop-down menu anymore as in the video). By default, the human genome is selected.<br />
<br> We can go to a position of our choice by entering the gene or chromosomal coordinates under <strong>Position/Search Term</strong> or immediatly go to the ‘Current position’ by clicking on <code>GO</code> on the right-hand side or <strong>Genome Browser</strong> in the menu bar on top.<br />
<br> <span style="color: goldenrod; font-family: Arial; font-size:16px;"><strong>Action</strong>: Ensure that “Human” is selected in the REPRESENTED SPECIES tree and “Dec. 2013 (GRCh38/hg38)” is selected under Human Assembly (red arrows in image below).</span><br />
<br> <span style="color: goldenrod; font-family: Arial; font-size:16px;"><strong>Action</strong>: Go to <em>ACTB</em> gene (gene coding for non-muscular actin part of cytoskeleton).</span><br />
<br> <em>Genomes page, ready to go to ACTB gene in Human genome GRCh38</em>:<br />
<img src="images/wk1_4b_ucsc_gb_actb_question_startpage.png" style="width:90.0%" /> </p>
<br>
<details>
<summary>
To change the Image width of the browser (to fit your screen):
</summary>
<blockquote>
<p><em>Tip</em>: You can adjust the width of the browser to fit your (laptop) screen under “Genome Browser” &gt; “Configure”. And adjust the “image width:” .. pixels. Hit <code>submit</code> to apply the changed settings and return to the browser:<br />
<img src="images/wk1_4_gb_configure.png" /><br />
<img src="images/wk1_4_gb_configureImage.png" style="width:80.0%" /></p>
</blockquote>
</details>
<br>
<details>
<summary>
Your browser should be showing chromosomal region:
</summary>
<blockquote>
<p><em>Hint</em>: chr7:5,527,148-5,530,601.</p>
</blockquote>
</details>
<p><br> As discussed in the video, in the top of the window we have navigation and zoom bottons. Right below we can read the chromosomal coordinates and length of the regions we are currently viewing (in this case the complete <em>ACTB</em> gene). Below that we see a miniature chromosome with the characteristic gray/black pattern from karyotyping. A red <span style="color: red;">|</span>-sign gives us an idea of our current positioning in the chromosome.<br />
<br> <em>Figure for Question 1</em>:<br />
<img src="images/wk1_4b_ucsc_gb_actb_locationinchromosome.png" style="width:80.0%" /> <br />
<br><br />
<br></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 1: What chromosome is ACTB located on? And is it located on the short or long arm of this chromosome?&quot;</span>,</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;chr7, short arm&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>, <span class="at">message =</span> <span class="st">&quot;The *ACTB* gene is indeed located on the short arm of chromosome 7. In the coordinates bar and in the &#39;ruler&#39; track we can see that we are looking at chromosome 7. By looking at the red |-sign in the chroosomal idiogram, we see that this gene is located to the left of the centromere (red diabolo within the idiogram) and thus ont the short arm.&quot;</span>),</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;chr7, long arm&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Where do you see the red |-sign in the miniature chromosome on top?&quot;</span>),</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;chr17, short arm&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Review the UCSC Genome Browser Basics video, or find *chr#* in the browser window&quot;</span>),</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;chr17, long arm&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Review the UCSC Genome Browser Basics video, or find *chr#* in the browser window&quot;</span>),</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><br> The data window displays various <em>tracks</em> or <em>track sets</em>. Tracks runs the complete length of the chromosome. Data includes most often:</p>
<ul>
<li><strong>Positional information</strong>, of a gene, SNP or cir-regulatory element for example, often displayed as horizontal bars the thickness and color of which can hold additional information depending on the type of track.<br />
</li>
<li><strong>Signal information</strong>, of gene expression, detection of histone marks or degree of conservation. Often displayed as histograms but can also be summarized in vertical bars with different levels of color intensity encoding the signal intensity.</li>
</ul>
<p><br> The method of display and whether individual tracks are displayed can be changed by:</p>
<ol style="list-style-type: decimal">
<li>Right-mouse click on the grey, vertical bar at the left-most area of the Genome Browser shows you options <code>hide</code>, <code>dense</code>, <code>squish</code>, <code>pack</code> and <code>full</code>. Additional adjustments to the track can be made by selecting <code>Configure [track name]</code> in this menu.<br />
<img src="images/wk1_4_gb_configure_track_display.png" style="width:60.0%" /><br />
</li>
<li>Scroll to the area below the data window, here you see a possibly overwhelming list of potential tracks, most are <code>hidden</code>. Using the pull down menu you can change display settings of multiple tracks at once. Always make sure to hit the <code>refresh</code> button below the browser window to ensure all changes are performed.</li>
</ol>
<p><br> <span style="color: goldenrod; font-family: Arial; font-size:16px;"><strong>Action</strong>: hide all tracks and tracksets except for the ruler and GENCODE v32 tracks.</span><br />
<br> <span style="color: goldenrod; font-family: Arial; font-size:16px;"><strong>Action</strong>: for GENCODE v32, via “Configure GENCODE v32”, make sure the option “splice variants” is unchecked.</span><br />
<br></p>
<details>
<summary>
Configuration window for GECODE v32 gene track with splice variants unselected:
</summary>
<img src="images/wk1_4b_ucsc_gb_unselectsplicevariants.png" style="width:90.0%" /> 
</details>
<br>
<details>
<summary>
Your GB session should look like this
</summary>
<img src="images/wk1_4b_ucsc_gb_actb_genetracksonly.png" style="width:90.0%" /> 
</details>
<p><br> We see the complete <em>ACTB</em> gene with exons as horizontally stretched rectangles and introns as horintal lines with arrowheads in them indicating the direction of transcription. The exons have different heights. The highest regions correspond to <em>coding</em> exons, the other regions are the 5- and 3UTR.<br />
<br> More information about <em>ACTB</em> can be obtained by clicking on the gene in the browser. Doing that brings you to a gene page like:<br />
<img src="images/wk1_4b_ucsc_gb_actb_geneinformation.png" style="width:80.0%" /> <br />
<br></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 2: How many exons does the *ACTB* gene have?&quot;</span>,</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>          <span class="fu">answer</span>(<span class="st">&quot;5&quot;</span>, <span class="at">message =</span> <span class="st">&quot;exons are displayed as thicker lines in the gene, introns as lines with arrows in them.&quot;</span>),</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>          <span class="fu">answer</span>(<span class="st">&quot;6&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>          <span class="fu">answer</span>(<span class="st">&quot;7&quot;</span>, <span class="at">message =</span> <span class="st">&quot;one exon can contain coding and non-coding (UTR) elements&quot;</span> ),</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>          <span class="fu">answer</span>(<span class="st">&quot;8&quot;</span>, <span class="at">message =</span> <span class="st">&quot;one exon can contain coding and non-coding (UTR) elements&quot;</span>),</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">allow_retry =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><br></p>
<p>Figure for Question 3:<br />
<img src="images/wk1_4b_ucsc_gb_actb_question_TSSquestion.png" /></p>
<p><br></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 3: Which letter points to the TSS?&quot;</span>,</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;A&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Pay attention to how directionality is visualized in the UCSC Genome Browser, discussed in video 1 at ±4:30&quot;</span>),</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;B&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Pay attention to how directionality is visualized in the UCSC Genome Browser, discussed in video 1 at ±4:30&quot;</span>),</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;C&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Pay attention to the visualization of UTRs, discussed in video 1 at ±5:10&quot;</span>),</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;D&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Pay attention to the visualization of UTRs, discussed in video 1 at ±5:10&quot;</span>),</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;E&quot;</span>, <span class="at">correct =</span> T))</span></code></pre></div>
<p><br></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>     <span class="fu">question</span>(<span class="st">&quot;Question 4: What statement about the *ACTB* gene is correct? (Select ALL that apply.) &quot;</span>,</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;ACTB gene is encoded on the minus-strand&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;The TTS of ACTB gene has a higher coordinate than its TSS&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Look at the direction of arrows in the introns.&quot;</span>),</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;The coding size of this gene is appr 2kb&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;If we want to look at the promoter of this gene, we have to move slightly to the right&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>              <span class="fu">answer</span>(<span class="st">&quot;The UTRs of this genes are distributed over 3 exons&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><br></p>
<p><span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Go to the gene <em>SPI1</em> </span><br />
<br></p>
<p><em>SPTI1</em> encodes the myeloid and B-cell–specific transcription factor PU.1 <a href="https://doi.org/10.1182/blood-2012-01-402453">Pham 2012 Blood</a>.</p>
<ul>
<li>Copy this sentence and fill in the blanks with numbers in the code:</li>
</ul>
<p><span style="color: red;">Are up- &amp; downstream discussed?</span><br />
<span style="color: red;">Klaas: no, I do not explicitly discuss up- and down-stream. Maybe explain this here very briefly. Maybe it is good practice in general to briefly explain key terms (e.g. TSS) in the practical as a reminder. This can be in the form of the &lt;U+2018&gt;background&lt;U+2019&gt; box you also used for explaining the CD markers/monocytes.</span></p>
<p>SPI1 gene is located on chr […]. It spans […] bp and is encoded on the [plus/minus] strand. It has […] exons and […] introns. It is an [zinc finger domain TF/ETS-domain transcription factor]. The first downstream protein-gene is [MYBPC3/SLC39A13].</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Fill in the blanks with numbers in the code:  </span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>SPI1 gene is located on chr ...  . It spans ... bp and is encoded on the [plus<span class="sc">/</span>minus] strand. It has ... coding exons and ... introns. It is an [zinc finger domain TF<span class="sc">/</span>ETS<span class="sc">-</span>domain transcription factor]. The first downstream protein<span class="sc">-</span>gene is [MYBPC3<span class="sc">/</span>SLC39A13].  </span></code></pre></div>
<p><strong>Wonderfull, you are now able to recognize genes, their location within the chromosome, and their coding elements as well as their orientation. These are important reference points used when we evaluate the quality of epigenomics data and to formulate hypothesis that we can test later on.</strong></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="viewing-blueprint-data" class="section level3">
<h3>1.4.3 Viewing BLUEPRINT data</h3>
<p><strong><em>Yes! Finally time for some real data! Using the UCSC Genome Browser, you will inspect characteristic histone PTM and DNaseI-seq combinations, what their location in relation to genes and - ultimately - what this may say about the gene’s transcription state.</em></strong></p>
<p><br> To upload the data we have already put them together in a so-called track hub, hosted by our own server. We can add this to our browser session by going to the tab “My Data” in the top menu and select “Track Hubs”. In the “My hubs” tab you can paste a URL that directs to the data you want to upload.<br />
<br></p>
<p><img src="images/wk1_4c_ucsc_gb_encode_fig1_gototrackhub.png" style="width:100.0%" alt="Go to Track Hubs" /> </p>
<p><img src="images/wk1_4c_ucsc_gb_encode_fig2_trackhub.png" style="width:100.0%" alt="My Hubs window" /> </p>
<p><br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Upload the required data using the following URL […].</span><br />
<br><br />
Note that, after adding the trackhub, an additional section is added below the browser, specfic to the tracks in this trackhub.<br />
<br> This trackhub contains histone PTM ChIP-seq and DNaseI-seq data of monocytes, to be precise of <strong>CD14+, CD16- classical monocytes</strong>. <br></p>
<blockquote>
<p><em>Background</em>: CD14 and CD16 are cell surface markers of the <strong>CD (Cluster of Differentiation) gene family</strong> which we will come across more often during this practical. These cell surface molecules and their differential expression are used for immunophenotyping of cells. CD molecules can function as receptors, ligands and, for example, activate signalling cascades, mediate cell adhesion, cell migration, regulate gene expression etc. <a href="https://www.genenames.org/data/genegroup/#!/group/471">HUGO, genegoups, CD</a></p>
</blockquote>
<p><br> We have the following datasets:</p>
<ul>
<li><p>ChIP-seq data histone PTMs:</p>
<ul>
<li>H3K9me3 ChIP-seq (gray)<br />
</li>
<li>H3K27me3 ChIP-seq (gray)<br />
</li>
<li>H3K4me3 ChIP-seq (red)<br />
</li>
<li>H3K4me1 ChIP-seq (yellow)<br />
</li>
<li>H3K27me1 ChIP-seq (peach)<br />
</li>
<li>H3K36me3 ChIP-seq (orange)</li>
</ul></li>
<li><p>DNaseI-seq data in a blue track.<br />
</p></li>
<li><p>polyA-selected, stranded RNA-seq data, again in green<br />
</p></li>
<li><p>Input ChIP-seq in black.</p></li>
</ul>
<blockquote>
<p><em>Tip</em>: What we haven’t discussed so far is the y-axis at the left side of the track with <em>signal</em> information such as the ones just uploaded. Using this y-axis we can evaluate signal intensities. The height is currently set to 30, 50 or 100. This corresponds to the number of reads mapping to that location per million sequenced reads. You can change the height of the y-axis by right-mouse click on the trakname, select the <code>Configure [trackname]</code> option and change under “Vertical viewing range”, the min" and “max”. Hit <code>APPLY</code> afterwards.</p>
</blockquote>
<p><br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Go back to the <em>ACTB</em> locus and zoom out 3x to see the surrounding region.</span></p>
<blockquote>
<p><em>Hint</em>: your browser should be showing chr7:5,523,694-5,534,055 approximately, a few bases more or less is not a problem.</p>
</blockquote>
<p><br></p>
<blockquote>
<p><span style="color: red;">UCSC GB session for the following exercises (tip: use right-mouse click to not loose this tutorial page): Monocyte wiggle data (blueprint in colors, encode in black): <a href="http://genome-euro.ucsc.edu/s/ctoenhake/hg38_monos_blueprint_and_encode" class="uri">http://genome-euro.ucsc.edu/s/ctoenhake/hg38_monos_blueprint_and_encode</a></span></p>
</blockquote>
<p>Use the RNA-seq data…</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 6: would you conclude that the ACTB gene is active?&quot;</span>,</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">answer</span>(<span class="st">&quot;Yes&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>       <span class="fu">answer</span>(<span class="st">&quot;No&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Look at the signal in RNA-seq data&quot;</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><br> <span style="color: red;">Klaas remark: would it be possible to have the students indicate why they conclude this? Perhaps by having them indicate the track on which they based their conclusion?</span><br />
<br></p>
<p>Within the Genomics tracks we can discriminate regions of high signal from the background. These are the peaks.<br />
<br></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 7: Which of the following regions points to a DNaseI-seq peak?)&quot;</span>,</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;chr7:5530101-5530501&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Nice try but this region has a H3K27ac ChIP-seq peak.&quot;</span>),</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;chr7:5530501-5530800&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>, <span class="at">message =</span> <span class="st">&quot;Yes. Note that this DNaseI-seq peak is located at the *ACTB* TSS/promoter&quot;</span>),</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;chr7:5532001-5532500&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Try again. If anything, you may be looking at the H3K4me1 signal.&quot;</span>, ),</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;None of the these&quot;</span>, <span class="at">message =</span> <span class="st">&quot;Try again. Please look again&quot;</span>),</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> <span class="cn">TRUE</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>     )</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 8: Which histone marks, are detected at the ACTB promoter? (Select ALL that apply)&quot;</span>,</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K9me3 is not detected here&quot;</span>),</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K27me3 is not detected here&quot;</span>),</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>, <span class="at">message =</span> <span class="st">&quot;correct&quot;</span>),</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K4me1&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K4me1 isn ot detected here&quot;</span>),</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>, <span class="at">message =</span> <span class="st">&quot;correct&quot;</span>),</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K36me3 is detected covering the gene body, not the promoter.&quot;</span>),</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;none&quot;</span>, <span class="at">message =</span> <span class="st">&quot;wrong&quot;</span>))</span></code></pre></div>
<p><br></p>
<p><span style="color: red;">Or is the promoter spanning 3kb and including the H3K4me1 peak?</span><br />
<span style="color: red;">Klaas remark: indeed H3K4me1 also seems to be detected at a distande of ~ 2kb from the TSS. Could be enhancer embedded in distal promoter, or could be proximal enhancer region. Maybe Simon can help distinguish this. Alternatively, we can define the promoter as a set number of bp from the TSS for this exercise</span></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="peaks-from-peak-calling-software" class="section level3">
<h3>1.4.4 Peaks from peak calling software</h3>
<p><strong><em>It would be cumbersume if we would have to identify all peaks manually, not to mention a highly biased procedure. Luckily we have peak calling software to help us with that. In the following section we will use those results along with the tracks.</em></strong><br />
<br> The peaks have been compiled in another trackhub: [URL].<br />
<br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Upload the peak data.</span><br />
<br> <em>Called</em> peaks are represented as horizontal bars. <br></p>
<p><span style="color: orange; font-family: Arial; font-size:18px;">ACTION: From <em>ACTB</em> locus, zoom out 100x.</span></p>
<blockquote>
<p><em>Hint</em>: your browser should be showing chr7:5,356,175-5,701,574 approximately, a few bases more or less is not a problem.</p>
</blockquote>
<p>A prominant H3K27me3 peaks comes in the viewing window now. It covers the promoter of <em>FSCN1</em> gene.</p>
<p><span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Zoom into that gene <em>uncluding</em> the prominant H3K27me3 peak.</span></p>
<blockquote>
<p><em>Hint</em>: your browser should be showing chr7:5,587,501-5,611,400 approximately, a few bases more or less is not a problem.</p>
</blockquote>
<blockquote>
<p><span style="color: red;">UCSC GB session for the following exercises (tip: use right-mouse click to not loose this tutorial page): Monocyte wiggle &amp; peak data, blueprint only: <a href="http://genome-euro.ucsc.edu/s/ctoenhake/hg38_monos_blueprint_wiglle_and_peaks" class="uri">http://genome-euro.ucsc.edu/s/ctoenhake/hg38_monos_blueprint_wiglle_and_peaks</a></span></p>
</blockquote>
<p><br></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 9: What additional histone marks, mark this region? (Select ALL that apply.)&quot;</span>,</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;we don&#39;t see H3K9me3&quot;</span>),</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K4me1&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>, <span class="at">message =</span> <span class="st">&quot;A very thin line, not a &#39;clear&#39; peak&quot;</span>),</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;we don&#39;t see H3K36me3 signal.&quot;</span>),</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;none&quot;</span>, <span class="at">message =</span> <span class="st">&quot;We are referring to region chr7:5,373,445-5,684,304.&quot;</span>)</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>         )</span></code></pre></div>
<p><br></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 10: Is *FSCN1* transcribed?&quot;</span>,</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;Yes&quot;</span>),</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;No&quot;</span>, <span class="at">correct =</span> T)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>         )</span></code></pre></div>
<p><br> This is a candidate bivalent promoter. <em>FSCN1</em> encodes a member of the fascin family (3 genes in humans) of actin-binding proteins. Fascin proteins organize filamentous actin polymers into parallel bundles. This is required for the formation of actin-based cellular protrusions during cell migration, motility, adhesion and cellular interactions of neurons, glial cells and also dendritic cells <a href="https://doi.org/10.4049/jimmunol.171.4.1825">Bros, 2003, J Immunol</a>. The bivalent marking of this promoter in monocytes makes sense, knowing that a proportion of monocytes differentiates into dendritic cell (also called antigen-presenting cell in this <a href="https://ctoen.github.io/FunctGen/#Hematopoiesis_scheme">scheme</a>).<br />
<br> The region inbetween <em>ACTB</em> and <em>FSCN1</em> is marked as well. What marks do you observe and what functional elements could they point to?<br />
<br></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 11: epigenetic marks inbetween *ACTB* and *FSCN1* (Select ALL that apply.)&quot;</span>,</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K9me3 is not detected here&quot;</span>),</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K27me3 is not detected here&quot;</span>),</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K4me1&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K36me3 is detected covering the gene body, not the promoter.&quot;</span>),</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;DNaseI-seq&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>))</span></code></pre></div>
<p><br></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 12: What functional element do these marks most likely reflect?&quot;</span>,</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;terminator&quot;</span>),</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;promoter&quot;</span>),</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;TSS&quot;</span>),</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;enhancer&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>, <span class="at">message =</span> <span class="st">&quot;At least the region spanning chr7:5560200-5564149 is reported a super-enhancer and controls among others the *ACTB* gene.&quot;</span>),</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;centromere&quot;</span>),</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;telomere&quot;</span>),</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;coding region&quot;</span>)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><br> Another highly expressed gene in this cell type is <em>CD14</em>, remember it’s protein was used to select these cells on. Answer the following question about this gene using the UCSC Genome Browser:<br />
<br></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 13: What statement about *CD14* is correct? (Select ALL that apply) &quot;</span>,</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;CD14 gene is encoded on the minus-strand&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;The TTS of CD14 gene has a higher coordinate than its TSS&quot;</span>), </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">asnwer</span>(<span class="st">&quot;*CD14* is encoded on the long arm of chr5&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">asnwer</span>(<span class="st">&quot;*CD14* is encoded on the short arm of chr5&quot;</span>),</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;The TSS of *CD14* is not accessible for DNaseI&quot;</span>), </span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;A DNaseI-peak appr 1.5 kb upstream of the TSS marks a candidate enhancer.&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><br> Approximately 20 kb downstream of <em>CD14</em> is another enhancer. What histone marks populate/mark this enhancer? (Downstream with respect to the <em>CD14</em> orientation.)</p>
<blockquote>
<p>For testing purposes: the enhancer I’m pointing to is located at chr5:140611060-140612601</p>
</blockquote>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 14: Which marks mark the enhancer 20kb downstream of *CD14*?&quot;</span>,</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;not H3K9me3&quot;</span>),</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;not H3K27me3&quot;</span>),</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;not H3K4me3&quot;</span>),</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K4me1&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;not H3K36me3&quot;</span>),</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;DNaseI&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> T)</span></code></pre></div>
<p><br> So far we have seen active promoters, bivalent promoters and enhancers. A region silenced by H3K9me3 is loacted appr. 120kb upstream of <strong>CD14</strong>.<br />
<br></p>
<blockquote>
<p>Tip: zoom out to chr5:140,147,501-141,605,000.</p>
</blockquote>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 15: The expression of what gene family is controlled by this marking?&quot;</span>,</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;PCDHA gene family&quot;</span>, <span class="at">correct =</span> T, <span class="at">message =</span> <span class="st">&quot;PCDHA gene cluster encodes a family of cadherin-like cell surface proteins that are expressed in neurons and are present at synaptic junctions.&quot;</span>),</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="fu">answer</span>(<span class="st">&quot;A group of LINC RNAs&quot;</span>),</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;Family of KRAB-ZNF encoding genes&quot;</span>, <span class="at">message =</span> <span class="st">&quot;not H3K27me3&quot;</span>),</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> T)</span></code></pre></div>
<p><br></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 16: What is the approximate size of the H3K9me3 silenced region? (*Tip2: hold shift and click-and-drag in the brwoser to capture and thereby measure the width of the H3K9me3 domain)&quot;</span>,</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;250kb&quot;</span>),</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;500kb&quot;</span>),</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;750kb&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>),</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;1Mb&quot;</span>),</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> T)</span></code></pre></div>
<p><br> Silencing does not necessarily have to be mediated by H3K9me3.<br />
<br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Go to the ACTA1 gene, encoding muscular actin. Zoom out 10x.</span></p>
<blockquote>
<p><em>Hint</em>: you should be at ± chr1:229,418,420-229,446,919.</p>
</blockquote>
<p><br></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 17: Which histone mark most likely mediates repression of *ACTA1* in these cells?&quot;</span>,</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K9me3 is not the most likely repressing mark here&quot;</span>),</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>,  <span class="at">message =</span> <span class="st">&quot;H3K4me3 is not associated with repression.&quot;</span>),</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K4me1&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K4me1 is generally not associated with repression.&quot;</span>),</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K27as is not associated with repression.&quot;</span>),</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K36me3 is not associated with repression.&quot;</span>),</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;none&quot;</span>, <span class="at">message =</span> <span class="st">&quot;wrong&quot;</span>),</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">allow_retry =</span> <span class="cn">TRUE</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>         )</span></code></pre></div>
<p><br> H3K27me3 is not exclusive for silencers (Huang, 2019, Gen Res)[<a href="https://genome.cshlp.org/content/29/4/657" class="uri">https://genome.cshlp.org/content/29/4/657</a>]</p>
<p><br></p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 18: Which other observations support that this gene is not active? (Select ALL that apply)&quot;</span>,</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;Absence of H3K36me3 marking covering the gene body.&quot;</span>, <span class="at">message =</span> <span class="st">&quot;H3K36me3 is indeed lacking but other observations are correct as well.&quot;</span>),</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;Absence of signal in RNA-seq signal&quot;</span>, <span class="at">message =</span> <span class="st">&quot;RNA-seq is indeed lacking but other observations are correct as well.&quot;</span>),</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;Absence of H3K4me3 signal at the *ACTA1* promoter&quot;</span>,  <span class="at">message =</span> <span class="st">&quot;correct&quot;</span>),</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>         <span class="fu">answer</span>(<span class="st">&quot;All the above&quot;</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>, <span class="at">message =</span> <span class="st">&quot;Indeed, the absence of signals in RNA-seq, H3K36me3 *and* H3K4me3 all support that this gene is not expressed. Observe that the H3K27me3 signal covers a relatively broad domain.&quot;</span> ),</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">allow_retry =</span> <span class="cn">TRUE</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>         )</span></code></pre></div>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="recognizing-blinded-genomics-data" class="section level3">
<h3>1.4.5 Recognizing blinded genomics data</h3>
<p>So far you have browsed through ChIP-, DNaseI- and RNA-seq data. These tracks had the correct track names attached to them.<br />
<br> Our collegue performed ChIPs for H3K9me3, H3K27me3, H3K4me3, H3K36me3, H3K4me1 and H3K27ac on CD4+ T cells but accidentally let the tubes fall before the tubes were properly labelled (always label your tubes first and <em>before</em> adding your sample!). <br> We know which ChIPs were performed, but which track (labelled a-f) is most likely which ChIP?<br />
<br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: upload the following track hub [URL].</span><br />
<br><br />
Which column from the following table reflects the most likely decoding?</p>
<table>
<thead>
<tr class="header">
<th align="center">ChIP</th>
<th align="center">encoding 1</th>
<th align="center">encoding 2</th>
<th align="center">encoding 3</th>
<th align="center">encoding 4</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">H3K9me3</td>
<td align="center">a</td>
<td align="center">f</td>
<td align="center">d</td>
<td align="center">b</td>
</tr>
<tr class="even">
<td align="center">H3K27me3</td>
<td align="center">b</td>
<td align="center">a</td>
<td align="center">e</td>
<td align="center">c</td>
</tr>
<tr class="odd">
<td align="center">H3K4me3</td>
<td align="center">c</td>
<td align="center">b</td>
<td align="center">f</td>
<td align="center">d</td>
</tr>
<tr class="even">
<td align="center">H3K4me1</td>
<td align="center">d</td>
<td align="center">c</td>
<td align="center">a</td>
<td align="center">e</td>
</tr>
<tr class="odd">
<td align="center">H3K27ac</td>
<td align="center">e</td>
<td align="center">d</td>
<td align="center">b</td>
<td align="center">f</td>
</tr>
<tr class="even">
<td align="center">H3K36me3</td>
<td align="center">f</td>
<td align="center">e</td>
<td align="center">c</td>
<td align="center">a</td>
</tr>
</tbody>
</table>
<blockquote>
<p><span style="color: red;">UCSC GB session: didn’t make this one yet!</span></p>
</blockquote>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 19: What is the most likely encoding?&quot;</span>,</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;encoding 1&quot;</span>),</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;encoding 2&quot;</span>),</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;encoding 3&quot;</span>),</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;encoding 4&quot;</span>),</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> <span class="cn">TRUE</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="summary" class="section level3">
<h3>1.4.6 Summary</h3>
<p>To summarize, we have seen various marks, finish the following statements based on your observations in the UCSC genome browser:</p>
<p><span style="color: red;">I prefer these be formulated as hypotheses with no correct answer but instead ‘save’ the answers to a file uploaded in week 2 so that students can re-evalue their answers. Perhaps ask the questions again in week 2 but that time answer them using the calculations of the whole genome/chr21</span></p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 20: Active promoters are recognized by markings ...&quot;</span>,</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;...&quot;</span>),</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;...&quot;</span>),</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;...&quot;</span>),</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;...&quot;</span>),</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> <span class="cn">TRUE</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><br></p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 21: The marks found at bivalent promoters are ...&quot;</span>,</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;...&quot;</span>),</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;...&quot;</span>),</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;...&quot;</span>),</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;...&quot;</span>),</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> <span class="cn">TRUE</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><br></p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 22: Active enhancers are marked by ...&quot;</span>,</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;...&quot;</span>),</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;...&quot;</span>),</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;...&quot;</span>),</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;...&quot;</span>),</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> <span class="cn">TRUE</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><br></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 23: H3K9me3-mediated silencing often spans ...&quot;</span>,</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;small domain, single genes&quot;</span>),</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;broad domain, multple genes&quot;</span>, <span class="at">correct =</span> T),</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> <span class="cn">TRUE</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><br></p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">question</span>(<span class="st">&quot;Question 24: The gene body of a transcribed gene (eg *ACTB* or *CD14* gene in  monocytes) is covered by ...&quot;</span>,</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>),</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>),</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>),</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K4me1&quot;</span>),</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>),</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>, <span class="at">correct =</span> T, <span class="at">message =</span> <span class="st">&quot;Indeed, H3K36me3 is generally associated active chromatin. Depending on where in the gene and when in the transcriptional prociess H3K36 is methylated (and to what degree), this modification has been found to have distinct roles, ranging from preventing aberrant transcriptional initiation within coding sequences to recruitment of splicing factors, dosage compensation and more. The first occrus in yeast, by recruitment of the deactylase complex enforcing a deacetylated chromatin state in the wake of transcribing RNAPII [Wagner and Carpenter, 2014, Nat Rev Mol Cell Biol](https://doi.org/10.1038/nrm3274)).&quot;</span>),</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">allow_retry =</span> <span class="cn">TRUE</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><br></p>
<hr />
<p><br></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
