<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>1. UCSC Genome Browser</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="custom.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Functional Genomics 2020</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="p2_learningObjectives.html">Learning Objectives</a>
</li>
<li>
  <a href="p3_script.html">Script</a>
</li>
<li>
  <a href="p4_week1.html">W1</a>
</li>
<li>
  <a href="p5_week2.html">W2</a>
</li>
<li>
  <a href="p6_intro_bioconductor.html">DC Bioconductor</a>
</li>
<li>
  <a href="p7_links.html">Links</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">1. UCSC Genome Browser</h1>

</div>


<style type="text/css">
.title {
  display: none;
}

#getting-started img {
  margin-right: 10px;
}

</style>
<div class="row" style="padding-top: 20px;">


<div id="week-1" class="section level1">
<h1>Week 1:</h1>
<ul>
<li>Raw ChIP-seq data processing and visualize epigenomics data in the UCSC Genome browser.<br />
</li>
<li>Global learning objectives #1-3.<br />
<br></li>
</ul>
<blockquote>
<h4 id="learning-objectives">Learning Objectives</h4>
<ol style="list-style-type: decimal">
<li>Know how raw reads (from eg ChIP-seq) are processed to obtain trustworthy data.<br />
</li>
<li>Understand how to move (“<em>browse</em>”) through the UCSC genome browser and know how to go to regions and genes of interest.<br />
</li>
<li>Recognize the following elements/regions in the UCSC genome browser: chromosome, centromere, telomere, gene, exon, intron, utr, promoter, enhancer/repressor.<br />
</li>
<li>Tell apart UCSC genome browser tracks of H3K27ac, H3K27me3, H3K36me3, H3K4me1, H3K4me3 and H3K9me3 histone ChIP-seq and ATAC-seq signal (and ChIP-seq input) based on their coverage profile and their location with respect to genes and other annotated genomic/chromosomal elements.</li>
<li>Can interpret - and thus understand - ChIP- and ATAC-seq signal in the UCSC genome browser.</li>
</ol>
</blockquote>
<p><br></p>
<div id="genomics-methods-recapdiscussion" class="section level3">
<h3>1.1 Genomics methods recap/discussion</h3>
<p><strong>Discuss</strong>:</p>
<ul>
<li>What is ChIP-seq? What is ATAC-seq? Presentation with ChIP-seq steps? Refer to book?<br />
</li>
<li>Sequencing samples. What happens in sequencer? What is the output?<br />
</li>
<li>Brief genomics recap/pre-quiz?</li>
</ul>
<p><br></p>
</div>
<div id="from-raw-data-to-science" class="section level3">
<h3>1.2 From raw data to science</h3>
<p><img src="images/week1/whatbioinformaticiancsdo.png" /> <br></p>
<p><a href="https://vanheeringen-lab.github.io/seq2science/index.html">Seq2science</a><br />
<em>fastq file, bam processing, bigwig, peak calling, count table</em></p>
<p><strong>Discuss</strong>:</p>
<ul>
<li>Modify Siebrens pptx.<br />
</li>
<li>Fastq files, what does the data look like?<br />
</li>
<li>4 lines of code with:
<ol style="list-style-type: decimal">
<li>unique_read_name (starts with @-sign)<br />
</li>
<li>nucleotide sequence<br />
</li>
<li><code>+</code> (previously, this line showed the same as line 1)<br />
</li>
<li>nucleotide quality score (Phred encoded)</li>
</ol></li>
<li>The neccessary processing steps:
<ol style="list-style-type: lower-alpha">
<li>Read quality and adapter trimming<br />
</li>
<li>Aligning fastq reads to reference genome<br />
</li>
<li>BAM sieving.<br />
</li>
</ol>
<ul>
<li>What considerations are important? PCR duplicates, multi mapping reaads<br />
</li>
<li>Discuss use of input sample<br />
</li>
</ul>
<ol start="4" style="list-style-type: lower-alpha">
<li>Peak calling. Use of control samples. Narrow and Broad peaks?<br />
</li>
<li>Count table: What does it look?<br />
<br></li>
</ol></li>
</ul>
<blockquote>
<h4 id="working-questions">Working questions</h4>
<ol style="list-style-type: decimal">
<li>What histone mark marks the promoter of TNF gene? Let’s go and look for it ;).<br />
</li>
<li>Now your turn, what about CD3 is this gene active in monocytes? (no it is not, also logical, it is part of the T cel receptor present on the cellsurface of T cells)<br />
</li>
<li>Look at the enhancer located 20kb upstream of …. [gene] …. Based on the histone and chromatin accessibility data would you say this enhancer is active or inactive? What histone marks do you find around it?<br />
</li>
<li><em>Summarizing</em> What marks are generally found at promoters, at gene bodies, active enhancers and at repressed domains?</li>
</ol>
</blockquote>
<p><strong>Week 1</strong>:</p>
<div id="the-ucsc-genome-browser" class="section level4">
<h4>The UCSC Genome Browser</h4>
<p>The UCSC Genome Browser is a visualization tool that uses the genomic coordinate system as reference<br />
- coordinates are written as chromosome:start-end.<br />
- 1-based<br />
- is a visualization tool of genomics data - hosted by the …<br />
- can be accessed via:<br />
- Depending on the reference genome of your interest, many or few additional tracks containing information are present</p>
<p>A <strong>track</strong> is a graph within the genome browser, spanning the complete genome</p>
<p><strong>Browsing</strong> theUCSC genome browser, scanning through the genome, can be done in various ways:<br />
- On top use the <code>&lt;&lt;&lt;</code>, <code>&lt;&lt;</code>, <code>&lt;</code> and <code>&gt;</code>, <code>&gt;&gt;</code> and <code>&gt;&gt;&gt;</code> buttons - Using a</p>
<ul>
<li><strong>Focus</strong>: how we make data out of a ChIP-seq experiment, how to use the UCSC Genome Browser, and familiarize ourselves with the UCSC Genome Browser visualization.<br />
</li>
<li><em>seq2science, UCSC Genome browser</em><br />
</li>
<li>Global objectives #1-3:
<ol style="list-style-type: decimal">
<li>Understand how we come from a ChIP-sequencing experiment to useful data.</li>
<li>Know how to display and browse through genomics data in the UCSC Genome Browser and formulate some basic hypotheses using this tool.<br />
</li>
<li>Tell apart different genomics datasets based on their pattern int he UCSC Genome Browser.<br />
</li>
</ol></li>
<li><p>Perform in H5P?<br />
<br></p></li>
<li>Discuss ChIP-seq in presentation<br />
</li>
<li>Follow seq2science workflow<br />
</li>
<li><p>Peakcalling: note that it is not black/white, not absolute, thresholds are set. NOT into technicalities and mathematics of read piling/peak models, narrow-wide peaks<br />
<br></p></li>
</ul>
</div>
</div>
<div id="look-at-the-read-density-yourself-in-a-genome-browser" class="section level3">
<h3>1.3 Look at the read density yourself in a genome browser</h3>
<p><em>Functional elements, ChIP signals, peaks, hypotheses about enrichment.</em></p>
<p><strong>Discuss</strong>:</p>
<ul>
<li>UCSC Genome Browser
<ul>
<li>how to switch off tracks. Hide all tracks except for <strong>GENCODE v32</strong> and <strong>Base Position</strong>.<br />
</li>
<li>how to browse trough the data<br />
</li>
<li>how to upload custom data. Upload the <span style="color: red;">processed chip-seq data of histone marks in <strong>heatlhy cells</strong>, input and a <strong>failed track</strong> as .bigwig file. Or use trackhub?</span>
<ul>
<li>bigwig is a special data format to display dense, continuous data as graphs. The main advantage of this format is that only the portions of the files needed to dispaly a particular regoin are transferred making it considerably faster in a genome browsing tool <a href="https://genome.ucsc.edu/goldenPath/help/bigWig.html">ref</a>.<br />
</li>
</ul></li>
<li>As functional genomic elements: <span style="color: red;">upload custom track or use <strong>ENCODE cCREs</strong> track?</span></li>
</ul></li>
</ul>
<p>Now browse through the data.</p>
<p>Peaks are …</p>
<p><strong>Assignments/Questions</strong>:</p>
<ol style="list-style-type: decimal">
<li>One of the tracks represents the input, which one and why?<br />
</li>
<li>What kind of data is visualized in track “A”?<br />
</li>
<li>Browse to these four regions, which one of them would you consider to display a peak (tip, make screenshot and paste in a temporary pptx. to compare the images)<br />
</li>
<li>Of the marks visualized which ones are associated with gene expression and which ones with repression (from lecture)?</li>
</ol>
<p><strong>Discuss</strong>:<br />
Peak calling software is used to identify peaks within a certain range of confidence. <span style="color: red;">and using replictes?</span></p>
<p><strong>Assignments/Questions</strong>:</p>
<ol start="5" style="list-style-type: decimal">
<li>Which of the following influences the number of peaks identified:<br />
</li>
</ol>
<ul>
<li>P-value threshold, the lower the fewer peaks<br />
</li>
<li>read depth<br />
</li>
<li>Q-value/false-discovery rate, the<br />
</li>
<li>the type of control sample …<br />
</li>
<li></li>
</ul>
<p><strong>Discuss</strong>:</p>
<p>Upload peaks called by the software.</p>
<p><strong>Assignments/Questions</strong>:</p>
<ol start="6" style="list-style-type: decimal">
<li>What are different functional genomic elements you have heard of during lectures?</li>
</ol>
<p>The custom track contains five tracks, numbered 1-5, representing different genomic elements.</p>
<ol start="7" style="list-style-type: decimal">
<li>Which # represent what kind of element?<br />
</li>
<li>Browse through the browser, estimate for each histone mark whether it is enriched in one or several of the genomic elements (1-5)</li>
</ol>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
