<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>2. Peaks in r shortened</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="custom.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">FG 2020</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="p3_week1.html">Week 1</a>
</li>
<li>
  <a href="p4_week2_v4.html">Week 2</a>
</li>
<li>
  <a href="prepdata.html">Appendix</a>
</li>
<li>
  <a href="example1_sequence_short_fastqc.html">FASTQC_exmpl1</a>
</li>
<li>
  <a href="example2_sequence_short_fastqc.html">FASTQC_exmpl2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">2. Peaks in r shortened</h1>

</div>


<style type="text/css">
.title {
  display: none;
}

</style>
<p><em>focus: genomic ranges, overlap functions, nearest functions.</em></p>
<div id="introduction" class="section level2">
<h2>2.1 Introduction</h2>
<p><br> In week 1 we discussed the major steps imvolved in obtaining genomics data, from experiment to raw data to normalized signal and peaks. We examined histone PTM ChIP-, DNaseI-, and RNA-seq data of monocytes in the UCSC genome browser and searched for regions with increased signal, <em>peaks</em>. We looked at the position of these peaks in the chromosome and with respect to genes as well as the co-occurence of different marks.<br />
<br> These observations showed you that some marks are associated with particular functional elements, as was also discussed in class (eg active enhancer, active promoter, silenced domains).<br />
<br> <strong>This week and next week we will put numbers to these observations and perform computational analyses to answer common questions including:</strong> “<em>How many peaks do I have?</em>”, “<em>Is this mark statistically enriched in a particular genomic element?</em>”, “<em>Which and how often do marks co-occur?</em>”,“<em>The expression of which gene could be affected by this mark?</em>”, and “<em>What is the signal of the mark around a particular element of interest like the TSS?</em>”<br />
<br></p>
<div id="learning-objectives" class="section level3">
<h3>2.1.1 Learning Objectives</h3>
<blockquote>
<p>At the end of week 2 you are able to:</p>
<ol style="list-style-type: decimal">
<li>Create <code>GRanges</code> and <code>GRangesList</code> objects.<br />
</li>
<li>Import ChIP-seq peaks into a <code>GRanges</code> object in r.<br />
</li>
<li>Apply common <code>base</code> and <code>plyranges</code> functions to summarize <code>GRanges</code> objects.<br />
</li>
<li>Detect and count overlap between two <code>GRanges</code> objects using the <code>findOverlaps()</code> function from the <code>GenomicRanges</code> package.<br />
</li>
<li>Visualize the number of overlaps in a vennDiagram using the <code>VennDiagram</code> package.<br />
</li>
<li>Visualize the number of overlaps with bar- and pieplots.<br />
</li>
<li>Statistically test for enrichment of histone marks in a particular genomic region.</li>
<li>Identify the nearest TSS to a histone ChIP-seq peak using the <code>distanceToNearest()</code> function from the <code>GenomicRanges</code> package.<br />
</li>
<li>Identify the genes that are associated with these TSSs.<br />
These refer to global learning objectives #4-#7.</li>
</ol>
</blockquote>
<p><br></p>
<hr />
<p><br></p>
<blockquote>
<p>TEASER… let’s look at some of the plots you will make.</p>
</blockquote>
<p><br></p>
<hr />
<p><br></p>
</div>
</div>
<div id="introducing-genomicranges" class="section level2">
<h2>2.2 Introducing GenomicRanges</h2>
<div id="background-bioconductor" class="section level3">
<h3>2.2.1 Background Bioconductor</h3>
<p><a href="https://www.bioconductor.org/">Bioconductor</a> is a special repository of packages for the analysis of high throughput genomic data in r. Packages are often heavily dependend on each other and regular releases and a special installer ensure that you install package version that can work together. Usually you would use the <code>biocManager::install()</code> command to install these packages. But in this tutorial, all required packages are already installed.<br />
<br> Bioconductor packages that we will be using include:</p>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Package</th>
<th>Purpose</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="https://bioconductor.org/packages/release/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html">GenomicRanges</a></td>
<td>Provides the <code>GRanges</code> data structures to store and handle genomic intervals.</td>
</tr>
<tr class="even">
<td><a href="https://bioconductor.org/packages/release/bioc/html/plyranges.html">plyranges</a></td>
<td><code>dplyr</code>-like interface for interacting and manipulating <code>Ranges</code> data structures.</td>
</tr>
<tr class="odd">
<td><a href="https://www.bioconductor.org/packages/release/bioc/html/GenomicFeatures.html">GenomicFeatures</a></td>
<td>Functions to retrieve and manage genomic features from database packages.</td>
</tr>
<tr class="even">
<td><a href="https://www.bioconductor.org/packages/release/bioc/html/rtracklayer.html">rtracklayer</a></td>
<td>Provides functions to import and work with annotation files with various formats (GFF, BED, bedGraph, BED15, WIG, BigWig and 2bit).</td>
</tr>
<tr class="odd">
<td><code>TxDb</code> packages</td>
<td>Provide an R representation of gene models, each genome has a separate package.</td>
</tr>
<tr class="even">
<td><code>OrgDb</code> packages</td>
<td>Contain mappings between a unique gene identifier and other kinds of identifiers for a certain genome (eg from Entrez gene identifier to Gene Symbol)</td>
</tr>
</tbody>
</table>
<p><br> Bioconductor packages can use <strong>data structures</strong> different from <code>vectors</code>,<code>matrices</code> or <code>data.frames</code>. The <code>GRanges</code> data structure provided by the <code>GenomicRanges</code> package is an example and is used by many. We will come to those shortly.<br />
<br> These “Bioconductor-specific-objects”, like <code>GRanges</code>, generally have their own set of functions or methods that you can explore using <code>methods(class = &quot;...&quot;)</code>. To find out the <code>class</code>, use <code>class(object_name)</code>.<br />
<br> Here’s a list of basic functions to explore objects and/or call the help on packages or functions:</p>
<ul>
<li>Use <code>class()</code> to found out what kind of data structure you are dealing with<br />
</li>
<li>Use <code>show()</code> or <code>print()</code> to have r print a summary of your data<br />
</li>
<li>use <code>methods(class = &quot;...&quot;)</code> to get a list of (default) methods or functions you can use to extract and manipulate the data, also called <em>assessor</em> functions or <em>getter</em> and <em>setter</em> functions<br />
</li>
<li>use <code>help(package = &quot;[packageName]&quot;)</code> to display the help page</li>
</ul>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="genomicranges-constructing-granges-objects" class="section level3">
<h3>2.2.2 GenomicRanges: Constructing GRanges objects</h3>
<p>You are already familiar with the <strong>data structures</strong> <code>vector</code>, <code>matrix</code>, <code>data.frame</code> and <code>lists</code> in R. In genomics we often work with <strong>interval</strong> data. Think of peaks, genes, exons, … any genomic region reported with tge genomic coordinates :<code>chr</code>, <code>start</code>, and <code>end</code>.<br />
<br> Storing these in a <code>data.frame</code> is possible but not very efficient. E.g. a simple manipulation such as shifting all reported intervals 2 bp to the right, requires you to manipulate 2 columns at the same time.<br />
<br> Interval data can be more efficiently handled with the <code>IRanges</code> package which works with a data structure especially developed for <strong>ranges of integers</strong>: <code>IRanges</code> objects.<br />
<br> To construct an <code>IRanges</code> object we require at least two of the following three values: a starting coordinate, a finishing coordinate or the width of the interval.<br />
<br> Here we create <code>IRranges</code> object <code>ir</code> with three intervals, starting at positions <code>3</code>, <code>5</code> and <code>17</code>, and ending at positions <code>10</code>, <code>20</code> and <code>30</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">ir &lt;-<span class="st"> </span><span class="kw">IRanges</span>(<span class="dt">start=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">17</span>), <span class="dt">end=</span><span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">20</span>,<span class="dv">30</span>))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">print</span>(ir)</a></code></pre></div>
<pre><code>## IRanges object with 3 ranges and 0 metadata columns:
##           start       end     width
##       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
##   [1]         3        10         8
##   [2]         5        20        16
##   [3]        17        30        14</code></pre>
<p>Depicting <code>ir</code> visually we see blocks representing the intervals along the horizontal axis:<br />
<img src="p4_week2_v4_files/figure-html/IRanges%20plot-1.png" width="480" /> <br> The <code>GRanges</code> objects of the <code>GenomicRanges</code> package are very similar but require a additional <code>sequence name</code> (in other words a chromosome) for every interval. <br> Here we reate a <code>GRanges</code> object using <code>ir</code> and call for it’s <code>class</code> and summary (as this prints 6 lines by default, the whole object is printed in this case):<br />
<br></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># create gr object: </span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">gr &lt;-<span class="st"> </span><span class="kw">GRanges</span>(<span class="dt">seqnames =</span> <span class="kw">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr2&quot;</span>), <span class="dt">ranges=</span>ir)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co"># find out the class:  </span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="kw">class</span>(gr)</a></code></pre></div>
<pre><code>## [1] &quot;GRanges&quot;
## attr(,&quot;package&quot;)
## [1] &quot;GenomicRanges&quot;</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># print a summary:</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw">print</span>(gr)</a></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr1      3-10      *
##   [2]     chr1      5-20      *
##   [3]     chr2     17-30      *
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<p><br> The <code>strand</code> column holds <code>*</code>, meaning the intervals are <em>unstranded</em>.<br />
<br> Here we create <code>gr</code> again but define the strand for every interval with <code>strand =</code></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># create gr object with strandinfo: </span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2">gr &lt;-<span class="st"> </span><span class="kw">GRanges</span>(<span class="dt">seqnames =</span> <span class="kw">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr2&quot;</span>), </a>
<a class="sourceLine" id="cb7-3" data-line-number="3">              <span class="dt">ranges=</span>ir, </a>
<a class="sourceLine" id="cb7-4" data-line-number="4">              <span class="dt">strand =</span> <span class="kw">c</span>(<span class="st">&quot;+&quot;</span>, <span class="st">&quot;-&quot;</span>, <span class="st">&quot;-&quot;</span>))</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">gr</a></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 0 metadata columns:
##       seqnames    ranges strand
##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;
##   [1]     chr1      3-10      +
##   [2]     chr1      5-20      -
##   [3]     chr2     17-30      -
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<p><br> We can add scores and names to these intervals with <code>score(object_name) &lt;-</code> and <code>names(object_name) &lt;-</code>. The names will replace the row idnentifiers. Additional custom columns can be defined with the <code>$</code>-sign.<br />
<br> Here we add names, scores and a gc% to each interval:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">set.seed</span>(<span class="dv">1234</span>) <span class="co"># set.seed to ensure that with the random number generation, we each time generate the same sequence of numbers</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co"># add names to `my_gr`</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="kw">names</span>(gr) &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;peak&quot;</span>, LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>], <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="co"># add a custom score</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="kw">score</span>(gr) &lt;-<span class="st"> </span><span class="kw">round</span>(<span class="kw">runif</span>(<span class="dv">3</span>, <span class="dv">0</span>, <span class="dv">106</span>), <span class="dt">digits =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="co"># add custom GC content</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7">gr<span class="op">$</span>gc &lt;-<span class="st"> </span><span class="kw">round</span>(<span class="kw">runif</span>(<span class="dv">3</span>, <span class="fl">0.35</span>, <span class="fl">0.60</span>), <span class="dt">digits =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co"># print the result</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="kw">print</span>(gr)</a></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 2 metadata columns:
##          seqnames    ranges strand |     score        gc
##             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;
##   peak_A     chr1      3-10      + |    12.053     0.506
##   peak_B     chr1      5-20      - |    65.964     0.565
##   peak_C     chr2     17-30      - |    64.583     0.510
##   -------
##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</code></pre>
<p><br> <code>GRanges</code> objects follow the <strong>tidy data principle</strong>: each row of a <code>Ranges</code> object corresponds to an interval, and each column will represent a variable about that interval, and generally each object will represent a single unit of an observation (like gene annotations). <br> Notice that the names replace the rowIDs and that this object now holds 2 <strong>metadata columns</strong> besides the genomic coordinates. Metadata is <strong>data about the data</strong>:<br />
<br></p>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Genomic coordinates</th>
<th>Metadata columns</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Printed on the left-hand side of the <code>|</code></td>
<td>Printed on the right-hand side of the <code>|</code></td>
</tr>
<tr class="even">
<td>Extract using <code>granges(object_name)</code></td>
<td>Extract as DataFrame with <code>mcols(object_name)</code> or <code>object_name$column_name</code> for a specific column</td>
</tr>
<tr class="odd">
<td>Columns are restricted to <code>seqnames</code>, <code>ranges</code> and <code>strand</code></td>
<td>Almost anything can be stored in the metadata, we defined <code>score</code> and <code>GC</code>-content here</td>
</tr>
</tbody>
</table>
<p><br> Information about the genome and sequences is stored in <code>seqinfo</code>. This currently tells us that the intervals are from 2 sequences of a unknown genome. Genome information can be appended using <code>seqinfo(my_gr) &lt;- Seqinfo(genome=&quot;hg38&quot;)</code>. This automatically loads the sequence information specified by <code>genome</code> by querying the UCSC database. We skip this step to limit the size of the object.<br />
<br></p>
<blockquote>
<p><strong>Exercise 1 - Finish the following code, creating <code>my_gr2</code> with:</strong></p>
<ul>
<li>Invervals A-E located on resp. chr1, chr2, chr3, chr4 and chr5 (one interval per chromosome);<br />
</li>
<li>from A to E, starts are respectively at position <code>3</code>, <code>5</code>, <code>13</code>, <code>18</code>, and <code>20</code>;<br />
</li>
<li>and a <code>width</code> of <code>5</code>, <code>7</code>, <code>6</code>, <code>9</code> and <code>11</code> bp;<br />
</li>
<li>the intervals A, B and C on the minus strand, D and E on the plus strand;<br />
</li>
<li>random scores in a variable <code>score</code>;<br />
</li>
<li>and a random GC% in the <code>GC</code>-column.<br />
</li>
<li>Then add names to each row with <code>names()</code><br />
Finish by printing the summary of <code>my_gr</code> using the <code>print()</code> function.</li>
</ul>
</blockquote>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">set.seed</span>(<span class="dv">1234</span>) <span class="co"># set.seed to ensure that with the random number generation, we each time generate the same sequence of numbers.  </span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"></a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="co"># create GRanges object &#39;my_gr&#39; intervals A-C: </span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4">my_gr &lt;-<span class="st"> </span><span class="kw">....</span>(<span class="dt">.. =</span> <span class="kw">c</span>(<span class="st">&quot;...&quot;</span>), </a>
<a class="sourceLine" id="cb11-5" data-line-number="5">              <span class="dt">.... =</span> <span class="kw">IRanges</span>(<span class="dt">start =</span> <span class="kw">c</span>(..), <span class="dt">... =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">7</span>, <span class="dv">9</span>, <span class="dv">11</span>)),</a>
<a class="sourceLine" id="cb11-6" data-line-number="6">              <span class="dt">.. =</span> <span class="kw">c</span>(<span class="st">&quot;-&quot;</span>, <span class="st">&quot;-&quot;</span>, ..)</a>
<a class="sourceLine" id="cb11-7" data-line-number="7">              <span class="dt">.. =</span> <span class="kw">round</span>(<span class="kw">runif</span>(<span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">106</span>), <span class="dt">digits =</span> <span class="dv">3</span>),</a>
<a class="sourceLine" id="cb11-8" data-line-number="8">              <span class="dt">gc =</span> <span class="kw">round</span>(<span class="kw">runif</span>(<span class="dv">5</span>, <span class="fl">0.35</span>, <span class="fl">0.60</span>), <span class="dt">digits =</span> <span class="dv">3</span>) </a>
<a class="sourceLine" id="cb11-9" data-line-number="9">              )</a>
<a class="sourceLine" id="cb11-10" data-line-number="10"></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="co"># add names to `my_gr`</span></a>
<a class="sourceLine" id="cb11-12" data-line-number="12"><span class="kw">..</span>(my_gr) &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;peak&quot;</span>, LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>], <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</a>
<a class="sourceLine" id="cb11-13" data-line-number="13"></a>
<a class="sourceLine" id="cb11-14" data-line-number="14"><span class="co"># print summary of my_gr</span></a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="kw">..</span>(my_gr)</a></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">set.seed</span>(<span class="dv">1234</span>) <span class="co"># set.seed to ensure that with the random number generation, we each time generate the same sequence of numbers.  </span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2"></a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="co"># create GRanges object &#39;my_gr&#39; intervals A-C: </span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4">my_gr &lt;-<span class="st"> </span><span class="kw">GRanges</span>(<span class="dt">seqnames =</span> <span class="kw">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr2&quot;</span>, <span class="st">&quot;chr3&quot;</span>, <span class="st">&quot;chr4&quot;</span>, <span class="st">&quot;chr5&quot;</span>), </a>
<a class="sourceLine" id="cb12-5" data-line-number="5">              <span class="dt">ranges =</span> <span class="kw">IRanges</span>(<span class="dt">start=</span><span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">5</span>, <span class="dv">13</span>, <span class="dv">18</span>, <span class="dv">20</span>), <span class="dt">width =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">7</span>, <span class="dv">9</span>, <span class="dv">11</span>)),</a>
<a class="sourceLine" id="cb12-6" data-line-number="6">              <span class="dt">strand =</span> <span class="kw">c</span>(<span class="st">&quot;-&quot;</span>, <span class="st">&quot;-&quot;</span>, <span class="st">&quot;-&quot;</span>,<span class="st">&quot;+&quot;</span>, <span class="st">&quot;+&quot;</span>),</a>
<a class="sourceLine" id="cb12-7" data-line-number="7">              <span class="dt">score =</span> <span class="kw">round</span>(<span class="kw">runif</span>(<span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">106</span>), <span class="dt">digits =</span> <span class="dv">3</span>),</a>
<a class="sourceLine" id="cb12-8" data-line-number="8">              <span class="dt">gc =</span> <span class="kw">round</span>(<span class="kw">runif</span>(<span class="dv">5</span>, <span class="fl">0.35</span>, <span class="fl">0.60</span>), <span class="dt">digits =</span> <span class="dv">3</span>) </a>
<a class="sourceLine" id="cb12-9" data-line-number="9">              )</a>
<a class="sourceLine" id="cb12-10" data-line-number="10"></a>
<a class="sourceLine" id="cb12-11" data-line-number="11"><span class="co"># add names to `my_gr`</span></a>
<a class="sourceLine" id="cb12-12" data-line-number="12"><span class="kw">names</span>(my_gr) &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;peak&quot;</span>, LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>], <span class="dt">sep =</span> <span class="st">&quot;_&quot;</span>)</a>
<a class="sourceLine" id="cb12-13" data-line-number="13"></a>
<a class="sourceLine" id="cb12-14" data-line-number="14"><span class="co"># print summary of my_gr</span></a>
<a class="sourceLine" id="cb12-15" data-line-number="15"><span class="kw">print</span>(my_gr)</a></code></pre></div>
<pre><code>## GRanges object with 5 ranges and 2 metadata columns:
##          seqnames    ranges strand |     score        gc
##             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;
##   peak_A     chr1       3-7      - |    12.053     0.510
##   peak_B     chr2      5-11      - |    65.964     0.352
##   peak_C     chr3     13-21      - |    64.583     0.408
##   peak_D     chr4     18-28      + |    66.078     0.517
##   peak_E     chr5     20-24      + |    91.257     0.479
##   -------
##   seqinfo: 5 sequences from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
</div>
<div id="working-with-granges" class="section level3">
<h3>2.3.2 Working with <code>GRanges</code></h3>
<p><strong>Accessor functions</strong> extract components from the <code>GRanges</code> object.</p>
<ul>
<li><code>start()</code>, <code>end()</code>, <code>width()</code>, <code>ranges()</code> and <code>strand()</code> accessors extract the respective elements from the genomic coordinates.<br />
</li>
<li><code>score()</code> prints the scores stored in metadata.</li>
<li><code>names()</code> prints the rownames.<br />
</li>
<li>Custom metadata columns can be accessed using the <code>$</code>- sign.</li>
</ul>
<p><br> Here’s a visual summary of a <code>GRanges</code> object called <code>my_gr</code> with the accessors to extract its content:</p>
<p><img src="images/wk2_2_GRanges_fields_my_gr.png" /></p>
<p><br> Additional common functions to summarize and view <code>GRanges</code> objects include:</p>
<ul>
<li><code>length()</code> prints the number of stored intervals.</li>
<li><code>head(object_name, n = ...)</code> and <code>tail(object_name, n = ...)</code> print <code>n</code> number of intervals in the head or tail of the object. Can be usefull for checking that the dataset is complete.<br />
</li>
<li>GRanges behave like vectors of ranges and can be subsetted using <code>[subset range, metadata columns]</code>. E.g. <code>my_gr[3:5, &quot;GC&quot;]</code> will print only rows 3 to 5 and of the metadata only the <code>GC</code>-column.</li>
</ul>
<p><br> <strong>dplyr-related functions</strong> are available for <code>GRanges</code> through the <code>plyranges</code> package. (The <code>dplyr</code> package it self is not compatable with <code>GRanges</code>.) As the function names of <code>dplyr</code> and <code>plyranges</code> overlap, we sometimes use the formulation <code>package::function</code> to clarify to you the package we are using.<br />
<br> A brief reminder of common <code>dplyr</code> functions and examples of <code>plyranges</code>-relatives on <code>GRanges</code> objects:</p>
<table>
<colgroup>
<col width="33%" />
<col width="33%" />
<col width="33%" />
</colgroup>
<thead>
<tr class="header">
<th>Function</th>
<th>Purpose</th>
<th>Example on <code>GRanges</code></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>select()</code></td>
<td>subset variables (=columns)</td>
<td><code>select(my_gr, GC)</code></td>
</tr>
<tr class="even">
<td><code>group_by()</code></td>
<td>group data into rows with the same value for the specified variable.</td>
<td><code>my_gr %&gt;% group_by(strand)</code>.</td>
</tr>
<tr class="odd">
<td><code>filter()</code></td>
<td>subset observations (= rows)</td>
<td><code>filter(my_gr, GC &lt; 0.4)</code> or <code>my_gr %&gt;% group_by(strand) %&gt;%  filter(GC == max(GC))</code>.</td>
</tr>
<tr class="even">
<td><code>summarize()</code></td>
<td>Summarise variables, often per group</td>
<td><code>group_by(my_gr, strand) %&gt;% summarize(n = n(), gc = max(GC))</code></td>
</tr>
</tbody>
</table>
<p><br> We can also use the <code>pipe</code> operator <code>%&gt;%</code> to combine functions in a workflow.<br />
<br> <strong>We will practice using these functions on real peak data, which we will import in the next section</strong>.<br />
<br></p>
<hr />
<p><br></p>
</div>
</div>
<div id="peak-files-in-r" class="section level2">
<h2>2.3 Peak files in R</h2>
<div id="data-import" class="section level3">
<h3>2.3.1 Data import</h3>
<p>Most genomic interval data comes in a tabular format that has the basic information about the location of the interval and some other information. Common file formats include BED, GFF or BigWig files. These can be imported with the base <code>object_name &lt;- read.table(&quot;location/of/file.bed&quot;)</code> function and converted to a <code>GRanges</code> object using <code>gr_object &lt;- as(object_name, &quot;GRanges&quot;)</code>.<br />
<br> A quicker method is to use the <code>import()</code> function from the <code>rtracklayer</code> package. This function parses the files directly into <code>GRanges</code> objects with genomic ranges and metadata.<br />
<br> You will import BED files with the genomic locations of the peaks from the histone ChIP-seq experiment in monocytes we looked at in week 1. On the <a href="http://dcc.blueprint-epigenome.eu/#/md/chip_seq_grch38">BLUEPRINT methods page</a> we can read that peak calling was performed with MACS2 software. We also note that for 4 ChIPs, the <code>-broad</code> option was included resulting in so-called <code>broadPeak</code> files. The others follow the <code>narrowPeak</code> file format. This is important because we need to specify this file format in our <code>import()</code> function.</p>
<table>
<thead>
<tr class="header">
<th>narrowPeak file</th>
<th>broadPeak file</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>H3K27ac</td>
<td>H3K27me3</td>
</tr>
<tr class="even">
<td>h3k4me3</td>
<td>H3K36me3</td>
</tr>
<tr class="odd">
<td>H3K9/14ac</td>
<td>H3K9me3</td>
</tr>
<tr class="even">
<td>H2A.Zac</td>
<td>H3K4me1</td>
</tr>
</tbody>
</table>
<p>These two files are much alike (see <a href="https://github.com/macs3-project/MACS">MACS2 github</a>). They hold the following information about the identified peaks:</p>
<table>
<colgroup>
<col width="25%" />
<col width="25%" />
<col width="25%" />
<col width="25%" />
</colgroup>
<thead>
<tr class="header">
<th>Column</th>
<th>Information</th>
<th>narrowPeak description</th>
<th>broadPeak description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1.</td>
<td>chromosome name</td>
<td>same</td>
<td>same</td>
</tr>
<tr class="even">
<td>2.</td>
<td>peak start</td>
<td>same</td>
<td>same</td>
</tr>
<tr class="odd">
<td>3.</td>
<td>peak end</td>
<td>same</td>
<td>same</td>
</tr>
<tr class="even">
<td>4.</td>
<td>peak name</td>
<td>same</td>
<td>same</td>
</tr>
<tr class="odd">
<td>5.</td>
<td>score</td>
<td>-10log(qvalue) * 10, rounded down to integer value)</td>
<td>mean of -10log(qvalue) * 10 <em>across all positions</em></td>
</tr>
<tr class="even">
<td>6.</td>
<td>strand</td>
<td>+, - or * for unstranded</td>
<td>same</td>
</tr>
<tr class="odd">
<td>7.</td>
<td>fold_enrichment</td>
<td>at peak summit</td>
<td>mean across all positions</td>
</tr>
<tr class="even">
<td>8.</td>
<td>-log10(qvalue) (<em>e.g.if qvalue = 1e-10, this value is 10</em>)</td>
<td>at peak summit</td>
<td>mean <em>across all positions</em></td>
</tr>
<tr class="odd">
<td>9.</td>
<td>-log10(pvalue)</td>
<td>at peak summit</td>
<td>mean <em>across all positions</em></td>
</tr>
<tr class="even">
<td>10.</td>
<td>relative summit</td>
<td>position relative to peak start</td>
<td>not reported</td>
</tr>
</tbody>
</table>
<p>We have restricted all peak files to chromosome 19 to limit their file size.</p>
<blockquote>
<p><strong>Exercise 2:</strong></p>
<ul>
<li>Read in the following peak files wih the <code>import()</code> function from the <code>rtracklayer</code> package.<br />
</li>
<li>Define the appropriate file format, i.e. “narrowPeak” or “broadPeak”.</li>
</ul>
</blockquote>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="co"># H3k4me1 </span></a>
<a class="sourceLine" id="cb15-2" data-line-number="2">mono_h3k4me1 &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import</span>(<span class="st">&quot;data/blueprint/bed/C000S5H2.ERX547981.H3K4me1.bwa.GRCh38.broad.20150527.chr19.bed&quot;</span>, <span class="dt">format =</span> <span class="st">&quot;...&quot;</span>)</a>
<a class="sourceLine" id="cb15-3" data-line-number="3"></a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="co"># h3k4me3  </span></a>
<a class="sourceLine" id="cb15-5" data-line-number="5">mono_h3k4me3 &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">...</span>(<span class="st">&quot;data/blueprint/bed/C000S5H2.ERX547984.h3k4me3.bwa.GRCh38.20150527.chr19.bed&quot;</span>, <span class="dt">format =</span> <span class="st">&quot;...&quot;</span>)</a>
<a class="sourceLine" id="cb15-6" data-line-number="6"></a>
<a class="sourceLine" id="cb15-7" data-line-number="7"><span class="co"># H3K9me3 </span></a>
<a class="sourceLine" id="cb15-8" data-line-number="8">mono_h3k9me3 &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">...</span>(<span class="st">&quot;data/blueprint/bed/C000S5H2.ERX547982.H3K9me3.bwa.GRCh38.broad.20150527.chr19.bed&quot;</span>, <span class="dt">format =</span> <span class="st">&quot;...&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="co"># H3k4me1 </span></a>
<a class="sourceLine" id="cb16-2" data-line-number="2">mono_h3k4me1 &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import</span>(<span class="st">&quot;data/blueprint/bed/C000S5H2.ERX547981.H3K4me1.bwa.GRCh38.broad.20150527.chr19.bed&quot;</span>, <span class="dt">format =</span> <span class="st">&quot;broadPeak&quot;</span>)</a>
<a class="sourceLine" id="cb16-3" data-line-number="3"></a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="co"># h3k4me3  </span></a>
<a class="sourceLine" id="cb16-5" data-line-number="5">mono_h3k4me3 &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import</span>(<span class="st">&quot;data/blueprint/bed/C000S5H2.ERX547984.h3k4me3.bwa.GRCh38.20150527.chr19.bed&quot;</span>, <span class="dt">format =</span> <span class="st">&quot;narrowPeak&quot;</span>)</a>
<a class="sourceLine" id="cb16-6" data-line-number="6"></a>
<a class="sourceLine" id="cb16-7" data-line-number="7"><span class="co"># H3K9me3 </span></a>
<a class="sourceLine" id="cb16-8" data-line-number="8">mono_h3k9me3 &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw">import</span>(<span class="st">&quot;data/blueprint/bed/C000S5H2.ERX547982.H3K9me3.bwa.GRCh38.broad.20150527.chr19.bed&quot;</span>, <span class="dt">format =</span> <span class="st">&quot;broadPeak&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<p><br> Awesome! Let’s take a quick look at one of these files.</p>
<blockquote>
<p><strong>Exercise 3:</strong><br />
Print the head 2 lines of <code>mono_h3k4me3</code> using the function <code>head()</code>.</p>
</blockquote>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw">head</span>(mono_h3k4me3, <span class="dt">n =</span> <span class="dv">2</span>)</a></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 6 metadata columns:
##       seqnames        ranges strand |                            name     score
##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; |                     &lt;character&gt; &lt;numeric&gt;
##   [1]    chr19 245884-246269      * | 3337.macs2_peak_call_peak_22416       317
##   [2]    chr19 266835-267653      * | 3337.macs2_peak_call_peak_22417      1805
##       signalValue    pValue    qValue      peak
##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;
##   [1]     14.0559    33.902   31.7868       190
##   [2]     50.0226   183.133  180.5450       544
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<p><br> The original .BED files lacked column headers. The <code>rtracklayer::import()</code> function appendended variable names and converted the data to the data type (eg numeric, integer, character etc.) according to the file format we defined.<br />
<br> <strong>Let’s do some exploratory analyses on these data.</strong>.<br />
<br></p>
<hr />
<p><br></p>
</div>
<div id="exploratory-analysis" class="section level3">
<h3>2.3.2 Exploratory analysis</h3>
<p>We will use the functions described for <code>GRanges</code> objects in section 2.2.3 to <code>mono_h3k4me3</code>.</p>
<blockquote>
<p><strong>Exercise 4:</strong></p>
<ul>
<li>Use the <code>length()</code> function to determine the number of h3k4me3 peaks in monocytes.<br />
</li>
<li>What is the distribution of peak sizes and scores? Use <code>summary()</code>, <code>width()</code> and <code>score()</code>.</li>
</ul>
</blockquote>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="co"># How many peaks do we have?  </span></a></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="co"># How many peaks do we have?  </span></a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="kw">length</span>(mono_h3k4me3)</a></code></pre></div>
<pre><code>## [1] 2838</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="kw">grade_result</span>(</a>
<a class="sourceLine" id="cb24-2" data-line-number="2">  <span class="kw">pass_if</span>(<span class="op">~</span><span class="kw">identical</span>(.result, <span class="dv">2838</span>))</a>
<a class="sourceLine" id="cb24-3" data-line-number="3">)</a></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="co"># What is the distribution of peak sizes?   </span></a>
<a class="sourceLine" id="cb25-2" data-line-number="2"><span class="kw">summary</span>(<span class="kw">...</span>(...))</a></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="co"># What is the distribution of peak sizes?   </span></a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="kw">summary</span>(<span class="kw">width</span>(mono_h3k4me3))</a></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   200.0   389.2   879.5  1198.6  1661.2 12705.0</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1"><span class="co"># What is the distribution of peak scores?</span></a></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1"><span class="co"># What is the distribution of peak scores?</span></a>
<a class="sourceLine" id="cb30-2" data-line-number="2"><span class="kw">summary</span>(<span class="kw">score</span>(mono_h3k4me3))</a></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##      15      84     374    1212    2415    5228</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<p><br></p>
</div>
<div id="exploratory-plotting" class="section level3">
<h3>2.3.3 Exploratory plotting</h3>
<p>Let’s plot some of the above generated outputs using <strong>base</strong> plotting functions.</p>
<blockquote>
<p><strong>Exercise 5:</strong><br />
Plot the h3k4me3 peaksizes as a histogram and as a boxplot using the base <code>hist()</code> and <code>boxplot()</code> functions.</p>
</blockquote>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="co"># plot the distribution of peak sizes for h3k4me3 as histogram</span></a>
<a class="sourceLine" id="cb33-2" data-line-number="2"><span class="kw">..</span>(<span class="kw">..</span>(mono_h3k4me3), <span class="dt">main =</span> <span class="st">&quot;Monocytes h3k4me3 ChIP-seq, chr19&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;peak size&quot;</span>, <span class="dt">col =</span> <span class="st">&quot;gray&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1"><span class="co"># plot the distribution of peak sizes for h3k4me3 as histogram </span></a>
<a class="sourceLine" id="cb34-2" data-line-number="2"><span class="kw">hist</span>(<span class="kw">width</span>(mono_h3k4me3), <span class="dt">main =</span> <span class="st">&quot;Monocytes h3k4me3 ChIP-seq, chr19&quot;</span>, <span class="dt">xlab =</span> <span class="st">&quot;peak size&quot;</span>, <span class="dt">col =</span> <span class="st">&quot;gray&quot;</span>)</a></code></pre></div>
<p><img src="p4_week2_v4_files/figure-html/q5a_exploreplot-solution-1.png" width="672" /></p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1"><span class="co"># plot the same data as boxplot, ensue that &quot;peak size&quot; labels the right axis (x or y)</span></a>
<a class="sourceLine" id="cb36-2" data-line-number="2"><span class="kw">..</span>(..., <span class="dt">main =</span> <span class="st">&quot;Monocytes h3k4me3 ChIP-seq, chr19&quot;</span>, <span class="dt">... =</span> <span class="st">&quot;peak size&quot;</span>, <span class="dt">col =</span> <span class="st">&quot;gray&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1"><span class="co"># plot the same data as boxplot, ensue that &quot;peak size&quot; labels the right axis (x or y)</span></a>
<a class="sourceLine" id="cb37-2" data-line-number="2"><span class="kw">boxplot</span>(<span class="kw">width</span>(mono_h3k4me3), <span class="dt">main =</span> <span class="st">&quot;Monocytes h3k4me3 ChIP-seq, chr19&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;peak size&quot;</span>, <span class="dt">col =</span> <span class="st">&quot;gray&quot;</span>)</a></code></pre></div>
<p><img src="p4_week2_v4_files/figure-html/q5b_exploreplot-solution-1.png" width="672" /></p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<p><br> &gt; <em>Note</em>: plotting with the <code>ggplots</code> packages is also possible, as long as we convert the data we are interested to <code>data.frame</code> objects by running for example: <code>meta_h3k4me3 &lt;- as.data.frame(mcols(mono_h3k4me3))</code>.</p>
<p><br> Several peaks have a width of &gt;10000 bp. You retrieve these peaks with: <code>mono_h3k4me3[width(mono_h3k4me3)&gt; 10000,]</code> or using <code>filter()</code> from <code>plyranges</code>.<br />
<br></p>
<blockquote>
<p><strong>Exercise 6:</strong><br />
Use [ ]-subsetting and <code>length()</code> to find the number of peaks with width &gt; 10kb.</p>
</blockquote>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1"><span class="co"># How many mono_h3k4me3 peaks have width &gt; 10kb&gt;</span></a></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1"><span class="co"># using [ ]-subsetting</span></a>
<a class="sourceLine" id="cb40-2" data-line-number="2"><span class="kw">length</span>(mono_h3k4me3[<span class="kw">width</span>(mono_h3k4me3) <span class="op">&gt;</span><span class="st"> </span><span class="dv">10000</span>,])</a></code></pre></div>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1"><span class="kw">grade_result</span>(</a>
<a class="sourceLine" id="cb41-2" data-line-number="2">  <span class="kw">pass_if</span>(<span class="op">~</span><span class="kw">identical</span>(.result, <span class="dv">4</span>))</a>
<a class="sourceLine" id="cb41-3" data-line-number="3">)</a></code></pre></div>
<blockquote>
<p><strong>Exercise 7:</strong><br />
Filter <code>mono_h3k4me3</code> for peaks with width &gt; 10kb using <code>filter()</code> from <code>plyranges</code></p>
</blockquote>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1"><span class="co"># filter mono_h3k4me3 for peaks with size above 10kb</span></a>
<a class="sourceLine" id="cb42-2" data-line-number="2">.. <span class="op">%&gt;%</span><span class="st"> </span>plyranges<span class="op">::</span><span class="kw">..</span>(.. <span class="op">&gt;</span><span class="st"> </span>..)</a></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1"><span class="co"># filter mono_h3k4me3 for peaks with size above 10kb</span></a>
<a class="sourceLine" id="cb43-2" data-line-number="2">mono_h3k4me3 <span class="op">%&gt;%</span><span class="st"> </span>plyranges<span class="op">::</span><span class="kw">filter</span>(width <span class="op">&gt;</span><span class="st"> </span><span class="dv">10000</span>)</a></code></pre></div>
<pre><code>## GRanges object with 4 ranges and 6 metadata columns:
##       seqnames            ranges strand |                            name
##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |                     &lt;character&gt;
##   [1]    chr19   4057708-4069982      * | 3337.macs2_peak_call_peak_22766
##   [2]    chr19 13094389-13105943      * | 3337.macs2_peak_call_peak_23309
##   [3]    chr19 35896740-35909444      * | 3337.macs2_peak_call_peak_24016
##   [4]    chr19 39404035-39414347      * | 3337.macs2_peak_call_peak_24149
##           score signalValue    pValue    qValue      peak
##       &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;
##   [1]      4368    100.8720   441.685   436.885      8901
##   [2]      3267     79.7881   330.122   326.784      8268
##   [3]      3088     46.5275   312.010   308.813     11185
##   [4]      4273     97.9021   431.897   427.309      4278
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<p><br></p>
<blockquote>
<p><strong>Exercise 8</strong><br />
Use the <code>plyranges::filter</code> and the UCSC genome browser to find the gene(s) whose promoter(s) is/are covered by the peak with the highest score.</p>
</blockquote>
<blockquote>
<p>hints:</p>
<p>1: condition on <code>score == max(score)</code></p>
</blockquote>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" data-line-number="1"><span class="co"># find the peak peak with highest score</span></a></code></pre></div>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" data-line-number="1"><span class="co"># find the peak peak with highest score</span></a>
<a class="sourceLine" id="cb47-2" data-line-number="2">mono_h3k4me3 <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(score <span class="op">==</span><span class="st"> </span><span class="kw">max</span>(score))</a></code></pre></div>
<pre><code>## GRanges object with 1 range and 6 metadata columns:
##       seqnames            ranges strand |                            name
##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |                     &lt;character&gt;
##   [1]    chr19 15377793-15380841      * | 3337.macs2_peak_call_peak_23451
##           score signalValue    pValue    qValue      peak
##       &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;
##   [1]      5228     116.995   530.186   522.803      1381
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;The promoters of which gene(s) are covered by the h3k4me3 peak with the highest score&quot;</span>?,</a>
<a class="sourceLine" id="cb49-2" data-line-number="2">         <span class="kw">answer</span>(<span class="st">&quot;*AKAP8*&quot;</span>, <span class="dt">correct =</span> T, <span class="dt">message =</span> <span class="st">&quot;While this peak spans parts of both genes, it covers the promoter of *AKAP8* but the 3&#39;UTR of *AKAP8L*.&quot;</span> ),</a>
<a class="sourceLine" id="cb49-3" data-line-number="3">         <span class="kw">answer</span>(<span class="st">&quot;*AKAP8L*&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Incorrect. Be aware of the directionaility of this gene by looking at the arrowsheads in the introns.&quot;</span>),</a>
<a class="sourceLine" id="cb49-4" data-line-number="4">         <span class="kw">answer</span>(<span class="st">&quot;Both&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Only one of the genes listed here. Look at the direction of transcription indicated by the arrowheads in the introns of the genes.&quot;</span>)</a>
<a class="sourceLine" id="cb49-5" data-line-number="5">)</a></code></pre></div>
<p><br> <strong>Later on in this tutorial we will learn how to annotate peaks with the gene that they most likely control. First we will calculate the fraction of overlap among two <code>GRanges</code> objects. </strong><br />
<br></p>
<hr />
<p><br> ## 2.4 Overlap analysis<br />
### 2.4.1 Detect overlap with GenomicRanges<br />
The <code>GenomicRanges</code> package has a family of functions to count and identify overlappig intervals in <code>GRanges</code> objects:<br />
<br> <strong>countOverlaps</strong>: <code>countOverlaps(query, subject)</code> returns a integer vector with the number of overlaps for each element in the <code>query</code><br />
<br> <strong>subsetByOverlaps</strong>: <code>subsetByOverlaps(query, subject)</code> extracts the elements in the <code>query</code> that overlap with at least one element in the subject.<br />
<br> <strong>findOverlaps</strong>: <code>findOverlaps(query, subject)</code> returns a <code>Hits</code> object containing the index pairings for the overlapping elements.</p>
<ul>
<li>The columns of indices can be accessed through <code>queryHits(overlap_object)</code> and <code>subjectHits(overlap_object)</code>.<br />
</li>
<li>If a peak in one of the inputs overlaps with mutiple peaks in the other, its index will appear multiple times in the output.</li>
</ul>
<p><br><br />
### 2.4.2 Overlap promoters vs h3k4me3 peaks<br />
We have used the package <code>TxDb.Hsapiens.UCSC.hg38.knownGene</code> to retrieve the genomic coordinates of all genes on human chromosome 19. This object, <code>genes</code>, thus contains all the coordinates of the outermost UTR boundaries and the Entrez gene identifier of each gene.<br />
<br> We define the promoter region as 1kb upstream and 200bp downstream the TSS as follows:<br />
<br></p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" data-line-number="1"><span class="co"># identify TSS (in this case 1 per gene even though we know that genes have multiple TSSs)</span></a>
<a class="sourceLine" id="cb50-2" data-line-number="2">tss &lt;-<span class="st"> </span><span class="kw">resize</span>(genes, <span class="dt">width =</span> <span class="dv">1</span>, <span class="dt">fix =</span> <span class="st">&quot;start&quot;</span>)</a>
<a class="sourceLine" id="cb50-3" data-line-number="3"></a>
<a class="sourceLine" id="cb50-4" data-line-number="4"><span class="co"># view TSS object</span></a>
<a class="sourceLine" id="cb50-5" data-line-number="5"><span class="kw">head</span>(tss, <span class="dt">n =</span> <span class="dv">3</span>)</a></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 1 metadata column:
##             seqnames    ranges strand |     gene_id
##                &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
##           1    chr19  58362751      - |           1
##   100049587    chr19  51646801      - |   100049587
##   100073347    chr19  56840866      + |   100073347
##   -------
##   seqinfo: 1 sequence from hg38 genome</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" data-line-number="1"><span class="co"># consider 1kb upstream and 200bp downstream the TSS as promoter</span></a>
<a class="sourceLine" id="cb52-2" data-line-number="2">promoters &lt;-<span class="st"> </span><span class="kw">resize</span>(tss, <span class="dt">width =</span> <span class="dv">1000</span>, <span class="dt">fix =</span> <span class="st">&quot;end&quot;</span>)</a>
<a class="sourceLine" id="cb52-3" data-line-number="3">promoters &lt;-<span class="st"> </span><span class="kw">resize</span>(promoters, <span class="dt">width =</span> <span class="dv">1200</span>, <span class="dt">fix =</span> <span class="st">&quot;start&quot;</span>)</a>
<a class="sourceLine" id="cb52-4" data-line-number="4"></a>
<a class="sourceLine" id="cb52-5" data-line-number="5"><span class="co"># show promoters object</span></a>
<a class="sourceLine" id="cb52-6" data-line-number="6"><span class="kw">head</span>(promoters, <span class="dt">n =</span> <span class="dv">3</span>)</a></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 1 metadata column:
##             seqnames            ranges strand |     gene_id
##                &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
##           1    chr19 58362551-58363750      - |           1
##   100049587    chr19 51646601-51647800      - |   100049587
##   100073347    chr19 56839867-56841066      + |   100073347
##   -------
##   seqinfo: 1 sequence from hg38 genome</code></pre>
<p><br> We want to know how many of the promoters overlap with a h3k4me3 peak and vise versa.<br />
<br></p>
<blockquote>
<p><strong>Exercise 10:</strong><br />
Use <code>findOverlaps()</code> to determine the overlap between (query) <code>mono_h3k4me3</code> and (subject) <code>promoters</code>.</p>
</blockquote>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb54-1" data-line-number="1"><span class="co"># Find overlap between mono_h3k4me3 peaks and promoters</span></a>
<a class="sourceLine" id="cb54-2" data-line-number="2">ovl &lt;-<span class="st"> </span><span class="kw">..</span>(<span class="dt">query =</span> .., <span class="dt">subject =</span> ..)</a>
<a class="sourceLine" id="cb54-3" data-line-number="3"></a>
<a class="sourceLine" id="cb54-4" data-line-number="4"><span class="co"># print the overlap output</span></a>
<a class="sourceLine" id="cb54-5" data-line-number="5"><span class="kw">show</span>(..)</a></code></pre></div>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" data-line-number="1"><span class="co"># Find overlap between mono_h3k4me3 peaks and promoters</span></a>
<a class="sourceLine" id="cb55-2" data-line-number="2">ovl &lt;-<span class="st"> </span><span class="kw">findOverlaps</span>(<span class="dt">query =</span> mono_h3k4me3, <span class="dt">subject =</span> promoters)</a>
<a class="sourceLine" id="cb55-3" data-line-number="3"></a>
<a class="sourceLine" id="cb55-4" data-line-number="4"><span class="co"># print the overlap output</span></a>
<a class="sourceLine" id="cb55-5" data-line-number="5"><span class="kw">show</span>(ovl)</a></code></pre></div>
<pre><code>## Hits object with 1256 hits and 0 metadata columns:
##          queryHits subjectHits
##          &lt;integer&gt;   &lt;integer&gt;
##      [1]         3        1609
##      [2]         5        1058
##      [3]         6         407
##      [4]         7         630
##      [5]         8         630
##      ...       ...         ...
##   [1252]      2833        1642
##   [1253]      2836        1341
##   [1254]      2836        1416
##   [1255]      2837        1341
##   [1256]      2837        1416
##   -------
##   queryLength: 2838 / subjectLength: 1729</code></pre>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<p><br> ‘ovl’ gives you the indices of <code>mono_h3k4me3</code> and <code>promoters</code> that overlap. If a peak or a promoter overlaps several times, each overlap will be reported on a new row.<br />
<br></p>
<blockquote>
<p><strong>Exercise 11:</strong><br />
How many of the promoters are part of the overlap? And how many of the h3k4me3 peaks?</p>
<ul>
<li>Use <code>queryHits()</code> and <code>subjectHits()</code> to extract the indices of overlapping peaks and promoters respectively.<br />
</li>
<li>Use <code>unique()</code> to minimize this output to unique peaks.<br />
</li>
<li>Use <code>length()</code> to count the number of unique intervals of the query and the subject.</li>
</ul>
</blockquote>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb58-1" data-line-number="1"><span class="co"># What is the number of unique h3k4me3 peaks reported in &#39;ovl&#39;?</span></a>
<a class="sourceLine" id="cb58-2" data-line-number="2"><span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">..</span>(..))) </a></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb59-1" data-line-number="1"><span class="co"># What is the number of unique h3k4me3 peaks reported in &#39;ovl&#39;?</span></a>
<a class="sourceLine" id="cb59-2" data-line-number="2"><span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">queryHits</span>(ovl)))</a></code></pre></div>
<pre><code>## [1] 1008</code></pre>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb61-1" data-line-number="1"><span class="kw">grade_result</span>(</a>
<a class="sourceLine" id="cb61-2" data-line-number="2">  <span class="kw">pass_if</span>(<span class="op">~</span><span class="kw">identical</span>(.result, <span class="dv">1008</span>))</a>
<a class="sourceLine" id="cb61-3" data-line-number="3">)</a></code></pre></div>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb62-1" data-line-number="1"><span class="co"># What is the proportion of h3k4me3 peaks that overlap?</span></a>
<a class="sourceLine" id="cb62-2" data-line-number="2"><span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">..</span>(..)))<span class="op">/</span><span class="kw">length</span>(..)</a></code></pre></div>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb63-1" data-line-number="1"><span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">queryHits</span>(ovl)))<span class="op">/</span><span class="kw">length</span>(mono_h3k4me3)</a></code></pre></div>
<pre><code>## [1] 0.3551797</code></pre>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb65-1" data-line-number="1"><span class="kw">grade_result</span>(</a>
<a class="sourceLine" id="cb65-2" data-line-number="2">  <span class="kw">pass_if</span>(<span class="op">~</span><span class="kw">identical</span>(<span class="kw">round</span>(.result, <span class="dt">digits=</span> <span class="dv">3</span>), <span class="kw">round</span>(<span class="fl">0.3551797</span>, <span class="dt">digits =</span> <span class="dv">3</span>)))</a>
<a class="sourceLine" id="cb65-3" data-line-number="3">)</a></code></pre></div>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb66-1" data-line-number="1"><span class="co"># What is the proportion of promoters that overlap?</span></a>
<a class="sourceLine" id="cb66-2" data-line-number="2"><span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">..</span>(..)))<span class="op">/</span><span class="kw">length</span>(..)</a></code></pre></div>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb67-1" data-line-number="1"><span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">subjectHits</span>(ovl)))<span class="op">/</span><span class="kw">length</span>(promoters)</a></code></pre></div>
<pre><code>## [1] 0.6495084</code></pre>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb69-1" data-line-number="1"><span class="kw">grade_result</span>(</a>
<a class="sourceLine" id="cb69-2" data-line-number="2">  <span class="kw">pass_if</span>(<span class="op">~</span><span class="kw">identical</span>(<span class="kw">round</span>(.result, <span class="dt">digits=</span> <span class="dv">3</span>), <span class="kw">round</span>(<span class="fl">0.6495084</span>, <span class="dt">digits =</span> <span class="dv">3</span>)))</a>
<a class="sourceLine" id="cb69-3" data-line-number="3">)</a></code></pre></div>
<p><br> Let’s plot this overlap in a venn diagram with the <code>plot.pairwise.venn()</code> function of the <code>VennDiagram</code> package.</p>
<ul>
<li>Use the minimum of the two ‘unique counts’ as the number of ‘common peaks’ for our venn diagram.</li>
</ul>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb70-1" data-line-number="1"><span class="co"># Identify the minimum of the two numbers of unique intervals among h3k4me3 peaks and promoters</span></a>
<a class="sourceLine" id="cb70-2" data-line-number="2">common_intervals &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">c</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">queryHits</span>(ovl))), <span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">subjectHits</span>(ovl)))))</a>
<a class="sourceLine" id="cb70-3" data-line-number="3"></a>
<a class="sourceLine" id="cb70-4" data-line-number="4"><span class="co"># call a new plotting area</span></a>
<a class="sourceLine" id="cb70-5" data-line-number="5"><span class="kw">grid.newpage</span>()</a>
<a class="sourceLine" id="cb70-6" data-line-number="6"></a>
<a class="sourceLine" id="cb70-7" data-line-number="7"><span class="co"># Plot the overlap in a venn diagram</span></a>
<a class="sourceLine" id="cb70-8" data-line-number="8"><span class="kw">draw.pairwise.venn</span>( </a>
<a class="sourceLine" id="cb70-9" data-line-number="9">   <span class="dt">area1=</span><span class="kw">length</span>(mono_h3k4me3),</a>
<a class="sourceLine" id="cb70-10" data-line-number="10">   <span class="dt">area2=</span><span class="kw">length</span>(promoters), </a>
<a class="sourceLine" id="cb70-11" data-line-number="11">   <span class="dt">cross.area=</span>common_intervals, </a>
<a class="sourceLine" id="cb70-12" data-line-number="12">   <span class="dt">category=</span><span class="kw">c</span>(<span class="st">&quot;h3k4me3&quot;</span>, <span class="st">&quot;Promoters&quot;</span>), </a>
<a class="sourceLine" id="cb70-13" data-line-number="13">   <span class="dt">fill=</span><span class="kw">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;gray&quot;</span>), </a>
<a class="sourceLine" id="cb70-14" data-line-number="14">   <span class="dt">cat.cex=</span><span class="fl">1.2</span>)</a></code></pre></div>
<p><img src="p4_week2_v4_files/figure-html/venndiagram-1.png" width="672" /></p>
<pre><code>## (polygon[GRID.polygon.11], polygon[GRID.polygon.12], polygon[GRID.polygon.13], polygon[GRID.polygon.14], text[GRID.text.15], text[GRID.text.16], text[GRID.text.17], text[GRID.text.18], text[GRID.text.19])</code></pre>
<p><br> Is there a significant enrichment of h3k4me3 in promoters?<br />
<br> To answer this question, we compare the fraction of promoters with a h3k4me3 peak with the chromosome 19-wide fraction of promoters.<br />
<br> If h3k4me3 is not enriched at promoters, we would expect that the fraction of promoters with a h3k4me3 peak is in the same range as the fraction of promoters on chromosome 19. (=This is our null hypothesis of no enrichment.<br />
<br> To calculate the chromosome 19-wide fraction of promoters we identify the total number of bps covered by a promoter. As some promoters may overlap we first <em>reduce</em> <code>promoters</code> to non-overlapping intervals. Visually, <code>reduce()</code> works like this: <img src="images/wk2_4_reduce_rockefelleruniversity.png" /> <br />
<br></p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb72-1" data-line-number="1"><span class="co"># total number of bps covered by promoters</span></a>
<a class="sourceLine" id="cb72-2" data-line-number="2">total_bp_prom &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">width</span>(GenomicRanges<span class="op">::</span><span class="kw">reduce</span>(promoters)))</a>
<a class="sourceLine" id="cb72-3" data-line-number="3"><span class="co"># chromosome 19-wide fraction of promoters </span></a>
<a class="sourceLine" id="cb72-4" data-line-number="4">(prom_fraction_chr19 &lt;-<span class="st"> </span>total_bp_prom<span class="op">/</span><span class="kw">seqlengths</span>(promoters)[<span class="st">&quot;chr19&quot;</span>])</a></code></pre></div>
<pre><code>##      chr19 
## 0.03404707</code></pre>
<p><br> Promoters make up 3.4% of this chromosome but nearly 65% of all promoters overlap with h3k4me3 peaks. This looks like a strong enrichment. We can test whether the observed fraction is indeed larger than expected with a binomial test, in r we use the function <code>binom.test(x, n, p)</code> for this:</p>
<ul>
<li><code>x</code> = number of successes, number of promters with h3k4me3 peak<br />
</li>
<li><code>n</code> = total number of trails, total number of h3k4me3 peaks<br />
</li>
<li><code>p</code> = expected probability of success, in this case the fraction of promoters in chromosome 19</li>
</ul>
<p><br></p>
<blockquote>
<p><em>Background</em>: The binomial test is run when an experiment has two possible outcomes (i.e. success/failure) and you have an idea about what the probability of success is. Success in this case is overlap and our expectation is that 3.4% of the cases show overlap. The test calculates the probability of getting a desired outcome with a specific sample size <code>n</code>.</p>
</blockquote>
<blockquote>
<p><strong>Exercise 12</strong><br />
Use a binomial test to test for enrichment of h3k4me3 at promoters.</p>
<ul>
<li>Use the function <code>binom.test(x, n, p, alternative = &quot;greater&quot;)</code> to call the test<br />
</li>
<li>We set <code>alternative = &quot;greater&quot;</code> because we test for <em>enrichment</em> and our alternative hypothesis is that the true probability is <em>larger</em> than the expected probability.</li>
</ul>
</blockquote>
<p><br></p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb74-1" data-line-number="1"><span class="co"># binomial test for enrichment of h3k4me3 peaks in promoters:  </span></a>
<a class="sourceLine" id="cb74-2" data-line-number="2"><span class="kw">binom.test</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">..</span>(ovl))), <span class="kw">length</span>(..), prom_fraction_chr19, <span class="dt">alternative =</span> <span class="st">&quot;greater&quot;</span> )</a></code></pre></div>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb75-1" data-line-number="1"><span class="kw">binom.test</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">subjectHits</span>(ovl))), <span class="kw">length</span>(mono_h3k4me3), prom_fraction_chr19, <span class="dt">alternative =</span> <span class="st">&quot;greater&quot;</span> )</a></code></pre></div>
<pre><code>## 
##  Exact binomial test
## 
## data:  length(unique(subjectHits(ovl))) and length(mono_h3k4me3)
## number of successes = 1123, number of trials = 2838, p-value &lt; 2.2e-16
## alternative hypothesis: true probability of success is greater than 0.03404707
## 95 percent confidence interval:
##  0.3805112 1.0000000
## sample estimates:
## probability of success 
##              0.3957012</code></pre>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb77-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<p><br> The binomial test shows a p-value &lt; 2.2e-16 and a confidence interval that excludes the expected 0.034. We therefore reject the H0 of no enrichment.<br />
<br> In the genome browser we also observed H3K27ac often at promoters. Is this mark enriched in these regions?</p>
<blockquote>
<p><strong>Exercise 13</strong><br />
Use a binomial test to test for enrichment of H3K27ac at promoters.</p>
<ul>
<li>First, use findOverlaps to detect the overlap between intervals reported in mono_h3k37 and promoters</li>
<li>Use the function <code>binom.test(x, n, p, alternative = &quot;greater&quot;)</code> to call the test<br />
</li>
<li>We set <code>alternative = &quot;greater&quot;</code> because we test for <em>enrichment</em> and our alternative hypothesis is that the true probability is <em>larger</em> than the expected probability.</li>
</ul>
</blockquote>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb78-1" data-line-number="1"><span class="co"># find overlap between mono_h3k27ac and promoters</span></a>
<a class="sourceLine" id="cb78-2" data-line-number="2">ovl2 &lt;-<span class="st"> </span>...</a></code></pre></div>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb79-1" data-line-number="1">ovl2 &lt;-<span class="st"> </span><span class="kw">findOverlaps</span>(<span class="dt">query =</span> mono_h3k27ac, <span class="dt">subject =</span> promoters)</a></code></pre></div>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb80-1" data-line-number="1"><span class="co"># binomial test for enrichment of H3K27ac peaks in promoters</span></a>
<a class="sourceLine" id="cb80-2" data-line-number="2"><span class="co"># `prom_fraction_chr19` holds the fractio of promoters in chromosome 19</span></a>
<a class="sourceLine" id="cb80-3" data-line-number="3"><span class="kw">..</span>(<span class="kw">..</span>(<span class="kw">..</span>(<span class="kw">..</span>(..))), <span class="kw">..</span>(..), prom_fraction_chr19, <span class="dt">alternative =</span> <span class="st">&quot;greater&quot;</span> )</a></code></pre></div>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb81-1" data-line-number="1"><span class="co"># binomial test for enrichment of H3K27ac peaks in promoters:  </span></a>
<a class="sourceLine" id="cb81-2" data-line-number="2"><span class="kw">binom.test</span>(<span class="kw">length</span>(<span class="kw">unique</span>(<span class="kw">subjectHits</span>(ovl2))), <span class="kw">length</span>(mono_h3k27ac), prom_fraction_chr19, <span class="dt">alternative =</span> <span class="st">&quot;greater&quot;</span> )</a></code></pre></div>
<pre><code>## 
##  Exact binomial test
## 
## data:  length(unique(subjectHits(ovl2))) and length(mono_h3k27ac)
## number of successes = 865, number of trials = 3793, p-value &lt; 2.2e-16
## alternative hypothesis: true probability of success is greater than 0.03404707
## 95 percent confidence interval:
##  0.2168716 1.0000000
## sample estimates:
## probability of success 
##              0.2280517</code></pre>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb83-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<p><br> &gt;<strong>Exercise 14:</strong><br />
Do you conclude that H3K27ac is enriched at promoters?</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb84-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Is H3K27ac enriched at promoters?&quot;</span>,</a>
<a class="sourceLine" id="cb84-2" data-line-number="2">         <span class="kw">answer</span>(<span class="st">&quot;*Yes*&quot;</span>, <span class="dt">correct =</span> T),</a>
<a class="sourceLine" id="cb84-3" data-line-number="3">         <span class="kw">answer</span>(<span class="st">&quot;*No*&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Incorrect. Look at the p-value of the previous test.&quot;</span>)</a>
<a class="sourceLine" id="cb84-4" data-line-number="4">)</a></code></pre></div>
</div>
<div id="overlap-with-genomic-features" class="section level3">
<h3>2.4.4 Overlap with genomic features</h3>
<p><span style="color: red;">remove?</span><br />
To understand the function of a ChIPped mark or factor we often want to know to which genes and genomic features it binds. <br> We will use a <code>TxDb</code> objects. Such an object is an R interface to prefabricated databases contained by specific annotation packages. The package TxDb.Hsapiens.UCSC.hg38.knownGene includes all human genes and transcripts from UCSC with coordinates for the hg18 genome assembly.</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb85-1" data-line-number="1"><span class="co"># we need the `TxDb` object and limits its active chromosomes to chromosome 19</span></a>
<a class="sourceLine" id="cb85-2" data-line-number="2"><span class="kw">library</span>(TxDb.Hsapiens.UCSC.hg38.knownGene)</a>
<a class="sourceLine" id="cb85-3" data-line-number="3">txdb &lt;-<span class="st"> </span>TxDb.Hsapiens.UCSC.hg38.knownGene</a>
<a class="sourceLine" id="cb85-4" data-line-number="4"><span class="kw">seqlevels</span>(txdb) &lt;-<span class="st"> &quot;chr19&quot;</span></a>
<a class="sourceLine" id="cb85-5" data-line-number="5"></a>
<a class="sourceLine" id="cb85-6" data-line-number="6"><span class="co"># laod package to annotate peaks with genes</span></a>
<a class="sourceLine" id="cb85-7" data-line-number="7"><span class="kw">require</span>(ChIPpeakAnno)   </a>
<a class="sourceLine" id="cb85-8" data-line-number="8"></a>
<a class="sourceLine" id="cb85-9" data-line-number="9"><span class="co"># calculate the overlap with features</span></a>
<a class="sourceLine" id="cb85-10" data-line-number="10">h3k4me3_features &lt;-<span class="st"> </span><span class="kw">assignChromosomeRegion</span>(mono_h3k4me3, <span class="dt">TxDb=</span>txdb, <span class="dt">nucleotideLevel=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb85-11" data-line-number="11"><span class="co"># show the results</span></a>
<a class="sourceLine" id="cb85-12" data-line-number="12">h3k4me3_features</a></code></pre></div>
<pre><code>## $percentage
## subjectHits
##               Exons             Introns            fiveUTRs           threeUTRs 
##            59.65469            79.21071            40.16913            13.42495 
##           Promoters immediateDownstream   Intergenic.Region 
##            53.52361            24.06624            13.03735 
## 
## $jaccard
##               Exons             Introns            fiveUTRs           threeUTRs 
##          0.05531595          0.12477798          0.19041256          0.03819932 
##           Promoters immediateDownstream   Intergenic.Region 
##          0.18327703          0.05151607          0.13049339</code></pre>
<blockquote>
<p><strong>Exercise 13</strong><br />
Plot the percentage of features with the mark as barplot and as piechart</p>
<ul>
<li>Access the percent values in <code>h3k4me3_features</code> using the $ operator.<br />
</li>
<li>For plotting use <code>pie()</code> and <code>barplot()</code>.</li>
</ul>
</blockquote>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb87-1" data-line-number="1"><span class="co"># make pie-chart</span></a></code></pre></div>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb88-1" data-line-number="1"><span class="co"># make pie-chart</span></a>
<a class="sourceLine" id="cb88-2" data-line-number="2"><span class="kw">pie</span>(h3k4me3_features<span class="op">$</span>percentage)</a></code></pre></div>
<p><img src="p4_week2_v4_files/figure-html/q13a_anno-solution-1.png" width="672" /></p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb89-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb90-1" data-line-number="1"><span class="co"># make barplot</span></a>
<a class="sourceLine" id="cb90-2" data-line-number="2">bp &lt;-<span class="st"> </span><span class="kw">..</span>(..<span class="op">$</span>.., <span class="dt">ylab=</span><span class="st">&quot;%&quot;</span>)</a>
<a class="sourceLine" id="cb90-3" data-line-number="3"><span class="co"># add percentages to the bars with text()  </span></a>
<a class="sourceLine" id="cb90-4" data-line-number="4"><span class="kw">text</span>(bp, h3k4me3_features<span class="op">$</span>percentage, <span class="kw">signif</span>(h3k4me3_features<span class="op">$</span>percentage, <span class="dv">4</span>), <span class="dt">pos=</span><span class="dv">1</span>, <span class="dt">main =</span> <span class="st">&quot;Mono h3k4me3 in features&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb91-1" data-line-number="1"><span class="co"># make barplot</span></a>
<a class="sourceLine" id="cb91-2" data-line-number="2">bp &lt;-<span class="st"> </span><span class="kw">barplot</span>(h3k4me3_features<span class="op">$</span>percentage, <span class="dt">ylab=</span><span class="st">&quot;%&quot;</span>)</a>
<a class="sourceLine" id="cb91-3" data-line-number="3"><span class="co"># add percentages to the bars with text()  </span></a>
<a class="sourceLine" id="cb91-4" data-line-number="4"><span class="kw">text</span>(bp, h3k4me3_features<span class="op">$</span>percentage, <span class="kw">signif</span>(h3k4me3_features<span class="op">$</span>percentage, <span class="dv">4</span>), <span class="dt">pos=</span><span class="dv">1</span>, <span class="dt">main =</span> <span class="st">&quot;Mono h3k4me3 in features&quot;</span>)</a></code></pre></div>
<p><img src="p4_week2_v4_files/figure-html/q13anno-solution-1.png" width="672" /></p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb92-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
</div>
<div id="overlap-with-grangeslist" class="section level3">
<h3>2.4.5 Overlap with GRangesList</h3>
<p><span style="color: red;">remove?</span><br />
So far we looked at h3k4me3 marking. How do the other marks overlap with genomic features? Instead of rerunning the above code for each mark individually, we combine the individual peak objects into a list of <code>GRanges</code> objects with the function <code>GRangesList()</code>.<br />
<br><br />
We have created a <code>GRangesList</code> <code>mono_list</code> by:</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb93-1" data-line-number="1"><span class="co"># make a list of your GRanges objects</span></a>
<a class="sourceLine" id="cb93-2" data-line-number="2">mono_list  &lt;-<span class="st"> </span><span class="kw">GRangesList</span>(mono_h3k4me1, mono_h3k4me3, mono_h3k9me3, mono_h3k27ac, mono_h3k27me3, mono_h3k36me3)</a>
<a class="sourceLine" id="cb93-3" data-line-number="3"><span class="co"># add names to each element in the list</span></a>
<a class="sourceLine" id="cb93-4" data-line-number="4"><span class="kw">names</span>(mono_list) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;h3k4me1&quot;</span>, <span class="st">&quot;h3k4me3&quot;</span>, <span class="st">&quot;h3k9me3&quot;</span>, <span class="st">&quot;h3k27ac&quot;</span>, <span class="st">&quot;h3k27me3&quot;</span>, <span class="st">&quot;h3k36me3&quot;</span>)</a>
<a class="sourceLine" id="cb93-5" data-line-number="5"><span class="co"># print the length of your list  </span></a>
<a class="sourceLine" id="cb93-6" data-line-number="6"><span class="kw">length</span>(mono_list)</a></code></pre></div>
<pre><code>## [1] 6</code></pre>
<p><br> The <code>length</code> function now applies to the list and not to the individual items any more.<br />
<br> We can access elements in the list using <code>[[]]</code> or <code>$</code>-sign.</p>
<ul>
<li>Use <code>[[]]</code> and the <em>index</em>: eg. <code>mono_list[[1]]</code> will extract the first item in the list.</li>
<li>Use <code>[[]]</code> and the name: eg <code>mono_list[[&quot;h3k36me3&quot;]]</code> will extract the item with this name.<br />
</li>
<li>Use <code>$item_name</code> to extract the item with that name.</li>
</ul>
<p>To calculate the length of individual items in the list use <code>lapply()</code>, which applies a function to each item in the list.<br />
<br> With the fuction <code>lapply(list_object, function)</code> we can apply or <em>loop</em> a function to each element in the list and returns the results as a list.<br />
<br> Here we apply <code>length()</code> to each item in <code>mono_list</code></p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb95-1" data-line-number="1"><span class="co"># print the length of each element at once with apply</span></a>
<a class="sourceLine" id="cb95-2" data-line-number="2"><span class="kw">lapply</span>(mono_list, length) </a></code></pre></div>
<pre><code>## $h3k4me1
## [1] 3282
## 
## $h3k4me3
## [1] 2838
## 
## $h3k9me3
## [1] 1893
## 
## $h3k27ac
## [1] 3793
## 
## $h3k27me3
## [1] 3059
## 
## $h3k36me3
## [1] 2084</code></pre>
<p><br> <code>unlist()</code> flattens the list to one object. Depending on the elements in the list this will return a <code>vector</code>, <code>matrix</code> or, if all elements have the same data structure and column names, one large version of that particular data structure.<br />
<br> E.g. <code>unlist(mono_list)</code> will give you one large <code>GRanges</code> object that is often not usefull because peaks of all marks are mixed up. <code>unlist(lapply(mono_list, length))</code>, on the other hand, returns a usefull vector of peak numbers per ChIP.<br />
<br></p>
<blockquote>
<p><strong>Exercise 14:</strong><br />
Make a barplot of peak numbers per ChIP:**</p>
<ul>
<li>Use the <code>mono_list</code> object<br />
</li>
<li>Obtain <code>counts_vector</code> with <code>lapply()</code>, and <code>unlist()</code><br />
</li>
<li>Use the <code>length</code> within <code>lapply()</code></li>
</ul>
</blockquote>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb97-1" data-line-number="1"><span class="co"># print the length of each element at once with apply</span></a>
<a class="sourceLine" id="cb97-2" data-line-number="2">counts_list &lt;-<span class="st"> </span><span class="kw">..</span> ( .. , .. )</a>
<a class="sourceLine" id="cb97-3" data-line-number="3"></a>
<a class="sourceLine" id="cb97-4" data-line-number="4"><span class="co"># unlist the resulting output</span></a>
<a class="sourceLine" id="cb97-5" data-line-number="5">counts_vector &lt;-<span class="st"> </span><span class="kw">..</span>(..)</a>
<a class="sourceLine" id="cb97-6" data-line-number="6"></a>
<a class="sourceLine" id="cb97-7" data-line-number="7"><span class="co"># make barplot of the number of peaks, store this in &#39;bp&#39;:  </span></a>
<a class="sourceLine" id="cb97-8" data-line-number="8">bp &lt;-<span class="st"> </span><span class="kw">..</span>(.., <span class="dt">ylab =</span> <span class="st">&quot;Number of peaks&quot;</span>, <span class="dt">main =</span> <span class="st">&quot;Monocytes ChIP-seq; chr19&quot;</span>)</a>
<a class="sourceLine" id="cb97-9" data-line-number="9"></a>
<a class="sourceLine" id="cb97-10" data-line-number="10"><span class="co"># add actual values as text lables to the plot</span></a>
<a class="sourceLine" id="cb97-11" data-line-number="11"><span class="kw">text</span>(bp, counts_vector, <span class="dt">labels=</span><span class="kw">names</span>(counts_vector), <span class="dt">pos=</span><span class="dv">1</span>)</a></code></pre></div>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb98-1" data-line-number="1"><span class="co"># print the length of each element at once with apply</span></a>
<a class="sourceLine" id="cb98-2" data-line-number="2">counts_list &lt;-<span class="st"> </span><span class="kw">lapply</span>(mono_list, length)</a>
<a class="sourceLine" id="cb98-3" data-line-number="3"></a>
<a class="sourceLine" id="cb98-4" data-line-number="4"><span class="co"># unlist the resulting output</span></a>
<a class="sourceLine" id="cb98-5" data-line-number="5">counts_vector &lt;-<span class="st"> </span><span class="kw">unlist</span>(counts_list)</a>
<a class="sourceLine" id="cb98-6" data-line-number="6"></a>
<a class="sourceLine" id="cb98-7" data-line-number="7"><span class="co"># barplot of the number of peaks, store this in &#39;bp&#39;:   </span></a>
<a class="sourceLine" id="cb98-8" data-line-number="8">bp &lt;-<span class="kw">barplot</span>(counts_vector, <span class="dt">ylab =</span> <span class="st">&quot;Number of peaks&quot;</span>, <span class="dt">main =</span> <span class="st">&quot;Monocytes ChIP-seq; chr19&quot;</span>)</a>
<a class="sourceLine" id="cb98-9" data-line-number="9"></a>
<a class="sourceLine" id="cb98-10" data-line-number="10"><span class="co"># add actual values as text lables to the plot</span></a>
<a class="sourceLine" id="cb98-11" data-line-number="11"><span class="kw">text</span>(bp, counts_vector, <span class="dt">labels =</span> counts_vector, <span class="dt">pos=</span><span class="dv">1</span>)</a></code></pre></div>
<p><img src="p4_week2_v4_files/figure-html/q14_barplot-solution-1.png" width="672" /></p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb99-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<p><br> Instead of the length, let’s look at the distribution of these peaks among features, as in section 2.4.3.<br />
<br></p>
<blockquote>
<p><strong>Exercise 15</strong></p>
<p>For each item in <code>mono_list</code> determine the overlap with genomic features. Use the <code>assignChromosomeRegion(object,TxDb=txdb, nucleotideLevel=FALSE)</code> to assign peaks to features.</p>
</blockquote>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb100-1" data-line-number="1"><span class="co"># laod package to annotate peaks with genes</span></a>
<a class="sourceLine" id="cb100-2" data-line-number="2"><span class="kw">require</span>(ChIPpeakAnno)   </a>
<a class="sourceLine" id="cb100-3" data-line-number="3"></a>
<a class="sourceLine" id="cb100-4" data-line-number="4"><span class="co"># calucluate the overlap with features</span></a>
<a class="sourceLine" id="cb100-5" data-line-number="5"><span class="co"># this code takes rather long to run!!!!</span></a>
<a class="sourceLine" id="cb100-6" data-line-number="6">allmarks_features &lt;-<span class="st"> </span><span class="kw">lapply</span>(mono_list, <span class="cf">function</span>(.object) <span class="kw">assignChromosomeRegion</span>(.object, <span class="dt">TxDb=</span>txdb, <span class="dt">nucleotideLevel=</span><span class="ot">FALSE</span>))</a>
<a class="sourceLine" id="cb100-7" data-line-number="7"><span class="kw">names</span>(allmarks_features) &lt;-<span class="st"> </span><span class="kw">names</span>(mono_list)</a>
<a class="sourceLine" id="cb100-8" data-line-number="8"></a>
<a class="sourceLine" id="cb100-9" data-line-number="9"><span class="co"># show the results</span></a>
<a class="sourceLine" id="cb100-10" data-line-number="10">allmarks_features</a></code></pre></div>
<pre><code>## $h3k4me1
## $h3k4me1$percentage
## subjectHits
##               Exons             Introns            fiveUTRs           threeUTRs 
##            59.59781            79.95125            34.06459            18.19013 
##           Promoters immediateDownstream   Intergenic.Region 
##            50.03047            30.37782            13.04083 
## 
## $h3k4me1$jaccard
##               Exons             Introns            fiveUTRs           threeUTRs 
##          0.06975003          0.16409230          0.17245103          0.06382979 
##           Promoters immediateDownstream   Intergenic.Region 
##          0.20771664          0.08202386          0.20336821 
## 
## 
## $h3k4me3
## $h3k4me3$percentage
## subjectHits
##               Exons             Introns            fiveUTRs           threeUTRs 
##            59.65469            79.21071            40.16913            13.42495 
##           Promoters immediateDownstream   Intergenic.Region 
##            53.52361            24.06624            13.03735 
## 
## $h3k4me3$jaccard
##               Exons             Introns            fiveUTRs           threeUTRs 
##          0.05531595          0.12477798          0.19041256          0.03819932 
##           Promoters immediateDownstream   Intergenic.Region 
##          0.18327703          0.05151607          0.13049339 
## 
## 
## $h3k9me3
## $h3k9me3$percentage
## subjectHits
##               Exons             Introns            fiveUTRs           threeUTRs 
##            41.94400            65.61014            13.99894            17.27417 
##           Promoters immediateDownstream   Intergenic.Region 
##            28.73745            30.26941            26.36027 
## 
## $h3k9me3$jaccard
##               Exons             Introns            fiveUTRs           threeUTRs 
##          0.02346613          0.06555819          0.03029264          0.03765112 
##           Promoters immediateDownstream   Intergenic.Region 
##          0.04463773          0.05019711          0.27578391 
## 
## 
## $h3k27ac
## $h3k27ac$percentage
## subjectHits
##               Exons             Introns            fiveUTRs           threeUTRs 
##           43.606644           80.859478           25.388874            8.700237 
##           Promoters immediateDownstream   Intergenic.Region 
##           42.209333           18.876878           10.123912 
## 
## $h3k27ac$jaccard
##               Exons             Introns            fiveUTRs           threeUTRs 
##          0.04843906          0.15092019          0.11776935          0.02947218 
##           Promoters immediateDownstream   Intergenic.Region 
##          0.14806252          0.04912521          0.09429594 
## 
## 
## $h3k27me3
## $h3k27me3$percentage
## subjectHits
##               Exons             Introns            fiveUTRs           threeUTRs 
##            41.41876            62.99444            14.22033            14.22033 
##           Promoters immediateDownstream   Intergenic.Region 
##            25.95619            24.91010            26.93691 
## 
## $h3k27me3$jaccard
##               Exons             Introns            fiveUTRs           threeUTRs 
##          0.03662167          0.09786694          0.04599281          0.04346089 
##           Promoters immediateDownstream   Intergenic.Region 
##          0.06257881          0.05769668          0.30443055 
## 
## 
## $h3k36me3
## $h3k36me3$percentage
## subjectHits
##               Exons             Introns            fiveUTRs           threeUTRs 
##           68.570058           96.017274           27.927063           39.635317 
##           Promoters immediateDownstream   Intergenic.Region 
##           41.218810           51.919386            1.103647 
## 
## $h3k36me3$jaccard
##               Exons             Introns            fiveUTRs           threeUTRs 
##          0.05958636          0.15607207          0.06879433          0.14721084 
##           Promoters immediateDownstream   Intergenic.Region 
##          0.07550321          0.13397722          0.04495472</code></pre>
</div>
</div>
<div id="annotation-of-chip-seq-peaks" class="section level2">
<h2>2.5 Annotation of ChIP-seq peaks</h2>
<div id="distance-to-tss" class="section level3">
<h3>2.5.1 Distance to TSS</h3>
<p>In this part of the analysis we will assign peaks to their closest TSS, assuming that it likely is involved in regulating the expression regulates that gene.<br />
<br> The <code>GenomicRanges</code> package has the function <code>distanceToNearest(x, subject)</code> to help you identify the distance between intervals in <code>x</code> and the nearest neighbor in <code>subject</code>. In this case, the intervals are the peaks and the subject are the TSSs. The function outputs a <code>Hits</code> objects with distances.<br />
<br></p>
<blockquote>
<p><strong>Exercise 15</strong><br />
Use <code>distanceToNearest()</code> to get the distance between h3k4me3 peaks and the nearest TSS.</p>
</blockquote>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb102-1" data-line-number="1"><span class="co"># calculate the distances from peaks to tss</span></a>
<a class="sourceLine" id="cb102-2" data-line-number="2">h3k4me3_to_tss &lt;-<span class="st"> </span><span class="kw">..</span>(.., ..)</a></code></pre></div>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb103-1" data-line-number="1"><span class="co"># calculate the distances from peaks to tss</span></a>
<a class="sourceLine" id="cb103-2" data-line-number="2">h3k4me3_to_tss &lt;-<span class="st"> </span><span class="kw">distanceToNearest</span>(mono_h3k4me3, tss)</a></code></pre></div>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb104-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<p><br> Let’s look at the resulting <code>Hits</code> object.<br />
<br></p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb105-1" data-line-number="1"><span class="co"># show object h3k4me3_to_tss</span></a>
<a class="sourceLine" id="cb105-2" data-line-number="2"><span class="kw">show</span>(h3k4me3_to_tss)</a></code></pre></div>
<pre><code>## Hits object with 2838 hits and 1 metadata column:
##          queryHits subjectHits |  distance
##          &lt;integer&gt;   &lt;integer&gt; | &lt;integer&gt;
##      [1]         1         924 |     36487
##      [2]         2        1609 |     23749
##      [3]         3        1609 |         0
##      [4]         4        1609 |      3603
##      [5]         5        1058 |         0
##      ...       ...         ... .       ...
##   [2834]      2834          25 |      2812
##   [2835]      2835          25 |      3247
##   [2836]      2836        1341 |         0
##   [2837]      2837        1416 |        53
##   [2838]      2838        1416 |      1545
##   -------
##   queryLength: 2838 / subjectLength: 1729</code></pre>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb107-1" data-line-number="1"><span class="co"># summary of distances</span></a>
<a class="sourceLine" id="cb107-2" data-line-number="2"><span class="kw">summary</span>(<span class="kw">mcols</span>(h3k4me3_to_tss)[,<span class="dv">1</span>])</a></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##       0       0    3422    9599   11567  447135</code></pre>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb109-1" data-line-number="1"><span class="co"># plot the distribution of distances</span></a>
<a class="sourceLine" id="cb109-2" data-line-number="2"><span class="kw">boxplot</span>(<span class="kw">mcols</span>(h3k4me3_to_tss)[,<span class="dv">1</span>])</a></code></pre></div>
<p><img src="p4_week2_v4_files/figure-html/look%20at%20hits-1.png" width="672" /> <br></p>
</div>
<div id="find-associating-gene" class="section level3">
<h3>2.5.2 Find associating gene</h3>
<p>We can use this <code>Hits</code> object to subset peaks to those that are less than 10 Kb away from the TSS. We can than use the function <code>subjectHits()</code> to get the genes associated to each peak.<br />
<br></p>
<blockquote>
<p><strong>Exercise 16</strong><br />
Subset the h3k4me3_to_tss for distances =&lt; 10kb and select the associated genes from <code>genes</code> object.</p>
</blockquote>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb110-1" data-line-number="1"><span class="co"># save the distances in a vector</span></a>
<a class="sourceLine" id="cb110-2" data-line-number="2">distances &lt;-<span class="st"> </span><span class="kw">mcols</span>(h3k4me3_to_tss)[,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb110-3" data-line-number="3"></a>
<a class="sourceLine" id="cb110-4" data-line-number="4"><span class="co"># subset h3k4me3_to_tss by distance</span></a>
<a class="sourceLine" id="cb110-5" data-line-number="5">h3k4me3_to_tss_close &lt;-<span class="st"> </span>..[..,..]</a></code></pre></div>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb111-1" data-line-number="1"><span class="co"># save the distances in a vector</span></a>
<a class="sourceLine" id="cb111-2" data-line-number="2">distances &lt;-<span class="st"> </span><span class="kw">mcols</span>(h3k4me3_to_tss)[,<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb111-3" data-line-number="3"></a>
<a class="sourceLine" id="cb111-4" data-line-number="4"><span class="co"># subset h3k4me3_to_tss for distances &lt;= 10000</span></a>
<a class="sourceLine" id="cb111-5" data-line-number="5">h3k4me3_to_tss_close &lt;-<span class="st"> </span>h3k4me3_to_tss[distances <span class="op">&lt;=</span><span class="st"> </span><span class="dv">10000</span>,]</a></code></pre></div>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb112-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb113-1" data-line-number="1"><span class="co"># obtain the genes</span></a>
<a class="sourceLine" id="cb113-2" data-line-number="2">mono_h3k4me3_genes &lt;-<span class="st"> </span><span class="kw">unique</span>(genes[<span class="kw">..</span>(..)])</a>
<a class="sourceLine" id="cb113-3" data-line-number="3"></a>
<a class="sourceLine" id="cb113-4" data-line-number="4"><span class="co"># look at the output</span></a>
<a class="sourceLine" id="cb113-5" data-line-number="5"><span class="kw">show</span>(mono_h3k4me3_genes)</a></code></pre></div>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb114-1" data-line-number="1"><span class="co"># obtain the genes</span></a>
<a class="sourceLine" id="cb114-2" data-line-number="2">mono_h3k4me3_genes &lt;-<span class="st"> </span><span class="kw">unique</span>(genes[<span class="kw">subjectHits</span>(h3k4me3_to_tss_close)])</a>
<a class="sourceLine" id="cb114-3" data-line-number="3"></a>
<a class="sourceLine" id="cb114-4" data-line-number="4"><span class="co"># look at the output</span></a>
<a class="sourceLine" id="cb114-5" data-line-number="5"><span class="kw">show</span>(mono_h3k4me3_genes)</a></code></pre></div>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb115-1" data-line-number="1"><span class="kw">grade_code</span>()</a></code></pre></div>
<p><br> Besides the distance, we are also (very!) interested in the distribution of the ChIP-seq signal around the TSS. We will look at that next week.</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
