<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>1. UCSC Genome Browser</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="custom.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Functional Genomics 2020</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="p2_learningObjectives.html">Learning Objectives</a>
</li>
<li>
  <a href="p4_week1.html">UCSC Genome Browser</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">1. UCSC Genome Browser</h1>

</div>


<style type="text/css">
.title {
  display: none;
}

</style>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">colorize &lt;-<span class="st"> </span><span class="cf">function</span>(x, color) {</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">  <span class="cf">if</span> (knitr<span class="op">::</span><span class="kw">is_latex_output</span>()) {</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">    <span class="kw">sprintf</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">textcolor{%s}{%s}&quot;</span>, color, x)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">  } <span class="cf">else</span> <span class="cf">if</span> (knitr<span class="op">::</span><span class="kw">is_html_output</span>()) {</a>
<a class="sourceLine" id="cb1-5" data-line-number="5">    <span class="kw">sprintf</span>(<span class="st">&quot;&lt;span style=&#39;color: %s;&#39;&gt;%s&lt;/span&gt;&quot;</span>, color, </a>
<a class="sourceLine" id="cb1-6" data-line-number="6">      x)</a>
<a class="sourceLine" id="cb1-7" data-line-number="7">  } <span class="cf">else</span> x</a>
<a class="sourceLine" id="cb1-8" data-line-number="8">}</a>
<a class="sourceLine" id="cb1-9" data-line-number="9"></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="co"># can be used as inline code, eg: `r colorize(&quot;Additional Ideas:&quot;, &quot;red&quot;)`  </span></a></code></pre></div>
<div id="learning-objectives" class="section level2">
<h2>1.1 Learning Objectives</h2>
<p><br> <strong>This week’s focus will be on how sequencing data is processed and familiarize ourselves with the UCSC Genome browser as a tool to visually inspect (epi-)genomics data.</strong><br />
<br></p>
<blockquote>
<ol style="list-style-type: decimal">
<li>Understand the steps taken to go from raw to useable data in a genomics experiment.<br />
</li>
<li>Understand how to move in the UCSC genome browser and know how to go to regions and genes of interest.<br />
</li>
<li>Recognize the following elements/regions in the UCSC genome browser: chromosome, centromere, telomere, gene, exon, intron, utr, promoter, and (candidate) enhancer/repressors.<br />
</li>
<li>Tell apart UCSC genome browser tracks of H3K27ac, H3K27me3, H3K36me3, H3K4me1, H3K4me3 and H3K9me3 histone ChIP-seq and DNaseI-seq signal (and ChIP-seq input) based on their coverage profile and their location with respect to genes and other annotated genomic/chromosomal elements.</li>
<li>Can interpret - and thus understand - ChIP- and DNaseI-seq signal in the UCSC genome browser.<br />
</li>
</ol>
<ul>
<li>These refer to Global learning objectives #1-3.</li>
</ul>
</blockquote>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="genomic-methods-recap" class="section level2">
<h2>1.2 Genomic methods recap</h2>
<ul>
<li>What is ChIP-seq? What is DNaseI-seq? Presentation with ChIP-seq steps? Refer to book?<br />
</li>
<li>What is a library and what do we mean with ‘library preparation’? What’s the use of adapters?<br />
</li>
<li>Sequencing samples. What happens in sequencer? What is the output?<br />
</li>
<li>Brief genomics recap/pre-quiz?</li>
</ul>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="from-raw-data-to-science" class="section level2">
<h2>1.3 From raw data to science</h2>
<center>
<img src="images/week1/2_seq2science/whatbioinformaticiancsdo.png" style="width:60.0%" />
</center>
<p><br><br></p>
<div id="sequencer-output" class="section level3">
<h3>1.3.1 Sequencer output</h3>
The sequencer determines the sequence of the DNA molecules in your sample and produces massive amounts of data. The read sequences are written to <code>.fastq</code> files that show 4 lines per DNA sequence. This is an example with 2 DNA sequences (8 lines):<br />
<br>
</center>
<img src="images/week1/2_seq2science/fastq_example3.png" /><br />

</center>
<p><br></p>
<p>The lines in the fastq file tell us:</p>
<ul>
<li>line 1: start with ‘@’-sign and contains sequence ID given by the sequencer including sequencer id, flowcell id and sample barcode<br />
</li>
<li>line 2: the DNA sequence in ACGTN-alphabet<br />
</li>
<li>line 3: ‘+’ (originally reported the sequence ID again)<br />
</li>
<li>line 4: quality score for each nucleotide</li>
</ul>
<p>Fastq files have the extension .fq or .fastq, or are compressed as .fq.gz or .fastq.gz</p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="post-sequencing-processing-steps" class="section level3">
<h3>1.3.2 Post-sequencing processing steps</h3>
<p>The chemical reaction and quality of the optical measurement that occur in the sequencer are not perfect and depend among others on the nucleotide composition of all samples, the length of the DNA molecules and the concentration of the samples. The Phred quality score (line 3) represents the nucleotide identification accuracy or base call accuray. This encoded score can be used downstream to remove nucleotides or complete reads with low quality.</p>
<p>In adddition, you want to know where the DNA molecule came from in your reference genome and, depending on your experimental set-up, likely want to identify regions with more reads than expected if reads were randomly distributed over the genome. E.g. identify peaks of TF binding, peaks of histone variants, expressed genes, or interacting chromosomal regions.</p>
<p>It is the bio-informaticians (or your) task to:</p>
<ol style="list-style-type: decimal">
<li><strong>Remove adapters and poor-quality nucleotides</strong>: The adapters should be removed, as they were not part of the original sample and low-quality nt’s are more likely to be wrong and we therefore remove those as well. Can result in that reads are removed completely because of their overall low quality or short length.<br />
<br></li>
<li><strong>Align reads to the genome</strong>: Identify the likely location in the reference genome where the original DNA molecule came from. Since the first publication of the human genome, successive versions (also called <em>assemblies</em> or <em>builds</em>) have been released with improvements in quality thanks to technological advancements as well as improvements in the representativeness of the genome with respect to the genetic diversity found in the human population, including historically underrepresentated populations. We will work with the (one of the) latests relasese referred to as GRCh38 (Genome Reference Consortium Human Build 38). The alignment procedure often adds flags for reads aligned to a unique region or to multiple regions which you can use to filter.<br />
<br></li>
<li><strong>Filter artefacts and reads aligning to multiple regions</strong>: Remove duplicated regions, remove PCR artefacts and optionally remove reads aligning non-uniquely.<br />
<br></li>
<li><strong>Visualize the data</strong>: .Bigwig files are created which can be uploaded to the UCSC Genome Browser which we will look at in more detail today as well.<br />
<br></li>
<li><strong>Call peaks</strong>: Computationally identify regions with significantly higher number of reads than expected. There are different tools out there to perform this task MACS2 is one of them and was used to call peaks in the datasets that we will be using. Peak calling is a mathemetical/statistical procedure that depends on set thresholds (p-value, false-discovery rate), the control sample used. It is always wise to visually inspect the resulting peaks in the UCSC genome browser.<br />
<br></li>
<li><strong>Read counting</strong>: Count reads that fall within genes, other genomic elements or peak regions. These countings and the actual peak locations are used for eg. differential gene expression in differentiated vs undifferentiated. Or control versus treatment. A well-known, widely used tool for this purpose is DESeq2.<br />
<br></li>
<li>Optionally <strong>DNaseI footprinting</strong>, <strong>variant calling</strong>, <strong>novel gene identification</strong>: Depending on your experimental set up … For DNaseI-seq, we can identify small depleted regions within peaks that correspond to the nucleotides that interact with the TF. This interaction protects these nucleotides from DNaseI attacks. These so-called footprints can be used to identify TF motifs and predict the likely TF bound. Although this sounds simple, and for a stable binding factor like CTCF it may be the case, it is not always that simple in reality.<br />
<br></li>
</ol>
<p><span style="color: red;"> Additional Ideas: </span></p>
<ul>
<li>Mention specific softwares/tools?<br />
</li>
<li>Show examples of the files? Might help students understand how “the data” looks like. Could be in this numbered list or in section below (1.3.4).<br />
</li>
<li>Elaborate on important/often-applied settings and their rationale?<br />
</li>
<li>Add an asignment here: to put steps in the right order with quality checks in between? Like in:<br />
<br> <!-- ![](images/week1/2_seq2science/seq2science_siebrenspptx.png){width=70%}\    
[Sec2science / adapted from Siebrens presentation / shows RNA-seq pipeline ](https://vanheeringen-lab.github.io/seq2science/content/workflows.html)      
<br>

![](images/week1/2_seq2science/encode_chipseq_readmapping.png){width=50%}\   
[The ENCODE Project, ChIP-seq read mapping ](https://www.encodeproject.org/pipelines/ENCPL220NBH/)   
<br>  --></li>
</ul>
<p><img src="images/week1/2_seq2science/Furay_2012_naturereviews_fig2_chipseqanalysis.png" style="width:70.0%" /> <br />
<a href="https://pubmed.ncbi.nlm.nih.gov/23090257/">Furay, 2012, Nat Rev</a></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="quality-evaluation" class="section level3">
<h3>1.3.3 Quality evaluation</h3>
<p>It is important to check the quality of your data along the way:</p>
<ul>
<li>FASTQC (Qualty Check) software visaully summarizes the attached Phred quality scores.<br />
</li>
<li>Evaluate the data yourself in the genome browser and compare replicate samples, samples of different histone marks, compare to a reference dataset or compare to the input material.<br />
</li>
<li>Clustering of samples; can reveal label swaps, failed experiments or contaminations.</li>
</ul>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="output-files-bigwigs-peaks-counts" class="section level3">
<h3><span style="color: red;">1.3.4 Output files: bigwigs, peaks &amp; counts</span></h3>
<p><span style="color: red;">Showing examples of how the output may look like or do this along the way in the numbered list above?</p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="alternatively-use-fair-data" class="section level3">
<h3><span style="color: red;">1.3.5 Alternatively: use FAIR data</h3>
<p><span style="color: red;">Add section discussion FAIR data and referring to several platforms where it can be found? Several reserach consortia provide easy access and clear description of their data through data portals. These files can be downloaded or immediately uploaded to the UCSC Genome browser using the URL. </span></p>
<p><br></p>
<hr />
<p><br></p>
</div>
<div id="exercises" class="section level3">
<h3>1.3.6 Exercises</h3>
<p>Ideas:</p>
<ul>
<li>Put steps in right order with options for quality checks.<br />
</li>
<li>Recognize a type of file?<br />
</li>
<li>Evaluate good/bad fastqc output?</li>
</ul>
<p><br></p>
<hr />
<p><br></p>
</div>
</div>
<div id="data-visualization-ucsc-genome-browser" class="section level2">
<h2>1.4 Data visualization: UCSC Genome Browser</h2>
<blockquote>
<p>UCSC GB sessions for making these exercises:</p>
<ul>
<li>Default should look like (if it doesn’t, I need to adjust the text): <a href="http://genome-euro.ucsc.edu/s/ctoenhake/hg38_default" class="uri">http://genome-euro.ucsc.edu/s/ctoenhake/hg38_default</a><br />
</li>
<li>Genes and ruler only: <a href="http://genome-euro.ucsc.edu/s/ctoenhake/hg38_genesonly" class="uri">http://genome-euro.ucsc.edu/s/ctoenhake/hg38_genesonly</a><br />
</li>
<li>Monocyte wiggle data (blueprint in colors, encode in black): <a href="http://genome-euro.ucsc.edu/s/ctoenhake/hg38_monos_blueprint_and_encode" class="uri">http://genome-euro.ucsc.edu/s/ctoenhake/hg38_monos_blueprint_and_encode</a><br />
</li>
<li>Monocyte wiggle + peak data, blueprint only: <a href="http://genome-euro.ucsc.edu/s/ctoenhake/hg38_monos_blueprint_and_encode/hg38_monos_blueprint_wiglle_and_peaks" class="uri">http://genome-euro.ucsc.edu/s/ctoenhake/hg38_monos_blueprint_and_encode/hg38_monos_blueprint_wiglle_and_peaks</a></li>
</ul>
</blockquote>
<div id="ucsc-genome-browser-intro" class="section level4">
<h4>1.4.1 UCSC Genome browser intro</h4>
<p>Genome browsers are great tools to display genomics data. You can walk, or <em>browse</em>, through all chromosomes, look at the location of genes, extract DNA sequences, examine pre-loaded data or upload your own data. The latter comes in quite handy when you want to, for example, have a visual (personal) confirmation of data quality, evaluate the effect of different settings for processing the raw reads, and/or confirm as computationally detected associations.<br />
<br></p>
<blockquote>
<p>If you didn’t do so already, watch the following youtube video’s to get familiar with the UCSC Genome Browser:</p>
<ol style="list-style-type: decimal">
<li><a href="https://www.youtube.com/watch?v=DNXI-M9oQl8">UCSC Genome Browser Basics</a><br />
</li>
<li><a href="https://www.youtube.com/watch?v=1Kpy5G_MSG8">UCSC Genome Browser Tracks</a></li>
</ol>
<p><strong>Optionally</strong>, finding the coordinates of a sequence using blat, handy for eg. primers, and using the Table Browser to, among others, obtain DNA, RNA or protein sequences:</p>
<ol start="3" style="list-style-type: decimal">
<li><a href="https://www.youtube.com/watch?v=E6DDNGDXH8A">UCSC Genome Browser Blat</a><br />
</li>
<li><a href="https://www.youtube.com/watch?v=Ueqt5Wgre9c&amp;ab">UCSC Genome Browser Table Browser</a></li>
</ol>
</blockquote>
<p><br></p>
</div>
<div id="go-to-the-ucsc-genome-browser" class="section level3">
<h3>1.4.2 Go to the UCSC Genome browser</h3>
<p><strong>First, we will familiarize ourselves with browsing through the UCSC genome browser and the recognition of genes, their coding elements and their orientation.</strong></p>
<p><span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Go to the <a href="genome.ucsc.edu">UCSC Genome Browser</a>.</span><br />
<br> The starting page has been updated since the video was made but still holds the same blue bar at the top with <strong>Genomes</strong>, <strong>Tools</strong>, etc.:</p>
<div class="figure">
<img src="images/week1/3b_ucsc_gb_looking_at_genes/ucsc_genome_browser_fig1_startpage_topmenubar.png" alt="Top menu bar in UCSC Genome Browser" style="width:100.0%" />
<p class="caption">Top menu bar in UCSC Genome Browser</p>
</div>
<p><br></p>
<blockquote>
<p><em>Tip</em>: In case you have been here before, it is wise to reset user settings such that you don’t continue with settings from a previous project. For this, hover over “Genome Browser” and select “Reser All User Settings”.</p>
</blockquote>
<p><span style="color: orange; font-family: Arial; font-size:18px;">ACTION: To access the human genome, select the “Genomes” tab.</span><br />
<br></p>
<blockquote>
<p><em>Tip</em>: A pop-up may appear now or one stap later, redirected you to the official European mirror site if those servers are geographically closer, that’s ok.</p>
<ul>
<li>European mirror site, servers are located at the Universität Bielefeld Center for Biotechnology in Bielefeld, Germany: <a href="http://genome-euro.ucsc.edu">http://genome-euro.ucsc.edu</a><br />
</li>
<li>Asian mirror site, located at the RIKEN Yokohama Campus, Japan: <a href="http://genome-asia.ucsc.edu">http://genome-asia.ucsc.edu</a></li>
</ul>
</blockquote>
<p><br> On the left you can browse and select the genome you are interested in (not a drop-down menu anymore as in the video). By default, the human genome is selected.<br />
<br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Ensure that “Human” is selected in the <strong>REPRESENTED SPECIES</strong> tree (left) and “Dec. 2013 (GRCh38/hg38)” is selected under <strong>Human Assembly</strong> (right).</span><br />
<br> We can go to a position of our choice by entering the gene or chromosomal coordinates under <strong>Position/Search Term</strong> or immediatly go to the ‘Current position’ by clicking on <code>GO</code> on the right-hand side or <strong>Genome Browser</strong> in the menu bar on top.<br />
<br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Go to <em>ACTB</em> gene (gene coding for non-muscular actin part of cytoskeleton).</span></p>
<div class="figure">
<img src="images/week1/3b_ucsc_gb_looking_at_genes/ucsc_gb_actb_question_startpage.png" alt="Genomes page, ready to go to ACTB gene in Human genome GRCh38" />
<p class="caption">Genomes page, ready to go to <em>ACTB</em> gene in Human genome GRCh38</p>
</div>
<p><br></p>
<blockquote>
<p><em>Hint</em> your browser should be showing chr7:5,527,148-5,530,601.</p>
</blockquote>
<p>As discussed in the video, in the top of the window we have navigation and zoom bottons. Right below we can read the chromosomal coordinates of and length of the gene and below that we see a miniature chromosome with the characteristic gray/black pattern from karyotyping. A red |-sign gives us an idea of our current positioning in the chromosome.<br />
<br> <img src="images/week1/3b_ucsc_gb_looking_at_genes/ucsc_gb_actb_locationinchromosome.png" style="width:100.0%" /> <br />
<br></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r exercise"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">     <span class="kw">question</span>(<span class="st">&quot;Question 1: What chromosome is ACTB located on? And is it located on the short or long arm of this chromosome?&quot;</span>,</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">              <span class="kw">answer</span>(<span class="st">&quot;chr7, short arm&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">              <span class="kw">answer</span>(<span class="st">&quot;chr7, long arm&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Where do you see the red |-sign in the miniature chromosome on top?&quot;</span>),</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">              <span class="kw">answer</span>(<span class="st">&quot;chr17, short arm&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Review the UCSC Genome Browser Basics video, or find *chr#* in the browser window&quot;</span>),</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">              <span class="kw">answer</span>(<span class="st">&quot;chr17, long arm&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Review the UCSC Genome Browser Basics video, or find *chr#* in the browser window&quot;</span>),</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">              <span class="dt">allow_retry =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">              )</a></code></pre></div>
<p><br> The data window displays various <em>tracks</em> or <em>track sets</em>. Tracks runs the complete length of the chromosome. Data includes most often:</p>
<ul>
<li><strong>positional information</strong>, of a gene, SNP or cir-regulatory element for example, often displayed as horizontal bars the thickness and color of which can hold additional information depending on the type of track.<br />
</li>
<li><strong>signal information</strong>, of gene expression, detection of histone marks or degree of conservation. Often displayed as histograms but can also be summarized in vertical bars with different levels of color intensity encoding the signal intensity.</li>
</ul>
<p><br> <span style="color: red;"> Question asign what the name of the track with gene locations is or poicutre with c/b/c/d/e/f etc. instead of track name and asking which one diplays the chromosomal coordinates and which one the gene locations?</span></p>
<p><br> The method of display and whether individual tracks are displayed can be changed by:</p>
<ol style="list-style-type: decimal">
<li>Right-mouse click on the grey, vertical bar at the left-most area of the Genome Browser shows you options <code>hide</code>, <code>dense</code>, <code>squish</code>, <code>pack</code> and <code>full</code>. Additional adjustments to the track can be made by selecting <code>Configure [track name]</code> in this menu.<br />
</li>
<li>Scroll to the area below the data window, here you see a possibly overwhelming list of potential tracks, most are <code>hidden</code>. Using the pull down menu you can change display settings of multiple tracks at once. Always make sure to hit the <code>refresh</code> button below the browser window to ensure all changes are performed.</li>
</ol>
<p><br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: hide all tracks and tracksets except for the ruler and GENCODE v32 tracks.</span><br />
<br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: for GENCODE v32, via <code>Configure GENCODE v32</code>, make sure the option “splice variants” is unchecked.</span></p>
<p><br> Your GB session should look like this: <img src="images/week1/3b_ucsc_gb_looking_at_genes/ucsc_gb_actb_genetracksonly.png" style="width:100.0%" /> <br />
<br> We see the complete <em>ACTB</em> gene with exons as horizontally stretched rectangles and introns as horintal lines with arrowheads in them indicating the coding direction/strand. The exons have different heights. The highest regions correspond to <em>coding</em> exons, the other regions are the 5- and 3UTR.<br />
<br> More information about <em>ACTB</em> can be obtained by clicking on the gene in the browser:<br />
<img src="images/week1/3b_ucsc_gb_looking_at_genes/ucsc_gb_actb_geneinformation.png" style="width:100.0%" /> <br />
<br></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r exercise"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 2: How many exons does the *ACTB* gene have?&quot;</span>,</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">          <span class="kw">answer</span>(<span class="st">&quot;5&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;exons are displayed as thicker lines in the gene, introns as lines with arrows in them.&quot;</span>),</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">          <span class="kw">answer</span>(<span class="st">&quot;6&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">          <span class="kw">answer</span>(<span class="st">&quot;7&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;one exon can contain coding and non-coding (UTR) elements&quot;</span> ),</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">          <span class="kw">answer</span>(<span class="st">&quot;8&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;one exon can contain coding and non-coding (UTR) elements&quot;</span>),</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">           <span class="dt">allow_retry =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><br><br />
<img src="images/week1/3b_ucsc_gb_looking_at_genes/ucsc_gb_actb_question_TSSquestion.png" alt="ACT locus for question 3" style="width:100.0%" /> <br />
<br></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r exercise"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 3: Which letter points to the start codon?&quot;</span>,</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">              <span class="kw">answer</span>(<span class="st">&quot;A&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Look at the directionality of the gene, this is more likely the end of the gene.&quot;</span>),</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">              <span class="kw">answer</span>(<span class="st">&quot;B&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Look at the directionality of the gene, this is more likely the end of the coding exons.&quot;</span>),</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">              <span class="kw">answer</span>(<span class="st">&quot;C&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">              <span class="kw">answer</span>(<span class="st">&quot;D&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Be aware of the small 5UTR in this exon!&quot;</span>),</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">              <span class="kw">answer</span>(<span class="st">&quot;E&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;This is more likely the TSS.&quot;</span>))</a></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r exercise"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">     <span class="kw">question</span>(<span class="st">&quot;Question 4: What statement about the *ACTB* gene is correct? (Select ALL that apply.) &quot;</span>,</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">              <span class="kw">answer</span>(<span class="st">&quot;ACTB gene is encoded on the minus-strand&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">              <span class="kw">answer</span>(<span class="st">&quot;The TTS of ACTB gene has a higher coordinate than its TSS&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Look at the direction of arrows in the introns.&quot;</span>),</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">              <span class="kw">answer</span>(<span class="st">&quot;The coding size of this gene is appr 2kb&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">              <span class="kw">answer</span>(<span class="st">&quot;If we want to look at the promoter of this gene, we have to move slightly to the left&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb5-6" data-line-number="6">              <span class="kw">answer</span>(<span class="st">&quot;This gene is not transcribed&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;you don&#39;t have data to say this is correct&quot;</span>)</a>
<a class="sourceLine" id="cb5-7" data-line-number="7">)</a></code></pre></div>
<p><br></p>
<p><span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Go to the gene <em>SPI1</em> </span><br />
<br></p>
<p><em>SPTI1</em> encodes the myeloid and B-cell–specific transcription factor PU.1 <a href="https://doi.org/10.1182/blood-2012-01-402453">Pham 2012 Blood</a>.</p>
<ul>
<li>Copy this sentence and fill in the blanks with numbers in the code:</li>
</ul>
<p><span style="color: red;">Are up- &amp; downstream discussed?&lt;&gt;</p>
<p>SPI1 gene is located on chr […]. It spans […] bp and is encoded on the [plus/minus] strand. It has […] exons and […] introns. It is an [zinc finger domain TF/ETS-domain transcription factor]. The first downstream protein-gene is [MYBPC3/SLC39A13].</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># Fill in the blanks with numbers in the code:  </span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">SPI1 gene is located on chr ...  . It spans ... bp and is encoded on the [plus<span class="op">/</span>minus] strand. It has ... coding exons and ... introns. It is an [zinc finger domain TF<span class="op">/</span>ETS<span class="op">-</span>domain transcription factor]. The first downstream protein<span class="op">-</span>gene is [MYBPC3<span class="op">/</span>SLC39A13].  </a></code></pre></div>
<p><strong>Wonderfull, you are now able to recognize genes, their location within the chromosome, and their coding elements as well as their orientation. These are important reference points used when we evaluate the quality of epigenomics data and to formulate hypothesis that we can test later on.</strong></p>
<hr />
<p><br></p>
</div>
<div id="viewing-blueprint-data" class="section level3">
<h3>1.4.3 Viewing BLUEPRINT data</h3>
<p><strong><em>Yes! Finally time for some real data! Using the UCSC Genome Browser, you will inspect characteristic histone PTM and DNaseI-seq combinations, what their location in relation to genes and - ultimately - what this may say about the gene’s transcription state.</em></strong></p>
<p><br> To upload the data we have already put them together in a so-called track hub, hosted by our own server. We can add this to our browser session by going to the tab “My Data” in the top menu and select “Track Hubs”. In the “My hubs” tab you can paste a URL that directs to the data you want to upload.<br />
<br></p>
<p><img src="images/week1/3c_ucsc_gb_looking_at_data/ucsc_gb_encode_fig1_gototrackhub.png" alt="Go to Track Hubs" style="width:100.0%" /> </p>
<p><img src="images/week1/3c_ucsc_gb_looking_at_data/ucsc_gb_encode_fig2_trackhub.png" alt="My Hubs window" style="width:100.0%" /> </p>
<p><br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Upload the required data using the following URL […].</span><br />
<br><br />
Note that, after adding the trackhub, an additional section is added below the browser, specfic to the tracks in this trackhub.<br />
<br> This trackhub contains histone PTM ChIP-seq and DNaseI-seq data of monocytes, to be precise of <strong>CD14+, CD16- classical monocytes</strong>. <br></p>
<blockquote>
<p><em>Background</em>: CD14 and CD16 are cell surface markers of the <strong>CD (Cluster of Differentiation) gene family</strong> which we will come across more often during this practical. These cell surface molecules and their differential expression are used for immunophenotyping of cells. CD molecules can function as receptors, ligands and, for example, activate signalling cascades, mediate cell adhesion, cell migration, regulate gene expression etc. <a href="https://www.genenames.org/data/genegroup/#!/group/471">HUGO, genegoups, CD</a></p>
</blockquote>
<p><br> We have the following datasets:</p>
<ul>
<li><p>ChIP-seq data histone PTMs:</p>
<ul>
<li>H3K9me3 ChIP-seq (gray)<br />
</li>
<li>H3K27me3 ChIP-seq (gray)<br />
</li>
<li>H3K4me3 ChIP-seq (red)<br />
</li>
<li>H3K4me1 ChIP-seq (yellow)<br />
</li>
<li>H3K27me1 ChIP-seq (peach)<br />
</li>
<li>H3K36me3 ChIP-seq (orange)</li>
</ul></li>
<li>DNaseI-seq data in a blue track.<br />
</li>
<li>polyA-selected, stranded RNA-seq data, again in green<br />
</li>
<li><p>Input ChIP-seq in black.</p></li>
</ul>
<blockquote>
<p><em>Tip</em>: What we haven’t discussed so far is the y-axis at the left side of the track with <em>signal</em> information such as the ones just uploaded. Using this y-axis we can evaluate signal intensities. The height is currently set to 30, 50 or 100. This corresponds to the number of reads mapping to that location per million sequenced reads. You can change the height of the y-axis by right-mouse click on the trakname, select the <code>Configure [trackname]</code> option and change under “Vertical viewing range”, the min&quot; and “max”. Hit <code>APPLY</code> afterwards.</p>
</blockquote>
<p><br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Go back to the <em>ACTB</em> locus and zoom out 3x to see the surrounding region.</span></p>
<blockquote>
<p><em>Hint</em>: your browser should be showing chr7:5,523,694-5,534,055 approximately, a few bases more or less is not a problem.</p>
</blockquote>
<p>Use the RNA-seq data…</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">     <span class="kw">question</span>(<span class="st">&quot;Question 6: would you conclude that the ACTB gene is active?&quot;</span>,</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">              <span class="kw">answer</span>(<span class="st">&quot;Yes&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">              <span class="kw">answer</span>(<span class="st">&quot;No&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Look at the signal in RNA-seq data&quot;</span>)</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">     )</a></code></pre></div>
<p><br> Within the Genomics tracks we can discriminate regions of high signal from the background. These are the peaks.<br />
<br></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 7: Which of the following regions points to a DNaseI-seq peak?)&quot;</span>,</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">  <span class="kw">answer</span>(<span class="st">&quot;chr7:5530101-5530501&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Nice try but this region has a H3K27ac ChIP-seq peak.&quot;</span>),</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">  <span class="kw">answer</span>(<span class="st">&quot;chr7:5530501-5530800&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>, <span class="dt">message =</span> <span class="st">&quot;Yes. Note that this DNaseI-seq peak is located at the *ACTB* TSS/promoter&quot;</span>),</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">  <span class="kw">answer</span>(<span class="st">&quot;chr7:5532001-5532500&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Try again. If anything, you may be looking at the H3K4me1 signal.&quot;</span>, ),</a>
<a class="sourceLine" id="cb8-5" data-line-number="5">  <span class="kw">answer</span>(<span class="st">&quot;None of the these&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Try again. Please look again&quot;</span>),</a>
<a class="sourceLine" id="cb8-6" data-line-number="6">  <span class="dt">allow_retry =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7">     )</a></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r exercise"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 8: Which histone marks, are detected at the ACTB promoter? (Select ALL that apply)&quot;</span>,</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">  <span class="kw">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;H3K9me3 is not detected here&quot;</span>),</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">  <span class="kw">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;H3K27me3 is not detected here&quot;</span>),</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">  <span class="kw">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>, <span class="dt">message =</span> <span class="st">&quot;correct&quot;</span>),</a>
<a class="sourceLine" id="cb9-5" data-line-number="5">  <span class="kw">answer</span>(<span class="st">&quot;H3K4me1&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;H3K4me1 isn ot detected here&quot;</span>),</a>
<a class="sourceLine" id="cb9-6" data-line-number="6">  <span class="kw">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>, <span class="dt">message =</span> <span class="st">&quot;correct&quot;</span>),</a>
<a class="sourceLine" id="cb9-7" data-line-number="7">  <span class="kw">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;H3K36me3 is detected covering the gene body, not the promoter.&quot;</span>),</a>
<a class="sourceLine" id="cb9-8" data-line-number="8">  <span class="kw">answer</span>(<span class="st">&quot;none&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;wrong&quot;</span>))</a></code></pre></div>
<p><span style="color: red"> or is the promoter spanning 3kb and including the H3K4me1 peak? </span><br />
<br></p>
</div>
<div id="peaks-from-peak-calling-software" class="section level3">
<h3>1.4.4 Peaks from peak calling software</h3>
<p><strong><em>It would be cumbersume if we would have to identify all peaks manually, not to mention a highly biased procedure. Luckily we have peak calling software to help us with that. In the following section we will use those results along with the tracks.</em></strong><br />
<br> The peaks have been compiled in another trackhub: [URL].<br />
<br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Upload the peak data.</span><br />
<br> <em>Called</em> peaks are represented as horizontal bars. <br></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r exercise"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 9: What determines the width of the *called* peaks? (Select ALL that apply.)&quot;</span>,</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">         <span class="kw">answer</span>(<span class="st">&quot;the False Discovery Rate used with peak calling&quot;</span>, <span class="dt">correct =</span>T ),</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">         <span class="kw">answer</span>(<span class="st">&quot;The local read density&quot;</span>),</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">         <span class="kw">answer</span>(<span class="st">&quot;The signal in the corresponding &#39;input&#39; dataset at these regions&quot;</span>),</a>
<a class="sourceLine" id="cb10-5" data-line-number="5">         <span class="kw">answer</span>(<span class="st">&quot;The surrounding &#39;noise&#39; signal&quot;</span>),</a>
<a class="sourceLine" id="cb10-6" data-line-number="6">         <span class="dt">allow_retry =</span><span class="ot">TRUE</span>)</a></code></pre></div>
<p><br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Zoom out 3x and then another 10x.</span></p>
<blockquote>
<p><em>Hint</em>: your browser should be showing chr7:5,373,445-5,684,304 approximately, a few bases more or less is not a problem.</p>
</blockquote>
<p>A prominant H3K27me3 peaks comes in the viewing window now. It covers the promoter of <em>FSCN1</em> gene. <br></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 10: What additional histone marks, mark this region? (Select ALL that apply.)&quot;</span>,</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">         <span class="kw">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;we don&#39;t see H3K9me3&quot;</span>),</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">         <span class="kw">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;we don&#39;t see H3K27me3&quot;</span>),</a>
<a class="sourceLine" id="cb11-4" data-line-number="4">         <span class="kw">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>, <span class="dt">correct =</span> T, <span class="dt">message =</span> <span class="st">&quot;correct&quot;</span>),</a>
<a class="sourceLine" id="cb11-5" data-line-number="5">         <span class="kw">answer</span>(<span class="st">&quot;H3K4me1&quot;</span>, <span class="dt">correct =</span> T, <span class="dt">message =</span> <span class="st">&quot;correct&quot;</span>),</a>
<a class="sourceLine" id="cb11-6" data-line-number="6">         <span class="kw">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;A very thin line, not a &#39;clear&#39; peak&quot;</span>),</a>
<a class="sourceLine" id="cb11-7" data-line-number="7">         <span class="kw">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;we don&#39;t see H3K36me3 signal.&quot;</span>),</a>
<a class="sourceLine" id="cb11-8" data-line-number="8">         <span class="kw">answer</span>(<span class="st">&quot;none&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;We are referring to region chr7:5,373,445-5,684,304.&quot;</span>)</a>
<a class="sourceLine" id="cb11-9" data-line-number="9">         )</a></code></pre></div>
<p><br> This is an exmaple of a bivalent promoter.<br />
<br></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 11: Based on the histone marks, would you predict that *FSCN1* is transcribed?&quot;</span>,</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">         <span class="kw">answer</span>(<span class="st">&quot;Yes&quot;</span>),</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">         <span class="kw">answer</span>(<span class="st">&quot;No&quot;</span>, <span class="dt">correct =</span> T)</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">         )</a></code></pre></div>
<p><br> The region inbetween <em>ACTB</em> and <em>FSCN1</em> is maked as well. What marks do you observe and what functional elements could they point to?<br />
<br></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 12: epigenetic marks inbetween *ACTB* and *FSCN1* (Select ALL that apply.)&quot;</span>,</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">  <span class="kw">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;H3K9me3 is not detected here&quot;</span>),</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">  <span class="kw">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;H3K27me3 is not detected here&quot;</span>),</a>
<a class="sourceLine" id="cb13-4" data-line-number="4">  <span class="kw">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb13-5" data-line-number="5">  <span class="kw">answer</span>(<span class="st">&quot;H3K4me1&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">  <span class="kw">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb13-7" data-line-number="7">  <span class="kw">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;H3K36me3 is detected covering the gene body, not the promoter.&quot;</span>),</a>
<a class="sourceLine" id="cb13-8" data-line-number="8">  <span class="kw">answer</span>(<span class="st">&quot;DNaseI-seq&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>))</a></code></pre></div>
<p><br></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 13: What functional element do these marks most likely reflect?&quot;</span>,</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">  <span class="kw">answer</span>(<span class="st">&quot;terminator&quot;</span>),</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">  <span class="kw">answer</span>(<span class="st">&quot;promoter&quot;</span>),</a>
<a class="sourceLine" id="cb14-4" data-line-number="4">  <span class="kw">answer</span>(<span class="st">&quot;TSS&quot;</span>),</a>
<a class="sourceLine" id="cb14-5" data-line-number="5">  <span class="kw">answer</span>(<span class="st">&quot;enhancer&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>, <span class="dt">message =</span> <span class="st">&quot;At least the region spanning chr7:5560200-5564149 is reported a super-enhancer and controls among others the *ACTB* gene.&quot;</span>),</a>
<a class="sourceLine" id="cb14-6" data-line-number="6">  <span class="kw">answer</span>(<span class="st">&quot;centromere&quot;</span>),</a>
<a class="sourceLine" id="cb14-7" data-line-number="7">  <span class="kw">answer</span>(<span class="st">&quot;telomere&quot;</span>),</a>
<a class="sourceLine" id="cb14-8" data-line-number="8">  <span class="kw">answer</span>(<span class="st">&quot;coding region&quot;</span>)</a>
<a class="sourceLine" id="cb14-9" data-line-number="9">)</a></code></pre></div>
<p><br> Another highly expressed gene in this cell type is <em>CD14</em>, remember it’s protein was used to select these cells on. Answer the following question about this gene using the UCSC Genome Browser:<br />
<br></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r exercise"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 11: What statement about *CD14* is correct? (Select ALL that apply) &quot;</span>,</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">  <span class="kw">answer</span>(<span class="st">&quot;CD14 gene is encoded on the minus-strand&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">  <span class="kw">answer</span>(<span class="st">&quot;The TTS of CD14 gene has a higher coordinate than its TSS&quot;</span>), </a>
<a class="sourceLine" id="cb15-4" data-line-number="4">  <span class="kw">asnwer</span>(<span class="st">&quot;*CD14* is encoded on the long arm of chr5&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb15-5" data-line-number="5">  <span class="kw">asnwer</span>(<span class="st">&quot;*CD14* is encoded on the short arm of chr5&quot;</span>),</a>
<a class="sourceLine" id="cb15-6" data-line-number="6">  <span class="kw">answer</span>(<span class="st">&quot;The TSS of *CD14* is not accessible for DNaseI&quot;</span>), </a>
<a class="sourceLine" id="cb15-7" data-line-number="7">  <span class="kw">answer</span>(<span class="st">&quot;A DNaseI-peak appr 1.5 kb upstream of the TSS marks a candidate enhancer.&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb15-8" data-line-number="8">  <span class="dt">allow_retry =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><br> Approximately 20 kb downstream of <em>CD14</em> is another enhancer. What histone marks populate/mark this enhancer? (Downstream with respect to the <em>CD14</em> orientation.)</p>
<blockquote>
<p>For testing purposes: the enhancer I’m pointing to is located at chr5:140611060-140612601</p>
</blockquote>
<div class="sourceCode" id="cb16"><pre class="sourceCode r exercise"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 12: Which marks mark the enhancer 20kb downstream of *CD14*?&quot;</span>,</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">  <span class="kw">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;not H3K9me3&quot;</span>),</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">  <span class="kw">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;not H3K27me3&quot;</span>),</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">  <span class="kw">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;not H3K4me3&quot;</span>),</a>
<a class="sourceLine" id="cb16-5" data-line-number="5">  <span class="kw">answer</span>(<span class="st">&quot;H3K4me1&quot;</span>, <span class="dt">correct =</span> T),</a>
<a class="sourceLine" id="cb16-6" data-line-number="6">  <span class="kw">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>, <span class="dt">correct =</span> T),</a>
<a class="sourceLine" id="cb16-7" data-line-number="7">  <span class="kw">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;not H3K36me3&quot;</span>),</a>
<a class="sourceLine" id="cb16-8" data-line-number="8">  <span class="kw">answer</span>(<span class="st">&quot;DNaseI&quot;</span>, <span class="dt">correct =</span> T),</a>
<a class="sourceLine" id="cb16-9" data-line-number="9">  <span class="dt">allow_retry =</span> T)</a></code></pre></div>
<p><br> So far we have seen active promoters, bivalent promoters and enhancers. A region silenced by H3K9me3 is loacted appr. 120kb upstream of <strong>CD14</strong>.<br />
<br></p>
<blockquote>
<p>Tip: zoom out to chr5:139,000,001-143,000,000.</p>
</blockquote>
<div class="sourceCode" id="cb17"><pre class="sourceCode r exercise"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 13: The expression of what gene family is controlled by this marking?&quot;</span>,</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">  <span class="kw">answer</span>(<span class="st">&quot;PCDHA gene family&quot;</span>, <span class="dt">correct =</span> T, <span class="dt">message =</span> <span class="st">&quot;PCDHA gene cluster encodes a family of cadherin-like cell surface proteins that are expressed in neurons and are present at synaptic junctions.&quot;</span>),</a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="kw">answer</span>(<span class="st">&quot;A group of LINC RNAs&quot;</span>),</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">  <span class="kw">answer</span>(<span class="st">&quot;Family of KRAB-ZNF encoding genes&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;not H3K27me3&quot;</span>),</a>
<a class="sourceLine" id="cb17-5" data-line-number="5">  <span class="dt">allow_retry =</span> T)</a></code></pre></div>
<p><br></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r exercise"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 14: What is the approximate size of the H3K9me3 silenced region? (*Tip2: click-and-drag in the brwoser to capture and thereby measure the width of the H3K9me3 domain)&quot;</span>,</a>
<a class="sourceLine" id="cb18-2" data-line-number="2">  <span class="kw">answer</span>(<span class="st">&quot;250kb&quot;</span>),</a>
<a class="sourceLine" id="cb18-3" data-line-number="3">  <span class="kw">answer</span>(<span class="st">&quot;500kb&quot;</span>),</a>
<a class="sourceLine" id="cb18-4" data-line-number="4">  <span class="kw">answer</span>(<span class="st">&quot;750kb&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>),</a>
<a class="sourceLine" id="cb18-5" data-line-number="5">  <span class="kw">answer</span>(<span class="st">&quot;1Mb&quot;</span>),</a>
<a class="sourceLine" id="cb18-6" data-line-number="6">  <span class="dt">allow_retry =</span> T)</a></code></pre></div>
<p><br> Silencing does not necessarily have to be mediated by H3K9me3.<br />
<br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: Go to the ACTA1 gene, encoding muscular actin. Zoom out 10x.</span></p>
<blockquote>
<p><em>Hint</em>: you should be at ± chr1:229,418,420-229,446,919.</p>
</blockquote>
<p><br></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 13=5: Which histone mark mediates repression of *ACTA1* in these cells?&quot;</span>,</a>
<a class="sourceLine" id="cb19-2" data-line-number="2">         <span class="kw">answer</span>(<span class="st">&quot;H3K9me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;wrong&quot;</span>),</a>
<a class="sourceLine" id="cb19-3" data-line-number="3">         <span class="kw">answer</span>(<span class="st">&quot;H3K27me3&quot;</span>, <span class="dt">correct =</span> T),</a>
<a class="sourceLine" id="cb19-4" data-line-number="4">         <span class="kw">answer</span>(<span class="st">&quot;H3K4me3&quot;</span>,  <span class="dt">message =</span> <span class="st">&quot;correct&quot;</span>),</a>
<a class="sourceLine" id="cb19-5" data-line-number="5">         <span class="kw">answer</span>(<span class="st">&quot;H3K4me1&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;correct&quot;</span>),</a>
<a class="sourceLine" id="cb19-6" data-line-number="6">         <span class="kw">answer</span>(<span class="st">&quot;H3K27ac&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;wrong&quot;</span>),</a>
<a class="sourceLine" id="cb19-7" data-line-number="7">         <span class="kw">answer</span>(<span class="st">&quot;H3K36me3&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;wrong&quot;</span>),</a>
<a class="sourceLine" id="cb19-8" data-line-number="8">         <span class="kw">answer</span>(<span class="st">&quot;none&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;wrong&quot;</span>),</a>
<a class="sourceLine" id="cb19-9" data-line-number="9">         <span class="dt">allow_retry =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb19-10" data-line-number="10">         )</a></code></pre></div>
<p><br></p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 16: Which other observations support that this gene is not active? (Select ALL that apply)&quot;</span>,</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">         <span class="kw">answer</span>(<span class="st">&quot;Absence of H3K36me3 marking covering the gene body.&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;H3K36me3 is indeed lacking but other observations are correct as well.&quot;</span>),</a>
<a class="sourceLine" id="cb20-3" data-line-number="3">         <span class="kw">answer</span>(<span class="st">&quot;Absence of signal in RNA-seq signal&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;RNA-seq is indeed lacking but other observations are correct as well.&quot;</span>),</a>
<a class="sourceLine" id="cb20-4" data-line-number="4">         <span class="kw">answer</span>(<span class="st">&quot;Absence of H3K4me3 signal at the *ACTA1* promoter&quot;</span>,  <span class="dt">message =</span> <span class="st">&quot;correct&quot;</span>),</a>
<a class="sourceLine" id="cb20-5" data-line-number="5">         <span class="kw">answer</span>(<span class="st">&quot;All the above&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span>, <span class="dt">message =</span> <span class="st">&quot;Indeed, the absence of signals in RNA-seq, H3K36me3 *and* H3K4me3 all support that this gene is not expressed. Observe that the H3K27me3 signal covers a relatively broad domain.&quot;</span> ),</a>
<a class="sourceLine" id="cb20-6" data-line-number="6">         <span class="dt">allow_retry =</span> <span class="ot">TRUE</span></a>
<a class="sourceLine" id="cb20-7" data-line-number="7">         )</a></code></pre></div>
<p><br></p>
</div>
<div id="recognizing-blinded-genomics-data" class="section level3">
<h3>1.4.5 Recognizing blinded genomics data</h3>
<p>So far you have browsed through ChIP-, DNaseI- and RNA-seq data. These tracks had the correct track names attached to them.<br />
<br> <span style="color: orange; font-family: Arial; font-size:18px;">ACTION: upload the following track hub [URL].</span><br />
<br> Our collegue accidently let the tubes fall before the tubes were properly labelled (always label your tubes first before adding your sample!). We know ChIPs were performed for H3K9me3, H3K27me3, H3K4me3, H3K36me3, H3K4me1 and H3K27ac. Which track (labelled a-f) is which?</p>
</div>
<div id="summary" class="section level3">
<h3>1.4.6 Summary</h3>
<p>Hypothesis forming, where do marks occur?</p>
</div>
</div>
<div id="old-text-ignore-please" class="section level2">
<h2>Old text, ignore please!!</h2>
<div class="sourceCode" id="cb21"><pre class="sourceCode r exercise"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw">question</span>(<span class="st">&quot;Question 9: The DNaseI-seq peak located *upstream* of the *ACTB* could be an enhancer. What observation gives additional supports for this statement?&quot;</span>,</a>
<a class="sourceLine" id="cb21-2" data-line-number="2">  <span class="kw">answer</span>(<span class="st">&quot;enhancer RNA-seq signal&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;There is no RNA-seq signal in this region so no.&quot;</span>),</a>
<a class="sourceLine" id="cb21-3" data-line-number="3">  <span class="kw">answer</span>(<span class="st">&quot;Absence of H3K4me1 signal in this region.&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;It is not the *absence* of H3K4me1 which is associated with enhancers... &quot;</span>),</a>
<a class="sourceLine" id="cb21-4" data-line-number="4">  <span class="kw">answer</span>(<span class="st">&quot;Presence of H3K4me1 signal in this region.&quot;</span>, <span class="dt">correct =</span> <span class="ot">TRUE</span> ),</a>
<a class="sourceLine" id="cb21-5" data-line-number="5">  <span class="kw">answer</span>(<span class="st">&quot;The location of this DNaseI-seq peak suggests it is as it lies outside a gene&quot;</span>, <span class="dt">message =</span> <span class="st">&quot;Enhancer do not exclusively lie outside genes but also in for example introns.&quot;</span>),</a>
<a class="sourceLine" id="cb21-6" data-line-number="6">  <span class="dt">allow_retry =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p>Other H3K37me3 marked promoters:</p>
<ul>
<li>MYOD1 gene, encoding tF for muscle cell differentiation<br />
</li>
<li>SLCA4A6 gene, encoding a serotonin (neurotransmitter) … bivalent? Has bit of H3K4me1.<br />
</li>
<li>OCE2 (eye color) - also bivalent?<br />
</li>
<li>NEUROG2 (tf for noerogenesis) - also bivalent?<br />
</li>
<li>OLIG2 (tf neuro/oligodendrite genesis) - also bivalent?</li>
</ul>
<p>Also look at a broad H3K9me3 domain? I see some but those have H3K4me3 and H3K36me3 marking as well….</p>
<ul>
<li><p>Background:</p>
<ul>
<li>H3K9me3 spans braod domains of 10ks of bp’s. Constitutive hetrochromatin and less often facultative heterochromatin.<br />
</li>
<li>H3K9me3 and H3K27me3 are mechanistically differen.<br />
</li>
<li>Not part of bivalent promoters?<br />
</li>
<li>reviews: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4698194/">Becker 2017 Trens in Genetics</a>; <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6803365/">Ninova 2019 Genomics</a></li>
</ul></li>
<li><p>Example locations:</p>
<ul>
<li>chr19 broad domains with KRAB-ZNF encoding genes. Spanning 0.4-1Mb chr19:28,115,617-58,617,616 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1665633/">Vogel 2005 Genome Res</a><br />
</li>
<li>H3K9me3 is associated with non-coding regions in the genome as well as repressing lineage-inappropriate genes <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4698194/">Becker 2015 Trends in Genetics</a></li>
</ul></li>
</ul>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
